---
output: pdf_document
header-includes:
    - \usepackage{setspace}\doublespacing
    #- \usepackage[left]{lineno}
    #- \linenumbers
    - \usepackage{helvet}
    - \renewcommand{\familydefault}{\sfdefault}
theme: Springer
title: \singlespace Identification of cell types involved in rare disease-associated human phenotypes
subtitle: \singlespace Using expression weighted celltype enrichment analysis to identify cell-phenotype relationships in scRNA-seq data from Descarets and phenotype associated genelists from the human phenotype ontology
#date: 03/08/2021
author: 
- Robert Gordon-Smith, Molecular and Cellular Biosciences MRes
- Dr Nathan Skene, Department of Brain Sciences, Imperial College London

bibliography: bibliography.bib
csl: elsevier-harvard.csl
fontsize: 12pt
linkcolor: blue
spacing: double
# toc: yes
params:
#  results_data: "data/Descartes_All_Results_extras.rda"
  results_data: "data/descartes_all_results_061021.rda"
abstract: \singlespace Despite the name, rare diseases (RD) contribute to a significant burden of disease globally. The increasing accessibility of genomic and biomedical datasets, and high throughput analytical techniques, has made it possible to uncover new insights into the genetic susceptibility and cell types involved. With the low prevalence of individual RDs, they have often not had the resource allocation they need. Here we attempt to overcome this by tackling the problem as a whole. The Human Phenotype Ontology (HPO) contains disease phenotypes annotated with associated risk genes. Here, these disease-associated gene lists, combined with human single-cell RNA sequence data, were used to identify the primary cell types involved in the HPO disease phenotypes. Expression weighted cell type enrichment (EWCE) was used for the analysis with 100,000 bootstrap reps. Over 8000 significant cell-phenotype associations were found (where *q*<0.05). The results were shown to be overrepresented by expected enrichments. For example, immune cells were more frequently enriched for phenotypes from the \"Abnormalities of the immune system\" ontology branch than all other cell types combined (*p*<0.05). The analysis was able to reproduce many previously known cell-phenotype relationships documented in the literature. There were also many novel and unexpected results, on examination, many of these were shown to have have a plausible mechanistic explanation. We can be confident that within these results are many previously unknown insights that could provide targets to future research. As too many results were produced to present in a single paper, the Rare Disease EWCE web app was developed to make the analysis available to clinicians and researchers, without the need for specialist knowledge and computational resources (https://ovrhuman.github.io/ewce_website/). \newpage


---

```{r echo = FALSE, message = FALSE, warning=FALSE}


library(patchwork)

library(ggplot2)
library(ontologyIndex)
library(ontologyPlot)
library(EWCE)
library(cowplot)
library(wesanderson)
library(dplyr)

data(hpo)
phenotype_to_genes = HPOExplorer::load_phenotype_to_genes("data/phenotype_to_genes.txt")
disease_descriptions = readRDS("data/disease_descriptions.Rda")
rownames(disease_descriptions) = disease_descriptions$HPO_id
load("data/Descartes_All_Results_extras.rda")
descartes_mappings = read.csv("data/DescartesHuman_celltype_mapping.csv")
descartes_mappings$level1 = gsub("_"," ",descartes_mappings$level1)
#tm_mappings = read.csv("data/TabulaMuris_celltype_mapping.csv")
ctd = readRDS("data/CTD_Descartes_withplot.rds")

source("source/cell_select_ggnetwork_plot.R")
source("source/phenotypes_per_cell_plot.R")
source("source/ewce_plot_function.R")

if (!require(HPOEWCE)) {
  #install.packages("rd_package/RareDiseaseEWCE", repos=NULL, type="source")
  devtools::install_github("ovrhuman/HPOEWCE")
  library(HPOEWCE)
}
if (!require(MultiEWCE)) {
  #install.packages("rd_package/RareDiseaseEWCE", repos=NULL, type="source")
  devtools::install_github("neurogenomics/MultiEWCE")
  library(MultiEWCE)
}
if (!require(HPOExplorer)) {
  #install.packages("rd_package/RareDiseaseEWCE", repos=NULL, type="source")
  devtools::install_github("neurogenomics/HPOExplorer")
  library(HPOExplorer)
}


savePlot <- function (plot, w = 1000, h = 1000, path = "savedPlot.png") {
  png(path,width = w, height = h)
  print(plot)
  dev.off()
}

ggloadImage <- function(path) {
  img <- png::readPNG(path)
  g <- grid::rasterGrob(img,interpolate = TRUE)
  plt <- qplot(1:10,1:10,geom="blank") +
    annotation_custom(g,xmin=-Inf,xmax=Inf,ymin = -Inf, ymax=Inf) +
    theme_blank()
  return(plt)
}


ggplotify <- function(plot_object, height_px = 1000, width_px = 1000) {
  fp <- "ggplotify_temp.png"
  savePlot(plot_object, width_px, height_px, fp)
  plt <- ggloadImage(fp)
  file.remove(fp)
  return(plt)
}



```


```{r rmd_stuff, message=FALSE, echo=FALSE}
# useful stuff
## Rmd commands
# 
# **Cross referencing**
# To reference a figure, place a label in the figure caption: `\\label{fig:figname}`.
# Then reference it in the text with `\ref{fig:figname}`
# 
# **Bibliography and citations**
# Include a .bib document in the yaml. copy bib format citaitons into this file from [zbib](zbib.org). The first parameter in the bib is the reference name of the citation. We can then use that reference name to cite in text like this `@HumanPhenotypeOntology2020` or at the end of a paragraph like this `[@HumanPhenotypeOntology2020]`. Check bibliogoraphy and citation cookbook docs for more options. see \ref{useful}
# 
# **Knit to word doc**
# change pdf_document to word_document in yaml. 
# 
# **laytex commands**
# use both laytex commands and markdown syntax to edit stuff. only laytex commands work in the Yaml header?
```


\newpage

# Introduction

## Rare diseases 
Although rare diseases (RD) have a prevalence of less than 1 in 2000, they are over 6000 in number [@orphanet2012]. As a conservative estimate, between 263 and 466 million patients suffer from RDs globally, contributing to a significant disease burden. For individual RDs, economic incentives are often not aligned with the need for research and drug development. However, despite the large number of distinct RDs, approximately 72% are genetic disorders, which may provide opportunities to approach them collectively [@estimatingRareDiseasePrevalence2020]. Given the rise of genome-wide association studies (GWAS), electronic healthcare records, and transcriptomic data, it is increasingly feasible to utilise high-throughput computational methods. Not only does increased understanding of RDs benefit patients directly, but RDs can also be used as disease models and they can give valuable insights into complex polygenetic conditions, further extending the impact of RD research beyond what is implied by the term "rare" [@LessonsFromMonogeneticDisease2006].

Here we utilise human single cell transcriptomic data from the Descartes human cell atlas, and gene-phenotype relationships from the Human Phenotype Ontology (HPO) to identify cell types associated with the primary disease pathology for `r length(unique(phenotype_to_genes$Phenotype))` RD phenotypes [@descartes_2020; @HumanPhenotypeOntology2020]. Information scarcity and poorly defined case definitions have historically presented significant obstacles to the diagnosis, treatment, and research of RDs. RDs patients often find their diagnosis too late or not at all. Additionally, given the broad scope of this research, it is impossible to present all of the results in a single paper. In light of these points, a crucial part of this project is to develop a publicly available web-based application for retrieval of results relevant to the users field of study. 


## Human Phenotype Ontology
The Human Phenotype Ontology (HPO) is an initiative founded in 2008 that consists of an ontology of $13\,000$ annotated clinically relevant phenotypes. They are connected in a directed acyclic graph with edges representing transitive "is-a" connections [@HumanPhenotypeOntology2020]. Each phenotype is a subclass of its parent phenotypes, all the way up to the root node of all phenotypes, which is "Phenotypic abnormality". This allows for computationally efficient and human interpretable representation of the complex relationships between phenotypes. For example, Figure \ref{fig:ADHD_ancestors} shows all ancestor terms for Attention Deficit Hyperactivity Disorder (ADHD), which is a subclass of both hyperactivity and short attention span, which themselves are subclasses of two nervous system disorders, movement and behavioural abnormalities, respectively.

```{r ADHD_ancestors,fig.cap = "\\label{fig:ADHD_ancestors}\\textbf{ADHD ancestor terms.} The ancestor terms of ADHD (HPO Id: HP:0007018) in the HPO. It is 5 generations down from the root node, \"Phenotypic abnormality\". The arrows represent \"is-a\" relationships, meaning that a child term is a sub-class of the term above it. This was plotted using R version 4.05 and the ontolgyX package [@RLanguage; @ontologyXpackage2017].", message=FALSE, echo=FALSE}
namer = function(term){
  return (hpo$name[term])
}
terms = get_term_property(hpo,property="ancestors" ,term ="HP:0007018",
                          as_names=FALSE)
print(onto_plot(hpo, terms= terms[2:length(terms)],label = namer,
          shape="circle",fontsize = 80, edge_attributes = list(color="grey")))
```

As well as conveying the relationship between nodes (phenotypes), the description logic of bio-ontologies can also describe properties of the node, including synonyms, descriptions, and associated genes and diseases. This data integration allows for implicit knowledge hidden in the data to become explicit and retrievable [@ClassificationOntologyPrecisionMed2018]. Currently, much of the HPO focuses on rare Mendelian diseases, many of which have extensive annotations taken from electronic health records and genotype data. The aggregation of data from heterogeneous sources makes the HPO an indispensable tool in phenotype-driven genomic research, precision medicine, and diagnostics. 

## Descartes Human-Cell atlas
The Descartes human-cell atlas is a single-cell gene expression reference atlas for the human body, created using three-level combinatorial indexing (sci-RNA-seq3) [@descartes_2020]. It includes 4 million cells representing 15 different organs from 121 human fetal samples. Uniform manifold approximation and projection (UMAP) and Louvain clustering were applied to the transcriptomic data. After accounting for common annotations across tissues, it clustered into 77 main cell types. Additionally, the Tabula Muris whole-body mouse scRNA dataset was analysed, which clustered into 38 distinct cell types [@tabulamuris_2018]. The Descartes results are the primary focus of this study.

## Expression weighted cell type enrichment (EWCE) 
EWCE is a technique developed by @skene_2016 that utilises single-cell gene expression data to determine whether a set of genes is significantly associated with a particular cell type with respect to expression. Unlike other enrichment analysis techniques, like population-specific expression analysis (PSEA), developed by @PSEA_2011, EWCE does not rely on quantitative gene expression data from disease tissue samples or predefined cell marker genes. Instead, EWCE takes a set of genes found to be associated with a phenotype or disease and, using scRNA-seq data, it identifies cells that express that gene set more than could be expected by chance from a random gene set of the same length.

Using scRNA-seq data allows for the use of techniques such as UMAP and t-distributed stochastic neighbour embedding (t-SNE) to classify the cell types involved. This allows EWCE to identify cell sub-types at a higher resolution than is possible with the predefined marker genes used in PSEA. Another advantage of EWCE is that the gene lists can be obtained from genome-wide association studies (GWAS) and therefore are involved in the primary genetic susceptibility. Conversely, PSEA results can be confounded by secondary "reactive" expression. For example, in PSEA you may find that inflammatory cytokines are overexpressed in disease tissue sample due to immune activation. It would then be unclear if this is part of the primary pathomechanism or is a protective effect, brought about by an appropriate immune response to the condition. Additionally, the lack of need for quantitative expression data from disease samples makes it applicable to many more publicly available data sets.

In this study, all `r length(unique(phenotype_to_genes$ID))` RD associated gene lists were taken from the HPO and, using the human whole-body scRNA-seq data from Descartes, EWCE was run to identify the key cell types involved in the phenotypes. This task would not have been feasible with other methods and without these large publicly available data sets. There are multiple objectives to this project. The first is to create a program, written in R, to pull data from the HPO and the Descartes Human cell atlas, clean and format the data, and then run EWCE to identify significantly enriched cell types for the RD phenotypes [@RLanguage]. A secondary objective is to make the code written for this project available for future research and follow up studies by storing it on the Neurogenomics lab GitHub, and submitting modifications to the EWCE R package that is published on Bioconductor, an open-source repository of software for bioinformatics. It is common practice to submit improvements in open source software development and helps to drive the improvement of future versions. Finally, a web-based application must be developed for exploring and retrieving the results as there are too many to be presented in one paper. This will allow domain experts to view results most relevant to their field, which will bring gene set enrichment analysis to a wider audience and enable the full impact of the study to be realised. 


\newpage
# Materials and methods {#method}

## EWCE overview
The EWCE R package from Bioconductor (version 1.1.0) is used for the analysis [@EWCE_Bioconductor_2021]. To briefly summarise the technique, as described by @skene_2016, EWCE takes a target gene list of length $n$, referred to as $T$, and a set of background genes referred to as $B$. A scRNA data set is then used to calculate a gene-cell specificity score for each gene in every cell type. Simply put, this is obtained by dividing the expression of a gene within a cell type by the total expression of the gene in all cell types. 

$$
e_{g,c} = \frac{\sum_{i=1}^{|L|}F(g,i,c)/N_c}{\sum_{r=1}^{k}(\sum_{i=1}^{|L|}F(g,i,r)/N_r)}
$$

$$
F(g,i,c) = \begin{cases}
  r_{g,i}, l_i = c\\
  0, l_i \neq c 
\end{cases}
$$

Where $e_{g,c}$ is the specific expression of gene $g$ in cell type $c$. $N_c$ is the number of $c$, and $r_{g,i}$ is expression of $g$ in cell $i$ (indexed from $c$). We can then sum the specificity scores of the genes in $T$ to get its total expression specificity score in a given cell ($\gamma$). This is done for all cells, enabling us to quantify the level of specific expression of gene list $T$ (indexed by $X$) in each cell type ($c$).

$$
\gamma(X,c) = \sum_{g \in X}e_{g,c}
$$

Bootstrapping is then used to determine if the level of expression of $T$ in a given cell is statistically significant, in other words, the probability of enrichment. To do this, $100\,000$ random gene sets of length $n$ are then randomly sampled without replacement from $B$. The same cell-specific expression calculation, as described above, is then calculated for all $100\,000$ random gene sets in each cell type "$c$". This gives a probability distribution of cell-specific expression for gene sets of length $n$ in any given $c$. The mean and standard deviation of this distribution is normalised to 0 and 1 respectively and is used to calculate a Z score. We can then determine the probability of enrichment of $T$ in $c$ based on the number of bootstrap gene lists that have a higher cell type specific expression than $T$.

$$
P(X\text{ enriched for }c) = \frac{\sum_{j=1}^{100000} \begin{cases}
1 \text{ } \gamma(X,c) > \gamma(D_j,c) \\
0 \text{ } \gamma(X,c) < \gamma(D_j,c) \end{cases} }{100000}
$$

Gene sets with higher specific expression than most random gene sets of the same length have a high probability of enrichment in a given cell type. The target gene sets used here are obtained from the HPO. Each gene set is known to be associated with a rare disease phenotype. If one of these gene sets is significantly enriched in a cell type, then it is likely that the cell type plays a role in the pathology, possibly making them valuable targets for future research.

Gene-cell specificity scores, rather than absolute expression, is used because it allows us to weight the expression of a gene by how commonly it is expressed in other cell types. In other words, genes that are only expressed highly in a few cell types are weighted higher and contribute more to the probability of enrichment than common genes. However, we must exclude genes with very low expression (mean < 0.2 in all cell types), as a small number of reads of the gene could make it appear highly specific. The Descartes human cell atlas scRNA dataset was used. It contained `r length(ctd[[1]]$annot)` cell samples that clustered into `r length(unique(ctd[[1]]$annot))` distinct cell types. The RD phenotype-associated gene lists were obtained from the HPO. There are `r length(unique(phenotype_to_genes$Phenotype))` phenotypes in the HPO, however, phenotypes with less than 4 associated genes were excluded from analysis, leaving `r length(unique(all_results_merged$list))` gene lists. The fold change of enrichment is calculated by dividing the specific expression of $T$ by the mean expression of the bootstrapped random gene sets. 

The EWCE package has a function for visualising results (`ewce_plot`) which returns a plot of the fold change of expression for a gene set in each cell type. It also gives a dendrogram of the clustering of the cell types in the plot. The dendrogram must then be manually aligned with the bar chart using image editing software in the current version. Given the many thousands of results that need to be visualised for this study, manual alignment of the plots would not be possible. A modified version of the `ewce_plot` function was created to enable automatic alignment of the dendrogram to the bar chart. This solution was then submitted to the EWCE developers and the modified `ewce_plot` function has now been published on [Bioconductor.org](http://bioconductor.org/packages/release/bioc/html/EWCE.html) in the latest version of the EWCE package.  

## Multiple testing adjustment
Due to the high multiple testing burden of this analysis, the Benjamini-Hochberg (BH) method was used to limit the false discovery rate (FDR). Traditional family-wise error rate based methods, like Bonferroni corrections, are too conservative when performing these types of high-throughput analyses, resulting in the rejection of many positive results. Instead, BH corrects the proportion of significant findings down to an acceptable FDR [@BenjaminiHochberg1995]. The p-values that have been adjusted for FDR are referred to as q-values.

## Analysis workflow [needs to be updated]
This analysis involved multiple steps, including pulling the data from online resources, cleaning and formatting it and running the analysis. A makefile (`Makefile.R`) was created to automate the whole process. This RD EWCE analysis makefile was then pushed to the UKDRI Neurogenomics lab [GitHub](https://github.com/neurogenomics/), a remote host for software development and version control. This allows it to be reused in follow up analysis, when more phenotypic data becomes available, as well as to be developed further for future studies by other researchers. 

A visual representation of the workflow can be seen in Figure \ref{fig:Figure2}. The latest scRNA-seq data set is downlaoded from Descartes and used to produce the cell type data (CTD) file by calculating specificity scores for each gene-cell combination. The CTD is structured as a list of matrices of normalised specificity scores, with columns representing the cell type and rows representing gene. The matrices are generated at various levels of cell-clustering resolution, so that the user can decide how specific they wish to be in distinguishing between cell sub-types. In this case, the levels generated were Level 1 (Cell lineage), Level 2 (Organ Cell lineage), Level 3 (Cell lineage Developmental day), and Level 4 (Organ Cell lineage Developmental day). The level 1 data was used as it provides sufficient resolution without over-complicating the results or increasing the multiple testing burden unnecessarily. The `gen_results` function is then called, which pulls the phenotype gene lists from the HPO website and runs EWCE on each, using the CTD file gene-cell specificity scores. 

In order to run the EWCE analysis with $100\,000$ bootstraps on `r length(unique(phenotype_to_genes$Phenotype))` phenotypes, significant computational resources were required. To make this possible, the EWCE R package was modified to support parallel processing. A secure shell connection (SSH) was then established with the Imperial College High Performance computing (HPC) service; and an 8 CPU machine with 96 GB of random access memory (RAM) was requested. The analysis ran at a rate of approximately 90 phenotypes per hour and took a little over 100 hours to complete. A terminal multiplexer (Tmux) was used so that the session would continue to run in the event of disconnection.
 
## Results Portal
A web-based application was made so that the study can have its maximum impact. It is important for researchers and clinicians to be able to retrieve relevant results without the need for substantial computational resources and specialist knowledge of enrichment analysis. As the results give cell-phenotype relationships, they need to be retrievable by both fields (cell type and phenotype), and it must also be possible to subset them by significance and effect size (q-value and fold change). Due to the heavy computation required to generate the figures and subset the results, the cell selection and phenotype selection features were developed as separate web applications to reduce loading times. The apps were created using the Shiny Web application Framework for R and deployed on the [ShinyApps](https://www.shinyapps.io/) server [@Shiny_package]. A landing page was created using HTML and CSS to provide links to the apps, data sources, and code repositories; as well as to give general information about the project. 

```{r Figure2,  fig.width = 20, fig.cap = "\\label{fig:Figure2}\\textbf{Workflow.} \\textbf{A} The scRNA-seq data was taken from the Descartes Human Cell Atlas and the phenotype associated genelists from the Human Phenotype Ontology. \\textbf{B} This represents the computation of results using EWCE, where a specificity score for the target gene list, in a given cell, is calculated and then compared with a distribution of 100,000 randomly sampled genelist of the same length. \\textbf{C} This is the interactive results portal. It allows the user to search for significant phenotype enricments for a given cell type. The figure is generated using a combination of bespoke functions and the ggplot2 and ggnetwork packages [@ggplot2_package; @ggnetwork_package]. The size of the nodes represents the direction of the \"is_a\" relationship between nodes, where parent terms are larger than their child terms. Expression fold change is represented by the colour. In this example, the user has selected Acinar cells, with a significance threshold of \\textit{q}<0.005 and a fold change threshold of > 5. The hover box is comprised of further results and a phenotype definition pulled from the HPO API.", message=FALSE, echo=FALSE, warning =FALSE, results = FALSE}
#figure_count = figure_count + 1; figures$EWCE_home = figure_count;
library(imager)
Figure1 = load.image("figures/rd_workflow2.png")
print(plot(Figure1, axes = FALSE))
```

A demonstration of the interactive plot feature from the cell-select app can be seen in Figure \ref{fig:Figure2} C. The ggnetowrk R package was used to give x and y coordinates to each phenotype in the network and the plotting was done using the ggplot2 R package [@ggnetwork_package; @ggplot2_package]. Mapping of the ontology level of the phenotypes to the size of the node was done with some bespoke implementation of graph algorithms so that phenotypes higher up the HPO hierarchy appear larger than those below (within a connected component of the graph). Functions were also made to generate the hover box by pulling data from the HPO API. This allows the user to see the results and disease definition when their mouse hovers over. These functions were then included in the [HPOExplorer](https://github.com/neurogenomics/HPOExplorer) R package for use in other projects. 

The interactive plot works well for exploring the data. However, it is less clear when printed as a static image. A second, more print-friendly version, was also created and users may download the figure from the app. This was created using the ontologyX R package `onto_plot` function, modified to allow a heat map of fold change. An example of these print friendly plots can be Figure \ref{fig:Figure2} A, which is a subset of the HPO network. 


\newpage
# Results



```{r pheno_by_branch_withd_dend, echo = FALSE, warning = FALSE , results=FALSE,message = FALSE }
palette_branch <- wesanderson::wes_palette("Darjeeling1")


plot_n_signif_phenos_per_cell_by_branch_den <- function (all_results_merged, plot_branches = c("Abnormality of the nervous system", 
                                                "Abnormality of the cardiovascular system", "Abnormality of the immune system"), 
          q_threshold = 0.05, hpo = hpo, phenotype_to_genes = phenotype_to_genes, 
          ctd = ctd, make_dendro = TRUE, palette_branch = c("#FF0000", "#00A08A", "#F2AD00")) 
{
  hpo_branches = hpo$children["HP:0000118"]
  hpo_branches_names = c()
  for (b in hpo_branches) {
    hpo_branches_names = c(hpo_branches_names, hpo$name[b])
  }
  main_branch_df = data.frame(hpo_id = hpo_branches, phenotype = hpo_branches_names)
  descendants = list()
  for (b in hpo_branches[[1]]) {
    descendants[[hpo$name[b]]] = ontologyIndex::get_descendants(hpo, 
                                                                b)
  }
  if (!"branch" %in% colnames(all_results_merged)) {
    phenotypes = unique(phenotype_to_genes$Phenotype)
    all_results_merged$branch = NA
    pheno_branches = c()
    for (p in phenotypes) {

      index = match(p, hpo$name)
      if (!is.na(index)) {
        id = hpo$id[[index]]
        for (i in seq(1, length(descendants))) {
          if (id %in% descendants[[i]]) {
            pheno_branches[p] = names(descendants[i])
          }
        }
      }
    }
    all_results_merged$branch = NA
    remaining = length(unique(all_results_merged$list))
    #cat("Asigning branch to phenotypes\nN remaining:\n")
    for (p in unique(all_results_merged$list)) {
      #cat("\r", remaining, "      ")
      remaining = remaining - 1
      all_results_merged$branch[all_results_merged$list == 
                                  p] = pheno_branches[p]
    }
  }
  signif_results = all_results_merged[all_results_merged$q <= 
                                        q_threshold & !is.na(all_results_merged$branch), ]
  n_signif_per_cell_by_branch = data.frame()
  for (b in unique(plot_branches)) {
    for (c in unique(all_results_merged$CellType)) {
      n = length(signif_results[signif_results$CellType == 
                                  c & signif_results$branch == b, ]$list)
      n_signif_per_cell_by_branch = rbind(n_signif_per_cell_by_branch, 
                                          data.frame(branch = b, CellType = c, n_signif = n))
    }
  }
  n_signif_per_cell_by_branch$hypergeo_p = NA
  total_signif = length(signif_results$q)
  for (i in seq(1, length(n_signif_per_cell_by_branch$CellType))) {
    cell = n_signif_per_cell_by_branch$CellType[i]
    branch = n_signif_per_cell_by_branch$branch[i]
    group1 = sum(n_signif_per_cell_by_branch$n_signif[n_signif_per_cell_by_branch$CellType == 
                                                        cell])
    group2 = sum(n_signif_per_cell_by_branch$n_signif[n_signif_per_cell_by_branch$branch == 
                                                        branch])
    overlap = n_signif_per_cell_by_branch$n_signif[i]
    n_signif_per_cell_by_branch$hypergeo_p[i] = stats::phyper(overlap - 
                                                                1, group2, total_signif - group2, group1, lower.tail = FALSE)
  }
  n_signif_per_cell_by_branch$hypergeo_q = stats::p.adjust(n_signif_per_cell_by_branch$hypergeo_p, 
                                                           method = "BH")
  n_signif_per_cell_by_branch$signif_asterics = ""
  n_signif_per_cell_by_branch$signif_asterics[n_signif_per_cell_by_branch$hypergeo_q < 
                                                0.05] = "*"
  n_signif_per_cell_by_branch$signif_asterics[n_signif_per_cell_by_branch$hypergeo_q < 
                                                0.005] = "**"
  n_signif_per_cell_by_branch$signif_asterics[n_signif_per_cell_by_branch$hypergeo_q < 
                                                5e-04] = "***"
  n_signif_per_cell_by_branch$signif_asterics[n_signif_per_cell_by_branch$hypergeo_q < 
                                                5e-05] = "****"
  cell_order = factor(gsub("_", " ", ctd[[1]]$plotting$cell_ordering))
  n_signif_per_cell_by_branch$cell_order = match(n_signif_per_cell_by_branch$CellType, 
                                                 cell_order)
  n_signif_per_cell_by_branch$CellType = stats::reorder(n_signif_per_cell_by_branch$CellType, 
                                                        n_signif_per_cell_by_branch$cell_order)
  n_signif_per_cell_by_branch$branch_f <- factor(n_signif_per_cell_by_branch$branch, 
                                                 levels = plot_branches)

  
  # maybe wont work
  cell_order <- ctd[[1]]$plotting$cell_ordering
neuronal_cell_types = cell_order[seq(14,32)]
immune_cell_types = cell_order[c(5,6,7,75,76,77)]
cardiac_cell_types = cell_order[c(42,4)]
#ns_colour = "#619CFF"; card_colour = "#F8766D"; immune_colour = "#00BA38";
ns_colour = palette_branch[1]; card_colour = palette_branch[2]; immune_colour = palette_branch[3];

cell_colours = c()
for (c in cell_order) {
  if (c %in% neuronal_cell_types) {
    cell_colours = append(cell_colours, ns_colour)
  } else if (c %in% immune_cell_types) (
    cell_colours = append(cell_colours, immune_colour)
  ) else if (c %in% cardiac_cell_types) {
    cell_colours = append(cell_colours, card_colour)
  } else {
    cell_colours = append(cell_colours,"black")
  }
}

named_colours = c()
for (i in seq(1, length(plot_branches))) {
  named_colours[plot_branches[i]] = palette_branch[i]
}
  

  facet_branch_plt <- ggplot(n_signif_per_cell_by_branch[n_signif_per_cell_by_branch$branch %in% 
                                                           plot_branches, ], aes(x = CellType, y = n_signif, fill = branch)) + 
    geom_col() + scale_colour_manual(values =  named_colours,aesthetics = "fill") +
    geom_text(mapping = aes(label = signif_asterics, 
                                         y = n_signif + 5)) + cowplot::theme_cowplot() + ylab("Enrichments (n)") + 
    xlab("Cell type") + theme(axis.text.x = element_text(angle = 90, 
                                                         hjust = 1), legend.position = "none") + facet_wrap(~branch_f, 
                                                                                                            ncol = 1)


  facet_branch_plt = facet_branch_plt + theme(axis.text.x = element_text(colour = cell_colours, angle = 45))
  
  if (make_dendro) {
    the_dendrogram = ctd[[1]]$plotting$ggdendro_horizontal +
      theme(plot.margin = unit(c(0, 0, 0, 0), units = "cm")) +
      #  scale_x_discrete(breaks = total_res$CellType)
      scale_x_discrete(breaks = ctd[[1]]$plotting$cell_ordering)
    # CHANGE ^: added scale_x_discrete to set the mapping of the dendro to the x axis scale
    
    facet_branch_plt = cowplot::plot_grid(the_dendrogram, facet_branch_plt,axis = "lr",
                                          align = "v", ncol = 1,
                                          rel_heights = c(0.5, 3))
    # CHANGE ^: align argument to "v" and rel_heights to c(0.3,1) to make dend and barchart closer
    
    #output$withDendro = combined_plot
  }
  
  
  
    return(facet_branch_plt)
}

################################################################################
# Create hashmap for cell names to assocaited colour
cell_order <- ctd[[1]]$plotting$cell_ordering
neuronal_cell_types = cell_order[seq(14,32)]
immune_cell_types = cell_order[c(5,6,7,75,76,77)]
cardiac_cell_types = cell_order[c(42,4)]
#ns_colour = "#619CFF"; card_colour = "#F8766D"; immune_colour = "#00BA38";
ns_colour = palette_branch[1]; card_colour = palette_branch[2]; immune_colour = palette_branch[3];
cell_colours = c()
for (c in cell_order) {
  if (c %in% neuronal_cell_types) {
    cell_colours[c] = ns_colour
  } else if (c %in% immune_cell_types) (
    cell_colours[c] = immune_colour
  ) else if (c %in% cardiac_cell_types) {
    cell_colours[c] = card_colour
  } else {
    cell_colours[c] = "black"
  }
}

  





phenos_per_cell_colourText <- function (all_results_merged, cell_mappings, fold = 1, q_val = 0.005, 
    dataset = "Descartes", cell_colours_hashmap = cell_colours) 
{
    phenos_per_cell = data.frame()
    for (c in unique(all_results_merged$CellType)) {
        cur_cell = c
        n_phenos = length(all_results_merged[all_results_merged$CellType == 
            c & all_results_merged$q < q_val & all_results_merged$fold_change > 
            fold, "list"])
        phenos_per_cell = rbind(phenos_per_cell, data.frame(Cell = cur_cell, 
            n_phenos = n_phenos))
    }
    n_pheno_text = phenos_per_cell
    phenos_per_cell$Tissue = rep(NA, length(phenos_per_cell$Cell))
    if (dataset == "Descartes") {
        phenos_per_cell_tissue = data.frame()
        for (i in seq(1:length(phenos_per_cell$Cell))) {
            cur = cell_mappings[cell_mappings$level1 == paste(phenos_per_cell$Cell[i]), 
                ]$tissue
            for (t in unique(cur)) {
                if (!is.na(t)) {
                  phenos_per_cell$Tissue[i] = t
                  phenos_per_cell_tissue = rbind(phenos_per_cell_tissue, 
                    phenos_per_cell[i, ])
                }
            }
        }
        phenos_per_cell = phenos_per_cell_tissue
    }
    phenos_per_cell$Cell = stats::reorder(phenos_per_cell$Cell, 
        phenos_per_cell$n_phenos)
    if (dataset == "Descartes") {
        for (i in seq(1, length(phenos_per_cell$Cell))) {
            cur = phenos_per_cell[phenos_per_cell$Cell == phenos_per_cell$Cell[i], 
                ]
            phenos_per_cell$n_phenos[i] = phenos_per_cell$n_phenos[i]/length(unique(cur$Tissue))
        }
    }
    
    
   ## Need to sort this out as an option and add arguments for it 
    phenos_per_cell$Cell_colour <- cell_colours_hashmap[phenos_per_cell$Cell]

    
    explan_plt <- ggplot(phenos_per_cell, aes(n_phenos, as.factor(Cell))) + 
        scale_x_continuous("", expand = c(0, 0), limits = c(0, 
            max(n_pheno_text$n_phenos) + 55)) + theme_classic() + 
        scale_y_discrete(phenos_per_cell$Cell) + labs(title = paste0("Significantly enriched phenotypes per cell (q < ", 
        q_val, ", fold change > ", fold, ")")) + 
        ylab("Cell Type")
    if (dataset == "Descartes") {
        explan_plt = explan_plt + geom_col(aes(fill = Tissue), 
            size = 2, position = "stack")
    }
    else {
        explan_plt = explan_plt + geom_segment(aes(xend = 0, 
            yend = Cell), size = 2)
    }
    explan_plt = explan_plt + geom_text(data = n_pheno_text, 
        aes(label = n_phenos, x = n_phenos + 26, y = Cell), size = 4, 
        color = "black") + ylab("Cell Type") + theme(axis.title.y = element_blank(), axis.text.y = element_text(colour = phenos_per_cell$Cell_colour))
    return(explan_plt)
}

################################################################################

# Phenos per cell plot
descartes_mappings = read.csv("data/DescartesHuman_celltype_mapping.csv")
descartes_mappings$level1 = gsub("_"," ",descartes_mappings$level1)
plot_npc = phenos_per_cell_colourText(all_results_merged,descartes_mappings, fold=1,q_val=0.05, cell_colours_hashmap = cell_colours)

# Main branches of hpo
#main_branch_plt <- plot_n_phenotypes_per_branch_hpo(phenotype_to_genes=phenotype_to_genes,hpo=hpo)
phenosPerBranch <- function (phenotype_to_genes, hpo, highlighted_branches = c("Abnormality of the nervous system", 
                                                            "Abnormality of the cardiovascular system", "Abnormality of the immune system"),
                             set_colors = c("#619CFF","#F8766D","#00BA38") ,
          background_branches = hpo$children["HP:0000118"][[1]], 
          wes_anderson_palette = "Moonrise3") 
{
  color_pal <- wesanderson::wes_palette(wes_anderson_palette, 
                                        2)
  highlighted_branches_ids <- hpo$id[match(highlighted_branches, 
                                           hpo$name)]
  phenos_per_branch <- data.frame()
  for (b in background_branches) {
    n <- length(ontologyIndex::get_descendants(hpo, b))
    if (b %in% highlighted_branches_ids) {
      target_branch <- "target"
    }
    else {
      target_branch <- "Other"
    }
    phenos_per_branch <- rbind(phenos_per_branch, data.frame(branch = hpo$name[b], 
                                                             n_phenos = n, target = target_branch))
  }
  phenos_per_branch$branch <- stats::reorder(phenos_per_branch$branch, 
                                             phenos_per_branch$n_phenos)
  
  phenos_per_branch$color <- "gray"
  for (i in seq(1, length(set_colors))) {
    phenos_per_branch$color[phenos_per_branch$branch == highlighted_branches[i]] <- set_colors[i]
  }
  
  phenos_per_branch_plt <- ggplot(phenos_per_branch, aes(x = n_phenos, 
                                                         y = branch)) + 
    geom_segment(mapping = aes(xend = 0,  yend = branch), size = 3, color = phenos_per_branch$color, fill = phenos_per_branch$color) + xlab("Descendants (n)") + 
    ylab(element_blank()) + 
    scale_color_manual(values = color_pal) + 
    theme(axis.line.x = element_blank(), panel.background = element_blank(), 
          panel.grid.major = element_blank(), panel.grid.minor = element_blank(), 
          axis.line.y = element_line(color = "black"), 
          axis.ticks.x = element_blank(), axis.text.x = element_blank(), 
          axis.text.y = element_text(size = 11),
          legend.position = "none") +
    scale_x_continuous(position = "top")
  return(phenos_per_branch_plt)
}
main_branch_plt <- phenosPerBranch(phenotype_to_genes, hpo, set_colors = palette_branch[1:3])



# Phenos per cell by branch


plot_branches = c("Abnormality of the nervous system","Abnormality of the cardiovascular system","Abnormality of the immune system")
facet_branch_plt = plot_n_signif_phenos_per_cell_by_branch_den(all_results_merged, plot_branches = plot_branches, q_threshold = 0.05, hpo=hpo, phenotype_to_genes=phenotype_to_genes, ctd=ctd, make_dendro =TRUE)


```

```{r Figure3, fig.height=20, fig.width = 20, fig.cap = "\\label{fig:Figure3}\\textbf{A} This shows how many significant phenotype enrichments were found for each cell type, where q<0.05 and fold change > 1. \\textbf{B} Shows the child terms of phenotypic abnormality (ID: HP:0000118). These constitute the main branches of the HPO, or the main classes of phenotypic abnormality. The bars represent the number of descendant terms within each branch. \\textbf{C} This is a representation of the main branches from phenotypic abnormality. The three branches that we will explore more depely in further examples have been colour coded. \\textbf{D} This shows the number of significant enrichments per cell from three main branches of the HPO. It can be seen that the greatest number of significant enrichments are seen in the cell types that we would expect. For example, the vast majority of enrichments in the nervous system branch are assocaited with cells related to the nervous system. An additional hypergeometric test was done to show where the number of significant enrichments in a branch is significantly elevated, asterics, ****, *** **, and *, indicate that \\textit{q}<0.00001, 0.0001, 0.001, and 0.05, respectively. The cells are ordered by their dendrogram grouping, which can be seen at the top.", message = FALSE, warning = FALSE, echo = FALSE }



namer = function(term){
  return (hpo$name[term])
}
terms = ontologyIndex::get_term_property(hpo,property="children" ,term = hpo$id[which(hpo$name == "Phenotypic abnormality")],as_names=FALSE)
terms = append(terms,hpo$id[which(hpo$name == "Phenotypic abnormality")])

terms <- data.frame("terms" = terms, "colour" = "gray")



plot_branches = c("Abnormality of the nervous system","Abnormality of the cardiovascular system", "Abnormality of the immune system")
named_colours <- c()
for (i in seq(length(plot_branches))) {
  named_colours[plot_branches[i]] <- palette_branch[i]
}

term_colours <- c()
term_names <- c()
term_labels <- c()
for (t in terms$terms) {
  cur_name <- hpo$name[t]
  term_names <- append(term_names, cur_name)
  
  if( cur_name == "Phenotypic abnormality") {
    term_labels <- append(term_labels, cur_name)
    term_colours <- append(term_colours, "lightblue")
  } else if (cur_name %in% plot_branches) {
    term_labels <- append(term_labels, "")
    term_colours <- append(term_colours, named_colours[cur_name])
  } else {
    term_labels <- append(term_labels, " ")
    term_colours <- append(term_colours, "gray")
  }
}
 
terms$colour <- term_colours 
terms$name <- term_names 
terms$labels <- term_labels


phenotypic_abnormality_children <- ontologyPlot::onto_plot(hpo, terms= terms$terms,label = terms$labels,
          shape="circle",fontsize = 80,fillcolor = terms$colour ,edge_attributes = list(color="grey"))





layout <- "
AA###BBBBB##
AA###BBBBB##
AACCCCCCCCCC
AACCCCCCCCCC
AADDDDDDDDDD
AADDDDDDDDDD
AADDDDDDDDDD
AADDDDDDDDDD
AADDDDDDDDDD
AADDDDDDDDDD
"


Fig2 = (  
  (plot_npc + theme(legend.position =c(0.7,0.3), axis.text.y = element_text(size = 12)))  +
  (main_branch_plt + theme(axis.text.y = element_text(size = 13))) +
          ggplotify(phenotypic_abnormality_children, height_px = 300, width_px = 1200) + # try increase height_px (orig 300,1200) 
        ( facet_branch_plt + theme(axis.title.y =element_text(margin = margin(l = -15, b = -10))) ) + 
        plot_layout(design = layout) + plot_annotation(tag_levels = LETTERS) )

#ggsave("fig2.png",Fig2, dpi = 400, width = 24, height = 20)
print(Fig2)

```

```{r phenos_per_cell_df, echo = FALSE, message = FALSE, warning = FALSE, results= FALSE} 
phenos_per_cell_df = dataframe_phenos_per_cell(all_results_merged,descartes_mappings, fold=1,q_val=0.05)
```

The Descartes human scRNA data clustered into `r length(unique(all_results_merged$CellType))` unique cell types. There were `r length(all_results_merged$CellType[all_results_merged$q < 0.05])` significant cell-phenotype associations across all cells (*q*<0.05). The number of significant phenotype associations for each individual cell type can be seen in Figure \ref{fig:Figure3} A. 

The greatest number of significant phenotypes enriched for a single cell type was `r max(phenos_per_cell_df$n_phenos)`, seen in `r phenos_per_cell_df$Cell[match(max(phenos_per_cell_df$n_phenos),phenos_per_cell_df$n_phenos)]`, (where *q*<0.05 and fold change > 1).

## Expected cell types are associated with the main HPO branches 
The root HPO term, phenotypic abnormality (HP:0000118), has `r length(hpo$children[["HP:0000118"]])` child phenotypes that represent the main classes of phenotypic abnormality in the HPO. These `r length(hpo$children[["HP:0000118"]])` branches can be seen in Figure \ref{fig:Figure3} B. Abnormality of the nervous system, abnormality of the cardiovascular system, and abnormality of the immune system have been highlighted, as these branches have clear cell types in which there is expected to be high numbers of enrichments (neurons, cardiomyocytes, and immune cells, respectively). These expected associations will be used as a means of validating the results.

As can be seen in Figure \ref{fig:Figure3} D, the phenotypes from these branches tend to have a greater number of significant enrichments in the expected cell types. A hypergeometric test was used to determine if the number of significantly enriched phenotypes for a cell type within a branch is significantly elevated. BH correction was used to account for multiple comparisons. A significant result from the hypergeometric test suggests that the branch as a whole is significantly associated with those cell types (the hypergeometric test results, seen in Fig. \ref{fig:expected_cell_branch}, are denoted by  \*\*\*\*, \*\*\* \*\*, and \*   which indicate that *q*<0.00001, 0.0001, 0.001, and 0.05, respectively). The cells are ordered by dendrogram grouping, so it can be seen that the large cluster of significant results in the "Abnormality of the nervous system" branch, ranging from astrocytes to sympathoblasts, are cells of the nervous system. The largest number of disease phenotypes in the nervous system branch are associated with limbic system neurons. In the immune system branch, there were a large number of significant enrichments in immune cells such as lymphoid cells, myeloid cells and antigen presenting cells. Finally, the cell type most significantly associated with abnormality of the cardiovascular system is cardiomyocytes (*q*<0.00001). These findings are in agreement with the predicted cell-phenotype relationships.

Additionally, there are many novel results that may provoke new lines of research. For example, a significant number of sub-phenotypes of "Abnormality of the cardiovascular system" are enriched in hepatoblasts.   

```{r proportion_expected_plt, echo = FALSE}
# modifying this function to be more specific about colours used. 
proportion_of_expected_enrichments_plot_2 <- function (all_results_merged, hpo, target_cells = c("Excitatory neurons", 
    "Limbic system neurons"), cell_type_description = "Neuronal cells", 
    HPO_Ids = ontologyIndex::get_descendants(hpo, hpo$id[match("Abnormality of the nervous system", 
        hpo$name)]), phenotype_description = "Nervous system phenotypes", 
    color_pal = c("#619CFF","#F8766D","#00BA38","gray"), color_expected_phenotypes = 1, 
    color_other_phenotypes = 4, blank_x_axis = FALSE) 
{
    cur_descendants_names = c()
    for (d in HPO_Ids) {
        cur_descendants_names = append(cur_descendants_names, 
            paste(hpo$name[d]))
    }
    all_results_merged$cur_branch = "Other"
    all_results_merged[all_results_merged$list %in% cur_descendants_names, 
        ]$cur_branch = phenotype_description
    all_results_merged$cur_branch = factor(all_results_merged$cur_branch)
    all_results_merged$minus_log_p = round(-log10(all_results_merged$p + 
        1e-06), digits = 0)
    proportional_results = data.frame()
    for (logp in unique(all_results_merged$minus_log_p)) {
        cur = all_results_merged[all_results_merged$minus_log_p == 
            logp, ]
        prop_branch = 100 * (length(cur$cur_branch[cur$cur_branch == 
            phenotype_description & cur$CellType %in% target_cells])/length(cur$cur_branch[cur$CellType %in% 
            target_cells]))
        prop_other = 100 * (length(cur$cur_branch[cur$cur_branch == 
            "Other" & cur$CellType %in% target_cells])/length(cur$cur_branch[cur$CellType %in% 
            target_cells]))
        proportional_results = rbind(proportional_results, data.frame(cur_branch = phenotype_description, 
            prop = prop_branch, minus_log_p = logp))
        proportional_results = rbind(proportional_results, data.frame(cur_branch = "Other", 
            prop = prop_other, minus_log_p = logp))
    }
    #color_pal = wesanderson::wes_palette(wes_color_palette, n_colors)
    prop_plot <- ggplot(proportional_results, aes(x = minus_log_p, 
        y = prop, color = cur_branch)) + geom_point(size = 3) + 
        geom_line(mapping = aes(linetype = cur_branch), size = 0.6) + 
        labs(color = "", linetype = "") + ylab(paste0("% Enrichments in \"", 
        cell_type_description, "\"")) + 
      scale_y_continuous(limits = c(0, 100)) + cowplot::theme_cowplot() + 
        scale_color_manual(values = c(color_pal[color_expected_phenotypes], 
            color_pal[color_other_phenotypes])) + theme(legend.position = c(0.5,0.95), 
        legend.text = element_text(size = 10), title = element_text(size = 11), axis.title.y = element_text(size=12))
      if (! blank_x_axis){
        prop_plot = prop_plot + xlab("-log10(p)")
      } else {
        prop_plot = prop_plot + theme(axis.text.x = element_blank(), axis.title.x = element_blank())
      } 
    return(list(prop_plot, proportional_results))
}

```

```{r patterns, message=FALSE, warning=FALSE, echo=FALSE}

# Significance and expected cell type ##########################################
plot_branches = c("Abnormality of the nervous system", "Abnormality of the cardiovascular system", "Abnormality of the immune system")
expected_cells = c("Excitatory neurons", "Cardiomyocytes","Antigen presenting cells")
correlation_results = data.frame()  
proportion_plots = list()
blank_x = c(TRUE, TRUE, FALSE)
for (i in seq(length(plot_branches))) {
  cur_plot <- proportion_of_expected_enrichments_plot_2(all_results_merged,
   hpo,
   target_cells=c(expected_cells[i]),
   cell_type_description = expected_cells[i],
   HPO_Ids = ontologyIndex::get_descendants(hpo,hpo$id[match(plot_branches[i],hpo$name)]),
   phenotype_description = plot_branches[i],
   color_pal =  c(palette_branch[1],palette_branch[2],palette_branch[3],"gray"),
   color_expected_phenotypes = i,
   color_other_phenotypes = 4, blank_x_axis = blank_x[i])
  proportion_plots[[i]] = cur_plot[[1]]
  correlation_results = rbind(correlation_results,cur_plot[[2]])
}

#sigplot <- cowplot::plot_grid(plotlist=proportion_plots, align = "h",nrow = 1,labels=c("A","B","C"))


# ont level plots (violin) #####################################################
blank_x <-   theme(axis.text.x = element_blank(), axis.title.x = element_blank())
      

ont_levels <- data.frame("phenotype"=unique(phenotype_to_genes$Phenotype),
                         "hpo_id"=hpo$id[match(unique(phenotype_to_genes$Phenotype),hpo$name)])
ont_levels <- ont_levels[complete.cases(ont_levels),]
lvls <- c()
for (id in ont_levels$hpo_id) {
  lvls = append(lvls, get_ont_level(hpo,id))
}
ont_levels$ont_lev <- lvls
rm(lvls)

n_associated_cells <- c()
for (p in ont_levels$phenotype) {
  #n_associated_cells <- append(n_associated_cells, length(all_results_merged$CellType[all_results_merged$list == p & all_results_merged$q < 0.05]))
  n_associated_cells <- append(n_associated_cells, length(all_results_merged$CellType[all_results_merged$list == p & all_results_merged$q < 0.05 & all_results_merged$fold_change > 1]))
}
ont_levels$n_associated_cells <- n_associated_cells
rm(n_associated_cells)

pal <- wesanderson::wes_palette("Darjeeling2", n=2)
ontlvl_ncells_plt <- ggplot(ont_levels, mapping=aes(x= factor(ont_lev), y=n_associated_cells)) +
  geom_jitter(color = pal[1]) +
  geom_violin(fill = NA) +
  geom_smooth(color = pal[2], method="loess",mapping = aes(x=ont_lev)) +
  cowplot::theme_cowplot() +
  labs(x="Ontology level", y="Associated cells/phenotype (n)",title = "n associated cells/phenotype by ontology level")+
  blank_x
#ontlvl_ncells_plt



# ont level facet

signif_res <- all_results_merged[all_results_merged$q < 0.05,]
ontlevz <- c()
for (p in unique(signif_res$HPO_id)) {
  ontlevz[p] <- get_ont_level(hpo,p)
}
signif_res$ontlvl <- ontlevz[signif_res$HPO_id]
signif_res <- signif_res[complete.cases(signif_res),]

pal <- wesanderson::wes_palette("Darjeeling2", n=2)
ontlvl_fold_plt <- ggplot(signif_res, aes(x = factor(ontlvl), y = fold_change)) +
  geom_jitter(color = pal[1]) +
  geom_violin(fill = NA) +
  geom_smooth(color = pal[2], method="loess",mapping = aes(x=ontlvl)) +
  cowplot::theme_cowplot() +
  labs(x="Ontology level", y="Fold change",title = "Fold change in specific expression by ontology level") +
  blank_x
  

# ngenes plt (stat smooth takes too long/doesnt work with all results so just use signif ?)
ngenes<-c()
for(p in unique(signif_res$list)) {
  ngenes[p] <- length(get_gene_list(p,phenotype_to_genes))
}
signif_res$ngenes <- ngenes[signif_res$list]
signif_res <- signif_res[complete.cases(signif_res),]
pal <- wesanderson::wes_palette("Darjeeling2", n=2)
ontlvl_ngenes_plt <- ggplot(signif_res, aes(x = factor(ontlvl), y = ngenes)) +
  geom_jitter(color = pal[1]) +
  geom_violin(fill = NA) +
  geom_smooth(color = pal[2], method="loess",mapping = aes(x=ontlvl)) +
  cowplot::theme_cowplot() +
  labs(x="Ontology level", y="Genes (n)",title = "Number of genes by ontology level")

#ontlvlplots <- cowplot::plot_grid(plotlist=list(ontlvl_ncells_plt,ontlvl_fold_plt,ontlvl_ngenes_plt),align = "hv",nrow = 1)



```

```{r Figure4, fig.height = 12, fig.width = 15, fig.cap = "\\label{fig:Figure4}\\textbf{A, B, and C}  \\textbf{D, E, and F} Show the relationship between significance threshold and the proportion of enrichments found in the expected HPO branch. As more stringent significance thresholds for enrichment in a particular cell type are used, the proportion of enrichments from an expected HPO branch increases. This validates the method as it shows that many of the strong phenotype-cell associations are in agreement with our current understanding. \\textbf{A} shows the proportion of enrichments for abnormality of the nervous system descendants in excitatory neurons as significance threshold is increased, \\textbf{B} looks at the cardiovascular system branch and cardiomyocytes, and \\textbf{C} looks at the immune system branch and antigen presenting cells.", message = FALSE, warning = FALSE , echo = FALSE}
blank_x_axis <- theme(x.axis.title =  element_blank(), axis.text.x = element_blank())
print(
  (
    ( ontlvl_ncells_plt /
       ontlvl_fold_plt /
       ontlvl_ngenes_plt
      ) 
    |
    ( proportion_plots[[1]]  /
       proportion_plots[[2]] /
       proportion_plots[[3]] 
      ) 
  ) +  plot_annotation(tag_levels = LETTERS) 
)

```

```{r signif_expected_cell_pearsons, message=FALSE, echo=FALSE, warning =FALSE, results = FALSE}


plot_branches = c("Abnormality of the nervous system", "Abnormality of the cardiovascular system", "Abnormality of the immune system")
expected_cells = c("Excitatory neurons", "Cardiomyocytes","Antigen presenting cells")
correlation_results = data.frame()  
# proportion_plots = list()
for (i in seq(length(plot_branches))) {
  cur_plot <- proportion_of_expected_enrichments_plot(all_results_merged,
   hpo,
   target_cells=c(expected_cells[i]),
   cell_type_description = expected_cells[i],
   HPO_Ids = ontologyIndex::get_descendants(hpo,hpo$id[match(plot_branches[i],hpo$name)]),
   phenotype_description = plot_branches[i],
   wes_color_palette="Darjeeling1",
   n_colors = 4,
   color_expected_phenotypes = i,
   color_other_phenotypes = 4)
#  proportion_plots[[i]] = cur_plot[[1]]
  correlation_results = rbind(correlation_results,cur_plot[[2]])
}

#print(cowplot::plot_grid(plotlist=proportion_plots, align = "h",nrow = 1,labels=c("A","B","C")))
pearsons_expected = cor.test(correlation_results$minus_log_p[correlation_results$cur_branch != "Other"],correlation_results$prop[correlation_results$cur_branch != "Other"],method="pearson")


```


To further demonstrate that the analysis finds expected phenotype-cell relationships, excitatory neurons, cardiomyocytes, and antigen presenting cells were again used. Figure \ref{fig:Figure4} D, E, and F show that if we take all results for a particular cell, the more significantly associated phenotypes disproportionately come from the expected branch of the HPO. In other words, as more stringent significance thresholds are used, the proportion of enriched phenotypes from the expected HPO branch increases (*r*~`r pearsons_expected$parameter`~=`r pearsons_expected$estimate`, *p*=`r pearsons_expected$p.value`).

## Low ontology level terms have high specific expression in enriched cells
It was then hypothesised that significant enrichments would have higher specific expression towards the leaf nodes of the HPO, compared to terms closer to the root. We quantify specific expression by fold change. This denotes how many times greater the specific expression is in a gene list, relative to the mean expression of random bootstrapped gene lists of the same length, in a given cell type, in other words, the strength of the enrichment. This effect can be seen in Figure \ref{fig:Figure4} A and B. Ontology level represents the position of a term in the HPO heirarchy. It denotes, how many generations of sub-phenotypes (descendants) a term has. For example, an ontology level 4 term like "Recurrent infections" has 4 generations of descendant terms below it, including "Recurrent gram-negative bacterial infections", which in turn has 2 generations of descendants, so it is at ontology level 2. The root term of the HPO (Phenotypic abnormality) is at ontology level 13. As ontology level approaches zero, the number of signifcantly assocaited cells goes down and the fold change in specific expression goes up. 

The relationship between ontology level and fold change shows that the phenotypes at higher ontology levels, which encompass a broad range of traits, generally a have lower specific expression. Conversely, the significant associations at lower HPO levels tend to have higher specific expression of the phenotype gene list in the associated cell type. Figure \ref{fig:Figure4} C shows that the higher ontology level phenotypes also tend to have longer gene lists. Conversely, the more narrowly defined phenotypes at lower ontology levels are generally associated with only a few genes. This further supports the idea that the gene lists at lower levels tend to be more specific.

```{r HPO_level_fold,fig.height=4,fig.width =8,fig.cap = '\\label{fig:HPO_level_fold}\\textbf{Relationship between ontology level and expression specificity. A:}  More narrowly defined phenotypes, at lower HPO levels, show more specific expression of their gene lists in significantly associated cells than phenotypes at higher HPO levels. Ontology level represents how many generations of sub-phenotypes a term has in the HPO. As we progress out towards these leaf-nodes (ontology level 0), the specific expression of the gene list tends to be higher in significantly enriched cell types. The level of specific expression is represented as fold change here. \\textbf{B:} At higher ontology levels, the phenotypes encompass a larger number of traits and are associated with larger lists of genes. The lower ontology level phenotypes are more narrowly defined and are often associated with only a few genes, possibly making them good targets for therapy and research.', message=FALSE, echo=FALSE}


print(ontlevel_to_foldchange_and_ngenes_plot(all_results_merged,hpo,phenotype_to_genes))

# some data for discussion below
n_desc_rec_inf = length(get_descendants(hpo, "HP:0002719"))
```


## Low ontology level terms are enriched in expected and novel cell types
While figures \ref{fig:Figure3} D and \ref{fig:Figure4} D, E, and F show that high-level HPO terms are generally associated with expected cell types, figure \ref{fig:recurrent_inf_n_per_cell} shows that expected cell-phenotype relationships are also found at the more specific phenotypes at lower HPO levels. As an example, we look at all descendant terms of Recurrent infections (HP:0002719), which includes `r n_desc_rec_inf` HPO terms at ontology levels ranging from 0 to 3. The hypothesis is that they will be primarily enriched in cells involved in the immune system. It was predicted that, from these sub-phenotypes, the number of significant enrichments in immune system associated cell types would be greater than the number of significant results in all other cell types combined.


```{r recurrent_inf_n_per_cell,fig.width=10,fig.height=6,fig.cap = '\\label{fig:recurrent_inf_n_per_cell}\\textbf{Significant enrichments per cell for all descendant terms of Recurrent infections.} This shows the number of significantly enriched phenotypes per cell for descendant terms of "Recurrent infections". As expected, the majority of enrichments are found in cells involved in the immune system, including antigen presenting cells, hematopoietic cells, yhymocytes, myeloid cells, and lymphoid cells. These collectively account for 68 significant enrichments. We also find novel enrichments in ciliated epithelial cells, ductal cells, ureteric bud cells, hepatoblasts, and some undefined cells. These may warrant further investigation.', message=FALSE, echo=FALSE}

library(wesanderson)
color_pal = wes_palette("Darjeeling1",4)
library(cowplot)

branch = "Recurrent infections"
branch_id = paste(hpo$id[match(branch,hpo$name)])
branch_descendants = paste(get_descendants(hpo,branch_id))
branch_descendants_names = paste(hpo$name[branch_descendants])
all_results_merged$list = paste(all_results_merged$list)
#all_results_merged$cur_branch = paste(all_results_merged$cur_branch)

all_results_merged$cur_branch = paste("Other")
all_results_merged$cur_branch[all_results_merged$list %in% branch_descendants_names] = branch



branch_signif_counts = data.frame()
for (c in unique(all_results_merged$CellType)) {
  n_signif = length(all_results_merged[all_results_merged$CellType == c & all_results_merged$cur_branch == branch & all_results_merged$q < 0.05, ]$q)
  branch_signif_counts = rbind(branch_signif_counts,
                                data.frame("branch"=branch,
                                           "CellType"=c,
                                           "n_signif"=n_signif))
}

cell_order = factor(gsub("_"," ",ctd[[1]]$plotting$cell_ordering))
branch_signif_counts$cell_order = match(branch_signif_counts$CellType, cell_order)
branch_signif_counts$CellType = reorder(branch_signif_counts$CellType, branch_signif_counts$cell_order)
branch_signif_counts$labels = branch_signif_counts$n_signif
branch_signif_counts$labels[branch_signif_counts$labels == 0] = ""

branch_plt <- ggplot(branch_signif_counts[branch_signif_counts$branch==branch,], aes(x=CellType,y=n_signif)) +
  geom_col( fill = color_pal[2], color = "black") +
  geom_text(mapping= aes(label = labels, y = n_signif + 3))+
  theme_cowplot()+
  ylab("N phenotypes") +
  scale_y_continuous(expand=c(0,0), limits= c(0,max(branch_signif_counts$n_signif)+5))+
  theme(axis.text.x = element_text(angle = 90, hjust=1, vjust =0.2),legend.position="none") +
  #coord_flip() +
  ggtitle("Significant enrichments per cell: Recurrent infections")


print(branch_plt)

immune_cells = c("Lymphoid cells","Myeloid cells","Antigen presenting cells","Thymocytes","Hematopoietic stem cells")
total_immune_enrich = sum(branch_signif_counts$n_signif[branch_signif_counts$CellType %in% immune_cells])
total_other_enrich = sum(branch_signif_counts$n_signif[!branch_signif_counts$CellType %in% immune_cells])
other_cells = unique(paste(branch_signif_counts$CellType[!branch_signif_counts$CellType %in% immune_cells & branch_signif_counts$n_signif > 0]))

immune_vs_other_t = t.test(branch_signif_counts$n_signif[branch_signif_counts$CellType %in% immune_cells],branch_signif_counts$n_signif[!branch_signif_counts$CellType %in% immune_cells])

```

As predicted, the majority of enrichments were found in cells associated with the immune system (`r tolower(immune_cells)`). `r total_immune_enrich` enrichments were found in immune system related cells, and only `r total_other_enrich` enrichments were found in other cell types (*t*~`r immune_vs_other_t$parameter`~=`r immune_vs_other_t$statistic`, *p*=`r round(immune_vs_other_t$p.value,digits=6)`). 

A selection of the sub-phenotypes of "Recurrent infections" were then isolated, to see where the significant effects lie, and to investigate some of the unexpected and novel associations. Figure \ref{fig:recurrent_bacterial_inf} shows the adjacent child phenotypes of "Recurrent bacterial infections", which is its self a child term of "Recurrent infections".

```{r recurrent_bacterial_inf,fig.width=10,fig.height=10, fig.cap = '\\label{fig:recurrent_bacterial_inf}\\textbf{Cell-phenotype associations for child terms of Recurrent bacterial infections.} Here we can see all cell-phenotype relationships for child phenotypes of the HPO term "Recurrent bacterial infections". Significance is indicated with asterics where *, **, ***, and **** represent textit{q} less than 0.05, 0.001, 0.0001, and 0.00001. respectively.', message=FALSE, echo=FALSE}

# Note: could probably just use ewce_plot function faceted with all the child terms of Recurrent bacterial infections.

branch = "Recurrent bacterial infections"
exclude_root = TRUE

branch_id = paste(hpo$id[match(branch,hpo$name)])
branch_descendants = hpo$children[[branch_id]]
branch_descendants_names = paste(hpo$name[branch_descendants])
if (exclude_root) {
  branch_descendants_names = branch_descendants_names[branch_descendants_names != branch]
} else {
  branch_descendants_names = c(branch, branch_descendants_names)
}
pheno_df = all_results_merged[all_results_merged$list %in% branch_descendants_names, ]

pheno_df$signif_asterics = ""
pheno_df$signif_asterics[pheno_df$q<0.05] = "*"
pheno_df$signif_asterics[pheno_df$q<0.001] = "**"
pheno_df$signif_asterics[pheno_df$q<0.0001] = "***"
pheno_df$signif_asterics[pheno_df$q<0.00001] = "****"

pheno_fold_plt <- ggplot(pheno_df, aes(x = CellType, y= fold_change, fill = list)) +
  ggtitle(paste0('Child nodes of HPO branch "',branch,'"')) +
  geom_col() +
  geom_text(label = pheno_df$signif_asterics, mapping = aes(y = fold_change + 1))+
  theme_cowplot() +
  scale_y_continuous(expand=c(0,0), limits= c(0,max(branch_signif_counts$n_signif)+2))+
  ylab("Fold change") +
  xlab("Cell type") +
  theme(axis.text.x = element_text(angle=90,hjust=1,vjust=0.2),legend.position = "none") +
  facet_wrap(~list, ncol=1)


print(pheno_fold_plt)

```

The analysis was able to reproduce a well known association with myeloid cells, a class of phagocytic immune cell often involved in defence against bacterial pathogens. Interestingly, a very strong association is seen between recurrent mycobacterial infection and ciliated epithelial cells (*q*<0.0001, fold change=`r round(all_results_merged[all_results_merged$CellType=="Ciliated epithelial cells" & all_results_merged$list =="Recurrent mycobacterial infections",]$fold_change,digits=2)`). 

Figure \ref{fig:recurrent_gramneg_inf} provides a closer look at the child phenotypes of Recurrent gram-negative bacterial infections. Recurrent neisserial infections were found to be enriched in Hepatoblasts (*q*=`r all_results_merged[all_results_merged$CellType=="Hepatoblasts" & all_results_merged$list == "Recurrent Neisserial infections", ]$q`, fold change=`r round(all_results_merged[all_results_merged$CellType=="Hepatoblasts" & all_results_merged$list == "Recurrent Neisserial infections", ]$fold_change,digits=2)`). As this is a potentially novel finding, this phenotype was also checked in the Tabula Muris data. Similarly, significant enrichment in hepatic cells (Kupfer cells and Hepatocytes) were found (*q*<0.0001).


```{r recurrent_gramneg_inf,fig.width=10,fig.height=10, fig.cap = '\\label{fig:recurrent_gramneg_inf}\\textbf{Enrichments for child phenotypes of "Recurrent gram-negative bacterial infections.} Here we can see all cell-phenotype relationships for child phenotypes of the HPO term "Recurrant gram-negative bacterial infections".', message=FALSE, echo=FALSE}
#Note: Again, I realized these plots are similar to ewce_plot so may be better to just use that.

branch = "Recurrent gram-negative bacterial infections"
exclude_root = TRUE

branch_id = paste(hpo$id[match(branch,hpo$name)])
branch_descendants = hpo$children[[branch_id]]
branch_descendants_names = paste(hpo$name[branch_descendants])
if (exclude_root) {
  branch_descendants_names = branch_descendants_names[branch_descendants_names != branch]
} else {
  branch_descendants_names = c(branch, branch_descendants_names)
}
pheno_df = all_results_merged[all_results_merged$list %in% branch_descendants_names, ]

pheno_df$signif_asterics = ""
pheno_df$signif_asterics[pheno_df$q<0.05] = "*"
pheno_df$signif_asterics[pheno_df$q<0.001] = "**"
pheno_df$signif_asterics[pheno_df$q<0.0001] = "***"
pheno_df$signif_asterics[pheno_df$q<0.00001] = "****"

pheno_fold_plt <- ggplot(pheno_df, aes(x = CellType, y= fold_change, fill = list)) +
  ggtitle(paste0('Child nodes of HPO branch "',branch,'"')) +
  geom_col() +
  geom_text(label = pheno_df$signif_asterics, mapping = aes(y = fold_change + 1))+
  theme_cowplot() +
  scale_y_continuous(expand=c(0,0), limits= c(0,max(branch_signif_counts$n_signif)+2))+
  ylab("Fold change") +
  xlab("Cell type") +
  theme(axis.text.x = element_text(angle=90,hjust=1,vjust=0.2),legend.position = "none") +
  facet_wrap(~list, ncol=1)


print(pheno_fold_plt)

```

To further validate the ability to find enrichments for specific RD phenotypes, an example from the abnormality of the nervous system branch was also examined (Impaired social interactions). This can be seen in figure \ref{fig:impared_social} (this figure also demonstrates the modified version of the `ewce_plot` function that was published in the EWCE Bioconductor package).


```{r impared_social,fig.width=10,fig.height=8, fig.cap = '\\label{fig:impared_social}\\textbf{Cell type enriched for Impared social interactions.} This shows the cell types that are significnatly enriched for the \"Impared social interactions\" gene list. The majority of significant enrichments are seen in cells of the nervous system, with particularly high specific expression in limbic system neurons and excitatory neurons. Interestingly, there is also a strong association with Amacrine cells, a retinal interneuron. Significance is denoted with asterics, where * signifies q<0.05', message=FALSE, echo=FALSE}


library(EWCE)
library(ggplot2)
library(cowplot)
source("source/ewce_plot_function.R")

# Dendrogram and signif cell types
pheno = "Impaired social interactions"
subset_results = all_results_merged[all_results_merged$list == pheno, ]
subset_results$CellType = gsub(" ","_",subset_results$CellType)
plt1 = ewce.plot(subset_results, mtc_method = "BH", ctd = ctd)$withDendro

print(plt1)


```


The child terms of impaired social interactions were then examined to get a higher resolution on the surprising association with amacrine cells. They were found to be significantly enriched for the Poor eye contact sub-term (see figure \ref{fig:poor_eye_contact}).


```{r poor_eye_contact,fig.width=10,fig.height=8, fig.cap = '\\label{fig:poor_eye_contact}\\textbf{Cell types enriched for Poor eye contact.} This shows the cells enriched for poor eye contact. It is interesting to note the surprising association of impaired social interaction with Amacrine cells, which are retinal interneurons, was found to come exclusively from a sub-phenotype related to the eye, "poor eye contact". Significance is denoted with astericks, where * signifies q<0.05', message=FALSE, echo=FALSE, erro=FALSE,warning=FALSE}


library(EWCE)
library(ggplot2)
library(cowplot)
source("source/ewce_plot_function.R")

# Dendrogram and signif cell types
pheno = "Poor eye contact"
subset_results = all_results_merged[all_results_merged$list == pheno, ]
subset_results$CellType = gsub(" ","_",subset_results$CellType)
plt2 = ewce.plot(subset_results, mtc_method = "BH", ctd = ctd)$withDendro

print(plt2)

```


\newpage
# Discussion
```{r hepatocyte_cardio, message=FALSE, echo=FALSE, warning =FALSE, results = FALSE}

cardio_id = hpo$id[match("Abnormality of the cardiovascular system", hpo$name)]
cardio_desc = get_descendants(hpo, cardio_id)
cardio_hepato = all_results_merged[all_results_merged$HPO_id %in% cardio_desc & all_results_merged$CellType == "Hepatoblasts" & all_results_merged$q < 0.05 & all_results_merged$fold_change > 7, c("CellType","list","fold_change","q")]
cardio_hepato$Phenotype = cardio_hepato$list
cardio_hepato = cardio_hepato[,c("CellType","Phenotype","fold_change","q")]


```

More than 8000 significant enrichments were found. These are spread across all HPO branches. High-level HPO terms, encompassing a broad range of traits, were significantly associated with expected cell types. This validates the method as it shows that the majority of cell-phenotype associations are in agreement with accepted knowledge. The number of expected enrichments also increases as more stringent significance thresholds are used, as the more well-understood cell-phenotype relationships tend to have a stronger enrichment.

It was found that, in significant cell-phenotype associations, the strength of enrichment (measured by fold change specific expression) tends to be higher at lower ontology levels. These more narrowly defined disease phenotypes also tend to have shorter, more specific gene lists. This potentially makes many of these low ontology level terms great targets for research. One of the roadblocks of gene therapy is the lack of understanding of the pleiotropic effects of many genes [@gene_therapy_2020]. Being able to isolate the problem down to a small set of genes in a very specific cell type goes a long way to alleviating this problem. This could also provide a way to prioritise different research possibilities. The results could be subset to show all low ontology level HPO terms that are severe enough to warrant treatment and have a strong association with a particular cell type. Other parameters could also be used in this way. For example, to prioritise long-lived cell types that may be the best candidates for gene therapy. 

Figure \ref{fig:expected_cell_branch} showed that the majority of significant enrichments for terms from the main branches of the HPO are in expected cell types. For example, terms that are a sub-class of "Abnormality of the cardiovascular system" are primarily enriched in cardiomyocytes. It was also noted that there are a number of enrichments in less obvious cell types. For example, a significant number of terms in the cardiovascular branch were enriched in hepatoblasts (*p*<0.0001). To explore this further, we retrieved all results from the cardiovascular branch that are enriched in hepatoblasts and have a fold change > 7, and *q* < 0.05. The phenotypes were primarily ones that often involve damage to arteries caused by lipid deposition (`r tolower(cardio_hepato$Phenotype)`). Given the large role that the liver plays in lipid metabolism, it is unsurprising that dysfunction of hepatocytes may be implicated in these cardiovascular diseases. Additionally, a brief literature search finds many recent studies exploring the link between hepatic cells and cardiovascular diseases, such as the paper by @xu_hepatocyte_2021 which shows that hepatocyte ATF3 is protective against atherosclerosis by regulating high-density lipoprotein metabolism. Furhter, @bell_hepatocyte_2018 showed that higher circulating hepatocyte growth factor is associated with elevated atherosclerosis progression measures. 

```{r cilia_recurrent_inf, message=FALSE, error=FALSE, warning=FALSE, echo=FALSE}
cil_inf = all_results_merged[all_results_merged$CellType == "Ciliated epithelial cells" & all_results_merged$HPO_id %in% get_descendants(hpo,hpo$id[match("Recurrent infections", hpo$name)] ) & all_results_merged$q < 0.05,]

rec_myco_fold = cil_inf[cil_inf$list == "Recurrent mycobacterial infections", ]$fold_change

```

More specific HPO terms from lower ontology levels were also primarily enriched in expected cell types. For example, it was found that the majority of significant enrichments for descendants of "Recurrent bacterial infections" were in immune cells. Again some less obvious enrichments were also found, such as ciliated epithelial cells. To investigate this further, the EWCE results for descendant terms of "Recurrent bacterial infections", that were also enriched in ciliated epithelial cells, were retrieved individually. It was found that the enrichments are primarily associated with the recurrent respiratory infections sub-branch (`r tolower(cil_inf$list)`). This relationship between structural defects of the cilia and recurrent respiratory tract infections has been documented in the literature [@immotile-cilia_1977]. The strongest association was found in mycobacterial infections (which includes tuberculosis), and this also primarily affects the lungs (fold change=`r round(rec_myco_fold,digits=2)`, *q*<0.00001). The only non-pulmonary phenotype was associated with ciliated cells was recurrent otitis media infections, which is also known to be a problem for people with immotile-cilia [@immotile_cilia_1983].

Another potentially novel finding was the significant enrichment in hepatoblasts for a term in the "Recurrent infections" branch. This was found to be specific to recurrent neisserial infections. For further confirmation, this phenotype was also analysed in the Tabula Muris data (mouse scRNA dataset), where it was also significantly enriched in hepatic cells (kupffer cells and hepatocytes). Whilst this is a seemingly surprising association, we found a number of plausible explanations for it in the literature. Fitz-Hugh-Curtis syndrome is a RD characterised by inflammation of the peritoneum and the tissues surrounding the liver, caused by *Neisseria gonorrhoeae* infection [@fitz-hugh-curtis_2017]. It is possible that an abnormality of hepatic cells leaves patients particularly susceptible to this problem. Though Fitz-Hugh_Curtis syndrome is a sub-phenotype of recurrent neisserial infections, at the time of writing, there was no entry for it in the HPO, so it was not possible to check if the enrichment was specific to this sub-phenotype. Perhaps the most likely explanation involves the complement system, a part of the innate immune system that plays a key role in defense against neisserial infections. Complement is synthesised primarily in the liver, and it was found that people with deficits in complement are at high risk for Neisserial infection [@lewis_complement_2020; @fijen_complement_1994]. 

```{r social_results, message=FALSE, echo=FALSE, warning=FALSE}
social_results = all_results_merged[all_results_merged$q < 0.05 & all_results_merged$list == "Impaired social interactions",]

# check descendants
library(ontologyIndex)
social_int_id = hpo$id[match("Impaired social interactions", hpo$name)]
social_int_desc = get_descendants(hpo, social_int_id )
social_int_desc_names = hpo$name[social_int_desc][hpo$name[social_int_desc] != "Impaired social interactions"]
social_desc_results = all_results_merged[all_results_merged$q < 0.05 & all_results_merged$list %in% social_int_desc_names & all_results_merged$CellType == "Amacrine cells",]


poor_eye_contact = all_results_merged[all_results_merged$q < 0.05 & all_results_merged$list == "Poor eye contact" & all_results_merged$CellType == "Amacrine cells",]


```

Significant enrichments for "Impaired social interactions" were then examined to take an example from another branch of the HPO. The expectation was that it would primarily involve neuronal cell types. There were strong enrichments in excitatory neurons (fold change=`r round(social_results[social_results$CellType == "Excitatory neurons","fold_change"],digits=3)`, *q*<0.00001) and limbic system neurons (fold change=`r round(social_results[social_results$CellType == "Limbic system neurons","fold_change"],digits=3)`, *q*<0.00001). While we may need to use higher resolution cell type data to explore the excitatory neurons result, which could encompass a lot of different cells, the association with limbic system neurons was found to be supported in the literature. The basolateral circuit encodes information about social signals, and the amygdala and cingulate gyrus are involved in social cognition, all of which are limbic system structures [@frith_brain_1996]. 

More surprisingly, an association between impaired social interaction and amacrine cells was found (fold change=`r round(social_results[social_results$CellType == "Amacrine cells","fold_change"],digits=3)`, *q*=`r social_results[social_results$CellType == "Amacrine cells","q"]`). These are a type of interneuron found in the retina. At first glance, this seemed like it may have been a false positive, possibly due to similarities to other cell types. In cases like this, it is possible to do a conditional enrichment analysis with EWCE that controls for expression in other cells, which may be worthwhile when investigating the more novel results in the future. However, to explore the association further, the descendant terms of impaired social interactions were checked to see if there was a more specific association between amacrine cells and one of the sub-phenotypes (`r tolower(hpo$name[get_descendants(hpo,hpo$id[match("Impaired social interactions", hpo$name)],exclude_roots=TRUE)])`). Interestingly, the significant effect was found exclusively in the Poor eye contact (`r poor_eye_contact$HPO_id`) sub-term (fold change=`r poor_eye_contact$fold_change`, *q*<0.05). These results may imply that, in some cases, poor eye contact could be caused by a physical problem with structures in the eye, rather than with the psychological origins normally associated with impaired social interaction related phenotypes. 

From these examples, it can be seen that when scrutinised, many of the more surprising cell-phenotype relationships are either previously known or at least have a plausible mechanistic basis. It is very likely that within the over 8000 results presented here, there are many previously unknown links between disease phenotypes and specific cell types that could lead to advances in the understanding and treatment of RDs. For the impact of the results to be fully realised, it was essential that they could be easily explored by domain experts and clinicians without the need for extensive computational resources and specialist knowledge of enrichment analysis and programming. The web-based applications were developed to facilitate this. Although further work is needed, a preliminary version has been deployed here [https://ovrhuman.github.io/ewce_website/]. Increased efficiency is needed in the cell select app, as the large data set and complex graph algorithms result in slow loading times.

This study would not have been possible with other methods such as hypergeometric tests and PSEA, which would not account for the specificity of gene expression, or would need quantitative expression data from the disease state, making them unable to distinguish between secondary "reactive" expression and primary expression associated with the main genetic susceptibility. The lack of requirements for disease tissue expression data made it possible to utilise large, publicly available lists of simple phenotype-gene associations. One disadvantage was the substantial amount of computation required for the analysis. The interactive HPC sessions had a time limit of 8 hours which was not long enough to complete the analysis in one pass, causing it to take substantially longer. In future analysis, we plan to use a 32 Core Threadripper, acquired by the Neurogenomics lab, which will be much faster and has no time limit. 

Another issue was that the HPO and Descartes data are updated and changed over time. Directly pulling the data from these resources for each analysis caused some problems with reproducing previous analysis. This was resolved by privately hosting the data in its current state, whilst still including an option in the makefile to download the latest data.

```{r abnormal_lip_similar, message=FALSE, echo=FALSE, warning=FALSE}
library(stats)
abnormal_lip <- readRDS("data/Abnormal_lip_morphology_similar_phenos.rds")
abnormal_lip$total_q = p.adjust(abnormal_lip$p_val, method="BH")

ns_branch = "Abnormality of the nervous system"
ns_branch_id = hpo$id[match(ns_branch, hpo$name)]
ns_desc = get_descendants(hpo,ns_branch_id)
ns_desc_names = hpo$name[ns_desc]
ns_lip_associations = abnormal_lip[abnormal_lip$Comparison_Phenotype %in% ns_desc_names, ]
ns_lip_associations$q = p.adjust(p=ns_lip_associations$p_val,method="BH")
ns_lip_associations = ns_lip_associations[ns_lip_associations$total_q < 0.05 & complete.cases(ns_lip_associations),]
ns_lip_associations$group = ns_branch
abnormal_lip$branch = "other"
abnormal_lip$branch[abnormal_lip$Comparison_Phenotype %in% ns_desc_names] = ns_branch

im_branch = "Abnormality of the immune system"
im_branch_id = hpo$id[match(im_branch, hpo$name)]
im_desc = get_descendants(hpo,im_branch_id)
im_desc_names = hpo$name[im_desc]
#im_desc_names = im_desc_names[!(im_desc_names %in% ns_desc_names)]
im_lip_associations = abnormal_lip[abnormal_lip$Comparison_Phenotype %in% im_desc_names, ]
im_lip_associations$q = p.adjust(p=im_lip_associations$p_val,method="BH")
im_lip_associations = im_lip_associations[im_lip_associations$total_q < 0.05 & complete.cases(im_lip_associations),]
im_lip_associations$group = im_branch

cv_branch = "Abnormality of the cardiovascular system"
cv_branch_id = hpo$id[match(cv_branch, hpo$name)]
cv_desc = get_descendants(hpo,cv_branch_id)
cv_desc_names = hpo$name[cv_desc]
#im_desc_names = im_desc_names[!(im_desc_names %in% ns_desc_names)]
cv_lip_associations = abnormal_lip[abnormal_lip$Comparison_Phenotype %in% cv_desc_names, ]
cv_lip_associations$q = p.adjust(p=cv_lip_associations$p_val,method="BH")
cv_lip_associations = cv_lip_associations[cv_lip_associations$total_q < 0.05 & complete.cases(cv_lip_associations),]
cv_lip_associations$group = cv_branch

# do bootstrap enrichment to show if there are significantly more significant results in the ns branch ?

```

One of the goals of bio-ontology data structures is the integration of information and making implicit knowledge become explicit. To this point, all of the cell-phenotype associations found in this study were already present in pre-existing data but had not yet been made accessible or explicitly stated. A future plan is to continue to bring out this type of information, thereby extending the description logic the bio-ontology. For example, the phenotypes are currently organised as a network of "is-a" connections, which is useful for understanding how the physical attributes of diseases are related. But, the phenotypes could also be connected in other ways; for example, the associated gene lists for each phenotype could be used to uncover genetic associations between phenotypes from completely separate "is-a" branches. As a proof of concept, a bootstrap enrichment analysis was done with 10,000 reps to test which phenotype gene lists are enriched for the "Abnormal lip morphology" gene list. In other words, to find phenotypes that are genetically similar or related to abnormal lip morphology. Though still in early development, the algorithm created for this phenotype gene list similarity analysis is shown below in supplementary - code example 3. 

\newpage
It was predicted that "Abnormal lip morphology" would have many genetically similar phenotypes in the "Abnormality of the nervous system" branch. This is because abnormal lip morphology can be caused by abnormal migration of neural crest cells during development and is often associated with abnormalities of the central nervous system. Additionally, both share an ectodermal origin [@mueller_central_2007]. Again for hypothesis testing purposes, we also looked at our other two example branches (immune and cardiovascular system). We found that there were `r length(ns_lip_associations$Z_score)` significantly similar gene lists in the nervous system branch, and only `r length(cv_lip_associations$Z_score)` and `r length(im_lip_associations$Z_score)` in the cardiovascular and immune system branches, respectively. Additionally, the slightly more related phenotypes we found in the cardiovascular branch may be explained by the fact that neural tube defects have also been associated with congenital heart diseases [@gardner_overdistention_1981].

The nervous system phenotypes that were found to be genetically similar to "Abnormal lip morphology" included many that would be very difficult to diagnose at birth, such as ADHD, Delayed speech and language development, and intellectual disability, and many other neurological abnormalities. These examples were all highly significant (*q*<0.0001 and fold change>15). These findings appear to be supported by the literature. For example, research by @conrad_incidence_2008 showed that children with isolated cleft lip and/or palate had significantly higher levels of neurological soft signs than controls. It was also found that the known association between abnormal lip morphology and abnormal neuronal migration was reproduced by this analysis (gene list similarity between "Abnormal lip morphology" and "Abnormal neuronal migration": *Z*=12.9, *q*<0.00001). 

The next step would be to perform the same enrichment analysis with 100,000 reps for all pairwise phenotype-phenotype relationships. This could extend the description logic of the HPO bio-ontology to have "has-a" connections, representing relationships between phenotypes that have a significantly similar gene list, rather than just the "is-a" connections that are currently available. The full impact of doing this is hard to predict, but an example use-case could be to find all instances of easily diagnosed diseases that are genetically similar to ones that are harder to diagnose. This could help to target diagnostic and screening resources to patients at higher risk of certain conditions, possibly leading to faster and more effective treatment. For example, it was found that the "Autism" gene list was significantly enriched for the "Abnormal lip morphology" gene list (*Z*=13.8, *q*<0.00001). Abnormal lip morphology is clearly easier to diagnose at birth than autism, and it has been shown that early autism spectrum disorder intervention (age < 3) provides significant benefit [@autism_early_2015]. Further work would need to be done in this case to determine if abnormal lip morphology is predictive of autism in human patients. However, it is easy to see how these kinds of findings could provide beneficial insights for RD patients, especially given that diagnosis is such a significant problem for many of them.

In addition to these plans to extend the analysis, much of the code written for the RD EWCE analysis will be useful for similar studies in the future, as well as for follow up RD studies when more data becomes available. A future goal for this project is to share the code in open-source software repositories. This would make this kind of analysis accessible to the wider research community. A start was made by submitting additional features to the EWCE Bioconductor package, and storing the main RD EWCE script on the UKDRI Neurogenomics lab GitHub repository, but much more remains to be done. 

Although individual RDs have a low prevalence, when taken as a whole, they impact the lives of a significant number of people worldwide. The increasing accessibility of genomic and phenotypic data, high throughput analytic techniques, and improved ways to organise and disseminate information have made this project possible. It is hoped that the results presented here can help to overcome some of the problems of resource allocation and information scarcity that have hindered RD research in the past.

\newpage
# Acknowledgments 
This work would not have been possible without the continued guidance and support from Dr Nathan Skeen, Brian M. Shlider, Alan Murphy and the rest of the UKDRI Neurogenomics lab from the Imperial college department of Brain Sciences. The generosity with lab resources and their personal time and energy made the project both enjoyable and an exceptional learning experience.

\newpage 
# Supplementary

## Code example 1 - RD EWCE shell script
\singlespace
```
#!/bin/bash
n=1
while (($n <= 20))
do
  echo "Running script (attempt = $n)"
  n=$(( n+1 ))
  module load anaconda3/personal
  source activate pyre 
  Rscript completed_phenotypes.R
  Rscript Makefile.R
done
```
 Code example 1: **Rare Disease EWCE HPC shell script.** This script loads the development environment and executes the code to run the analysis. It is placed in a while loop so that it automatically restarts if it terminates prematurely.
\doublespace

## Code example 2 - Find relative ont level
\singlespace
```{r code_example, message = FALSE, results='hide'}
 find_parent <- function (phenotype,phenoAdj,hpo){
   pos_parents = hpo$parents[phenotype]
   phenotypes = rownames(phenoAdj)
   paths = list()
   for (p in phenotypes){
     if (phenoAdj[p,phenotype] == 1) {
       if (p %in% pos_parents) {
         paths[p] = 1 + find_parent(p,phenoAdj,hpo) # <- recursion
       }}}
   if (length(paths) == 0) {
     return (0)
   } else {
     parents = 0
     for (i in seq(length(paths))) {
       if (paths[[i]] > parents) {
         parents = paths[[i]]
       }}}
   return (parents)}
```
Code example 2: **Find number of generations of ancestors for an HPO term.** This function finds the number of generations of ancestor terms present for each term in a connected component of a subset of the results. This is used to map the relative ontology level of terms to node size in the interactive plot. `phenoAdj` is an adjacency matrix of all the phenotypes to be plotted where 0 and 1 is used to indicate if `phenotype[i]` is a parent of `phenotype[j]`. `hpo` is the HPO ontology data object.


\doublespace

## Code example 3 - genelist similarity bootstrap test 

\singlespace
```{r}
similar_phenotypes <- function(phenotype,phenotype_to_genes,bootstrap=1000) {
  phenotype_similarity_scores = data.frame()  
  # Primary Phenotype gene list 
  pheno_genes = unique(phenotype_to_genes[
    phenotype_to_genes$Phenotype == phenotype, "Gene"])
  for (p in unique(phenotype_to_genes$Phenotype)) {
      # Comparison Phenotype gene list 
      comparison_pheno_genes=unique(
        phenotype_to_genes[phenotype_to_genes$Phenotype==p,"Gene"])
      if (length(comparison_pheno_genes)>4) {
        n_matches=length(pheno_genes[pheno_genes%in%comparison_pheno_genes])
        # Create bootstrap distribution for current comparison phenotype
        all_genes = unique(phenotype_to_genes$Gene)
        bootstrap_distribution = c()
        for (i in seq(1:bootstrap)){
          bootstrap_genelist=sample(
            all_genes,size=length(comparison_pheno_genes),replace=FALSE)
          bootstrap_n_matches=length(
            pheno_genes[pheno_genes %in% bootstrap_genelist])
          bootstrap_distribution=append(
            bootstrap_distribution,bootstrap_n_matches)}
        # calculate Z score for the comparison
        Z_score=(n_matches-mean(bootstrap_distribution))/sd(
          bootstrap_distribution)
        p_val = 2*(length(bootstrap_distribution[
          bootstrap_distribution>=n_matches])/length(bootstrap_distribution))
      } else {
        Z_score = NA 
        p_val = NA }
      phenotype_similarity_scores = rbind(phenotype_similarity_scores,
        data.frame("Primary_Phenotype"=phenotype,"Comparison_Phenotype"=p,
                   "Z_score"=Z_score,"p_val" = p_val)) }
  return(phenotype_similarity_scores)}
```
Code example 3: **Phenotype-phenotype bootstrap enrichment analysis function.** This is the algorithm currently being developed for the bootstrap enrichment analysis. It determines whether one gene list is enriched in another. Bootstrapping is used rather than hypergeometric test because it accounts for the specificity of genes to phenotypes. Simply put, it allows us to identify genetically similar RDs whilst accounting for how frequently individual genes are associated with RDs. 

\doublespace



# References



