---
output: pdf_document
header-includes:
    - \usepackage{setspace}\doublespacing
    #- \usepackage[left]{lineno}
    #- \linenumbers
    - \usepackage{helvet}
    - \renewcommand{\familydefault}{\sfdefault}
theme: Springer
title: \singlespace Identification of cell types involved in rare disease-associated human phenotypes
subtitle: \singlespace Using expression weighted celltype enrichment analysis to identify cell-phenotype relationships in scRNA-seq data from Descarets and phenotype associated genelists from the human phenotype ontology
#date: 03/08/2021
author: 
- Robert Gordon-Smith, Molecular and Cellular Biosciences MRes
- Dr Nathan Skene, Department of Brain Sciences, Imperial College London

bibliography: bibliography.bib
csl: elsevier-harvard.csl
fontsize: 12pt
linkcolor: blue
spacing: double
# toc: yes
params:
#  results_data: "data/Descartes_All_Results_extras.rda"
  results_data: "data/descartes_all_results_061021.rda"
abstract: \singlespace Despite the name, rare diseases (RD) contribute to a significant burden of disease globally. The increasing accessibility of genomic and biomedical datasets, and high throughput analytical techniques, has made it possible to uncover new insights into the genetic susceptibility and cell types involved. With the low prevalence of individual RDs, they have often not had the resource allocation they need. Here we attempt to overcome this by tackling the problem as a whole. The Human Phenotype Ontology (HPO) contains disease phenotypes annotated with associated risk genes. Here, these disease-associated gene lists, combined with human single-cell RNA sequence data, were used to identify the primary cell types involved in the HPO disease phenotypes. Expression weighted cell type enrichment (EWCE) was used for the analysis with 100,000 bootstrap reps. Over 8000 significant cell-phenotype associations were found (where *q*<0.05). The results were shown to be overrepresented by expected enrichments. For example, immune cells were more frequently enriched for phenotypes from the \"Abnormalities of the immune system\" ontology branch than all other cell types combined (*p*<0.05). The analysis was able to reproduce many previously known cell-phenotype relationships documented in the literature. There were also many novel and unexpected results, on examination, many of these were shown to have have a plausible mechanistic explanation. We can be confident that within these results are many previously unknown insights that could provide targets to future research. As too many results were produced to present in a single paper, the Rare Disease EWCE web app was developed to make the analysis available to clinicians and researchers, without the need for specialist knowledge and computational resources (https://ovrhuman.github.io/ewce_website/). \newpage


---

```{r echo = FALSE, message = FALSE, warning=FALSE}


library(patchwork)

library(ggplot2)
library(ontologyIndex)
library(ontologyPlot)
library(EWCE)
library(cowplot)
library(wesanderson)
library(dplyr)

data(hpo)
phenotype_to_genes = HPOExplorer::load_phenotype_to_genes("data/phenotype_to_genes.txt")
disease_descriptions = readRDS("data/disease_descriptions.Rda")
rownames(disease_descriptions) = disease_descriptions$HPO_id
load("data/Descartes_All_Results_extras.rda")
descartes_mappings = read.csv("data/DescartesHuman_celltype_mapping.csv")
descartes_mappings$level1 = gsub("_"," ",descartes_mappings$level1)
#tm_mappings = read.csv("data/TabulaMuris_celltype_mapping.csv")
ctd = readRDS("data/CTD_Descartes_withplot.rds")

source("source/cell_select_ggnetwork_plot.R")
source("source/phenotypes_per_cell_plot.R")
source("source/ewce_plot_function.R")

if (!require(HPOEWCE)) {
  #install.packages("rd_package/RareDiseaseEWCE", repos=NULL, type="source")
  devtools::install_github("ovrhuman/HPOEWCE")
  library(HPOEWCE)
}
if (!require(MultiEWCE)) {
  #install.packages("rd_package/RareDiseaseEWCE", repos=NULL, type="source")
  devtools::install_github("neurogenomics/MultiEWCE")
  library(MultiEWCE)
}
if (!require(HPOExplorer)) {
  #install.packages("rd_package/RareDiseaseEWCE", repos=NULL, type="source")
  devtools::install_github("neurogenomics/HPOExplorer")
  library(HPOExplorer)
}


savePlot <- function (plot, w = 1000, h = 1000, path = "savedPlot.png") {
  png(path,width = w, height = h)
  print(plot)
  dev.off()
}

ggloadImage <- function(path) {
  img <- png::readPNG(path)
  g <- grid::rasterGrob(img,interpolate = TRUE)
  plt <- qplot(1:10,1:10,geom="blank") +
    annotation_custom(g,xmin=-Inf,xmax=Inf,ymin = -Inf, ymax=Inf) +
    theme_blank()
  return(plt)
}


ggplotify <- function(plot_object, height_px = 1000, width_px = 1000) {
  fp <- "ggplotify_temp.png"
  savePlot(plot_object, width_px, height_px, fp)
  plt <- ggloadImage(fp)
  file.remove(fp)
  return(plt)
}



```


```{r rmd_stuff, message=FALSE, echo=FALSE}
# useful stuff
## Rmd commands
# 
# **Cross referencing**
# To reference a figure, place a label in the figure caption: `\\label{fig:figname}`.
# Then reference it in the text with `\ref{fig:figname}`
# 
# **Bibliography and citations**
# Include a .bib document in the yaml. copy bib format citaitons into this file from [zbib](zbib.org). The first parameter in the bib is the reference name of the citation. We can then use that reference name to cite in text like this `@HumanPhenotypeOntology2020` or at the end of a paragraph like this `[@HumanPhenotypeOntology2020]`. Check bibliogoraphy and citation cookbook docs for more options. see \ref{useful}
# 
# **Knit to word doc**
# change pdf_document to word_document in yaml. 
# 
# **laytex commands**
# use both laytex commands and markdown syntax to edit stuff. only laytex commands work in the Yaml header?
```


\newpage

# Introduction
Although rare diseases (RD) have a prevalence of less than 1 in 2000, they are over 6000 in number [@orphanet2012]. As a conservative estimate, between 263 and 466 million patients suffer from RDs globally, contributing to a significant disease burden. For individual RDs, economic incentives are often not aligned with the need for research and drug development. However, despite the large number of distinct RDs, approximately 72% are genetic disorders, which may provide opportunities to approach them collectively [@estimatingRareDiseasePrevalence2020]. Given the rise of genome-wide association studies (GWAS), electronic healthcare records, and transcriptomic data, it is increasingly feasible to utilise high-throughput computational methods. Not only does increased understanding of RDs benefit patients directly, but RDs can also be used as disease models and they can give valuable insights into complex polygenetic conditions, further extending the impact of RD research beyond what is implied by the term "rare" [@LessonsFromMonogeneticDisease2006].

The Human Phenotype Ontology (HPO) is an initiative founded in 2008 that consists of an ontology of $13\,000$ annotated clinically relevant phenotypes. They are connected in a directed acyclic graph with edges representing transitive "is-a" connections [@HumanPhenotypeOntology2020]. Each phenotype is a subclass of its parent phenotypes, all the way up to the root node of all phenotypes, which is "Phenotypic abnormality". This allows for computationally efficient and human interpretable representation of the complex relationships between phenotypes. The description logic of bio-ontologies can also describe properties of each term, including synonyms, descriptions, and associated genes and diseases. This data integration allows for implicit knowledge hidden in the data to become explicit and retrievable [@ClassificationOntologyPrecisionMed2018]. Currently, much of the HPO focuses on rare Mendelian diseases, many of which have extensive annotations taken from electronic health records and genotype data. Here we utilised the `r length(unique(phenotype_to_genes$ID))` phenotypes that have been annotated with lists of assocated genes along with the Descartes human-cell atlas, which is a single-cell gene expression reference atlas for the human body. It includes 4 million cells represetning 15 different organs, from 121 human fetal samples. It was created using three-level combinatorial indexing (sci-RNA-seq3) [@descartes_2020].

EWCE is a technique developed by @skene_2016 that utilises single-cell gene expression data to determine whether a set of genes is significantly associated with a particular cell type. It does not rely on quantitative gene expression data from disease tissue samples, or predefined cell marker genes. Instead, EWCE takes a set of genes found to be associated with a phenotype or disease and, using scRNA-seq data, it identifies cells that express that gene set more than could be expected by chance from a random gene set of the same length. This makes it particularly well suited to these large publicly available datasets. Here we created a pipeline for extracting and preparing this type of data and running EWCE to uncover cell-phenotype relationships. The tools for the analysis were made available as R packages for reproducability and to facilitate future analysis as more genotypic and phenotypic data becomes available. Finally, we created a web-based results portal that allows patients and domain experts to explore the results most relevant to them. This was intended to bring the analysis to a wider audience and enable the full impact of the study to be realised. 

# Methodology 
The EWCE R package from Bioconductor (version 1.1.0) is used for the analysis [@EWCE_Bioconductor_2021]. Additionally, further tools were created to facilitate the analysis of multiple gene lists in parallel, and manage ontology data. Theses were then made available as the [HPOExplorer](https://github.com/neurogenomics/HPOExplorer) and [MultiEWCE](https://github.com/neurogenomics/MultiEWCE) R packages. 

The EWCE technique is described in detail by @skene_2016, but in brief, a scRNA data set is used to calculate gene-cell specificity scores for all pairwise gene-cell combinations. This is obtained by dividing the expression of a gene within a cell type by the total expression of the gene in all cell types. EWCE then takes a target gene list of length $n$, referred to as $T$, and a set of background genes referred to as $B$. The total expression specificity of the gene list is calculated for each cell, using the specificity scores for each gene. Then, 100,000 gene lists of lenght $n$ are randomly sampled from $B$, and specificity scores are calculated for all of these gene lists. This distribution of specificity scores can then be used to determine the probabillity of enrichment. In other words, if the specificity of expression of $T$ in a given cell is much higher than most randomly sampled gene lists, it is likely that there is a significant assocaition.  

EWCE requires gene lists of length 4 or greater. `r length(unique(all_results_merged$list))` met the criteria. This resulted in a large multiple testing burden. Benjamini-Hochberg (BH) method was used to limit the false discovery rate (FDR).  

The production of R packages and analysis pipeline allows for both reproducibility and ease of incorporating newly available datasets into the analysis as they become available. By standardising the format of the generated results it also makes them all compatible with our web-based results portal, so the user can explore and compare the output of different analysis. The pipeline was primarily created in R with some bash scripting to automate certain tasks on the High performance computing cluster. The landing page for the website was made using HTML and CSS, and the web apps themselves were created with the Shiny Web application Framework for R and deployed on the [ShinyApps](https://www.shinyapps.io/) server [@Shiny_package]. A outline of this work flow can be seen in Figure \ref{fig:Figure1}.

```{r Figure1,  fig.width = 20,fig.height=20, fig.cap = "\\label{fig:Figure1}\\textbf{Work flow.} \\textbf{A} The scRNA-seq data was taken from the Descartes Human Cell Atlas and the phenotype associated genelists from the Human Phenotype Ontology. \\textbf{B} This represents the computation of results using EWCE, where a specificity score for the target gene list, in a given cell, is calculated and then compared with a distribution of 100,000 randomly sampled genelist of the same length. \\textbf{C} This is the interactive results portal. It allows the user to search for significant phenotype enricments for a given cell type. The figure is generated using a combination of bespoke functions and the ggplot2 and ggnetwork packages [@ggplot2_package; @ggnetwork_package]. The size of the nodes represents the direction of the \"is_a\" relationship between nodes, where parent terms are larger than their child terms. Expression fold change is represented by the colour. In this example, the user has selected Acinar cells, with a significance threshold of \\textit{q}<0.005 and a fold change threshold of > 5. The hover box is comprised of further results and a phenotype definition pulled from the HPO API.", message=FALSE, echo=FALSE, warning =FALSE, results = FALSE}
#figure_count = figure_count + 1; figures$EWCE_home = figure_count;
library(imager)
Figure1 = load.image("figures/rd_workflow2.png")
print(plot(Figure1, axes = FALSE))
```

\newpage
# Results

```{r pheno_by_branch_withd_dend, echo = FALSE, warning = FALSE , results=FALSE,message = FALSE }
palette_branch <- wesanderson::wes_palette("Darjeeling1")



phenos_per_cell_colourText2 <- function (all_results_merged, cell_mappings, fold = 1, q_val = 0.005, 
                                        dataset = "Descartes", cell_colours_hashmap = cell_colours) 
{
  phenos_per_cell = data.frame()
  for (c in unique(all_results_merged$CellType)) {
    cur_cell = c
    n_phenos = length(all_results_merged[all_results_merged$CellType == 
                                           c & all_results_merged$q < q_val & all_results_merged$fold_change > 
                                           fold, "list"])
    phenos_per_cell = rbind(phenos_per_cell, data.frame(Cell = cur_cell, 
                                                        n_phenos = n_phenos))
  }
  n_pheno_text = phenos_per_cell
  phenos_per_cell$Tissue = rep(NA, length(phenos_per_cell$Cell))
  if (dataset == "Descartes") {
    phenos_per_cell_tissue = data.frame()
    for (i in seq(1:length(phenos_per_cell$Cell))) {
      cur = cell_mappings[cell_mappings$level1 == paste(phenos_per_cell$Cell[i]), 
      ]$tissue
      for (t in unique(cur)) {
        if (!is.na(t)) {
          phenos_per_cell$Tissue[i] = t
          phenos_per_cell_tissue = rbind(phenos_per_cell_tissue, 
                                         phenos_per_cell[i, ])
        }
      }
    }
    
    phenos_per_cell = phenos_per_cell_tissue
  }
  
  
  if (dataset == "Descartes") {
    for (i in seq(1, length(phenos_per_cell$Cell))) {
      cur = phenos_per_cell[phenos_per_cell$Cell == phenos_per_cell$Cell[i], 
      ]
      phenos_per_cell$n_phenos[i] = phenos_per_cell$n_phenos[i]/length(unique(cur$Tissue))
    }
  }
  
  
  ## Need to sort this out as an option and add arguments for it 
  phenos_per_cell$Cell_colour <- cell_colours_hashmap[phenos_per_cell$Cell]
  phenos_per_cell$title <- "n significant enrichments per cell (q<0.05)"
  
  explan_plt <- ggplot(phenos_per_cell, aes(n_phenos, as.factor(Cell))) + 
    scale_x_continuous("", expand = c(0, 0), limits = c(0, 
                                                        max(n_pheno_text$n_phenos) + 55)) + theme_classic() + 
    scale_y_discrete(breaks = ctd[[1]]$plotting$cell_ordering) + labs(title= "Significant enrichments all branches (n)")  #labs(title = paste0("Significantly enriched phenotypes per cell (q < ", q_val, ", fold change > ", fold, ")")) + 
    
    ylab("Cell Type")
  if (dataset == "Descartes") {
    explan_plt = explan_plt + geom_col(aes(fill = Tissue), 
                                       size = 2, position = "stack")
  }
  else {
    explan_plt = explan_plt + geom_segment(aes(xend = 0, 
                                               yend = Cell), size = 2)
  }
  explan_plt = explan_plt + geom_text(data = n_pheno_text, 
                                      aes(label = n_phenos, x = n_phenos + 26, y = Cell), size = 4, 
                                      color = "black") +  ylab("Cell Type") + 
    theme(axis.title.y = element_blank(), axis.text.y = element_text(colour = phenos_per_cell$Cell_colour),
          axis.title.x = element_text("significant enrichments (n)"), plot.title = element_text(hjust = 0.5, size =12))
  return(explan_plt)
}





palette_branch <- wesanderson::wes_palette("Darjeeling1")


plot_n_signif_phenos_per_cell_by_branch_den <- function (all_results_merged, plot_branches = c("Abnormality of the nervous system", 
                                                                                               "Abnormality of the cardiovascular system", "Abnormality of the immune system"), 
                                                         q_threshold = 0.05, hpo = hpo, phenotype_to_genes = phenotype_to_genes, 
                                                         ctd = ctd, make_dendro = TRUE, palette_branch = c("#FF0000", "#00A08A", "#F2AD00")) 
{
  hpo_branches = hpo$children["HP:0000118"]
  hpo_branches_names = c()
  for (b in hpo_branches) {
    hpo_branches_names = c(hpo_branches_names, hpo$name[b])
  }
  main_branch_df = data.frame(hpo_id = hpo_branches, phenotype = hpo_branches_names)
  descendants = list()
  for (b in hpo_branches[[1]]) {
    descendants[[hpo$name[b]]] = ontologyIndex::get_descendants(hpo, 
                                                                b)
  }
  if (!"branch" %in% colnames(all_results_merged)) {
    phenotypes = unique(phenotype_to_genes$Phenotype)
    all_results_merged$branch = NA
    pheno_branches = c()
    for (p in phenotypes) {
      
      index = match(p, hpo$name)
      if (!is.na(index)) {
        id = hpo$id[[index]]
        for (i in seq(1, length(descendants))) {
          if (id %in% descendants[[i]]) {
            pheno_branches[p] = names(descendants[i])
          }
        }
      }
    }
    all_results_merged$branch = NA
    remaining = length(unique(all_results_merged$list))
    #cat("Asigning branch to phenotypes\nN remaining:\n")
    for (p in unique(all_results_merged$list)) {
      #cat("\r", remaining, "      ")
      remaining = remaining - 1
      all_results_merged$branch[all_results_merged$list == 
                                  p] = pheno_branches[p]
    }
  }
  signif_results = all_results_merged[all_results_merged$q <= 
                                        q_threshold & !is.na(all_results_merged$branch), ]
  n_signif_per_cell_by_branch = data.frame()
  for (b in unique(plot_branches)) {
    for (c in unique(all_results_merged$CellType)) {
      n = length(signif_results[signif_results$CellType == 
                                  c & signif_results$branch == b, ]$list)
      n_signif_per_cell_by_branch = rbind(n_signif_per_cell_by_branch, 
                                          data.frame(branch = b, CellType = c, n_signif = n))
    }
  }
  n_signif_per_cell_by_branch$hypergeo_p = NA
  total_signif = length(signif_results$q)
  for (i in seq(1, length(n_signif_per_cell_by_branch$CellType))) {
    cell = n_signif_per_cell_by_branch$CellType[i]
    branch = n_signif_per_cell_by_branch$branch[i]
    group1 = sum(n_signif_per_cell_by_branch$n_signif[n_signif_per_cell_by_branch$CellType == 
                                                        cell])
    group2 = sum(n_signif_per_cell_by_branch$n_signif[n_signif_per_cell_by_branch$branch == 
                                                        branch])
    overlap = n_signif_per_cell_by_branch$n_signif[i]
    n_signif_per_cell_by_branch$hypergeo_p[i] = stats::phyper(overlap - 
                                                                1, group2, total_signif - group2, group1, lower.tail = FALSE)
  }
  n_signif_per_cell_by_branch$hypergeo_q = stats::p.adjust(n_signif_per_cell_by_branch$hypergeo_p, 
                                                           method = "BH")
  n_signif_per_cell_by_branch$signif_asterics = ""
  n_signif_per_cell_by_branch$signif_asterics[n_signif_per_cell_by_branch$hypergeo_q < 
                                                0.05] = "*"
  n_signif_per_cell_by_branch$signif_asterics[n_signif_per_cell_by_branch$hypergeo_q < 
                                                0.005] = "**"
  n_signif_per_cell_by_branch$signif_asterics[n_signif_per_cell_by_branch$hypergeo_q < 
                                                5e-04] = "***"
  n_signif_per_cell_by_branch$signif_asterics[n_signif_per_cell_by_branch$hypergeo_q < 
                                                5e-05] = "****"
  cell_order = factor(gsub("_", " ", ctd[[1]]$plotting$cell_ordering))
  n_signif_per_cell_by_branch$cell_order = match(n_signif_per_cell_by_branch$CellType, 
                                                 cell_order)
  n_signif_per_cell_by_branch$CellType = stats::reorder(n_signif_per_cell_by_branch$CellType, 
                                                        n_signif_per_cell_by_branch$cell_order)
  n_signif_per_cell_by_branch$branch_f <- factor(n_signif_per_cell_by_branch$branch, 
                                                 levels = plot_branches)
  
  
  # maybe wont work
  cell_order <- ctd[[1]]$plotting$cell_ordering
  neuronal_cell_types = cell_order[seq(14,32)]
  immune_cell_types = cell_order[c(5,6,7,75,76,77)]
  cardiac_cell_types = cell_order[c(42,4)]
  #ns_colour = "#619CFF"; card_colour = "#F8766D"; immune_colour = "#00BA38";
  ns_colour = palette_branch[1]; card_colour = palette_branch[2]; immune_colour = palette_branch[3];
  
  cell_colours = c()
  for (c in cell_order) {
    if (c %in% neuronal_cell_types) {
      cell_colours = append(cell_colours, ns_colour)
    } else if (c %in% immune_cell_types) (
      cell_colours = append(cell_colours, immune_colour)
    ) else if (c %in% cardiac_cell_types) {
      cell_colours = append(cell_colours, card_colour)
    } else {
      cell_colours = append(cell_colours,"black")
    }
  }
  
  named_colours = c()
  for (i in seq(1, length(plot_branches))) {
    named_colours[plot_branches[i]] = palette_branch[i]
  }
  
  
  facet_branch_plt <- ggplot(n_signif_per_cell_by_branch[n_signif_per_cell_by_branch$branch %in% 
                                                           plot_branches, ], aes(x = CellType, y = n_signif, fill = branch)) + 
    geom_col() + scale_colour_manual(values =  named_colours,aesthetics = "fill") +
    geom_text(mapping = aes(label = signif_asterics, 
                            y = n_signif + 3)) + cowplot::theme_cowplot() + ylab("Enrichments (n)") + 
    geom_hline(yintercept=0) +
    xlab("Cell type") + theme(axis.text.x = element_text(angle = 90, 
                                                         hjust = 1), legend.position = "none") + 
    facet_wrap(~branch_f,  ncol = 1) + #, scales = "free_y") + # <- add back in?
    theme(strip.background = element_rect(colour = "white", fill= "white")) +
    scale_y_continuous(expand = c(0,0), limits = c(0,max(n_signif_per_cell_by_branch$n_signif)+6))
  
  descartes_mappings = read.csv("data/DescartesHuman_celltype_mapping.csv")
  descartes_mappings$level1 = gsub("_"," ",descartes_mappings$level1)
  plot_npc = phenos_per_cell_colourText2(all_results_merged,descartes_mappings, fold=1,q_val=0.05, cell_colours_hashmap = cell_colours) + theme(axis.text.y = element_text("Significant enrichments (n)"))  + coord_flip()+ theme(axis.text.x = element_blank(), axis.title.x = element_blank(), axis.title.y =  element_text("Significant enrichments (n)"))
  
  facet_branch_plt = facet_branch_plt + theme(axis.text.x = element_text(colour = cell_colours, angle = 45))
  
  if (make_dendro) {
    the_dendrogram = ctd[[1]]$plotting$ggdendro_horizontal +
      theme(plot.margin = unit(c(0, 0, 0, 0), units = "cm")) +
      #  scale_x_discrete(breaks = total_res$CellType)
      scale_x_discrete(breaks = ctd[[1]]$plotting$cell_ordering)
    # CHANGE ^: added scale_x_discrete to set the mapping of the dendro to the x axis scale
    
    facet_branch_plt = cowplot::plot_grid(the_dendrogram, plot_npc,facet_branch_plt,axis = "lr",
                                          align = "v", ncol = 1,
                                          rel_heights = c(0.5, 1, 4),labels=c("A","B","C"))
    # CHANGE ^: align argument to "v" and rel_heights to c(0.3,1) to make dend and barchart closer
    
    #output$withDendro = combined_plot
  }
  
  return(facet_branch_plt)
}






cell_order <- ctd[[1]]$plotting$cell_ordering
neuronal_cell_types = cell_order[seq(14,32)]
immune_cell_types = cell_order[c(5,6,7,75,76,77)]
cardiac_cell_types = cell_order[c(42,4)]
#ns_colour = "#619CFF"; card_colour = "#F8766D"; immune_colour = "#00BA38";
ns_colour = palette_branch[1]; card_colour = palette_branch[2]; immune_colour = palette_branch[3];
cell_colours = c()
for (c in cell_order) {
  if (c %in% neuronal_cell_types) {
    cell_colours[c] = ns_colour
  } else if (c %in% immune_cell_types) (
    cell_colours[c] = immune_colour
  ) else if (c %in% cardiac_cell_types) {
    cell_colours[c] = card_colour
  } else {
    cell_colours[c] = "black"
  }
}


plot_branches = c("Abnormality of the nervous system","Abnormality of the cardiovascular system","Abnormality of the immune system")

facet_branch_plt = plot_n_signif_phenos_per_cell_by_branch_den(all_results_merged, plot_branches = plot_branches, q_threshold = 0.05, hpo=hpo, phenotype_to_genes=phenotype_to_genes, ctd=ctd, make_dendro =TRUE)


```

```{r Figure2, fig.height=20, fig.width = 20, fig.cap = "\\label{fig:Figure2}\\textbf{A Dendrogram} showing the clustering of cell types in from the scRNA-seq data. The cells on the x axis are ordered by this dendrogram grouping. \\textbf{B Number of significant enrichments per cell type} This shows how many significant phenotype enrichments were found for each cell type from all branches in the HPO, where significant is defined as q<0.05 and fold change > 1. The colours represent the tissues of origin. \\textbf{C} This shows the number of significant enrichments per cell from three chosen branches of the HPO. It can be seen that the greatest number of significant enrichments are seen in the cell types that we would expect. For example, the vast majority of enrichments in the nervous system branch are assocaited with cells related to the nervous system. An additional hypergeometric test was done to show where there is an over representation of enrichments (asterics, ****, *** **, and *, indicate that \\textit{q}<0.00001, 0.0001, 0.001, and 0.05).", message = FALSE, warning = FALSE, echo = FALSE }

print(facet_branch_plt)


```

```{r phenos_per_cell_df, echo = FALSE, message = FALSE, warning = FALSE, results= FALSE} 
phenos_per_cell_df = dataframe_phenos_per_cell(all_results_merged,descartes_mappings, fold=1,q_val=0.05)
```

The Descartes human scRNA data clustered into `r length(unique(all_results_merged$CellType))` unique cell types. `r length(unique(all_results_merged$list))` gene lists from the HPO met the critereia for EWCE and `r length(all_results_merged$CellType[all_results_merged$q < 0.05])` significant cell-phenotype associations were found (*q*<0.05). The number of significant phenotype associations for each individual cell type can be seen in Figure \ref{fig:Figure2} B. 

The root HPO term, phenotypic abnormality (HP:0000118), has `r length(hpo$children[["HP:0000118"]])` child phenotypes that represent the main classes of phenotypic abnormality in the HPO. Of these, abnormality of the nervous system, cardiovascular system, and the immune system have been singled out for examples in \ref{fig:Figure2} C, as these branches have clear cell types in which there is expected to be high numbers of enrichments (neurons, cardiomyocytes, and immune cells, respectively). These expected associations can be used as a means of validating the results. To this end, a hypergeometric test was performed to identify cases where a cell type has a larger than expected number of enrichments from a given branch. The results from this are denoted by the "\*" above the bars. For example, it can be seen that significant enrichments for terms from the abnormality of the cardiovascular system branch are over represented in in cardiomyocytes (*q*<0.0001). A significant result like this suggests that the branch as a whole is broadly associated with the particular cell type. It can also be seen from the dendrogram grouping that large clusters of cell types are associated with a branch. For example, the group of nervous system related cells which all have large numbers of enrichments in the nervous system branch. The x axis text has been coloured red to help highlight this point. 

Additionally, there are many novel results that may provoke new lines of research. For example, a significant number of sub-phenotypes of "Abnormality of the cardiovascular system" are enriched in hepatoblasts.   

```{r proportion_expected_plt, echo = FALSE}
# modifying this function to be more specific about colours used. 
proportion_of_expected_enrichments_plot_2 <- function (all_results_merged, hpo, target_cells = c("Excitatory neurons", 
    "Limbic system neurons"), cell_type_description = "Neuronal cells", 
    HPO_Ids = ontologyIndex::get_descendants(hpo, hpo$id[match("Abnormality of the nervous system", 
        hpo$name)]), phenotype_description = "Nervous system phenotypes", 
    color_pal = c("#619CFF","#F8766D","#00BA38","gray"), color_expected_phenotypes = 1, 
    color_other_phenotypes = 4, blank_x_axis = FALSE) 
{
    cur_descendants_names = c()
    for (d in HPO_Ids) {
        cur_descendants_names = append(cur_descendants_names, 
            paste(hpo$name[d]))
    }
    all_results_merged$cur_branch = "Other"
    all_results_merged[all_results_merged$list %in% cur_descendants_names, 
        ]$cur_branch = phenotype_description
    all_results_merged$cur_branch = factor(all_results_merged$cur_branch)
    all_results_merged$minus_log_p = round(-log10(all_results_merged$p + 
        1e-06), digits = 0)
    proportional_results = data.frame()
    for (logp in unique(all_results_merged$minus_log_p)) {
        cur = all_results_merged[all_results_merged$minus_log_p == 
            logp, ]
        prop_branch = 100 * (length(cur$cur_branch[cur$cur_branch == 
            phenotype_description & cur$CellType %in% target_cells])/length(cur$cur_branch[cur$CellType %in% 
            target_cells]))
        prop_other = 100 * (length(cur$cur_branch[cur$cur_branch == 
            "Other" & cur$CellType %in% target_cells])/length(cur$cur_branch[cur$CellType %in% 
            target_cells]))
        proportional_results = rbind(proportional_results, data.frame(cur_branch = phenotype_description, 
            prop = prop_branch, minus_log_p = logp))
        proportional_results = rbind(proportional_results, data.frame(cur_branch = "Other", 
            prop = prop_other, minus_log_p = logp))
    }
    #color_pal = wesanderson::wes_palette(wes_color_palette, n_colors)
    prop_plot <- ggplot(proportional_results, aes(x = minus_log_p, 
        y = prop, color = cur_branch)) + geom_point(size = 3) + 
        geom_line(mapping = aes(linetype = cur_branch), size = 0.6) + 
        labs(color = "", linetype = "") + ylab(paste0("% Enrichments in \"", 
        cell_type_description, "\"")) + 
      scale_y_continuous(limits = c(0, 100)) + cowplot::theme_cowplot() + 
        scale_color_manual(values = c(color_pal[color_expected_phenotypes], 
            color_pal[color_other_phenotypes])) + theme(legend.position = c(0.5,0.95), 
        legend.text = element_text(size = 10), title = element_text(size = 11), axis.title.y = element_text(size=12))
      if (! blank_x_axis){
        prop_plot = prop_plot + xlab("-log10(p)")
      } else {
        prop_plot = prop_plot + theme(axis.text.x = element_blank(), axis.title.x = element_blank())
      } 
    return(list(prop_plot, proportional_results))
}

```

```{r patterns, message=FALSE, warning=FALSE, echo=FALSE}

# Significance and expected cell type ##########################################
plot_branches = c("Abnormality of the nervous system", "Abnormality of the cardiovascular system", "Abnormality of the immune system")
expected_cells = c("Excitatory neurons", "Cardiomyocytes","Antigen presenting cells")
correlation_results = data.frame()  
proportion_plots = list()
blank_x = c(TRUE, TRUE, FALSE)
for (i in seq(length(plot_branches))) {
  cur_plot <- proportion_of_expected_enrichments_plot_2(all_results_merged,
   hpo,
   target_cells=c(expected_cells[i]),
   cell_type_description = expected_cells[i],
   HPO_Ids = ontologyIndex::get_descendants(hpo,hpo$id[match(plot_branches[i],hpo$name)]),
   phenotype_description = plot_branches[i],
   color_pal =  c(palette_branch[1],palette_branch[2],palette_branch[3],"gray"),
   color_expected_phenotypes = i,
   color_other_phenotypes = 4, blank_x_axis = blank_x[i])
  proportion_plots[[i]] = cur_plot[[1]]
  correlation_results = rbind(correlation_results,cur_plot[[2]])
}

#sigplot <- cowplot::plot_grid(plotlist=proportion_plots, align = "h",nrow = 1,labels=c("A","B","C"))


# ont level plots (violin) #####################################################
blank_x <-   theme(axis.text.x = element_blank(), axis.title.x = element_blank())
      

ont_levels <- data.frame("phenotype"=unique(phenotype_to_genes$Phenotype),
                         "hpo_id"=hpo$id[match(unique(phenotype_to_genes$Phenotype),hpo$name)])
ont_levels <- ont_levels[complete.cases(ont_levels),]
lvls <- c()
for (id in ont_levels$hpo_id) {
  lvls = append(lvls, get_ont_level(hpo,id))
}
ont_levels$ont_lev <- lvls
rm(lvls)

n_associated_cells <- c()
for (p in ont_levels$phenotype) {
  #n_associated_cells <- append(n_associated_cells, length(all_results_merged$CellType[all_results_merged$list == p & all_results_merged$q < 0.05]))
  n_associated_cells <- append(n_associated_cells, length(all_results_merged$CellType[all_results_merged$list == p & all_results_merged$q < 0.05 & all_results_merged$fold_change > 1]))
}
ont_levels$n_associated_cells <- n_associated_cells
rm(n_associated_cells)

pal <- wesanderson::wes_palette("Darjeeling2", n=2)
ontlvl_ncells_plt <- ggplot(ont_levels, mapping=aes(x= factor(ont_lev), y=n_associated_cells)) +
  geom_jitter(color = pal[1]) +
  geom_violin(fill = NA) +
  geom_smooth(color = pal[2], method="loess",mapping = aes(x=ont_lev)) +
  cowplot::theme_cowplot() +
  labs(x="Ontology level", y="Associated cells/phenotype (n)",title = "n associated cells/phenotype by ontology level")+
  blank_x
#ontlvl_ncells_plt



# ont level facet

signif_res <- all_results_merged[all_results_merged$q < 0.05,]
ontlevz <- c()
for (p in unique(signif_res$HPO_id)) {
  ontlevz[p] <- get_ont_level(hpo,p)
}
signif_res$ontlvl <- ontlevz[signif_res$HPO_id]
signif_res <- signif_res[complete.cases(signif_res),]

pal <- wesanderson::wes_palette("Darjeeling2", n=2)
ontlvl_fold_plt <- ggplot(signif_res, aes(x = factor(ontlvl), y = log10(fold_change))) +
  geom_jitter(color = pal[1]) +
  geom_violin(fill = NA) +
  geom_smooth(color = pal[2], method="loess",mapping = aes(x=ontlvl)) +
  cowplot::theme_cowplot() +
  labs(x="Ontology level", y="log10(Fold change)",title = "Fold change in specific expression by ontology level") +
  blank_x
  

# ngenes plt (stat smooth takes too long/doesnt work with all results so just use signif ?)
ngenes<-c()
for(p in unique(signif_res$list)) {
  ngenes[p] <- length(get_gene_list(p,phenotype_to_genes))
}
signif_res$ngenes <- ngenes[signif_res$list]
signif_res <- signif_res[complete.cases(signif_res),]
pal <- wesanderson::wes_palette("Darjeeling2", n=2)
ontlvl_ngenes_plt <- ggplot(signif_res, aes(x = factor(ontlvl), y = ngenes)) +
  geom_jitter(color = pal[1]) +
  geom_violin(fill = NA) +
  geom_smooth(color = pal[2], method="loess",mapping = aes(x=ontlvl)) +
  cowplot::theme_cowplot() +
  labs(x="Ontology level", y="Genes (n)",title = "Number of genes by ontology level")

#ontlvlplots <- cowplot::plot_grid(plotlist=list(ontlvl_ncells_plt,ontlvl_fold_plt,ontlvl_ngenes_plt),align = "hv",nrow = 1)



```

```{r Figure3, fig.height = 12, fig.width = 15, fig.cap = "\\label{fig:Figure3} \\textbf{A, B, and C} Show the relationship between significance threshold and the proportion of enrichments found in the expected HPO branch. As more stringent significance thresholds for enrichment are used, the proportion of enrichments from an expected HPO branch increases. This validates the method as it shows that many of the strong phenotype-cell associations are in agreement with our current understanding. \\textbf{D, E, and F} describe features of the HPO that change with ontology level. We define ontology level as the number of generations of terms below a term, with the most general \"phenotypic abnormality\" at the top, and very specific phenotypes at level 0 with no sub-types below. in \\textbf{D} we can se that teh number of cell types associated with a phenotype goes down as we approach more specific phenotypes at leaf nodes. \\textbf{E} shows that these low level terms also typically have a higher fold change in specific expression. \\textbf{F} low level terms also typicaly have shorter gene lists than more general phenotypes.", message = FALSE, warning = FALSE , echo = FALSE}
blank_x_axis <- theme(x.axis.title =  element_blank(), axis.text.x = element_blank())
print(
  (
    ( proportion_plots[[1]]  /
       proportion_plots[[2]] /
       proportion_plots[[3]] 
      ) 
    |
      ( ontlvl_ncells_plt /
       ontlvl_fold_plt /
       ontlvl_ngenes_plt
      ) 
  ) +  plot_annotation(tag_levels = LETTERS) 
)

```

```{r signif_expected_cell_pearsons, message=FALSE, echo=FALSE, warning =FALSE, results = FALSE}


plot_branches = c("Abnormality of the nervous system", "Abnormality of the cardiovascular system", "Abnormality of the immune system")
expected_cells = c("Excitatory neurons", "Cardiomyocytes","Antigen presenting cells")
correlation_results = data.frame()  
# proportion_plots = list()
for (i in seq(length(plot_branches))) {
  cur_plot <- proportion_of_expected_enrichments_plot(all_results_merged,
   hpo,
   target_cells=c(expected_cells[i]),
   cell_type_description = expected_cells[i],
   HPO_Ids = ontologyIndex::get_descendants(hpo,hpo$id[match(plot_branches[i],hpo$name)]),
   phenotype_description = plot_branches[i],
   wes_color_palette="Darjeeling1",
   n_colors = 4,
   color_expected_phenotypes = i,
   color_other_phenotypes = 4)
#  proportion_plots[[i]] = cur_plot[[1]]
  correlation_results = rbind(correlation_results,cur_plot[[2]])
}

#print(cowplot::plot_grid(plotlist=proportion_plots, align = "h",nrow = 1,labels=c("A","B","C")))
pearsons_expected = cor.test(correlation_results$minus_log_p[correlation_results$cur_branch != "Other"],correlation_results$prop[correlation_results$cur_branch != "Other"],method="pearson")

n_desc_rec_inf = length(get_descendants(hpo, "HP:0002719"))



```


To further demonstrate that the analysis finds expected phenotype-cell relationships, excitatory neurons, cardiomyocytes, and antigen presenting cells were taken as examples. Figure \ref{fig:Figure3} A, B and C show that if we take all results for a particular cell, the more significantly associated phenotypes disproportionately come from the expected branch of the HPO. In other words, as more stringent significance thresholds are used, the proportion of enriched phenotypes from the expected HPO branch increases (*r*~`r pearsons_expected$parameter`~=`r pearsons_expected$estimate`, *p*=`r pearsons_expected$p.value`).

Figure \ref{fig:Figure3} D, E, and F show relationships related to ontology level of HPO terms. As ontology level approaches 0 (the most specific, narrowly defined phenotypes found at the leaf nodes) the number of significantly associated cell types per term goes down, the level of specific expression of the gene list in those cells goes up, and the number of genes in the phenotype gene list goes down. This all supports the idea that significant enrichments in low ontology level terms tend to be highly specific cell-phenotype relationships. This perhaps makes these results more actionable avenues for further research. 


## Low ontology level terms are enriched in expected and novel cell types

```{r Figure4,fig.width=20,fig.height=20, fig.cap = '\\label{fig:Figure4} \\textbf{A} Gives the number of significant enrichemnts per cell for descendants of recurrant infection. As expected, cells associated with the immune system account for large numbers of enrichemnts. Aditionally, there are some surprising enrichments, for example, there is one in hepatoblasts. \\textbf{B} Here we take a closer look at some of the specific phenotypes descending from recurrent infections. This shows child terms of recurrent bacterial infection. We can see a strong association with cilliated epithelial cells in mycobacterial infections. This is unsurprising, given that this is primarily a respiratory disease, and dysfunctional cillia leave people prone to respiratory infection \\textbf{C} This now goes a step further in specificity and looks at child terms of recurrent gram-negative bacterial infection. We can see that the surprising enrichemnt in hepatoblasts is actually from recurrent neiserial infections. In both \\textbf{B} and \\textbf{C}, significance is indicated with asterics (*, **, ***, and **** represent textit{q} less than 0.05, 0.001, 0.0001, and 0.00001. respectively).', message=FALSE, echo=FALSE}

library(wesanderson)
color_pal = wes_palette("Darjeeling1",4)
library(cowplot)

branch = "Recurrent infections"
branch_id = paste(hpo$id[match(branch,hpo$name)])
branch_descendants = paste(get_descendants(hpo,branch_id))
branch_descendants_names = paste(hpo$name[branch_descendants])
all_results_merged$list = paste(all_results_merged$list)
#all_results_merged$cur_branch = paste(all_results_merged$cur_branch)

all_results_merged$cur_branch = paste("Other")
all_results_merged$cur_branch[all_results_merged$list %in% branch_descendants_names] = branch



branch_signif_counts = data.frame()
for (c in unique(all_results_merged$CellType)) {
  n_signif = length(all_results_merged[all_results_merged$CellType == c & all_results_merged$cur_branch == branch & all_results_merged$q < 0.05, ]$q)
  branch_signif_counts = rbind(branch_signif_counts,
                               data.frame("branch"=branch,
                                          "CellType"=c,
                                          "n_signif"=n_signif))
}

cell_order = factor(gsub("_"," ",ctd[[1]]$plotting$cell_ordering))
branch_signif_counts$cell_order = match(branch_signif_counts$CellType, cell_order)
branch_signif_counts$CellType = reorder(branch_signif_counts$CellType, branch_signif_counts$cell_order)
branch_signif_counts$labels = branch_signif_counts$n_signif
branch_signif_counts$labels[branch_signif_counts$labels == 0] = ""

branch_plt <- ggplot(branch_signif_counts[branch_signif_counts$branch==branch,], aes(x=CellType,y=n_signif)) +
  geom_col( fill = color_pal[2], color = "black") +
  geom_text(mapping= aes(label = labels, y = n_signif + 3))+
  theme_cowplot()+
  ylab("N phenotypes") +
  scale_y_continuous(expand=c(0,0), limits= c(0,max(branch_signif_counts$n_signif)+5))+
  theme(axis.text.x = element_blank(), legend.position = "none")+#= element_text(angle = 90, hjust=1, vjust =0.2),legend.position="none") +
  #coord_flip() +
  ggtitle("Significant enrichments per cell: Recurrent infections")


#branch_plt

# pheno_fold_plt #########################
pal <- wesanderson::wes_palette("GrandBudapest2")
branch = "Recurrent bacterial infections"
exclude_root = TRUE

branch_id = paste(hpo$id[match(branch,hpo$name)])
branch_descendants = hpo$children[[branch_id]]
branch_descendants_names = paste(hpo$name[branch_descendants])
if (exclude_root) {
  branch_descendants_names = branch_descendants_names[branch_descendants_names != branch]
} else {
  branch_descendants_names = c(branch, branch_descendants_names)
}
pheno_df = all_results_merged[all_results_merged$list %in% branch_descendants_names, ]

pheno_df$signif_asterics = ""
pheno_df$signif_asterics[pheno_df$q<0.05] = "*"
pheno_df$signif_asterics[pheno_df$q<0.001] = "**"
pheno_df$signif_asterics[pheno_df$q<0.0001] = "***"
pheno_df$signif_asterics[pheno_df$q<0.00001] = "****"

recurrentBact_fold_plt <- ggplot(pheno_df, aes(x = CellType, y= fold_change, fill = list)) +
  ggtitle(paste0('Child nodes of HPO branch "',branch,'"')) +
  geom_col() +
  geom_hline(yintercept=0) +
  geom_text(label = pheno_df$signif_asterics, mapping = aes(y = fold_change + 1))+
  theme_cowplot() +
  scale_y_continuous(expand=c(0,0), limits= c(0,max(branch_signif_counts$n_signif)+2))+
  ylab("Fold change") +
  xlab("Cell type") +
  theme(axis.text.x = element_blank(), legend.position = "none")+#= element_text(angle=90,hjust=1,vjust=0.2),legend.position = "none") +
  facet_wrap(~list, ncol=1, scales = "free_y") +
  scale_fill_manual(values = pal)

#################
pal2 <- append(wesanderson::wes_palette("Darjeeling1"), wesanderson::wes_palette("Darjeeling2")[1])
branch = "Recurrent gram-negative bacterial infections"
exclude_root = TRUE

branch_id = paste(hpo$id[match(branch,hpo$name)])
branch_descendants = hpo$children[[branch_id]]
branch_descendants_names = paste(hpo$name[branch_descendants])
if (exclude_root) {
  branch_descendants_names = branch_descendants_names[branch_descendants_names != branch]
} else {
  branch_descendants_names = c(branch, branch_descendants_names)
}
pheno_df = all_results_merged[all_results_merged$list %in% branch_descendants_names, ]

pheno_df$signif_asterics = ""
pheno_df$signif_asterics[pheno_df$q<0.05] = "*"
pheno_df$signif_asterics[pheno_df$q<0.001] = "**"
pheno_df$signif_asterics[pheno_df$q<0.0001] = "***"
pheno_df$signif_asterics[pheno_df$q<0.00001] = "****"

recurrentGram_plt <- ggplot(pheno_df, aes(x = CellType, y= fold_change, fill = list)) +
  ggtitle(paste0('Child nodes of HPO branch "',branch,'"')) +
  geom_col() +
  geom_text(label = pheno_df$signif_asterics, mapping = aes(y = fold_change + 1))+
  theme_cowplot() +
  scale_y_continuous(expand=c(0,0) ,limits= c(0,max(branch_signif_counts$n_signif)+2))+
  ylab("Fold change") +
  xlab("Cell type") +
  theme(axis.text.x = element_text(angle=90,hjust=1,vjust=0.2),legend.position = "none") + 
  
  facet_wrap(~list, ncol=1, scales = "free_y") +
  scale_fill_manual(values= pal2)


layout <- "
AAAAAAAAA
BBBBBBBBB
BBBBBBBBB
BBBBBBBBB
BBBBBBBBB
CCCCCCCCC
CCCCCCCCC
CCCCCCCCC
CCCCCCCCC
CCCCCCCCC
CCCCCCCCC
"


plot <-( (branch_plt + theme(axis.title.x = element_blank(),plot.title = element_text(size =18))) +  # center title: hjust = 0.5
           (recurrentBact_fold_plt + theme(legend.position = "none", axis.title.x = element_blank(), strip.background = element_rect(fill ="white", colour = "white"))) + 
           (recurrentGram_plt+theme(axis.title.x = element_blank(),axis.text.x = element_text(angle=45,hjust=1,vjust=1),legend.position = "none", strip.background = element_rect(fill = "white", colour = "white"))) + geom_hline(yintercept=0)  ) + 
  plot_layout(design = layout) +  plot_annotation(tag_levels = LETTERS) 
#ggsave("test.png", plot, dpi = 400, width = 20, height = 25, units = "in")

print(plot)



immune_cells = c("Lymphoid cells","Myeloid cells","Antigen presenting cells","Thymocytes","Hematopoietic stem cells")
total_immune_enrich = sum(branch_signif_counts$n_signif[branch_signif_counts$CellType %in% immune_cells])
total_other_enrich = sum(branch_signif_counts$n_signif[!branch_signif_counts$CellType %in% immune_cells])
other_cells = unique(paste(branch_signif_counts$CellType[!branch_signif_counts$CellType %in% immune_cells & branch_signif_counts$n_signif > 0]))

immune_vs_other_t = t.test(branch_signif_counts$n_signif[branch_signif_counts$CellType %in% immune_cells],branch_signif_counts$n_signif[!branch_signif_counts$CellType %in% immune_cells])


```


While figures \ref{fig:Figure2} C and \ref{fig:Figure3} A, B, and C show that phenotypes from some of the main HPO branches are generally associated with expected cell types, figure \ref{fig:Figure4} shows that expected cell-phenotype relationships are also found in the more specific phenotypes at lower HPO levels. As an example, we look at all descendant terms of Recurrent infections (HP:0002719), which includes `r n_desc_rec_inf` HPO terms at ontology levels ranging from 0 to 3. The hypothesis is that they will be primarily enriched in cells involved in the immune system. It was predicted that, from these sub-phenotypes, the number of significant enrichments in immune system associated cell types would be greater than the number of significant results in all other cell types combined.


As predicted, the majority of enrichments were found in cells associated with the immune system (`r tolower(immune_cells)`). `r total_immune_enrich` enrichments were found in immune system related cells, and only `r total_other_enrich` enrichments were found in other cell types (*t*~`r immune_vs_other_t$parameter`~=`r immune_vs_other_t$statistic`, *p*=`r round(immune_vs_other_t$p.value,digits=6)`). 

A selection of the sub-phenotypes of "Recurrent infections" were then isolated, to see where the significant effects lie, and to investigate some of the unexpected and novel associations. Figure \ref{fig:Figure4} C shows the child phenotypes of "Recurrent bacterial infections", which is its self a child term of "Recurrent infections".

The analysis was able to reproduce a well known association with myeloid cells and bacterial skin infections, as well as staphlyocococal infection (also typically a skin infection). Myeloid cells are a class of phagocytic immune cell often involved in defence against bacterial pathogens. Interestingly, a very strong association is seen between recurrent mycobacterial infection and ciliated epithelial cells (*q*<0.0001, fold change=`r round(all_results_merged[all_results_merged$CellType=="Ciliated epithelial cells" & all_results_merged$list =="Recurrent mycobacterial infections",]$fold_change,digits=2)`). Mycobacteral infection (Tuberculosis being the most well known), often affects the respiratory system. Dysfunctional cilia leads to reduced mucus clearance from lungs and increased susceptibility to respiratory infection.

Figure \ref{fig:Figure4} C provides a closer look at the child phenotypes of Recurrent gram-negative bacterial infections. Recurrent neisserial infections were found to be enriched in Hepatoblasts (*q*=`r all_results_merged[all_results_merged$CellType=="Hepatoblasts" & all_results_merged$list == "Recurrent Neisserial infections", ]$q`, fold change=`r round(all_results_merged[all_results_merged$CellType=="Hepatoblasts" & all_results_merged$list == "Recurrent Neisserial infections", ]$fold_change,digits=2)`). As this is a potentially novel finding, this phenotype was also checked in the Tabula Muris data. Similarly, significant enrichment in hepatic cells (Kupfer cells and Hepatocytes) were found (*q*<0.0001).


\newpage
# Discussion
```{r hepatocyte_cardio, message=FALSE, echo=FALSE, warning =FALSE, results = FALSE}

cardio_id = hpo$id[match("Abnormality of the cardiovascular system", hpo$name)]
cardio_desc = get_descendants(hpo, cardio_id)
cardio_hepato = all_results_merged[all_results_merged$HPO_id %in% cardio_desc & all_results_merged$CellType == "Hepatoblasts" & all_results_merged$q < 0.05 & all_results_merged$fold_change > 7, c("CellType","list","fold_change","q")]
cardio_hepato$Phenotype = cardio_hepato$list
cardio_hepato = cardio_hepato[,c("CellType","Phenotype","fold_change","q")]


```

More than 8000 significant enrichments were found. These are spread across all HPO branches. High-level HPO terms, encompassing a broad range of traits, were significantly associated with expected cell types. This validates the method as it shows that the majority of cell-phenotype associations are in agreement with accepted knowledge. The number of expected enrichments also increases as more stringent significance thresholds are used, as the more well-understood cell-phenotype relationships tend to have a stronger enrichment.

It was found that, in significant cell-phenotype associations, the strength of enrichment (measured by fold change specific expression) tends to be higher at lower ontology levels. These more narrowly defined disease phenotypes also tend to have shorter, more specific gene lists. This potentially makes many of these low ontology level terms great targets for research. One of the roadblocks of gene therapy is the lack of understanding of the pleiotropic effects of many genes [@gene_therapy_2020]. Being able to isolate the problem down to a small set of genes in a very specific cell type goes a long way to alleviating this problem. This could also provide a way to prioritise different research possibilities. The results could be subset to show all low ontology level HPO terms that are severe enough to warrant treatment and have a strong association with a particular cell type. Other parameters could also be used in this way. For example, to prioritise long-lived cell types that may be the best candidates for gene therapy. 

Figure \ref{fig:Figure2} C showed that the majority of significant enrichments for terms from the main branches of the HPO are in expected cell types. For example, terms that are a sub-class of "Abnormality of the cardiovascular system" are primarily enriched in cardiomyocytes. It was also noted that there are a number of enrichments in less obvious cell types. For example, a significant number of terms in the cardiovascular branch were enriched in hepatoblasts (*p*<0.0001). To explore this further, we retrieved all results from the cardiovascular branch that are enriched in hepatoblasts and have a fold change > 7, and *q* < 0.05. The phenotypes were primarily ones that often involve damage to arteries caused by lipid deposition (`r tolower(cardio_hepato$Phenotype)`). Given the large role that the liver plays in lipid metabolism, it is unsurprising that dysfunction of hepatocytes may be implicated in these cardiovascular diseases. Additionally, a brief literature search finds many recent studies exploring the link between hepatic cells and cardiovascular diseases, such as the paper by @xu_hepatocyte_2021 which shows that hepatocyte ATF3 is protective against atherosclerosis by regulating high-density lipoprotein metabolism. Furhter, @bell_hepatocyte_2018 showed that higher circulating hepatocyte growth factor is associated with elevated atherosclerosis progression measures. 

```{r cilia_recurrent_inf, message=FALSE, error=FALSE, warning=FALSE, echo=FALSE}
cil_inf = all_results_merged[all_results_merged$CellType == "Ciliated epithelial cells" & all_results_merged$HPO_id %in% get_descendants(hpo,hpo$id[match("Recurrent infections", hpo$name)] ) & all_results_merged$q < 0.05,]

rec_myco_fold = cil_inf[cil_inf$list == "Recurrent mycobacterial infections", ]$fold_change

```

More specific HPO terms from lower ontology levels were also primarily enriched in expected cell types. For example, it was found that the majority of significant enrichments for descendants of "Recurrent bacterial infections" were in immune cells. Again some less obvious enrichments were also found, such as ciliated epithelial cells. To investigate this further, the EWCE results for descendant terms of "Recurrent bacterial infections", that were also enriched in ciliated epithelial cells, were retrieved individually. It was found that the enrichments are primarily associated with the recurrent respiratory infections sub-branch (`r tolower(cil_inf$list)`). This relationship between structural defects of the cilia and recurrent respiratory tract infections has been documented in the literature [@immotile-cilia_1977]. The strongest association was found in mycobacterial infections (which includes tuberculosis), and this also primarily affects the lungs (fold change=`r round(rec_myco_fold,digits=2)`, *q*<0.00001). The only non-pulmonary phenotype was associated with ciliated cells was recurrent otitis media infections, which is also known to be a problem for people with immotile-cilia [@immotile_cilia_1983].

Another potentially novel finding was the significant enrichment in hepatoblasts for a term in the "Recurrent infections" branch. This was found to be specific to recurrent neisserial infections. For further confirmation, this phenotype was also analysed in the Tabula Muris data (mouse scRNA dataset), where it was also significantly enriched in hepatic cells (kupffer cells and hepatocytes). Whilst this is a seemingly surprising association, we found a number of plausible explanations for it in the literature. Fitz-Hugh-Curtis syndrome is a RD characterised by inflammation of the peritoneum and the tissues surrounding the liver, caused by *Neisseria gonorrhoeae* infection [@fitz-hugh-curtis_2017]. It is possible that an abnormality of hepatic cells leaves patients particularly susceptible to this problem. Though Fitz-Hugh_Curtis syndrome is a sub-phenotype of recurrent neisserial infections, at the time of writing, there was no entry for it in the HPO, so it was not possible to check if the enrichment was specific to this sub-phenotype. Perhaps the most likely explanation involves the complement system, a part of the innate immune system that plays a key role in defense against neisserial infections. Complement is synthesised primarily in the liver, and it was found that people with deficits in complement are at high risk for Neisserial infection [@lewis_complement_2020; @fijen_complement_1994]. 

From these examples, it can be seen that when scrutinised, many of the more surprising cell-phenotype relationships are either previously known or at least have a plausible mechanistic basis. It is very likely that within the over 8000 results presented here, there are many previously unknown links between disease phenotypes and specific cell types that could lead to advances in the understanding and treatment of RDs. For the impact of the results to be fully realised, it was essential that they could be easily explored by domain experts and clinicians without the need for extensive computational resources and specialist knowledge of enrichment analysis and programming. The web-based applications were developed to facilitate this. Although further work is needed, a preliminary version has been deployed here [https://ovrhuman.github.io/ewce_website/]. Increased efficiency is needed in the cell select app, as the large data set and complex graph algorithms result in slow loading times.

This study would not have been possible with other methods such as hypergeometric tests and PSEA, which would not account for the specificity of gene expression, or would need quantitative expression data from the disease state, making them unable to distinguish between secondary "reactive" expression and primary expression associated with the main genetic susceptibility. The lack of requirements for disease tissue expression data made it possible to utilise large, publicly available lists of simple phenotype-gene associations. One disadvantage was the substantial amount of computation required for the analysis. The interactive HPC sessions had a time limit of 8 hours which was not long enough to complete the analysis in one pass, causing it to take substantially longer. In future analysis, we plan to use a 32 Core Threadripper, acquired by the Neurogenomics lab, which will be much faster and has no time limit. 

Another issue was that the HPO and Descartes data are updated and changed over time. Directly pulling the data from these resources for each analysis caused some problems with reproducing previous analysis. This was resolved by privately hosting the data in its current state, whilst still including an option in the makefile to download the latest data.

Much of the code written for the RD EWCE analysis will be useful for similar studies in the future, as well as for follow up RD studies when more data becomes available. The code is packaged as MultiEWCE and HPOExploer and was shared in open-source software repositories. This makes this kind of analysis accessible to the wider research community.

Although individual RDs have a low prevalence, when taken as a whole, they impact the lives of a significant number of people worldwide. The increasing accessibility of genomic and phenotypic data, high throughput analytic techniques, and improved ways to organise and disseminate information have made this project possible. It is hoped that the results presented here can help to overcome some of the problems of resource allocation and information scarcity that have hindered RD research in the past.

\newpage
# References



