---
title: "Cell type-specific contextualisation of the phenomic landscape: a comprehensive and scalable approach towards the diagnosis, prognosis and treatment of all rare diseases"
author:
  - name: Brian M. Schilder
    orcid: 0000-0001-5949-2191
    corresponding: true
    email: brian_schilder@alumni.brown.edu
    roles:
      - Investigation
      - Project administration
      - Software
      - Visualization
    affiliations:
      - Imperial College London
  - name: Kitty B. Murphy
    orcid: 0000-0002-8669-3076
    corresponding: false
    roles: []
    affiliations:
      - Imperial College London
  - name: Robert Gordon-Smith
    orcid: 0000-0001-6698-7387
    corresponding: false
    roles: []
    affiliations:
      - Imperial College London
  - name: Jai Chapman
    corresponding: false
    roles: []
    affiliations:
      - Imperial College London
  - name: Momoko Otani
    corresponding: false
    roles: []
    affiliations:
      - Imperial College London
  - name: Nathan G. Skene
    orcid: 0000-0002-6807-3180
    corresponding: true
    email: n.skene@imperial.ac.uk
    roles:
      - Project administration
    affiliations:
      - Imperial College London
keywords:
  - rare disease
  - phenotype
  - single-cell
  - gene therapy
plain-language-summary: |
  We identified the cell types underlying the symptoms of all rare diseases.
key-points:
  - We used the Human Phenotype Ontology and single-cell RNA-seq references to characterise the phenome.
  - We then demonstrated how these results can be applied to clinical diagnosis, prognosis and therapeutics development.
date: last-modified
bibliography: references.bib
citation:
  container-title: Nature Medicine
format:
  nature-pdf:
    journal: "sn-nature"
    keep-tex: true
    execute:
      cache: true
      echo: false
  docx:
    execute:
      cache: true
      echo: false
---

```{r setup, cache=FALSE, include=TRUE}
library(data.table)
```

```{r load_phenotype_to_genes}
hpo <- HPOExplorer::get_hpo()
p2g <- HPOExplorer::load_phenotype_to_genes()
annot <- HPOExplorer::load_phenotype_to_genes(3)
per_disease <- p2g[,list(ng=data.table::uniqueN(gene_symbol),
                         np=data.table::uniqueN(hpo_id)),by="disease_id"]
per_phenotype <- p2g[,list(ng=data.table::uniqueN(gene_symbol),
                           nd=data.table::uniqueN(disease_id)),by="hpo_id"]
```

```{r load_example_results}
## Import precomputed results for reporting summaries
results <- MSTExplorer::load_example_results()
results <- HPOExplorer::add_hpo_name(results, hpo = hpo)
results <- HPOExplorer::add_ont_lvl(results)
results <- HPOExplorer::add_ancestor(results, hpo = hpo)
results <- MSTExplorer::map_celltype(results)
MSTExplorer::add_logfc(results)
results[,effect:=estimate]
## Substitute B for \beta for now since Quarto doesn't seem to support
## Greek letters after they've been stored in a data.table...
results[,summary:=paste0(
  "$",
  "FDR_{p,c}=",format(q,digits=3),",",
  "B=",format(estimate,digits=3),
  "$"
)] 

```

```{r summarise_results}
res_summ <- MSTExplorer::summarise_results(results = results)
res_summ_all <- res_summ$tmerged[ctd=="all"] 
```

```{r load_example_ctd, cache=FALSE}
## Must use `cache.lazy=FALSE` because sparse matrices not yet supported for caching
ctd_list <- MSTExplorer::load_example_ctd(c("ctd_DescartesHuman.rds",
                                           "ctd_HumanCellLandscape.rds"),
                                          multi_dataset=TRUE)
```

```{r validate_associations_mkg}
validate_associations_mkg_out <- MSTExplorer::validate_associations_mkg(results = results)
kg_hp <- validate_associations_mkg_out$kg[grepl("^HP:",from)]
X <- "`**!!RECOMPUTE!!**`"
Y <- "`**!!RECOMPUTE!!**`"
```

## Abstract

Rare diseases (RDs) are an extremely heterogeneous and underserved category of medical conditions.
While the majority of RDs are strongly genetic, it remains largely unknown via which physiological mechanisms genetics cause RD.
Therefore, we sought to systematically characterise the cell type-specific mechanisms underlying all RD phenotypes with a known genetic cause by leveraging the Human Phenotype Ontology and transcriptomic single-cell atlases of the entire human body from embryonic, foetal, and adult samples.
In total we identified significant associations between `r res_summ_all[["cell types significant"]]` cell types and `r res_summ_all[["phenotypes significant"]]`/`r res_summ_all[["phenotypes tested"]]` (`r res_summ_all[["phenotypes significant (%)"]]`%) unique phenotypes across `r  res_summ_all[["diseases significant"]]` RDs. We estimate that this represents an over `r floor(validate_associations_mkg_out$increased_knowledge/100)*100`-fold increase in the collective knowledge of RD phenotype-cell type mechanisms.

Next, we demonstrated how these results may be used for personalised patient diagnosis and prognosis, as well as the development of novel therapeutics.
Finally, we take a data-driven approach to highlight several of the most promising gene/cell therapy candidates with the highest probability of animal model-to-human patient translation.
Furthermore, we have made these results entirely reproducible and freely accessible to the global community to maximise their impact.
To summarise, this work represents a significant step forward in the mission to treat patients across an extremely diverse spectrum of serious RDs.

## Introduction {#sec-introduction}

While rare diseases (RDs) are individually uncommon, they collectively account for an enormous global disease burden with over 10,000 recognised RDs affecting at least 300-400 million people globally [@Ferreira2019-jp] (1 in 10-20 people) [@Zhu2020-vo] . Over 75% of RDs primarily affect children with a 30% mortality rate by 5 years of age [@noauthor_undated-kp]. Despite the prevalence and severity of RDs, patients suffering from these conditions are vastly underserved due to several contributing factors. First, diagnosis is extremely challenging due to the highly variable clinical presentations of many of these diseases. The diagnostic odyssey can take patients and their families decades, with an average time to diagnosis of 5 years [@Marwaha2022-uy].
Of those, \~46% receive at least one incorrect diagnosis and over 75% of all patients never receive any diagnosis @Molster2016-da.
Second, prognosis is also made difficult by high variability in disease course and outcomes which makes matching patients with effective and timely treatment plans even more challenging.
Finally, even for patients who receive an accurate diagnosis/prognosis, treatments are currently only available for less than 5% of all RDs [@Halley2022-pd].
In addition to the scientific challenges of understanding RDs, there are strong financial disincentives for pharmaceutical and biotechnology companies to develop expensive therapeutics for exceedingly small RD patient populations with little or no return on investment [@Institute_of_Medicine_US_Committee_on_Accelerating_Rare_Diseases_Research_and_Orphan_Product_Development2010-vj; @Yates2022-ra].
Those that have been produced are amongst the world’s most expensive drugs, greatly limiting patients’ ability to access it [@Nuijten2022-yc; @Thielen2022-ud], The provision of timely, effective and affordable care for RD patients will require substantive transformations to our existing scientific, clinical, and regulatory frameworks.

A major challenge in both healthcare and scientific research is the scalable exchange of information.
Even in the age of electronic healthcare records (EHR) much of the information about an individual’s history is currently fractured across healthcare providers, often with differing nomenclatures for the same conditions.
The Human Phenotype Ontology (HPO) is a hierarchically organised set of controlled clinical terms that provides a much needed common framework by which clinicians and researchers can precisely communicate patient conditions [@Gargano2024-fc; @Kohler2019-pc @Robinson2008-ys; @Kohler2021-wk].
The HPO spans all domains of human physiology and currently describes `r hpo@n_terms` phenotypes across 10,300 RDs.
Each phenotype and disease is assigned its own unique identifier and organised as a hierarchical graph, such that higher-level terms describe broad phenotypic categories or *branches* (e.g. *HP:0033127*: 'Abnormality of the musculoskeletal system' which contains `r data.table::uniqueN(KGExplorer::get_ontology_descendants(ont = hpo, terms = "HP:0033127")|>unlist())` unique phenotypes) and lower-level terms describe increasingly precise phenotypes (e.g. *HP:0030675*: "Contracture of proximal interphalangeal joints of 2nd-5th fingers").
It has already been integrated into healthcare systems and clinical diagnostic tools around the world, with increasing adoption over time [@Gargano2024-fc].
Common ontology-controlled frameworks like the HPO open a wealth of new opportunities, especially when addressing RDs.
Services such as the Matchmaker Exchange [@Osmond2022-ml; @Philippakis2015-dq] have enabled the discovery of hundreds of underlying genetic etiologies, and led to the diagnosis of many patients.
This also opens the possibility of gathering cohorts of geographically dispersed patients to run clinical trials, the only viable option for treatment in many individuals.
To further increase the number of individuals who qualify for these treatments, as well as the trial sample size, proposals have been made deviate from the traditional single-disease clinical trial model and instead perform basket trials on groups of RDs with shared molecular etiologies (SaME) [@Zanello2023-zd].
However this approach, and indeed much of RD patient care, hinges upon first characterising the molecular mechanisms underlying each RD.

Over 80% of RDs have a known genetic cause [@Nguengang_Wakap2020-cz; @noauthor_2022-ok].
Despite this our knowledge of the physiological mechanisms via which genetics cause pathogenesis is lacking for most RDs, severely hindering our ability to effectively diagnose, prognose and treat RD patients.
The availability of standardised, ontology-controlled databases presents opportunities to systematically investigate RDs at scale.
Since 2008, the HPO has been continuously updated using knowledge from the medical literature, as well as by integrating databases of expert validated gene-phenotype relationships, such as OMIM [@Amberger2019-vl; @Amberger2017-tg; @McKusick2007-di], Orphanet [@Maiella2013-oo; @Weinreich2008-wm], and DECIPHER @Firth2009-qg.
A subset of the HPO contains gene annotations for `r res_summ_all[["phenotypes"]]` phenotypes across `r res_summ_all[["diseases"]]` diseases.
Yet genes alone do not tell the full story of how RDs come to be, as their expression and functional relevance varies drastically across the multitude of tissues and cell types contained within the human body.

Our knowledge of single-cell-resolution biology has exploded over the course of the last decade and a half, with numerous applications in both scientific and clinical practices [@Baysoy2023-vt; @Haque2017-bn; @Qi2023-ev].
More recently, comprehensive single-cell transcriptomic atlases across tissues have also emerged [@CZI_Single-Cell_Biology_Program2023-fs; @Svensson2020-lg].
In particular, the Descartes Human @Cao2020-qz and Human Cell Landscape @Han2020-iq projects provide comprehensive multi-system single-cell RNA-seq (scRNA-seq) atlases in embryonic, foetal, and adult human samples from across the human body.
These datasets provide data-driven gene signatures for hundreds of cell subtypes.
They also allow us to investigate disease mechanisms in the context of specific life stages.

Here, we combine and extend several of the most comprehensive genomic and transcriptomic resources currently available to systematically uncover the cell types underlying granular phenotypes across `r res_summ_all[["diseases significant"]]` diseases.
We then go on to highlight thousands of novel phenotype-cell type associations which collectively expand our knowledge of cell type-resolved phenotypes by an estimated `r round(validate_associations_mkg_out$increased_knowledge,0)`-fold.
Next, we present several potential avenues for real world applications of these results in the context of RD patient diagnosis, prognosis, treatment, and therapeutics development.

## Results {#sec-results}

### Phenotype-cell type associations

```{r run_phenomix, eval=FALSE}
## Create phenotype-gene matrix filled with aggregated GenCC evidence scores
ymat <- HPOExplorer::hpo_to_matrix(formula = "gene_symbol ~ hpo_id")
## Run phenomix with DescartesHuman CellTypeDataset
lm_res1 <- MSTExplorer::run_phenomix(ctd_name = "DescartesHuman",
                                     annotLevel = 2, 
                                     test_method = "glm_univariate",
                                     ymat = ymat)
## Run phenomix with HumanCellLandscape CellTypeDataset
lm_res2 <- MSTExplorer::run_phenomix(ctd_name = "HumanCellLandscape",
                                     annotLevel = 3, 
                                     test_method = "glm_univariate",
                                     ymat = ymat)
## Merge results
results <- data.table::rbindlist(list(DescartesHuman=lm_res1,
                                      HumanCellLandscape=lm_res2),
                                idcol = "ctd")
## Apply multiple testing correction
results[,q:=stats::p.adjust(p,method="fdr")]
```

In this study we systematically investigated the cell types underlying phenotypes across the HPO. 
A summary of the phenome-wide results stratified by single-cell atlas can be found in @tbl-summary.
Within the results using the Descartes Human single-cell atlas, `r res_summ$tmerged[ctd=="DescartesHuman"][["tests significant"]]`/`r res_summ$tmerged[ctd=="DescartesHuman"][["tests"]]` (`r res_summ$tmerged[ctd=="DescartesHuman"][["tests significant (%)"]]`%) tests across `r res_summ$tmerged[ctd=="DescartesHuman"][["cell types significant"]]`/`r res_summ$tmerged[ctd=="DescartesHuman"][["cell types"]]` (`r res_summ$tmerged[ctd=="DescartesHuman"][["cell types significant (%)"]]`%) cell types and `r res_summ$tmerged[ctd=="DescartesHuman"][["phenotypes significant"]]`/`r res_summ$tmerged[ctd=="DescartesHuman"][["phenotypes"]]` (`r res_summ$tmerged[ctd=="DescartesHuman"][["phenotypes significant (%)"]]`%) phenotypes revealed significant phenotype-cell type associations after multiple-testing correction ($FDR_{p,c}<0.05$). Using the Human Cell Landscape single-cell atlas, `r res_summ$tmerged[ctd=="HumanCellLandscape"][["tests significant"]]`/`r res_summ$tmerged[ctd=="HumanCellLandscape"][["tests"]]` (`r res_summ$tmerged[ctd=="HumanCellLandscape"][["tests significant (%)"]]`%) tests across `r res_summ$tmerged[ctd=="HumanCellLandscape"][["cell types significant"]]`/`r res_summ$tmerged[ctd=="HumanCellLandscape"][["cell types"]]` (`r res_summ$tmerged[ctd=="HumanCellLandscape"][["cell types significant (%)"]]`%) cell types and `r res_summ$tmerged[ctd=="HumanCellLandscape"][["phenotypes significant"]]`/`r res_summ$tmerged[ctd=="HumanCellLandscape"][["phenotypes"]]` (`r res_summ$tmerged[ctd=="HumanCellLandscape"][["phenotypes significant (%)"]]`%) phenotypes showed significant phenotype-cell type associations ($FDR_{p,c}<0.05$). The median number of significantly associated phenotypes per cell type was `r res_summ$tmerged[ctd=="DescartesHuman"][["phenotypes per cell type (median)"]]` (Descartes Human) and `r res_summ$tmerged[ctd=="HumanCellLandscape"][["phenotypes per cell type (median)"]]` (Human Cell Landscape), respectively.

Across both single-cell references, the median number of significantly associated cell types per phenotype was `r res_summ_all[["cell types per phenotype (median)"]]`, suggesting reasonable specificity of the testing strategy.
`r res_summ_all[["diseases significant"]]`/`r res_summ_all[["diseases"]]` (\~`r res_summ_all[["diseases significant (%)"]]`%) of diseases within the HPO gene annotations showed significant cell type associations for at least one of their respective phenotypes.

### Validation of expected phenotype-cell type relationships

```{r plot_bar_dendro, fig.height=16, fig.width=13}
plot_bar_dendro_out <- MSTExplorer::plot_bar_dendro(
  results = results, 
  show_plot = FALSE) 
overrep_dat <- plot_bar_dendro_out$ggbars_out$data_summary

overrep_dat[,summary:=paste0(ancestor_name,": ", 
                             n_celltypes_sig,"/",n_celltypes,
                             " types of ",shQuote(target_celltypes),
                             " were overrepresented",
                             " ($N_{p}$=",phenotypes_per_ancestor,").")] 
```

Within each high-level branch in the HPO shown in [Fig. @fig-summary]b, we tested whether each cell type was more often associated with phenotypes in that branch relative to those in all other branches (including those not shown).
We then checked whether each cell type was overrepresented (at $FDR_{b,c}<0.05$) within its respective on-target HPO branch, where the number of phenotypes within that branch ($N_{p}$).
`r paste(overrep_dat$summary,collapse=" ")`.

As an additional form of validation ([Fig. @fig-summary]d), we tested for a relationship between phenotype-cell type association significance ($-log_{e}(p_{p,c})$ where $log_{e}$ denotes natural log and and $p_{p,c}$ denotes uncorrected phenotype-cell type association p-values) and the proportion of on-target cell types. The list of on-target cell types were determined by matching each high-level HPO branch to a corresponding CL branch. These cross-ontology mappings can be found in @tbl-celltypes.
For this analysis we used raw p-values ($p_{p,c}$) rather than multiple-testing corrected p-values ($FDR_{p,c}$) to provide a more dynamic range of values (as the latter can drive values to 1).
All `r length(unique(plot_bar_dendro_out$ggprop_out$data$branch))`/`r length(unique(plot_bar_dendro_out$ggprop_out$data$branch))` high-level HPO branches showed a consistent upwards trend towards greater proportions of on-target cell types with increasing degrees of significance.
Furthermore, all branches also showed a proportion of on-target cell types above that expected by chance (baseline = on-target cell types / total cell types) at $-log_{e}(p_{p,c})>1$.

```{r fig-summary, fig.height=16, fig.width=13}
#| label: fig-summary
#| fig-cap: Summary of significant associations between phenotypes and cell types, aggregated by HPO branch. Here we show **a**, the total number of significant phenotype enrichments per cell type ($FDR_{p,c}<0.05$) across all branches of the HPO. **b**, Number of phenotype association related to several high-level branches of the HPO. Asterisks above each bar indicate whether that cell type was significantly more often enriched in that branch relative to all other HPO branches, including those not shown here, as a proxy for how specifically that cell type is associated with that branch; $FDR _{b,c}<1e-04$ (\*\*\*\*), $FDR _{b,c}<0.001$ (\*\*\*), $FDR_{b,c}<0.01$ (\*\*), $FDR _{b,c}<0.05$ (\*). **c**, Dendrogram derived from the Cell Ontology (CL) showing the relatedness of all tested cell types to one another. For simplicity, cell type labels shown here are aligned to the CL [@Diehl2016-gt] and can therefore encompass one or more cell types annotated by the original authors of scRNA-seq datasets [@Cao2020-qz; @Han2020-iq]. **d**, Percentage of significant phenotype associations with on-target cell types (second row of facet labels), respective to the HPO branch. As significance increases ($-log_{10}(p)$ along the *x-axis*) the percentage of on-target enriched cell types also increases (*y-axis*).

plot_bar_dendro_out$plot
```

### Validation of inter- and intra-dataset consistency

```{r validate_associations_correlate_ctd}
library(data.table)
## Across CTD
validate_associations_correlate_ctd_out <- MSTExplorer::validate_associations_correlate_ctd(
  results=results, 
  group_var="ctd")
## Replace p-values of exactly 0 with smallest number R can compute
validate_associations_correlate_ctd_out$data_stats$p.all$summary_data$p.value <- max(validate_associations_correlate_ctd_out$data_stats$p.all$summary_data$p.value,
                  .Machine$double.xmin)

## Within CTD: across developmental stages
validate_associations_correlate_ctd_out_hcl <- MSTExplorer::validate_associations_correlate_ctd(
  results=results,
  filters= list(ctd=c("HumanCellLandscape"), 
                stage=c("Fetus","Adult")),
  group_var="stage")
```

Next, we sought to validate the consistency of our results across the two single-cell reference datasets (Descartes Human vs. Human Cell Landscape) across the subset of overlapping cell types [Fig. @fig-ctd-correlation].
In total there were `r nrow(validate_associations_correlate_ctd_out$data$all)` phenotype-cell type associations to compare across the two datasets (across `r length(unique(validate_associations_correlate_ctd_out$data$all$hpo_id))` phenotypes and `r length(unique(validate_associations_correlate_ctd_out$data$all$cl_name))` cell types annotated to the exact same CL term.
We found that the correlation between p-values of the two datasets was high (`r paste0("$rho=",format(validate_associations_correlate_ctd_out$data_stats$p.all$summary_data$estimate, digits=3),", p=",format(validate_associations_correlate_ctd_out$data_stats$logFC.significant$summary_data$p.value, digits = 3),"$")`).
Within the subset of results that were significant in both single-cell datasets ($FDR_{p,c}<0.05$), we found that correlation of the association effect size were even stronger (`r paste0("$rho=",format(validate_associations_correlate_ctd_out$data_stats$logFC.significant$summary_data$estimate, digits=3),", p=",format(validate_associations_correlate_ctd_out$data_stats$logFC.significant$summary_data$p.value, digits = 3),"$")`).
We also checked for the intra-dataset consistency between the p-values of the foetal and adult samples in the Human Cell Landscape, showing a very similar degree of correlation as the inter-dataset comparison (`r paste0("$rho=",format(validate_associations_correlate_ctd_out_hcl$data_stats$p.all$summary_data$estimate, digits=3),", p=",format(validate_associations_correlate_ctd_out_hcl$data_stats$logFC.significant$summary_data$p.value, digits = 3),"$")`).
Together, these results suggest that our approach to identifying phenotype-cell type associations is highly replicable and generalisable to new datasets.

### More specific phenotypes are associated with fewer genes and cell types

```{r plot_ontology_levels, cache=FALSE}
plot_ontology_levels_out <- MSTExplorer::plot_ontology_levels(
  results = results, 
  ctd_list = ctd_list,
  show_plot = FALSE)

plot_ontology_levels_out_stats <- plot_ontology_levels_out$data_stats$summary_data|>
  data.table::setkeyv("parameter2")
## replace pvalues of exactly 0 with the minimum computable number in R
## This avoids creating -Inf when logging values.
plot_ontology_levels_out_stats[p.value==0, p.value:=.Machine$double.xmin]
plot_ontology_levels_out_stats[q.value==0, q.value:=.Machine$double.xmin]
plot_ontology_levels_out_stats[,summary:=paste0(
  "$",
  "p=",format(p.value,digits=3),", ", 
  "q=",format(q.value,digits=3),", ",
  "rho=",format(estimate,digits=3),
  "$"
  )][,summary:=gsub("=[ ]","=",summary)]
```

First, we found that phenotype ontology showed a significant negative correlation with the number of genes annotated to that phenotype in the HPO data ([Fig. @fig-ontology-lvl]a; `r plot_ontology_levels_out_stats["genes"]$summary`).
This is expected as broader phenotypes tend to have large gene set annotations.
Next, we reasoned that lower HPO ontology levels representing more specific phenotypes were likely to be associated with fewer, more specific subsets of cell types.
This was indeed the case, as we observed a strongly significant negative correlation between the two variables ([Fig. @fig-ontology-lvl]b; `r plot_ontology_levels_out_stats["cell types"]$summary`).
We also found that the effect size of significant phenotype-cell type associations ($FDR_{p,c}<0.05$) increased with greater phenotype specificity, though the relationship was rather weak ([Fig. @fig-ontology-lvl]c; `r plot_ontology_levels_out_stats["estimate"]$summary`).
Finally, we found that the mean expression specificity of phenotype-associated genes (within the cell types significantly associated with those respective phenotypes at $FDR_{p,c}<0.05$) was positively correlated phenotype ontology depth ([Fig. @fig-ontology-lvl]d; `r plot_ontology_levels_out_stats["mean_specificity"]$summary`).

```{r fig-ontology-lvl, fig.height=11, fig.width=12}
#| label: fig-ontology-lvl
#| fig-cap: More specific phenotypes are associated with fewer, more specific genes and cell types. Box plots showing relationship between HPO phenotype level and **a**, the number of genes annotated to each phenotype, **b**, the number of significantly enriched cell types, **c**, the effect size of phenotype-cell type association tests at $FDR_{p,c}<0.05$, and **d**, the mean expression specificity of phenotype-associated genes in the cell types significantly associated with those respective phenotypes ($FDR_{p,c}<0.05$). Ontology level 0 represents the most inclusive HPO term 'All', while higher ontology levels (max=16) indicate progressively more specific HPO terms (e.g. 'Contracture of proximal interphalangeal joints of 2nd-5th fingers'). Boxes are coloured by the mean value (respective to the subplot) within each HPO level.

plot_ontology_levels_out$plot
```

### Hepatoblasts have a unique role in recurrent Neisserial infections

```{r plot_bar_dendro_facets}
results_tmp <- HPOExplorer::add_ancestor(data.table::copy(results),
                                         lvl = 7,
                                         force_new = TRUE)
target_branches <- list("Recurrent bacterial infections"="leukocyte")
infections_out <- MSTExplorer::plot_bar_dendro_facets(
  results=results_tmp,
  target_branches=target_branches,
  facets = "hpo_name",
  legend.position="top",
  lvl=9,
  ncol=2,
  vlines="hepatoblast",
  fill_var="ancestor_name_original",
  facets_n=NULL,
  q_threshold=0.05,
  background_full=FALSE)
remove(results_tmp)

staph_res <- infections_out$data[hpo_name=="Recurrent staphylococcal infections"]
staph_res_top <- staph_res[,.SD[p %in% head(sort(p), 1)], by=c("hpo_id")]
   
recurrent_infections_ids <- KGExplorer::get_ontology_descendants(ont = hpo,
                                                                 terms = "Recurrent infections")[[1]]
hepatoblast_res <- results[q<0.05 &hpo_id %in% recurrent_infections_ids & cl_name=="hepatoblast"]
hepatocyte_res <- results[q<0.05 & ancestor_name=="Abnormality of the immune system" & grepl("hepatocyte",CellType,ignore.case = TRUE)] 
```

We selected the HPO term 'Recurrent bacterial infections' and all of its descendants (`r data.table::uniqueN(infections_out$data$hpo_id)-1` phenotypes) as an example of how investigations at the level of granular phenotypes can reveal different cell type-specific mechanisms ([Fig. @fig-rni]).
As expected, these phenotypes are primarily associated with immune cell types (e.g. macrophages, dendritic cells, T cells, monocytes, neutrophils).
Some associations confirm relationships previously suggested in the literature, such as that between 'Recurrent staphylococcal infections' and myeloid cells [@Heim2014-du; @Pidwill2020-le; @Stoll2018-dc; @Tebartz2015-xs].
Specifically, our results pinpoint `r staph_res_top$cl_name`s as the most strongly associated cell subtypes (`r staph_res_top$summary`).

In contrast to all other recurrent infection types, 'Recurrent Neisserial infections' highlighted a novel association with hepatoblasts (Descartes Human : `r hepatoblast_res[hpo_name=="Recurrent Neisserial infections" & CellType=="Hepatoblasts"]$summary`).
Whilst unexpected, a convincing explanation involves the complement system, a key driver of innate immune response to Neisserial infections.
Hepatocytes, which derive from hepatoblasts, produce the majority of complement proteins [@Zhou2016-kq], and Kupffer cells express complement receptors @Dixon2013-ok.
In addition, individuals with deficits in complement are at high risk for Neisserial infections [@Ladhani2019-nf; @Rosain2017-ih], and a genome-wide association study in those with a Neisserial infection identified risk variants within complement proteins [@The_International_Meningococcal_Genetics_Consortium2010-if] .
While the potential of therapeutically targeting complement in RDs (including Neisserial infections) has been proposed previously [@Lung2019-il; @Reis2015-yz], performing this in a gene- and cell type-specific manner may help to improve efficacy and reduce toxicity (e.g. due to off-target effects).
Importantly, there are over 56 known genes within the complement system [@Seal2023-pa], highlighting the need for a systematic, evidence-based approach to identify effective gene targets.

Also of note, despite the fact that our datasets contain both hepatoblasts and their mature counterpart, hepatocytes, only the hepatoblasts showed this association.
This suggests that the genetic factors that predispose individuals for risk of Neisserial infections are specifically affecting hepatoblasts before they become fully differentiated.
It is also notable that these phenotypes were the only ones within the 'Recurrent bacterial infections' branch, or even the broader 'Recurrent infections' branch, perhaps indicating a unique role for hepatoblasts in recurrent infectious disease.
The only phenotypes within the even broader 'Abnormality of the immune system' HPO branch that significantly associated with mature hepatocytes were `r shQuote(hepatocyte_res[1,]$hpo_name)` (`r hepatocyte_res[1,]$summary`) and `r shQuote(hepatocyte_res[2,]$hpo_name)` (`r hepatocyte_res[2,]$summary`) both of which are well-known to involve the liver [@Al-Hamoudi2009-le; @Brewer2018-dg; @Eshchar1973-tz].

```{r fig-rni, fig.height=11, fig.width=11}
#| label: fig-rni
#| fig-cap: Hepatoblasts have a unique role in recurrent Neisserial infections. Significant phenotype-cell type tests for phenotypes within the branch 'Recurrent bacterial infections'. Amongst all different kinds of recurrent bacterial infections, hepatoblasts (highlighted by vertical dotted lines) are exclusively enriched in 'Recurrent gram−negative bacterial infections'. Note that terms from multiple levels of the same ontology branch are shown as separate facets (e.g. 'Recurrent bacterial infections' and 'Recurrent gram−negative bacterial infections').

infections_out$plot + ggplot2::guides(fill=ggplot2::guide_legend(ncol=2))
```

```{r prioritise_targets_network-RNI}
## Annotate results with disease/symptom-level and gene-level information
## filtering q-values at this step yields the same results as filtering at the next step, 
## albeit with much fast computation.
results_annot <- HPOExplorer::add_genes(results[q<0.05],
                                        allow.cartesian = TRUE)
results_annot <- MSTExplorer::add_symptom_results(results = results_annot, 
                                                  ctd_list = ctd_list)
## Plot multi-scale mechanisms as an interactive network
phenotype <- "Recurrent Neisserial infections"
vn_rni <- MSTExplorer::prioritise_targets_network(
  top_targets = results_annot[hpo_name==phenotype], 
  main = NULL, 
  height = "400px",
  width = "500px",
  submain = NULL)
```

Next, we sought to link multi-scale mechanisms at the levels of disease, phenotype, cell type, and gene and visualise these as a network ([Fig. @fig-network-rni]).
This revealed that genetic deficiencies in different complement system genes (*C5*, *C8*, and *C7*) are primarily mediated by different cell types (hepatoblasts, stratified epithelial cells, and stromal cells, respectively).
While genes of the complement system are expressed throughout many different tissues and cell types, these results indicate that different subsets of these genes may mediate their effects through different cell types.
This finding suggests that investigating (during diagnosis) and targeting (during treatment) different cell types may be critical for the diagnosis and treatment of these closely related, yet mechanistically distinct, diseases.

```{r fig-network-rni, fig.height=5, fig.width=7, results='asis'}
#| label: fig-network-rni
#| fig-cap: Multi-scale mechanisms of Recurrent Neisserial infections. Starting from the bottom of the plot, one can trace how causal genes (yellow boxes) mediate their effects through cell types (orange circles), phenotypes (pruple cylinders) and ultimately diseases (blue cylinders). Cell types are connected to phenotypes via association testing ($FDR_{p,c}<0.05$), and to diseases when the symptom gene set overlap is >25%. Nodes were spatially arranged using the Sugiyama algorithm [@Sugiyama1981-ev].  

vn_rni$plot
```

### Monarch Knowledge Graph recall

Next, we used the Monarch Knowledge Graph (MKG) as a proxy for the field's current state of knowledge of phenotype-cell type associations.
We evaluated the proportion of MKG associations that were recapitulation by our results.
In total, our results contained at least one significant cell type associations for \>`r format(validate_associations_mkg_out$proportion_phenotypes_captured*100,digits=1)`% of the phenotypes described in the MKG.
Of these phenotypes, we captured \>45% of the MKG phenotype-cell associations when only considering exact overlap of CL-aligned cell type annotations.
This proportion increased with greater flexibility in the matching of cell type annotations, reaching a maximum of `r X`% at a ontology graph distance of `r Y` when considering the overlap of cell type annotations at the level of cell type ontology terms.
This suggests that our results are in line with the current state of knowledge, and that our approach can be used to identify novel phenotype-cell type associations.

### Annotation of phenotypes using generative large language models

```{r gpt_annot_codify}
gpt_check <- HPOExplorer::gpt_annot_check()
gpt_annot <- HPOExplorer::gpt_annot_codify()

gpt_annot$annot_weighted[,hpo_name:=gsub("^obsolete ","",hpo_name)]
least_severe_phenotype <- gpt_annot$annot_weighted[hpo_name=="Thin toenail" & severity_score_gpt==0,]
```

Severity annotations were gathered from GPT-4 for `r length(unique(gpt_check$annot$hpo_id))`/`r hpo@n_terms` (`r format(length(unique(gpt_check$annot$hpo_id))/hpo@n_terms*100,2)`%) HPO phenotypes.
In our companion study, benchmarking tests of these results using ground-truth HPO branch annotations.
For example, phenotypes within the 'Blindness' HPO branch (*HP:0000618*) were correctly annotated as causing blindness by GPT-4.
Across all annotations, the recall rate of GPT-4 annotations was `r format(mean(gpt_check$true_pos_rate)*100,digits=4)`% (min=`r format(min(gpt_check$true_pos_rate)*100,digits=4)`%, max=`r format(max(gpt_check$true_pos_rate)*100,digits=4)`%, SD=`r format(sd(gpt_check$true_pos_rate)*100,digits=4)`) with a mean consistency score of `r format(mean(gpt_check$consistency_rate)*100,digits=4)`% (min=`r format(min(gpt_check$consistency_rate)*100,digits=4)`%, max=`r format(max(gpt_check$consistency_rate)*100,digits=4)`%, SD=`r format(sd(gpt_check$consistency_rate)*100,digits=4)`) for phenotypes whose annotation were collected more than once.
This clearly demonstrates the ability of GPT-4 to accurately annotate phenotypes.
This allowed us to begin using these annotations to compute systematically collected severity scores for all phenotypes in the HPO.

From these annotations we computed a weighted severity score metric for each phenotype ranging from 0-100 (100 being the theoretical maximum severity of a phenotype that always causes every annotation).
Within our annotations, the most severe phenotype was `r shQuote(gpt_annot$annot_weighted[1,]$hpo_name)` (*`r gpt_annot$annot_weighted[1,]$hpo_id`*) with a severity score of `r format(gpt_annot$annot_weighted[1,]$severity_score_gpt,digits=4)`, followed by `r shQuote(gpt_annot$annot_weighted[2,]$hpo_name)` (*`r gpt_annot$annot_weighted[2,]$hpo_id`*) with a severity score of `r format(gpt_annot$annot_weighted[2,]$severity_score_gpt,digits=4)`.
There were `r data.table::uniqueN(gpt_annot$annot_weighted[severity_score_gpt==0]$hpo_name)` phenotypes with a severity score of 0 (e.g. `r paste(shQuote(least_severe_phenotype$hpo_name),collapse=", ")`).
The mean severity score across all phenotypes was `r format(mean(gpt_annot$annot_weighted$severity_score_gpt),digits=4)` (median=`r format(median(gpt_annot$annot_weighted$severity_score_gpt),digits=4)`, standard deviation=`r format(sd(gpt_annot$annot_weighted$severity_score_gpt),digits=4)`).

### Enrichment of foetal cell types in congenital phenotypes

```{r plot_congenital_annotations}
plot_congenital_annotations_out <- MSTExplorer::plot_congenital_annotations(
  results = results)

plot_congenital_annotations_out_data <- 
  data.table::data.table(plot_congenital_annotations_out$data, key="congenital_onset")
plot_congenital_annotations_out_data[,summary:=paste0(
  shQuote(congenital_onset),"=",.label,
  " (n=",counts," associations)"
)]
plot_congenital_annotations_out_stats <- data.table::data.table(
  plot_congenital_annotations_out$data_stats$summary_data
  )[,summary:=paste0(
  "$",
  "p=",format(p.value,digits=2),",",
  "\\chi^2_{Pearson}=",format(statistic,digits=2),",",
  "\\hat{V}_{Cramer}=",format(estimate,digits=2),
  "$"
)]
```

The frequency of congenital onset with each phenotype (as determined by GPT-4 annotations) was strongly predictive with the proportion of significantly associated foetal cell types in our results (`r plot_congenital_annotations_out_stats$summary`).
Furthermore, increasing congenital frequency annotation (on an ordinal scale) corresponded to an increase in the proportion of foetal cell types: `r paste(plot_congenital_annotations_out_data[fetal_celltype==TRUE]$summary,collapse=", ")`.
This is consistent with the expected role of foetal cell types in development and the aetiology of congenital disorders.

```{r fig-congenital, fig.height=6, fig.width=7}
#| label: fig-congenital
#| fig-cap: Congenital phenotypes are more often associated with foetal cell types. As a phenotype is more often congenital in nature, the greater proportion of foetal cell types are significantly asscoaited with it.

plot_congenital_annotations_out$plot
```

### Diagnosis via cell type-specific disease prediction

```{r predict_celltypes}
## Define input phenotypes/genes
phenotypes_diagnose <- c("Generalized neonatal hypotonia",
                         "Scrotal hypospadias",
                         "Increased circulating progesterone")
phenotypes_diagnose <- HPOExplorer::map_phenotypes(phenotypes_diagnose,
                                                   to="id",
                                                   hpo=hpo)
genes_include <- c("HSD3B2","HERC2")
genes_exclude <- c("SNORD115-1")
## Predict cell types
predict_celltypes_out <- MSTExplorer::predict_celltypes(
  phenotypes = names(phenotypes_diagnose),
  genes_include = genes_include, 
  genes_exclude = genes_exclude,
  phenotype_to_genes = p2g,
  show_plot = FALSE)
```

Using the function `MSTExplorer::predict_celltypes` we input `r length(phenotypes_diagnose)` inclusion phenotypes (`r paste(shQuote(names(phenotypes_diagnose)),paste0("(*",phenotypes_diagnose,"*)"),collapse = ", ")`), `r length(genes_include)` genes in which the patient is known to have deleterious mutations (`r paste0("*",genes_include,"*", collapse = ", ")`) and `r length(genes_exclude)` gene in which the patient is known not to have any deleterious mutations (`r paste0("*",genes_exclude,"*", collapse = ", ")`). This predicted that `r predict_celltypes_out$data_agg$cl_name[1]` (score sum=`r format(predict_celltypes_out$data_agg[1,]$score_sum,digits = 3)`, score mean=`r format(predict_celltypes_out$data_agg[1,]$score_mean,digits = 3)`, score standard deviation=`r format(predict_celltypes_out$data_agg[1,]$score_sd,digits = 3)`) were the most probable cell types underlying this combination of phenotypes and genotypes ([Fig. @fig-diagnosis]), which is highly consistent with existing evidence that adrenal insufficiency can cause both phenotypes via mutations in these genes [@Srivastava2023-ge; @Utsch2004-re]. This was the only cell type to receive a score two standard deviations from the mean score of all cell types (mean score: `r format(mean(predict_celltypes_out$data$score_mean),digits = 3)`).

```{r fig-diagnosis, fig.height=5, fig.width=7}
#| label: fig-diagnosis
#| fig-cap: Diagnosis - Observed phenotypes/genotypes can be used to identify causal cell types in individuals. Our phenotype-cell type association results can be used to make predictions about which cell types are underlying a set of phenotypes observed in a given patient. Here we input three inclusion phenotypes, two inclusion genes, and one exclusion gene into the function `MSTExplorer::predict_celltypes`. The output is a ranked list of the top 10 most probable cell types (*x-axis*) underlying this combination of phenotypes/genotypes (highest to lowest rank from left to right). The score on the *y-axis* is computed by aggregating phenotype-celltype association summary statistics and evidence-weighted phenotype-gene associations. In this simple example, cortical cells of the adrenal gland were predicted as the most probable cell type. The mean of the score sum is shown as a dashed line, while one standard deviation (SD) above this is shown as a dotted line. Each bar is coloured by its mean.

predict_celltypes_out$plot
```

### Prognosis via cell type-mediated differential outcomes

```{r plot_differential_outcomes_heatmap, fig.height=8, fig.width=12}
## Count number of diseases associated with these phenotypes
keep_descendants <- "Hypotonia" ## HP:0001252
hypotonia_results <- HPOExplorer::filter_descendants(results, 
                                                     keep_descendants = keep_descendants) 
hypotonia_results <- HPOExplorer::add_death(hypotonia_results,
                                            allow.cartesian = TRUE,
                                            agg_by = c("disease_id","hpo_id"))
hypotonia_results <- MSTExplorer::map_celltype(hypotonia_results) 
## Generate plot
plot_differential_outcomes_heatmap_out <- MSTExplorer::plot_differential_outcomes_heatmap( 
  results = hypotonia_results, 
  print_phenotypes = TRUE,
  fill_limits = c(1,8),
  show_plot = FALSE)
min_ageofdeath_score_celltypes <- unique(plot_differential_outcomes_heatmap_out$data_agg[which(AgeOfDeath_score_min==min(AgeOfDeath_score_min))]$celltype_symptom)
```

Hypotonia (*HP:0001252*) is a very broad phenotype containing `r length(unique(hypotonia_results$hpo_id))-1` subterms (e.g. "Generalised neonatal hypotonia") and is associated with `r length(unique(hypotonia_results$disease_id))` unique diseases in the HPO gene annotations.
Together, these hypotonia phenotypes were significantly associated with `r length(unique(hypotonia_results[q<0.05]$cl_name))`/`r length(unique(results$cl_name))` (`r format(length(unique(hypotonia_results[q<0.05]$cl_name))/length(unique(results$cl_name))*100, digits=4)`%) unique CL-aligned cell types.
This reflects the highly variable set of disease etiologies that can cause this broad-level phenotype.
Across all diseases, hypotonia phenotypes tended to be most consistently severe (lower mean age of death score) when associated with the cell type `r plot_differential_outcomes_heatmap_out$data_agg[1,]$celltype_symptom`.
While other cell types were associated with lower mean age of death scores (e.g. `r paste(min_ageofdeath_score_celltypes[seq(2)],collapse=", ")`), the severity of the outcomes were more variable.

```{r fig-prognosis, fig.height=8, fig.width=12}
#| label: fig-prognosis
#| fig-cap: Prognosis - Cell types predict the probability of deadly diseases. The broad phenotype 'Hypotonia' and its descendants occur in many different diseases (1,832 diseases in the HPO annotations).Therefore, it can be difficult to prognose clinical outcomes of a newborn individual with hypotonia. With additional knowledge of the particular cell types underlying a patient's hypotonia phenotype, one can greatly narrow down the range of potential outcomes (e.g. age of death). **a**, Here, we show the various cell types by which hypotonia phenotypes confer disease risk. **b**, We also computed the mean age of death score for each cell type across hypotonia-associated diseases, revealing that disrupted inhibitory neurons confer the greatest risk of early death. Ordinal age of death categories from the HPO disease annotations were encoded numerically and averaged ([Table @tbl-death]) to produce mean Age of Death scores for each disease (on a scale from 1-8). For example, a score of 1 corresponds to prenatal death, while a score of 8 corresponds to death in late adulthood.

plot_differential_outcomes_heatmap_out$plot
```

### Therapeutic target identification

```{r prioritise_targets}
prioritise_targets_out <- MSTExplorer::prioritise_targets(
  results = results, 
  ctd_list = ctd_list,
  phenotype_to_genes = p2g,
  hpo = hpo,
  
  keep_deaths=NULL,
  keep_onsets=NULL,
  keep_specificity_quantiles = seq(30,40), ## NULL:70, 30-40:64 
  keep_mean_exp_quantiles = seq(30,40), ## NULL:65, 10:55
  info_content_threshold=8, ## 8:55, 5:64 
  effect_threshold=NULL, ## 1:39
  severity_score_gpt_threshold=NULL, ## 10:78, NULL:82
  symptom_intersection_threshold=.25, ## .25:57
  evidence_score_threshold=3,  ## 5:47, 4:47, 3:64
  top_n = 10, ## 5:38, 20:42, 30:45, 40:52, 50:55
  group_vars = "hpo_id")
 
genes_per_pheno <- prioritise_targets_out$top_targets[,list(n=data.table::uniqueN(gene_symbol)),by="hpo_id"]
p2g <- HPOExplorer::add_ont_lvl(p2g)
min_ont_lvl <- 3
genes_per_pheno_all <- p2g[ontLvl>min_ont_lvl,
                           list(n=data.table::uniqueN(gene_symbol)),by="hpo_id"]


top_celltypes <- prioritise_targets_out$top_targets[,list(
  np=data.table::uniqueN(hpo_id)),
  by="cl_name"]|>data.table::setorderv("np",-1)
top_ancestors <- prioritise_targets_out$top_targets[,list(
  np=data.table::uniqueN(hpo_id),
  nc=data.table::uniqueN(CellType),
  ng=data.table::uniqueN(gene_symbol)
  ),
  by="ancestor_name"]|>
  data.table::setorderv("np",-1)
```

Next, we identified putative cell type-specific gene targets for several severe disease phenotypes.
This yielded putative therapeutic targets for `r prioritise_targets_out$report[step=="end"][["Phenotypes"]]` phenotypes across `r prioritise_targets_out$report[step=="end"][["Diseases"]]` diseases in `r prioritise_targets_out$report[step=="end"][["Cell types"]]` cell types and `r prioritise_targets_out$report[step=="end"][["Genes"]]` genes ([Fig. @fig-therapy-filter]).
While this constitutes a large number of genes in total, each phenotype was assigned a median of `r median(genes_per_pheno$n)` gene targets (mean=`r format(mean(genes_per_pheno$n),digits=3)`, min=`r min(genes_per_pheno$n)`, max=`r max(genes_per_pheno$n)`).
Relative to the number of genes annotations per phenotype in the HPO overall (median=`r median(genes_per_pheno_all$n)`, mean=`r format(mean(genes_per_pheno_all$n),digits=4)`, min=`r min(genes_per_pheno_all$n)`, max=`r max(per_phenotype$ng)`) this represents a substantial decrease in the number of candidate target genes, even when excluding high-level phenotypes (HPO level\>`r min_ont_lvl`).
It is also important to note that the phenotypes in the prioritised targets list are ranked by their severity, allowing us to distinguish between phenotypes with a high medical urgency (e.g. `r shQuote(unique(prioritise_targets_out$top_targets[order(severity_score_gpt, decreasing = TRUE)]$hpo_name)[1])`) from those with lower medical urgency (e.g. `r shQuote(unique(prioritise_targets_out$top_targets[order(severity_score_gpt)]$hpo_name)[1])`).
This can be useful for both clinicians, biomedical scientists, and pharmaceutical manufacturers who wish to focus their research efforts on phenotypes with the greatest need for intervention.

Across all phenotypes, `r top_celltypes[1,]$cl_name` were most commonly implicated (`r top_celltypes[1,]$np` phenotypes), followed by `r top_celltypes[2,]$cl_name` (`r top_celltypes[2,]$np` phenotypes), `r top_celltypes[2,]$cl_name` (`r top_celltypes[2,]$np` phenotypes), `r top_celltypes[3,]$cl_name` (`r top_celltypes[3,]$np` phenotypes), `r top_celltypes[4,]$cl_name` (`r top_celltypes[4,]$np` phenotypes), and `r top_celltypes[5,]$cl_name` (`r top_celltypes[5,]$np` phenotypes).
Grouped by higher-order ontology category, `r shQuote(top_ancestors[1,]$ancestor_name)` had the greatest number of enriched phenotypes (`r top_ancestors[1,]$np` phenotypes, `r top_ancestors[1,]$ng` genes), followed by `r shQuote(top_ancestors[2,]$ancestor_name)` (`r top_ancestors[2,]$np` phenotypes, `r top_ancestors[2,]$ng` genes), `r shQuote(top_ancestors[3,]$ancestor_name)` (`r top_ancestors[3,]$np` phenotypes, `r top_ancestors[3,]$ng` genes), `r shQuote(top_ancestors[4,]$ancestor_name)` (`r top_ancestors[4,]$np` phenotypes, `r top_ancestors[4,]$ng` genes), and `r shQuote(top_ancestors[5,]$ancestor_name)` (`r top_ancestors[5,]$np` phenotypes, `r top_ancestors[5,]$ng` genes).

### Therapeutic target validation

```{r ttd_check, fig.height=10, fig.width=10}
## Gene therapy only
ttd_check_out <- MSTExplorer::ttd_check(
  top_targets=prioritise_targets_out$top_targets, 
  drug_types = "Gene therapy",
  allow.cartesian = TRUE,
  show_plot = FALSE)

## All therapy types
ttd_check_all_out <- MSTExplorer::ttd_check(
  top_targets=prioritise_targets_out$top_targets,   
  allow.cartesian = TRUE,
  show_plot = FALSE)

```

To determine whether the genes prioritised by our therapeutic targets pipeline were plausible, we checked what percentage of gene therapy targets we recapitulated.
Data on therapeutic approval status was gathered from the Therapeutic Target Database (TTD; release `r Sys.Date()`) [@Liu2011-qd].
Overall, we prioritised `r format(sum(ttd_check_out$data[failed==FALSE]$prioritised)/nrow(ttd_check_out$data[failed==FALSE])*100,digits=1)`% of all non-failed existing gene therapy targets.
A hypergeometric test confirmed that our prioritised targets were significantly enriched for non-failed gene therapy targets (`r paste0("$p=",format(ttd_check_out$ttd_hypergeo_out$nonfailed_enrichment,digits=3),"$")`).
Importantly, we did not prioritise any of the failed therapeutics (`r format(sum(ttd_check_out$data[failed==TRUE]$prioritised)/nrow(ttd_check_out$data[failed==TRUE])*100, digits=1)`%), defined as having been terminated or withdrawn from the market.
The hypergeometric test for depletion of failed targets did not reach significance (`r paste0("$p=",format(ttd_check_out$ttd_hypergeo_out$failed_depletion,digits=3),"$")`), but this is to be expected as there was only one failed gene therapy target in the TTD database.

Even when considering therapeutics of any kind ([Fig. @fig-therapy-validate-all]), not just gene therapies, we recapitulated `r format(sum(ttd_check_all_out$data[failed==FALSE]$prioritised)/nrow(ttd_check_all_out$data[failed==FALSE])*100,digits=1)`% of the non-failed therapeutic targets and `r format(sum(ttd_check_all_out$data[failed==TRUE]$prioritised)/nrow(ttd_check_all_out$data[failed==TRUE])*100, digits=1)`% of the terminated/withdrawn therapeutic targets (n=`r nrow(ttd_check_all_out$data[failed==TRUE])`).
Here we found that our prioritised targets were significantly enriched for non-failed therapeutics (`r paste0("$p=",format(ttd_check_all_out$ttd_hypergeo_out$nonfailed_enrichment,digits=1),"$")`), and highly significantly depleted for failed therapeutics (`r paste0("$p=",format(ttd_check_all_out$ttd_hypergeo_out$failed_depletion,digits=1),"$")`).
This suggests that our multi-scale evidence-based prioritisation pipeline is capable of selectively identifying genes that are likely to be effective therapeutic targets.

```{r fig-therapy-validate, fig.height=5, fig.width=8}
#| label: fig-therapy-validate
#| fig-cap: Therapeutics - Validation of prioritised therapeutic targets. The proportion of existing gene therapy targets (documented in the Therapeutic Target Database) recapitulated by our prioritisation pipeline. Therapetics are stratified by the stage of clinical development they were at during the time of writing.

ttd_check_out$plot
```

### Selected example targets

```{r}
top_targets <- prioritise_targets_out$top_targets[,n_genes:=data.table::uniqueN(gene_symbol),
                                                  by="hpo_id"][n_genes<5 & proportion_driver_genes_symptom>.25]

top_phenotypes <- unique(top_targets$hpo_name)
```

```{r prioritise_targets_network-examples} 
height <- "80vh"
width <- "100vw"
phenotypes_network <- c("Lethal skeletal dysplasia",
                        "GM2-ganglioside accumulation",
                        "Alzheimer disease",
                        "Parkinson disease")

phenotype <- phenotypes_network[1]
vn_therapy_eg1 <- MSTExplorer::prioritise_targets_network(
  top_targets = top_targets[hpo_name==phenotype], 
  main = NULL,
  height = height,
  width = width,
  submain = NULL)

phenotype <- phenotypes_network[2]
vn_therapy_eg2 <- MSTExplorer::prioritise_targets_network(
  top_targets = top_targets[hpo_name==phenotype], 
  main = NULL,
  height = height,
  width = width,
  submain = NULL)

phenotype <- phenotypes_network[3]
vn_therapy_eg3 <- MSTExplorer::prioritise_targets_network(
  top_targets = top_targets[grepl(paste(phenotype,collapse = "|"), disease_name,ignore.case = TRUE)], 
  main = NULL, 
  height = height,
  width = width,
  submain = NULL)

phenotype <- phenotypes_network[4]
vn_therapy_eg4 <- MSTExplorer::prioritise_targets_network(
  top_targets = top_targets[grepl(paste(phenotype,collapse = "|"), disease_name,ignore.case = TRUE)], 
  main = NULL, 
  height = height,
  width = width,
  submain = NULL)
```


```{r fig-therapy-examples, fig.height=7, fig.width=12, results='asis'}
#| label: fig-therapy-examples
#| fig-cap: Example cell type-specific gene therapy targets for several severe phenotypes and their associated diseases. Each disease (blue cylinders) is connected to its phenotype (purple cylinders) based on well-established clinical observations recorded within the HPO [@Gargano2024-fc]. Phenotypes are connected to cell types (red circles) via association testing between weighted gene sets ($FDR_{p,c}<0.05$). Each cell type is connected to the prioritised gene targets (yellow boxes) based on the driver gene analysis.The thickness of the edges connecting the nodes represent the (mean) fold-change from the bootstrapped enrichment tests. Nodes were spatially arranged using the Sugiyama algorithm [@Sugiyama1981-ev].
#| layout: "[[1,1], [1], [1]]" 
#| fig-subcap:
#| - 'Lethal skeletal dysplasia'
#| - 'GM2-ganglioside accumulation'
#| - 'Alzheimer disease'
#| - 'Parkinson disease'

vn_therapy_eg1$plot
vn_therapy_eg2$plot
vn_therapy_eg3$plot
vn_therapy_eg4$plot
```

From our prioritised targets, we selected the following four sets of phenotypes or diseases as examples: `r paste(shQuote(phenotypes_network), collapse=", ")`.

Skeletal dysplasia is a heterogeneous group of over 450 disorders that affect the growth and development of bone and cartilage.
This phenotype can be lethal when deficient bone growth leads to the constriction of vital organs such as the lungs.
Even after surgical interventions, these complications continue to arise as the child develops.
Pharmacological interventions to treat this condition have largely been ineffective.
While there are various cell types involved in skeletal system development, our pipeline nominated chondrocytes as the causal cell type underlying the lethal form of this condition.
Assuringly, we found that the disease 'Achondrogenesis Type 1B' is caused by the genes *SLC26A2* and *COL2A1* via chondrocytes.
We also found that 'Platyspondylic lethal skeletal dysplasia, Torrance type'.
Thus, in cases where surgical intervention is insufficient, targeting these genes within chondrocytes may prove a viable long-term solution for children suffering from lethal skeletal dysplasia.

Tay-Sachs disease is a devastating disease in which children are born appearing healthy, which gradually degrades leading to death after 3-5 years.
The underlying cause is the toxic accumulation of gangliosides in the nervous system due to a loss of the enzyme produced by *HEXA*.
While this could in theory be corrected with gene editing technologies, there remain some outstanding challenges.
One of which is early detection and diagnosis, before irreversible damage has occurred.
Our pipeline implicated extravillous trophoblasts of the placenta in 'GM2-ganglioside accumulation'.
While not necessarily a target for gene therapy, checking these cells *in utero* for an absence of *HEXA* may serve as a viable biomarker as these cells normally express the gene at high levels.
Early detection of Tay-Sachs disease may lengthen the window of opportunity for therapeutic intervention, especially when genetic sequencing is not available or variants of unknown significance are found within *HEXA*.

Alzheimer disease (AD) is the most common neurodegenerative condition. It is characterised by a set of variably penetrant phenotypes including memory loss, cognitive decline, cerebral proteinopathy. Interestingly, we found that different forms of early onset AD (which are defined by the presence of a specific disease gene) are each associated with different cell types via different phenotypes. For example, AD 3 and AD 4 are primarily associated with cells of the digestive system (`r paste(shQuote(unique(top_targets[disease_name=="Alzheimer disease 4"]$cl_name)), collapse=", ")`) and are implied to be responsible for the phenotypes `r paste(shQuote(unique(top_targets[disease_name=="Alzheimer disease 4"]$hpo_name)), collapse=", ")`. Meanwhile, early-onset autosomal dominant
AD and AD 2 are primarily associated with immune cells (`r paste(shQuote(unique(top_targets[disease_name %in% c("Early-onset autosomal dominant Alzheimer disease","Alzheimer disease 2")]$cl_name)[1:2]), collapse=", ")`) and are implied to be responsible for the phenotypes `r paste(shQuote(unique(top_targets[disease_name %in% c("Early-onset autosomal dominant Alzheimer disease","Alzheimer disease 2")]$hpo_name)), collapse=", ")`. This suggests that different forms of AD may be driven by different cell types and phenotypes, which may help explain its variability in onset and clinical presentation. 

Finally, Parkinson disease (PD) is characterised by motor symptoms such as tremor, rigidity, and bradykinesia. However there are a number of additional phenotypes associated with the disease that span multiple physiological systems. PD 19a and PD 8 seemed to align most closely with the canonical understanding of PD as a disease of the central nervous system in that they implicated oligodendrocytes and neurons. Though the reference datasets being used in this study were not annotated at sufficient resolution to distinguish between different subtypes of neurons, in particular dopaminergic neurons. PD 19a/8 also suggested that risk variants in *LRRK2* mediate their effects on PD through both myeloid cells and oligodendrocytes by causing gliosis of the substantia nigra. The remaining clusters of PD mechanisms revolved around chondrocytes (PD 20), amacrine cells of the eye (hereditary late-onset PD), and the respiratory/immune system (PD 14). While the diversity in cell type-specific mechanisms is somewhat surprising, it may help to explain the wide variety of cross-system phenotypes frequently observed in PD.

It should be noted that the HPO only includes gene annotations for the monogenic forms of AD and PD. However it has previously been shown that there is at least partial overlap in their phenotypic and genetic aetiology with respect to their common forms. Thus understanding the monogenic forms of these diseases may shed light onto their more common counterparts.

### Experimental model translatability

```{r map_upheno_data}
pheno_map_genes_match <- KGExplorer::map_upheno_data()
pheno_map_targets <- pheno_map_genes_match[
  id1 %in% unique(prioritise_targets_out$top_targets$hpo_id)
  ]|>
  data.table::setorderv("phenotype_genotype_score",-1)
taxa_count <- sort(table(pheno_map_targets$gene_taxon_label2), decreasing = TRUE)

pheno_map_targets_severe <- pheno_map_targets[
  id1 %in% unique(prioritise_targets_out$top_targets[severity_score_gpt>10,]$hpo_id)
  ]

pheno_map_targets_severe[,summary:=paste0(
  shQuote(object_label1), 
  " ($SIM_{o,g}=",round(phenotype_genotype_score,3),"$)"
  )]
```

We computed interspecies translatability scores using a combination of both ontological ($SIM_{o}$) and genotypic ($SIM_{g}$) similarity relative to each homologous human phenotype and its associated genes [Fig. @fig-animal-models]. In total, we mapped `r length(unique(pheno_map_genes_match$id2))` non-human phenotypes (in `r paste0("*",sort(unique(pheno_map_genes_match$gene_taxon_label2)),"*",collapse=", ")`) to `r length(unique(pheno_map_genes_match$id1))` homologous human phenotypes. Amongst the `r length(unique(prioritise_targets_out$top_targets$hpo_id))` phenotype within our prioritised therapy targets, `r length(intersect(prioritise_targets_out$top_targets$hpo_id, pheno_map_genes_match$id1))` had viable animal models in at least on non-human species. Per species, the number of homologous phenotypes was: `r paste0("*",names(taxa_count),"*"," (n=",taxa_count,")")`. Amongst our prioritised targets with a GPT-4 severity score of >10, the phenotypes with the greatest animal model similarity were `r paste(pheno_map_targets_severe$summary[1:5],collapse = ", ")`.

## Discussion {#sec-discussion}

Across the `r res_summ_all[["cell types"]]` cell types and `r res_summ_all[["phenotypes"]]` RD-associated phenotypes investigated, more than `r res_summ_all[["tests significant"]]` significant phenotype-cell type relationships were discovered.
The examples we have highlighted above recapitulate well-known relationships, provide additional cellular context to many of these known relationships, and discover novel relationships at multiple biological scales.

Investigating RDs at the level of phenotypes offers several key advantages.
First, the vast majority of RDs only have one associated gene (`r nrow(per_disease[ng==1])`/`r nrow(per_disease)` diseases = `r format(nrow(per_disease[ng==1])/nrow(per_disease)*100,digits=1)`%).
Aggregating gene sets across diseases into phenotype-centric "buckets" permits sufficiently well-powered analyses, with an average of \~`r format(mean(per_phenotype$ng), digits=1)` genes per phenotype (median=`r median(per_phenotype$ng)`) [see Fig. @fig-diagram].
Second, we hypothesise that these phenotype-level gene sets converge on a limited number of molecular and cellular pathways.
Perturbations to these pathways manifest as one or more phenotypes which, when considered together, tend to be clinically diagnosed as a certain disease.
Third, RDs are often highly heterogeneous in their clinical presentation across individuals, leading to the creation of an ever increasing number of disease subtypes (some of which only have a single documented case).
In contrast, a phenotype-centric approach enables us to more accurately describe a particular individual’s version of a disease without relying on the generation of additional disease subcategories.
By characterising an individual’s precise phenotypes over time, we may better understand the underlying biological mechanisms that have caused their condition.
However, in order to achieve a truly precision-based approach to clinical care, we must first characterise the molecular and cellular mechanisms that cause the emergence of each phenotype.
Here, we provide a highly reproducible framework that enables this at the scale of the entire phenome.
This presents an opportunity to design basket trials of patients with different diseases but overlapping phenotypes and cellular mechanisms [@Zanello2023-zd].
It may be especially helpful for complex patients with diagnostically ambiguous sets of phenotypes who would otherwise be excluded from traditional clinical trials [@Diaz-Santiago2020-ep].

It was paramount to the success of this study to ensure our results were anchored in ground-truth benchmarks, generated falsifiable hypotheses, and rigorously guarded against false-positive associations.
Extensive validation using multiple approaches demonstrated that our methodology consistently recapitulates expected phenotype-cell type associations ([Fig. @fig-summary]-[Fig. @fig-congenital]).
This was made possible by the existence of comprehensive, structured ontologies for all phenotypes (HPO) and cell types (CL), which provide an abundance of clear and falsifiable hypotheses for which to test our predictions against.
Several key examples include 1) strong enrichment of associations between cell types and phenotypes within the same anatomical systems ([Fig. @fig-summary]b-d), 2) a strong relationship between phenotype-specificity and the strength and number of cell type associations ([Fig. @fig-ontology-lvl]), 3) identification of the precise cell subtypes involved in susceptibility to various subtypes of recurrent bacterial infections ([Fig. @fig-rni]), 4) a strong positive correlation between the frequency of congenital onset of a phenotype and the proportion of developmental cell types associated with it ([Fig. @fig-congenital])), and 5) consistent phenotype-cell type associations across multiple independent single-cell datasets ([Fig. @fig-ctd-correlation]).
Having validated our phenotype-cell type associations, we then went on to demonstrate how these results may be used in each stage of clinical care: diagnosis ([Fig. @fig-diagnosis]), prognosis ([Fig. @fig-prognosis]), treatment, and therapeutics development ([Fig. @fig-therapy-examples]).

Diagnosis is an essential but challenging step in RD patient care.
Additional phenotypes that emerge over time may assist a clinician to reach a more confident disease diagnosis.
However many of these phenotypes can have a serious impact on patient quality of life or survival and avoiding them would be far better for patient outcomes.
Often times phenotypes alone cannot clearly pinpoint the disease and thus a diagnosis is never reached.
Having a more complete understanding of the mechanisms underlying observed phenotypes allows clinicians to far more effectively make predictions about what additional, less obvious phenotypes they should search for to confirm or reject their hypothesis of disease diagnosis (e.g. with imaging or biomarker tests).

Consider the following hypothetical scenario.
A clinician observes that a newborn patient has several phenotypes (`r paste(shQuote(names(phenotypes_diagnose)),collapse = ", ")`), none of which conclusively point to a single disease diagnosis.
Under the strong suspicion that the phenotypes are genetic in origin, the clinician orders whole-genome sequencing (WGS) on the patient as well as the patient’s family.
The clinician finds that the patient has a number of putative causal mutations, narrowing down the number of potential diseases from hundreds to just 10.
Further narrowing down the possibilities at this stage can be extremely challenging even for expert clinical geneticists.
However, additional knowledge of which tissues and cell types are primarily affected allow the clinician to make a series of testable hypotheses that they may begin to investigate.
For example, two of the putative diseases are known to cause aberrant splicing events in a gene that is only expressed in adrenocortical cells ([Fig. @fig-diagnosis]), providing justification to order a needle biopsy of the adrenal gland.
RNA sequencing is performed on the tissue biopsy and it is discovered that the patient does indeed have high expression of the dysfunctional transcript, confirming the disease diagnosis [@Lord2021-rf].
This opens new avenues for the patient to receive timely and effective treatments for their specific condition, which is important as their version of the disease tends to lead to death in early childhood if left untreated ([Fig. @fig-prognosis]).
Fortunately, their diagnosis now qualifies them to participate in a clinical trial of a novel gene therapy with promising preliminary results.
Furthermore, it is predicted that this patient would respond especially well to this treatment given that the mechanisms of action of the gene therapy primarily acts on adrenocortical cells ([Fig. @fig-therapy-examples]).

Unfortunately, there are currently only treatments available for less than 5% of RDs [@Halley2022-pd].
Novel technologies including CRISPR, prime editing, antisense oligonucleotides, viral vectors, and/or lipid nanoparticles, have been undergone significant advances in the last several years [@Bueren2023-ma; @Bulaklak2020-ta; @Godbout2023-uo; @Kohn2023-vh; @Zhao2023-qy] and proven remarkable clinical success in an increasing number of clinical applications [@Darrow2019-om; @Mendell2017-kg; @Mueller2017-fz; @Russell2017-dh].
The U.S. Food and Drug Administration (FDA) recently announced an landmark program aimed towards improving the international regulatory framework to take advantage of the evolving gene/cell therapy technologies @Lu2024-kl with the aim of bringing dozens more therapies to patients in a substantially shorter timeframe than traditional pharmaceutical product development (typically 5-20 years with a median of 8.3 years) [@Brown2022-ye].
While these technologies have the potential to revolutionise RD medicine, their successful application is dependent on first understanding the mechanisms causing each disease.

To address this critical gap in knowledge, we used our results to create a reproducible and customisable pipeline to nominate cell type-resolved therapeutic targets ([Fig. @fig-therapy-filter]-[Fig. @fig-therapy-examples]).
Targeting cell type-specific mechanisms underlying granular RD phenotypes can improve therapeutic effectiveness by treating the causal root of an individual's conditions [@Bulaklak2020-ta; @Moffat2017-al].
A cell type-specific approach also helps to reduce the number of harmful side effects caused by unintentionally delivering the therapeutic to off-target tissues/cell types (which may induce aberrant gene activity), especially when combined with technologies that can target cell surface antigens (e.g viral vectors) [@Zhou2013-wx].
This has the additional benefit of reducing the minimal effective dose of a therapeutic, which can be both immunogenic and extremely financially costly [@Bueren2023-ma; @Kohn2023-vh; @Nuijten2022-yc; @Thielen2022-ud].
Here, we demonstrate the utility of a high-throughput evidence-based approach to RD therapeutics discovery by highlighting several of the most promising therapeutic candidates.
Our pipeline takes into account a myriad of factors, including the strength of the phenotype-cell type associations, symptom-cell type associations, cell type-specificity of causal genes, the severity and frequency of the phenotypes, suitability for gene therapy delivery systems (e.g. recombinant adeno-associated viral vectors (rAAV)), as well as a quantitative analysis of phenotypic and genetic animal model translatability ([Fig. @fig-animal-models]).
We validated these candidates by comparing the proportional overlap with gene therapies that are presently in the market or undergoing clinical trials, in which we recovered `r format(sum(ttd_check_out$data[failed==FALSE]$prioritised)/nrow(ttd_check_out$data[failed==FALSE])*100,digits=1)`% of all active gene therapies and `r format(sum(ttd_check_out$data[failed==TRUE]$prioritised)/nrow(ttd_check_out$data[failed==TRUE])*100, digits=1)`% of failed gene therapies ([Fig. @fig-therapy-validate], [Fig. @fig-therapy-validate-all]). Despite nominating a large number of putative targets, hypergeometric tests confirmed that our targets were strongly enriched for targets of existing therapies that are either approved or currently undergoing clinical trials.

It should be noted that our study has several key limitations.
First, while our cell type datasets are amongst the most comprehensive human scRNA-seq references currently available, they are nevertheless missing certain tissues, cell types (e.g. spermatocytes, oocytes), and life stages (post-natal childhood, senility).
It is also possible that we have not captured certain cell state signatures that only occur in disease (e.g. disease-associated microglia \[**CITATION**\]).
Though we reasoned that using only control cell type signatures would mitigate bias towards any particular disease, and avoid degradation of gene signatures due to loss of function mutations.
Second, the collective knowledge of gene-phenotype and gene-disease associations is far from complete and we fully anticipate that these annotations will continue to expand and change well into the future.
It is for this reason we designed this study to be easily reproduced within a single containerised script so that we (or others) may rerun it with updated datasets at any point.
Finally, causality is notoriously difficult to prove definitively from associative testing alone, and our study is not exempt from this rule.
Despite this, there are several reasons to believe that our approach is able to better approximate causal relationships than traditional approaches.
First, we did not intentionally preselect any subset of phenotypes or cell types to investigate here.
Along with a scaling prestep during linear modelling, this means that all the results are internally consistent and can be directly compared to one another (in stark contrast to literature meta-analyses).
Furthermore, for the phenotype gene signatures we used expert-curated GenCC annotations [@DiStefano2022-ao; @DiStefano2023-np] to weight the current strength of evidence supporting a causal relationship between each gene and phenotype.
This is especially important for phenotypes with large genes lists (thousands of annotations) for which some of the relationships may be tenuous.
Within the cell type references, we deliberately chose to use specificity scores (rather than raw gene expression) as this normalisation procedure has previously been demonstrated to better distinguish between signatures of highly similar cell types/subtypes [@Skene2016-rb].

Moving forward, we are now actively seeking industry and academic partnerships to begin experimentally validating our multi-scale target predictions and exploring their potential for therapeutic translation.
Nevertheless, there are more promising therapeutic targets here than our research group could ever hope to pursue by ourselves.
In the interest of accelerating research and ensuring RD patients are able to benefit from this work as quickly as possible, we have decided to publicly release all of the results described in this study.
These can be accessed in multiple ways, including through a suite of R packages as well as a web app, the  [Rare Disease Celltyping Portal](https://neurogenomics.github.io/rare_disease_celltyping_apps/home/).
The latter allows our results to be easily queried, filtered, visualised, and downloaded without any knowledge of programming.
Through these resources we aim to make our findings useful to a wide variety of RD stakeholders including subdomain experts, clinicians, advocacy groups, and patients.

## Conclusions {#sec-conclusions}

Ultimately, our primary objective was to develop a methodology capable of generating high-throughput phenome-wide predictions while preserving the accuracy and clinical utility typically associated with more narrowly focused studies.
With the rapid advancement of gene therapy technologies, and a regulatory landscape that is evolving to better meet the needs of a large and diverse patient population, there is finally momentum to begin to realise the promise of personalised medicine.
This has especially important implications for the global RD community which has remained relatively neglected.
Here, we lay out the groundwork necessary for this watershed moment by providing a scalable, cost-effective, and fully reproducible means of resolving the multi-scale, cell-type specific mechanisms of virtually all rare diseases.

## Methods {#sec-methods}

### Human Phenotype Ontology

```{r get_gencc}
gencc <- KGExplorer::get_gencc(agg_by = NULL)
gencc_version <- KGExplorer::get_version(gencc, return_version = TRUE)
```

The latest version of the HPO (release `r KGExplorer::get_version(hpo, return_version = TRUE)`) was downloaded from the EMBL-EBI Ontology Lookup Service [@Cote2010-gp] and imported into R using the `HPOExplorer` package.
This R object was used to extract ontological relationships between phenotypes as well as to assign absolute and relative ontological levels to each phenotype.
The latest version of the HPO phenotype-to-gene mappings and phenotype annotations were downloaded from the official HPO GitHub repository and imported into R using `HPOExplorer`.
This contains lists of genes associated with phenotypes via particular diseases, formatted as three columns in a table (gene, phenotype, disease).

However, not all genes have equally strong evidence of causality with a disease or phenotype, especially when considering that the variety of resources used to generate these annotations (OMIM, Orphanet, DECIPHER) use variable methodologies (e.g. expert-curated review of the medical literature vs. automated text mining of the literature).
Therefore we imported data from the Gene Curation Coalition (GenCC) [@DiStefano2022-ao; @DiStefano2023-np], which (as of `r gencc_version`) `r nrow(gencc)` evidence scores across `r data.table::uniqueN(gencc$disease_curie)` diseases and `r data.table::uniqueN(gencc$gene_curie)` genes.
Evidence scores are defined by GenCC using a standardised ordinal rubric which we then encoded as a semi-quantitative score ranging from 0 (no evidence of disease-gene relationship) to 6 (strongest evidence of disease-gene relationship) (see @tbl-gencc).
We then summed evidence scores per disease, merged this table with the HPO disease-phenotype-gene annotation table, and then cast the data into a gene-by-phenotype matrix filled with the aggregated mean evidence score.
This can be expressed as the following equations.

Let us denote:

-   $D$ as the set of $d$ diseases.

-   $p$ as a phenotype.

-   $g$ as a gene.

The final evidence-weighted gene-by-phenotype matrix ($M_{g,p}$) can be expressed as:

::: {#eq-evidence-scores .content-hidden unless-format="html"}
![](equations/eq1.png){height="300px"}

Construction of the evidence-weighted gene-by-phenotype matrix.
:::

\
\

::: {.content-visible unless-format="html"}
```{=tex}
\begin{equation*}
  \eqnmarkbox[NavyBlue]{n1}{M_{g,p}}
  =
  \frac{
    \eqnmarkbox[Cerulean]{n3a}{\sum_{d \in D}}
    \eqnmarkbox[blue]{n4a}{R(g,p,d)} 
    \times 
    \eqnmarkbox[BlueViolet]{n5}{E(g,d)}  
  }{
    \eqnmarkbox[Cerulean]{n3b}{\sum_{d \in D}}
    \eqnmarkbox[blue]{n4b}{R(g,p,d)}
  }
\end{equation*}
\annotate[yshift=1em]{left}{n1}{Weighted gene-by-phenotype \\evidence score matrix} 
\annotate[yshift=-2em]{below,left}{n3a,n3b}{Iterate over all diseases}
\annotate[yshift=-2.5em,xshift=2.5em]{below,right}{n4a,n4b}{Binary gene-by-phenotype \\relationship matrix,\\ (1=relationship, 0=no relationship)}
\annotate[yshift=2em]{left}{n5}{Weighted gene-by-disease \\evidence score matrix}
```
:::

\

Histograms of evidence score distributions at each step in processing can be found in [Fig. @fig-evidence-histograms].

### Single-cell transcriptomic atlases

In this study, the gene by cell type specificity matrix was constructed using the Descartes Human transcriptome atlas of foetal gene expression, which contains a mixture of single-nucleus and single-cell RNA-seq data (collected with sci-RNA-seq3) [@Cao2020-qz].
This dataset contains 377,456 cells representing 77 distinct cell types across 15 tissues.
All 121 human foetal samples ranged from 72 to 129 days in estimated postconceptual age.
To independently replicate our findings, we also used the Human Cell Landscape which contains single-cell transcriptomic data (collected with microwell-seq) from embryonic, foetal, and adult human samples across 49 tissues [@Han2020-iq].

Specificity matrices were generated separately for each transcriptomic atlas using the R package `EWCE` (v1.11.3) [@Skene2016-rb].
Within each atlas, cell types were defined using the authors’ original freeform annotations in order to preserve the granularity of cell subtypes as well as incorporate expert-identified rare cell types.
Cell types were only aligned and aggregated to the level of corresponding Cell Ontology (CL) [@Diehl2016-gt] annotations afterwards when generating summary figures and performing cross-atlas analyses.
Using the original gene-by-cell count matrices from each single-cell atlas, we computed gene-by-cell type expression specificity matrices as follows.

Let us denote: $g$ as a gene, $c$ as a cell type, and $i$ as a single cell.
Genes with very no expression across any cell types were considered to be uninformative and were therefore removed from the input gene-by-cell matrix $F(g,i,c)$.

::: {#eq-ctd-filter .content-hidden unless-format="html"}
![](equations/eq2.png){height="200px"}

Filtering of the gene-by-cell expression matrices.
:::

\

::: {.content-visible unless-format="html"}
```{=tex}
\begin{equation*}
  \eqnmarkbox[purple]{f1}{F(g,i,c)}
  =
  \begin{cases}
  \eqnmarkbox[WildStrawberry]{f2}{r_{g,i}},
  \text{ }l_i = c\\0,
  \text{ }l_i \neq c 
  \end{cases}
\end{equation*}
\annotate[yshift=1em]{left}{f1}{Filtered gene-by-cell expression matrix} 
\annotate[yshift=2em]{left}{f2}{Expression of gene $g$ in cell $i$} 
```
:::

\

Next, we calculated the mean expression per cell type and normalised the resulting matrix to transform it into a gene-by-cell type expression specificity matrix ($S_{g,c}$).
In other words, each gene in each cell type had a 0-1 score where 1 indicated the gene was mostly specifically expressed in that particular cell type relative to all other cell types.
This procedure was repeated separately for each of the single-cell atlases and can be summarised as:

::: {#eq-ctd-specificity .content-hidden unless-format="html"}
![](equations/eq3.png){height="300px"}

Construction of the gene-by-cell type specificity matrix.
:::

\

::: {.content-visible unless-format="html"}
```{=tex}
\begin{equation*}
  \eqnmarkbox[orange]{s1}{S_{g,c}}
  =
  \frac{
    \eqnmarkbox[purple]{s3a}{
      \frac{
        \sum_{i=1}^{|L|} F(g,i,c)
      }{
        N_c  
      }
    } 
  }{
   \eqnmarkbox[OrangeRed]{s6}{\sum_{r=1}^{k}}(
     \eqnmarkbox[purple]{s3b}{
      \frac{
        \sum_{i=1}^{|L|} F(g,i,c)
      }{
        N_c  
      }
    } 
   ) 
  }
\end{equation*}
\annotate[yshift=1em]{left}{s1}{Gene-by-cell type specificity matrix} 
\annotate[yshift=2em]{left}{s3a,s3b}{Compute mean expression of each gene per cell type} 
\annotate{below,left}{s6}{Compute row sums of \\mean gene-by-cell type matrix}
```
:::

\

### Phenotype-cell type associations

To test for relationships between each pairwise combination of phenotype (n=`r res_summ_all[["phenotypes"]]`) and cell type (n=`r res_summ_all[["cell types"]]`) we ran a series of univariate generalised linear models implemented via the `stats::glm` function in R.
First, we filtered the gene-by-phenotype evidence score matrix ($M _{g,p}$) and the gene-by-cell type expression specificity matrix ($S _{g,c}$) to only include genes present in both matrices (n=4,949 genes in the Descartes Human analyses; n=4,653 genes in the Human Cell Landscape analyses).
Then, within each matrix any rows or columns with a sum of 0 were removed as these were uninformative data points that did not vary.
To improve interpretability of the results $\beta$ coefficient estimates across models (i.e. effect size), we performed a scaling prestep on all dependent and independent variables.
Initial tests showed that this had virtually no impact on the total number of significant results or any of the benchmarking metrics based on p-value thresholds [Fig. @fig-summary].
This scaling prestep improved our ability to rank cell types by the strength of their association with a given phenotype as determined by separate linear models.

We repeated the aforementioned procedure separately for each of the single-cell references.
Once all results were generated using both cell type references (`r res_summ_all[["tests"]]` association tests total), we applied Benjamini-Hochberg false discovery rate [@Benjamini1995-vo] (denoted as $FDR_{p,c}$) to account for multiple testing.
Of note, we applied this correction across all results at once (as opposed to each single-cell reference separately) to ensure the $FDR_{p,c}$ was stringently controlled for across all tests performed in this study.

### Symptom-cell type associations

Here we define a symptom as a phenotype as it presents within the context of the specific disease.
The features of a given symptom can be described as the subset of genes annotated to phenotype $p$ via a particular disease $d$, denoted as $G_{d,p}$ ([see Fig. @fig-diagram]).
To attribute our phenotype-level cell type enrichment signatures to specific diseases, we first identified the gene subset that was most strongly driving the phenotype-cell type association by computing the intersect of genes that were both in the phenotype annotation and within the top 25% specificity percentile for the associated cell type.
We then computed the intersect between symptom genes ($G_{d,p}$) and driver genes ($G_{p,c}$), resulting in the gene subset $G_{d \cap p \cap c}$.
Only $G_{d \cap p \cap c}$ gene sets with 25% or greater overlap with the symptom gene subset ($G_{d,p}$) were kept.
This procedure was repeated for all phenotype-cell type-disease triads, which can be summarised as follows:

::: {#eq-symptoms .content-hidden unless-format="html"}
![](equations/eq4.png){height="300px"}
:::

\

::: {.content-visible unless-format="html"}
```{=tex}
\begin{equation*}
  \frac{
     \eqnmarkbox[Chartreuse3]{g1}{|G_{d \cap p \cap c} |}
    }{
       \eqnmarkbox[Emerald]{g2}{|G_{d,p}|}} 
  \geq \eqnmarkbox[SeaGreen]{g3}{.25} 
\end{equation*}
\annotate[yshift=1em]{left}{g1}{Intersect between \\symptom genes ($G_{d,p}$) and driver genes ($G_{p,c}$)} 
\annotate[yshift=-1em]{below,left}{g2}{Symptom genes \\(i.e. genes annotated to a phenotype\\ via a specific disease)} 
\annotate[yshift=-1em]{below,right}{g3}{Minimum proportion of overlap \\between $G_{d,p,c}$ and $G_{d,p}$}
```
:::

\

### Validation of expected phenotype-cell type relationships

We first sought to confirm that our tests (across both single-cell references) were able to recover expected phenotype-cell type relationships across seven high-level branches within the HPO ([Fig. @fig-summary]), including abnormalities of the cardiovascular system, endocrine system, eye, immune system, musculoskeletal system, nervous system, and respiratory system.
Within each branch the number of significant tests in a given cell type were plotted ([Fig. @fig-summary]b).
Mappings between freeform annotations (the level at which we performed our phenotype- cell type association tests) provided by the original atlas authors and their closest CL term equivalents were provided by CellxGene [@CZI_Single-Cell_Biology_Program2023-fs].
CL terms along the *x-axis* of [Fig. @fig-summary]b were assigned colours corresponding to which HPO branch showed the greatest number of enrichments (after normalising within each branch to account for differences in scale).
The normalised colouring allows readers to quickly assess which HPO branch was most often associated with each cell type, while accounting for differences in the number of phenotypes across branches.
We then ran a series of Analysis of Variance (ANOVA) tests to determine whether (within a given branch) a given cell type was more often enriched ($FDR_{p,c}<0.05$) within that branch relative to all of the other HPO branches of an equivalent level in the ontology (including all branches not shown in [Fig. @fig-summary]b).
After applying Benjamini-Hochberg multiple testing correction [@Benjamini1995-vo] (denoted as $FDR _{b,c}$), we annotated each respective branch-by-cell type bar according to the significance (\*\*\*\* : $FDR _{b,c}<1e-04$, \*\*\* : $FDR _{b,c}<0.001$, \*\* : $FDR _{b,c}<0.01$, \* : $FDR _{b,c}<0.05$).
Cell types in [Fig. @fig-summary]a-b were ordered along the *x-axis* according to a dendrogram derived from the CL ontology ([Fig. @fig-summary]c), which provides ground-truth semantic relationships between all cell types (e.g. different neuronal subtypes are grouped together).

As an additional measure of the accuracy of our phenotype-cell types test results we identified conceptually matched branches across the HPO and the CL ([Fig. @fig-summary]d and @tbl-celltypes).
For example, 'Abnormality of the cardiovascular system' in the HPO was matched with 'cardiocytes' in the CL which includes all cell types specific to the heart.
Analogously, 'Abnormality of the nervous system' in the HPO was matched with 'neural cell' in the CL which includes all descendant subtypes of neurons and glia.
This cross-ontology matching was repeated for each HPO branch and can be referred to as on-target cell types.
Within each branch, the $-log_{10}(FDR _{p,c})$ values of on-target cell types were binned by rounding to the nearest integer (*x-axis*) and the percentage of tests for on-target cell types relative to all cell types were computed at each bin (*y-axis*) ([Fig. @fig-summary]d).
The baseline level (dotted horizontal line) illustrates the percentage of on-target cell types relative to the total number of observed cell types.
Any percentages above this baseline level represent greater than chance representation of the on-target cell types in the significant tests.

### Monarch Knowledge Graph recall

Finally, we gathered known phenotype-cell type relationships from the Monarch Knowledge Graph (MKG), a comprehensive database of links between many aspects of disease biology [@Putman2024-et].
This currently includes `r nrow(kg_hp)` links between HPO phenotypes (n=`r data.table::uniqueN(kg_hp$from)`) and CL cell types (n=`r data.table::uniqueN(kg_hp$to)`).
Of these, we only considered the `r data.table::uniqueN(validate_associations_mkg_out$kg_filt)` phenotypes that we were able to test given that our approach was reliant on gene annotations.
We considered instances where we found a significant relationship between exactly matching pairs of HPO-CL terms as a hit.

However, as the cell types in MKG were not necessarily annotated at the same level as our single-cell references, we also considered instances where the MKG cell type was an ancestor term of our cell type (e.g. 'myeloid cell' vs. 'monocyte'), or *vice versa*, as hits.
Using these criteria, we determined our results recapitulated `r X`% of known phenotype-cell type relationships in the MKG.
We next computed how far along the CL ontological tree we would need to travel in order to reach a common ancestor between the MKG cell type and our cell type, for each phenotype-cell type link in the MKG.
This provides a metric of not just whether we recapitulated the exact cell types, but how dissimilar our identified cell types were for a given phenotype-cell type association ([Fig. @fig-monarch-recall]).

### Annotation of phenotypes using generative large language models

```{r code_dict}
gpt_codes <- formals(HPOExplorer::gpt_annot_codify)
code_dict <- paste0(shQuote(names(eval(gpt_codes$code_dict))),"=",
                     eval(gpt_codes$code_dict), collapse = ", ")
tiers_dict <- paste0(shQuote(names(eval(gpt_codes$tiers_dict))),"=",
                     eval(gpt_codes$tiers_dict), collapse = ", ") 
```

Only a small fraction of the the phenotypes in HPO (\<1%) have metadata annotations containing information on their time course, consequences, and severity.
This is due to the time-consuming nature of manually annotating thousands of phenotypes.
To generate such annotations at scale, we used Generative Pre-trained Transformer 4 (GPT-4), a large language model (LLM) as implemented within OpenAI’s chatGPT Application Programming Interface (API).
After extensive prompt engineering and ground-truth benchmarking, we were able to acquire annotations on how often each phenotype directly causes intellectual disability, death, impaired mobility, physical malformations, blindness, sensory impairments, immunodeficiency, cancer, reduced fertility, or is associated with a congenital onset.
These criteria were previously defined in surveys of medical experts as a means of systematically assessing phenotype severity [@Lazarin2014-we].
Responses for each metric were provided in a consistent one-word format which could be one of: `r paste0(shQuote(names(eval(gpt_codes$code_dict))), collapse=", ")`.
This procedure was repeated in batches (to avoid exceeding token limits) until annotations were gathered for `r length(unique(gpt_annot$annot$hpo_id))`/`r hpo@n_terms` HPO phenotypes.

We then encoded these responses into a semi-quantitative scoring system (`r code_dict`), which were then weighted by multiplying a semi-subjective scoring of the relevance of each metric to the concept of severity on a scale from `r min(unlist(eval(gpt_codes$tiers_dict)))`-`r max(unlist(eval(gpt_codes$tiers_dict)))`, with `r max(unlist(eval(gpt_codes$tiers_dict)))` being the most severe (`r tiers_dict`).
Finally, the product of the score was normalised to a quantitative severity score ranging from 0-100, where 100 is the theoretical maximum severity score.
This phenotype severity scoring procedure can be expressed as follows.

::: {#eq-gpt .content-hidden unless-format="html"}
![](equations/eq5.png){height="300px"}

Computing normalised severity score from encoded GPT-4 annotations.
:::

\
\

::: {.content-visible unless-format="html"}
```{=tex}
\begin{equation*}
  \eqnmarkbox[Brown4]{nss}{NSS_p}
  =
  \frac{ 
    \eqnmarkbox[Goldenrod]{nss2}{\sum_{j=1}^{m}} 
    (
      \eqnmarkbox[Goldenrod4]{nss3}{F_{pj}}
      \times 
      \eqnmarkbox[IndianRed4]{nss4}{W_j}
    )
    }{
    \eqnmarkbox[Tan]{nss5}{\sum_{j=1}^{m}(\max\{F_j\} \times W_j)} 
  } \times 100
\end{equation*}
\annotate[yshift=1em]{left}{nss}{Normalised Severity Score \\for each phenotype}
\annotate[yshift=3em]{left}{nss2}{Sum of weighted annotation values \\across all metrics}
\annotate[yshift=3em]{right}{nss3}{Numerically encoded annotation value \\of metric $j$ for phenotype $p$}
\annotate[yshift=1em]{right}{nss4}{Weight for metric $j$} 
\annotate[yshift=-1em]{below,right}{nss5}{Theoretical maximum severity score}
```
:::

\

### Enrichment of foetal cell types in congenital phenotypes

```{r fetal_keywords}
fetal_keywords <- shQuote(eval(formals(MSTExplorer::plot_congenital_annotations)$fetal_keywords) )
```

The GPT-4 annotations also enabled us to assess whether foetal cell types were more often significantly associated with congenital phenotypes in our Human Cell Landscape results as this single-cell reference contained both adult and foetal versions of cell types ([Fig. @fig-congenital]).
To do this, we performed a chi-squared ($\chi^2$) test on the proportion of significantly associated cell types containing any of the substrings `r paste(paste0(fetal_keywords[-length(fetal_keywords)], collapse = ", "),"or",tail(fetal_keywords,1))` (within cell types annotations from the original Human Cell Landscape authors [@Han2020-iq]) vs. those associated without, stratified by how often the corresponding phenotype had a congenital onset according to the GPT phenotype annotations (including `r paste0(shQuote(names(eval(gpt_codes$code_dict))), collapse=", ")`).
In addition, a series of $\chi^2$ tests were performed within each congenital onset frequency strata, to determine whether the observed proportion of foetal cell types vs. non-foetal cell types significantly deviated from the proportions expected by chance.

### Diagnosis via cell type-specific disease prediction

We designed an algorithm that uses our results to predict the most likely cell types underlying a set of phenotypic and genotypic traits observed in a patient ([Fig. @fig-diagnosis]).
This is implemented within `MSTExplorer::predict_celltypes` and takes HPO phenotypes as inputs.
It can optionally take included risk genes, excluded risk genes, included diseases and/or excluded diseases as additional inputs.
It then computes the It then outputs a weighted ranking of cell types, where higher ranking indicates a higher likelihood of being the underlying mechanism of the patient’s particular form of disease(s).

### Prognosis via cell type-mediated differential outcomes

The phenotype hypotonia is associated with diseases that range in severity from benign to debilitating to fatal [@Ahmed2016-ag].
In the absence of additional information, making an accurate diagnosis is extremely challenging even for experienced physicians.
The magnitude of this challenge is highlighted by the fact that each disease is associated with anywhere between `r min(per_disease$np)`-`r max(per_disease$np)` unique phenotypes (median=`r median(per_disease$np)`, mean=`r format(mean(per_disease$np),digits=4)`) within the HPO.
Conversely, each phenotype is associated with `r min(per_phenotype$nd)`-`r max(per_phenotype$nd)` diseases (median=`r median(per_phenotype$nd)`, mean=`r format(mean(per_phenotype$nd),digits=4)`).
We addressed this challenge by applying our phenotype-cell type association results in combination with expert-curated HPO annotations of clinical outcomes associated with each phenotype-disease pairing ([Fig. @fig-prognosis]).
We first extracted results for the phenotype 'Hypotonia' (*HP:0001252*) and its `r length(unique(hypotonia_results$hpo_id))-1` descendant subterms from our phenotype-cell type association analyses.
Next, we encoded the "Age of Death" categories associated with each disease in an ordinal scale ranging from 1, corresponding to prenatal death, to 8, corresponding to death in late adulthood (@tbl-death).
To determine whether cell type identity significantly predicted the age of death, we conducted an ANOVA where cell type was the predictor and "Age of Death score" was the outcome.

### Therapeutic target identification

We developed a systematic and automated strategy for identifying putative cell type-specific gene targets for each phenotype based on a series of filters at phenotype, cell type, and gene levels.
The entire target prioritisation procedure can be replicated with a single function: `MSTExplorer::prioritise_targets`.
This function automates all of the reference data gathering (e.g. phenotype metadata, cell type metadata, cell type signature reference, gene lengths, severity tiers) and takes a variety of arguments at each step for greater customisability.

### Therapeutic target validation

To assess whether our prioritised therapeutic targets were likely to be viable, we computed the overlap between our gene targets and those of existing gene therapies at various stages of clinical development ([Fig. @fig-therapy-validate]).
Gene targets were obtained for each therapy from the Therapeutic Target Database (TTD; release `r Sys.Date()`) and mapped onto standardised HUGO Gene Nomenclature Committee (HGNC) gene symbols using the `orthogene` R package.
We stratified our overlap metrics according to whether the therapies had failed (unsuccessful clinical trials or withdrawn), or were non-failed (successful or ongoing clinical trials).
We then conducted hypergeometric tests to determine whether the observed overlap between our prioritised targets and the non-failed therapy targets was significantly greater than expected by chance (i.e. enrichment).
We also conducted a second hypergeometric test to determine whether the observed overlap between our prioritised targets and the failed therapy targets was significantly less than expected by chance (i.e. depletion).
Finally, we repeated the analysis against all therapeutic targets, not just those of gene therapies, to determine whether our prioritised targets had relevance to other therapeutic modalities.

### Experimental model translatability

To improve the likelihood of successful translation between preclinical animal models and human patients, we created an interspecies translatability prediction tool for each phenotype nominated by our gene therapy prioritised pipeline ([Fig. @fig-animal-models]).
First, we extracted ontological similarity scores of homologous phenotypes across species from the MKG [@Putman2024-et].
Briefly, the ontological similarity scores ($SIM_o$) are computed for each homologous pair of phenotypes across two ontologies by calculating the overlap in homologous phenotypes that are ancestors or descendants of the target phenotype.
Next, we generated genotypic similarity scores ($SIM_g$) for each homologous phenotype pair by computing the proportion of 1:1 orthologous genes using gene annotation from their respective ontologies.
Interspecies orthologs were also obtained from the MKG.
Finally, both scores are multiplied together to yield a unified ontological-genotypic similarity score ($SIM_{o,g}$).

### Novel R packages

To facilitate all analyses described in this study and to make them more easily reproducible by others, we created several open-source R packages.
[`KGExplorer`](https://github.com/neurogenomics/KGExplorer) imports and analyses large-scale biomedical knowledge graphs and ontologies.
[`HPOExplorer`](https://github.com/neurogenomics/HPOExplorer) aids in managing and querying the directed acyclic ontology graph within the HPO.
[`MSTExplorer`](https://github.com/neurogenomics/MSTExplorer) facilitates the efficient analysis of many thousands of phenotype-cell type association tests, and provides a suite of multi-scale therapeutic target prioritisation and visualisation functions.
These R packages also include various functions for distributing the post-processed results from this study in an organised, tabular format.
Of note, `MSTExplorer::load_example_results` loads all summary statistics from our phenotype-cell type tests performed here.

### Rare Disease Celltyping Portal

To further increase the ease of access for stakeholders in the RD community without the need for programmatic experience, we developed a series of web apps to interactively explore, visualise, and download the results from our study.
Collectively, these web apps are called the Rare Disease Celltyping Portal.
The landing page for the website was made using HTML, CSS, and javascript and the web apps were created using the Shiny Web application framework for R and deployed on the [shinyapps.io](https://www.shinyapps.io) server.
The website can be accessed [here](https://neurogenomics.github.io/rare_disease_celltyping_apps/home). All code used to generate the website can be found [here](https://github.com/neurogenomics/rare_disease_celltyping_apps).

## Data and Code Availability

All data and code is made freely available through preexisting databases and/or GitHub repositories / software associated with this publication.

-   [Human Phenotype Ontology](https://hpo.jax.org)
-   [GenCC](https://thegencc.org/)
-   [Descartes Human scRNA-seq atlas](https://cellxgene.cziscience.com/collections/c114c20f-1ef4-49a5-9c2e-d965787fb90c)
-   [Human Cell Landscape scRNA-seq atlas](https://cellxgene.cziscience.com/collections/38833785-fac5-48fd-944a-0f62a4c23ed1)
-   [Rare Disease Celltyping Portal](https://neurogenomics.github.io/rare_disease_celltyping_apps/home)
-   [`KGExplorer`](https://github.com/neurogenomics/KGExplorer)
-   [`HPOExplorer`](https://github.com/neurogenomics/HPOExplorer)
-   [`MSTExplorer`](https://github.com/neurogenomics/MSTExplorer)
-   [Code to replicate analyses](https://github.com/neurogenomics/rare_disease_celltyping)
-   [Cell type-specific gene target prioritisation](https://neurogenomics.github.io/RareDiseasePrioritisation/reports/prioritise_targets)  
-   [Complement system gene list](https://www.genenames.org/data/genegroup/#!/group/492)


## Acknowledgements

We would like to thank the following individuals for their insightful feedback and assistance with data resources: Sarah J. Marzi, Gerton Lunter, Peter Robinson, Melissa Haendel, Ben Coleman, Nico Matentzoglu, Shawn T. O'Neil, Alan E. Murphy, Sarada Gurung.

### Funding

This work was supported by a UK Dementia Research Institute (UK DRI) Future Leaders Fellowship \[MR/T04327X/1\] and the UK DRI which receives its funding from UK DRI Ltd, funded by the UK Medical Research Council, Alzheimer’s Society and Alzheimer's Research UK.


## References {.unnumbered}

:::{#refs}

:::

\

{{< pagebreak >}}

## Supplementary Materials

### Supplementary Figures

```{r fig-evidence-histograms, fig.height=6, fig.height=8}
#| label: fig-evidence-histograms
#| fig-cap: Distribution of evidence scores at each processing step.

evidence_plot <- HPOExplorer::plot_evidence(phenotype_to_genes = p2g,
                                            show_plot = FALSE)
evidence_plot$plot
```


::: {#fig-diagram}
![](img/fig-diagram.png)

Diagrammatic overview of multi-scale disease investigation strategy.
Here we provide an abstract example of differential disease aetiology across multiple scales: diseases ($D$), phenotypes ($P$), cell types ($C$), genes ($G$), and clinical outcomes ($O$).
In the HPO, genes are assigned to phenotypes via particular diseases ($G_{d,p}$).
Therefore, the final gene list for each phenotype is aggregated from across multiple diseases ($G_{p}$).
We performed association tests for all pairwise combinations of cell types and phenotypes and filtered results after multiple testing corrections ($FDR_{p,c}<0.05$).
Each phenotype in the context of a given disease is referred to here as a symptom.
Links were established between symptoms and cell types through proportional gene set overlap at a minimum threshold of 25%.
:::

```{r fig-ctd-correlation}
#| label: fig-ctd-correlation
#| layout-ncol: 2
#| fig-cap: "Inter- and intra-dataset validation across the different CellTypeDataset (CTD) and developmental stages. Correlations are computed using Pearson's correlation coefficient. Point density is plotted using a 2D kernel density estimate."
#| fig-subcap: 
#| - "Correlation between the uncorrected p-values from all phenotype-cell type association tests using the Descartes Human vs. Human Cell Landscape CTDs."
#| - "Correlation between the $log_{10}(fold-change)$ from significant phenotype-cell type association tests ($FDR_{p,c}<0.05$) using the Descartes Human vs. Human Cell Landscape CTDs."
#| - "Correlation between the uncorrected p-values from all phenotype-cell type association tests using the Human Cell Landscape fetal samples vs. Human Cell Landscape adult samples."
#| - "Correlation between the $log_{10}(fold-change)$ from significant phenotype-cell type association tests ($FDR_{p,c}<0.05$) using the Human Cell Landscape fetal samples vs. Human Cell Landscape adult samples."

validate_associations_correlate_ctd_out$plot$p.all
validate_associations_correlate_ctd_out$plot$logFC.significant
validate_associations_correlate_ctd_out_hcl$plot$p.all
validate_associations_correlate_ctd_out_hcl$plot$logFC.significant
```

```{r plot_report}
### Generate filtering report plot
plot_report_out <- MSTExplorer::plot_report(
  results = results,
  rep_dt = prioritise_targets_out$report,
  show_plot = FALSE)
```

```{r fig-therapy-filter, fig.height=10, fig.width=15}
#| label: fig-therapy-filter
#| fig-cap: Therapeutics - Prioritised target filtering steps. This plot visualises the number of unique phenotype-cell type associations, cell types, genes, and phenotypes (*y-axis*) at each filtering step (*x-axis*) within the multi-scale therapeutic target prioritisation pipeline. Each step in the pipeline can be easily adjusted according to user preference and use case. See **Methods** for descriptions and criterion of each filtering step.**a**, The percentage of phenotypes belonging to each severity Tier after each filtering step (Tier 1 being the most severe). **b**, The number of phenotypes, cell types, associated diseases and genes remaining after each filtering step during the gene prioritisation pipeline.

plot_report_out
```

```{r fig-monarch-recall}
#| label: fig-monarch-recall
#| fig-cap: Recall of Monarch Knowledge Graph phenotype-cell type relationships.

cat("!!!RECOMPUTE!!!")
```

```{r fig-therapy-validate-all, fig.height=5, fig.width=8}
#| label: fig-therapy-validate-all
#| fig-cap: Therapeutics - Validation of prioritised therapeutic targets. Proportion of existing all therapy targets (documented in the Therapeutic Target Database) recapitulated by our prioritisation pipeline.

ttd_check_all_out$plot
```


```{r fig-animal-models, fig.height=12, fig.width=10}
#| label: fig-animal-models
#| fig-cap:  Identification of translatable experimental models. Interspecies translatability of human phenotypes nominated by our gene therapy prioritised pipeline. Above, our combined ontological-genotypic similarity score ($SIM_{o,g}$) is displayed as the heatmap fill colour stratified by the model organism (*x-axis*). An additional column (“n_genes_db1” on the far left) displays the total number of unique genes annotated to the phenotypic within the HPO. Phenotypes are clustered according to their ontological similarity in the HPO (*y-axis*).

library(ggplot2) # <-- Necessary due to bug in one of the plotting dependencies 

top_ids <- unique(prioritise_targets_out$top_targets$hpo_id)[1:1000]
plot_upheno_out <- KGExplorer::plot_upheno(
  pheno_map_genes_match = pheno_map_genes_match, 
  filters=list(id1=top_ids)
  )

plot_upheno_out$heatmap
```

```{r prioritise_targets_network-supp}
top_targets <- prioritise_targets_out$top_targets[,n_genes:=data.table::uniqueN(gene_symbol),
                                                  by="hpo_id"][n_genes<5 & proportion_driver_genes_symptom>.25]

height <- "60vh"

phenotype <- "respiratory failure"
vn_therapy_eg1 <- MSTExplorer::prioritise_targets_network(
  top_targets = top_targets[grepl(paste(phenotype,collapse = "|"), disease_name,ignore.case = TRUE)],
  main = phenotype, 
  height = height,
  submain = NULL)

phenotype <- "amyotrophic lateral sclerosis"
vn_therapy_eg2 <- MSTExplorer::prioritise_targets_network(
  top_targets = top_targets[grepl(paste(phenotype,collapse = "|"), disease_name,ignore.case = TRUE)],
  main = phenotype,
  height = height,
  submain = NULL)

phenotype <- "neurodegeneration"
vn_therapy_eg3 <- MSTExplorer::prioritise_targets_network(
  top_targets = top_targets[grepl(paste(phenotype,collapse = "|"), disease_name,ignore.case = TRUE)], 
  main = phenotype, 
  height = height,
  submain = NULL)

phenotype <- "small vessel disease"
vn_therapy_eg4 <- MSTExplorer::prioritise_targets_network(
  top_targets = top_targets[grepl(paste(phenotype,collapse = "|"), disease_name,ignore.case = TRUE)], 
  main = phenotype, 
  height = height,
  submain = NULL)
```

```{r fig-therapy-examples-supp, fig.height=7, fig.width=12, results='asis'}
#| label: fig-therapy-examples-supp
#| fig-cap: "Example cell type-specific gene therapy targets for several severe phenotypes and their associated diseases. Each disease (blue cylinders) is connected to its phenotype (purple cylinders) based on well-established clinical observations recorded within the HPO [@Gargano2024-fc].Phenotypes are connected to cell types (red circles) via association testing between weighted gene sets ($FDR_{p,c}<0.05$). Each cell type is connected to the prioritised gene targets (yellow boxes) based on the driver gene analysis.The thickness of the edges connecting the nodes represent the (mean) fold-change from the bootstrapped enrichment tests. Nodes were spatially arranged using the Sugiyama algorithm [@Sugiyama1981-ev]."
#| layout: "[[1,1], [1], [1]]" 
#| fig-subcap:
#| - 'Respiratory failure'
#| - 'Amyotrophic lateral sclerosis' 
#| - 'Neurodegeneration'
#| - 'Small vessel disease'

vn_therapy_eg1$plot
vn_therapy_eg2$plot
vn_therapy_eg3$plot
vn_therapy_eg4$plot
```

### Supplementary Methods

#### Therapeutics: Gene therapy target identification

Descriptions of each step in the prioritisation pipeline are as follows:

1.  **start**: All phenotype-cell type association results.

2.  **q_threshold**: Keep only results that were significant after multiple-testing correction (q\<0.05).

3.  **fold_threshold**: Keep only results with fold change\>=1.

4.  **keep_ont_levels**: Keep only phenotypes at certain absolute ontology levels within the HPO.

5.  **keep_onsets**: Keep only phenotypes with postnatal age of onsets to circumvent technical and ethical challenges associated with antenatal gene therapeutics delivery.

6.  **keep_tiers**: Keep only phenotypes with high severity Tiers.

    1.  We used a combination of manual curation and automated text-based substring queries to assign each phenotype a severity Tier as characterised in a survey of healthcare professionals [@Lazarin2014-we].

    2.  Tier 1: Diseases that shortened life span in adolescence or earlier or resulted in intellectual disability.

    3.  Tier 2: Diseases that shortened lifespan prematurely in adulthood, or resulted in impaired mobility or internal physical malformation.

    4.  Tier 3: Diseases causing sensory impairments (hearing, vision, touch, pain, or other), immunodeficiency/cancer, mental illness, or dysmorphic features.

    5.  Tier 4: Diseases that reduce fertility.
        Of the 49 phenotypes that were available in this severity ranking, we selected three that were classified as Tier 1 (the most severe disease category): mental deterioration, coma and respiratory failure.

7.  **severity_threshold**: Keep only phenotypes with mean severity score equal to or below the threshold.

    1.  Severity scores were computed by assigning each severity modifier term found in the HPO annotations a numerical value.
        In order of increasing severity:

    2.  HP:0012825 "Mild" (Severity_score=4)

    3.  HP:0012827 "Borderline" (Severity_score=3)

    4.  HP:0012828 "Severe" (Severity_score=2)

    5.  HP:0012829 "Profound" (Severity_score=1)

8.  **pheno_frequency_threshold**: Keep only phenotypes with mean frequency equal to or above the threshold (i.e. how frequently a phenotype is associated with any diseases in which it occurs).

    1.  Keep phenotypes with a mean frequency ≥10% or are NA by default.

9.  **keep_celltypes**: Keep only terminally differentiated cell types.

    1.  Of the 77 cell types tested in the Descartes cell type reference, the 40 terminally differentiated cell types were identified through a literature search. Of these, three (extravillous trophoblasts, syncytiotrophoblasts and trophoblast giant cells) were excluded as they only played a role in pregnancy [@Chang2018-qj; @Fogarty2011-ph; @Hu2010-eh], which would raise additional technical and ethical challenges as rAAV therapy has not yet been used to target foetuses in clinical trials.

10. **keep_seqnames**: Remove genes on non-standard chromosomes.

    1.  Only keep chromosomes 1-22, X, and Y.

11. **gene_size**: Keep only genes \<4.3kb in length.

    1.  Due to limitations in the length of the gene that can be carried by the rAAV vector, genes with a length of \>4.3kb were excluded.

12. **keep_biotypes**: Keep only genes belonging to certain biotypes (e.g. "protein_coding", "processed_transcript", "snRNA", "lincRNA", "snoRNA", "IG_C_gene").

    1.  Keep all biotypes by default.

13. **gene_frequency_threshold**: Keep only genes at or above a certain mean frequency threshold (i.e. how frequently a gene is associated with a given phenotype when observed within a disease).

    1.  Keep genes with a mean frequency ≥10% or are NA by default.

14. **keep_specificity_quantiles**: Keep only genes in top specificity quantiles from the cell type dataset.

    1.  To further narrow down genes, we extracted relevant metrics from the Descartes reference for each gene in each cell type. These included mean expression, specificity, and specificity quantiles (using 40 bins). Only genes with the most specific quantiles (39-40) were included for further analysis, as cell type-specific genes may be less likely to have off-target effects in other cell types.

15. **keep_mean_exp_quantiles**: Keep only genes in top mean expression quantiles from the cell type dataset

16. **end**: Final table of prioritised cell type- / phenotype-specific gene targets.

Finally, for more comprehensive target search, the we removed the filters for onsets (keep_onsets=NULL), Tier (keep_tiers=NULL), severity (severity_threshold=NULL), as well as relaxed the filters for phenotype frequency threshold (pheno_frequency_threshold=c(10,NA)), gene frequency threshold (gene_frequency_threshold = c(10,NA)), gene specificity quantiles (keep_specificity_quantiles = seq(20,40)), and gene expression quantiles (keep_mean_exp_quantiles = seq(20,40)).
 
