---
title: "Cell type-specific contextualisation of the human phenome: towards the systematic treatment of all rare diseases"
author:
  - name: Brian M. Schilder
    orcid: 0000-0001-5949-2191
    corresponding: true
    email: brian_schilder@alumni.brown.edu
    roles:
      - Investigation
      - Project administration
      - Software
      - Visualization
    affiliations:
      - name: Imperial College London
        city: London
        country: United Kingdom
        department: Department of Brain Sciences
      - name: UK Dementia Research Institute
        city: London
        country: United Kingdom
  - name: Kitty B. Murphy
    orcid: 0000-0002-8669-3076
    corresponding: false
    roles:
      - Investigation 
      - Software
      - Visualization
    affiliations:
      - name: Imperial College London
        city: London
        country: United Kingdom
        department: Department of Brain Sciences
      - name: UK Dementia Research Institute
        city: London
        country: United Kingdom
  - name: Hiranyamaya Dash
    orcid: 0009-0005-5514-505X
    corresponding: false
    roles:
      - Software
      - Visualization
    affiliations:
      - name: Imperial College London
        city: London
        country: United Kingdom
        department: Department of Brain Sciences
      - name: UK Dementia Research Institute
        city: London
        country: United Kingdom
  - name: Yichun Zhang
    corresponding: false
    roles:
      - Software
      - Visualization
    affiliations:
      - name: Imperial College London
        city: London
        country: United Kingdom
        department: Department of Brain Sciences
      - name: UK Dementia Research Institute
        city: London
        country: United Kingdom
  - name: Robert Gordon-Smith
    orcid: 0000-0001-6698-7387
    corresponding: false
    roles:
      - Investigation 
      - Software
      - Visualization
    affiliations:
      - name: Imperial College London
        city: London
        country: United Kingdom
        department: Department of Brain Sciences
      - name: UK Dementia Research Institute
        city: London
        country: United Kingdom
  - name: Jai Chapman
    corresponding: false
    roles:
      - Investigation 
      - Software
      - Visualization
    affiliations:
      - name: Imperial College London
        city: London
        country: United Kingdom
        department: Department of Brain Sciences
      - name: UK Dementia Research Institute
        city: London
        country: United Kingdom
  - name: Momoko Otani
    corresponding: false
    roles:
      - Investigation
    affiliations:
      - name: Imperial College London
        city: London
        country: United Kingdom
        department: National Heart and Lung Institute
  - name: Nathan G. Skene
    orcid: 0000-0002-6807-3180
    corresponding: true
    email: n.skene@imperial.ac.uk
    roles:
      - Project administration
    affiliations:
      - name: Imperial College London
        city: London
        country: United Kingdom
        department: Department of Brain Sciences
      - name: UK Dementia Research Institute
        city: London
        country: United Kingdom
keywords:
  - rare disease
  - phenotype
  - single-cell
  - gene therapy
key-points:
  - We used the Human Phenotype Ontology and single-cell RNA-seq references to characterise the phenome.
  - We then demonstrated how these results can be applied to clinical diagnosis, prognosis and therapeutics development.
date: last-modified
citation:
  container-title: medRxiv
bibliography: references.bib
editor:
  markdown:
    wrap: sentence
format:
  pdf: 
    include-in-header: 
      - fixes.tex
      
---

```{r setup, cache=FALSE, include=TRUE}
library(data.table)
require("ggdendro")
# ?formatdown::formatdown_options()

options(digits=2)
formatter <- function(x, ...) {
  # prettyNum(x, digits=2)
  # paste0("$",knitr:::format_sci(x),"$")
  if(is.numeric(x)){
    if(!is.integer(x)){
      formatdown::format_sci(x = x, digits = 2)    
    } else {
      formatC(x, big.mark = ",", mode = "integer")
    }
  } else {
    x
  }
}
knitr::knit_hooks$set(document = formatter, 
                      chunk = formatter,
                      inline = formatter)

# Quarto doesn't respect here::here. https://github.com/orgs/quarto-dev/discussions/9287
# Account for this:
if(!file.exists(here::here("index.qmd"))){
  # warning("Directory not set up properly. PWD:", here::here())
  setwd("manuscript/")
}
# Check that you can import file early
img_fetus <- magick::image_read(here::here("img","fetus.png"))
```

```{r load_phenotype_to_genes}
tag <- "v2024-02-08" ## For version control
cl_tag <- "v2023-09-21"
p2g <- HPOExplorer::load_phenotype_to_genes(tag=tag)
annot <- HPOExplorer::load_phenotype_to_genes(3, tag=tag)
per_disease <- p2g[,list(ng=data.table::uniqueN(gene_symbol),
                         np=data.table::uniqueN(hpo_id)),by="disease_id"]
per_phenotype <- p2g[,list(ng=data.table::uniqueN(gene_symbol),
                           nd=data.table::uniqueN(disease_id)),by="hpo_id"]
```

```{r load_example_results}
hpo <- HPOExplorer::get_hpo(tag=tag)
cl <- KGExplorer::get_ontology(name = "cl", tag = cl_tag, remove_rings=TRUE)
## Import precomputed results for reporting summaries
results <- MSTExplorer::load_example_results()
results <- HPOExplorer::add_hpo_name(results, hpo = hpo)
results <- HPOExplorer::add_ont_lvl(results)
results <- HPOExplorer::add_ancestor(results, hpo = hpo)
results <- MSTExplorer::map_celltype(results)
results <- MSTExplorer::add_logfc(results)
results[,effect:=estimate]
```

```{r summarise_results}
res_summ <- MSTExplorer::summarise_results(results = results)
res_summ_all <- res_summ$tmerged[ctd=="all"]  
```

```{r load_example_ctd, cache=FALSE}
## Must use `cache.lazy=FALSE` because sparse matrices not yet supported for caching
ctd_list <- MSTExplorer::load_example_ctd(c("ctd_DescartesHuman.rds",
                                            "ctd_HumanCellLandscape.rds"),
                                          multi_dataset=TRUE)
```

```{r validate_associations_mkg}
validate_associations_mkg_out <- MSTExplorer::validate_associations_mkg(results = results)
kg_hp <- validate_associations_mkg_out$kg[grepl("^HP:",from)] 
```

\newpage

## Abstract

Rare diseases (RDs) are a highly heterogeneous and underserved group of conditions.
Most RDs have a strong genetic basis but their causal pathophysiological mechanisms remain poorly understood.
We therefore systematically characterised the cell type–specific mechanisms for all genetically defined RD phenotypes by integrating the Human Phenotype Ontology with whole-body single-cell transcriptomic atlases from embryonic, foetal, and adult samples.
This revealed significant associations between `r res_summ_all[["cell types significant"]]` cell types and `r res_summ_all[["phenotypes significant"]]`/`r res_summ_all[["phenotypes tested"]]` (`r res_summ_all[["phenotypes significant (%)"]]`%) phenotypes across `r res_summ_all[["diseases significant"]]` RDs, substantially expanding knowledge of phenotype–cell type links. We then prioritised phenotypes for clinical impact based on severity (e.g. lethality, motor/mental impairment) and gene therapy compatibility (e.g. cell type specificity, postnatal treatability). All results are reproducible and freely available, including via an interactive web portal (<https://neurogenomics-ukdri.dsi.ic.ac.uk/>), representing a major advance toward treating patients across a broad spectrum of serious RDs.

## Introduction {#sec-introduction}

```{r afb_cells}
afb_cells <- KGExplorer::prune_ancestors(
  dat = results[ctd=="DescartesHuman" & q<.05 & CellType=="AFP_ALB_positive_cells"],
  id_col = "hpo_id",
  ont=hpo) |>
  data.table::setorderv("p") 

afb_cell_hepato <- rbind(
  HPOExplorer::filter_descendants(
  afb_cells, 
  keep_descendants = "Abnormality of hepatobiliary system physiology", 
  hpo = hpo),
  afb_cells[grepl("hepat|liver",hpo_name, ignore.case = TRUE)] 
) |> unique() 

```

Rare diseases (RDs) are individually uncommon but collectively this class of over 10,000 conditions affects 300–400 million people worldwide (1 in 10–20 individuals) [@Ferreira2019-jp; @Zhu2020-vo], 75% of RD patients are children, with a 30% rate mortality by age five [@noauthor_undated-kp].
Diagnosis is challenging due to highly variable presentations, averaging five years [@Marwaha2022-uy], with \~46% misdiagnosed and \>75% never diagnosed [@Molster2016-da].
Prognosis is similarly difficult.
Treatments exist for \<5% of RDs [@Halley2022-pd] and high development costs for small patient populations deter investment [@Institute_of_Medicine_US_Committee_on_Accelerating_Rare_Diseases_Research_and_Orphan_Product_Development2010-vj; @Yates2022-ra], making these therapies among the world’s most expensive [@Nuijten2022-yc; @Thielen2022-ud].
High-throughput therapeutic discovery could lower costs and speed delivery.

A major barrier in research and clinical care of diseases is inconsistent medical terminology.
The Human Phenotype Ontology (HPO) provides a unified, hierarchical framework of `r hpo@n_terms` phenotypes spanning 10,300 RDs [@Gargano2024-fc; @Kohler2019-pc; @Robinson2008-ys], integrated into diagnostics and linked to other ontologies (e.g. SNOMED CT, UMLS, ICD).
Over 80% of RDs have known genetic causes [@Nguengang_Wakap2020-cz], with HPO gene annotations curated from OMIM, Orphanet, DECIPHER, and case reports.
Yet gene lists alone lack the tissue and cell type context essential for understanding pathogenesis and improving diagnosis, prognosis, and treatment.

Single-cell RNA-seq (scRNA-seq) now enables transcriptome-wide profiling at cellular resolution [@Baysoy2023-vt; @Haque2017-bn; @Qi2023-ev].
Comprehensive atlases such as Descartes Human [@Cao2020-qz] and Human Cell Landscape [@Han2020-iq] cover embryonic to adult stages across tissues, providing gene signatures for hundreds of cell subtypes.
Integrating RD gene annotations with these profiles reveals the specific cell types through which genes act, including understudied cell types.

Cell type-specific mechanisms are critical for guiding the development of effective therapeutics, especially virally-mediated gene therapies [@kawabataImprovingCellspecificRecombination2024; @ocarrollAAVTargetingGlial2021].
Knowledge of the specific causal cell types can enhance efficacy and improve safety by avoiding off-target effects.
To facilitate these key insights, we developed a high-throughput pipeline to nominate cell type–resolved gene therapy targets across thousands of RD phenotypes, ranked by composite phenotype severity scores [@murphyHarnessingGenerativeAI2024].
This work expands knowledge of the cell types, organ systems, and life stages underlying RDs, with direct applications to precision therapeutic development.

::: {#fig-study-design fig-height="1500px"}
![](img/study_design.png){height="1500px"}

**Multi-modal data fusion reveals the cell types underlying thousands of human phenotypes.** Schematic overview of study design in which we numerically encoded the strength of evidence linking each gene and each phenotype (using the Human Phenotype Ontology and GenCC databases).
We then created gene signature profiles for all cell types in the Descartes Human and Human Cell Landscape scRNA-seq atlases.
Finally, we iteratively ran generalised linear regression tests between all pairwise combinations of phenotype gene signatures and cell type gene signatures.
The resulting associations were then used to nominate cell type-resolved gene therapy targets for thousands of rare diseases.
:::

## Results {#sec-results}

### Phenotype-cell type associations

```{r run_phenomix, eval=FALSE}
## Create phenotype-gene matrix filled with aggregated GenCC evidence scores
ymat <- HPOExplorer::hpo_to_matrix(formula = "gene_symbol ~ hpo_id")

## Run phenomix with DescartesHuman CellTypeDataset
lm_res1 <- MSTExplorer::run_phenomix(ctd_name = "DescartesHuman",
                                     annotLevel = 2, 
                                     test_method = "glm_univariate",
                                     ymat = ymat)
## Run phenomix with HumanCellLandscape CellTypeDataset
lm_res2 <- MSTExplorer::run_phenomix(ctd_name = "HumanCellLandscape",
                                     annotLevel = 3, 
                                     test_method = "glm_univariate",
                                     ymat = ymat)
## Merge results
results <- data.table::rbindlist(list(DescartesHuman=lm_res1,
                                      HumanCellLandscape=lm_res2),
                                idcol = "ctd")
## Apply multiple testing correction
results[,q:=stats::p.adjust(p,method="fdr")]
```

We systematically investigated cell types underlying HPO phenotypes, hypothesising that genes with cell type–specific expression are most relevant to those cell types, and that disrupting such genes will have variable effects across cell types.
More precisely, for each phenotype we created a list of associated genes weighted by the strength of the evidence supporting those associations, imported from the Gene Curation Coalition (GenCC) [@distefanoGeneCurationCoalition2022].
Analogously, we created mean gene expression profiles for each cell type using scRNA-seq atlases and then normlaized them to compute cell type specificity of gene expression (see **Methods** subsection *Single-cell transcriptomic atlases* for further details).

For comprehensiveness, we used two pan-tissue scRNA-seq atlases: Descartes Human (\~4 million single-nuclei and single-cells from `r length(unique(MSTExplorer::tissue_maps[ctd=="DescartesHuman"]$uberon_name))` foetal tissues) [@Cao2020-qz] and Human Cell Landscape (\~703,000 single-cells from `r length(unique(MSTExplorer::tissue_maps[ctd=="HumanCellLandscape"]$uberon_name))` embryonic, foetal and adult tissues) [@Han2020-iq].
For every unique combination of phenotype and cell type, we trained a generalized linear regression model to test for association between the respective gene-phenotype association scores and gene-cell type expression specificity scores ([Fig. @fig-study-design]).
We then applied stringent multiple testing correction to control the false discovery rate (FDR) across all tests, and significant phenotype–cell type associations were identified at FDR\<0.05.

In Descartes Human, `r res_summ$tmerged[ctd=="DescartesHuman"][["tests significant"]]`/`r res_summ$tmerged[ctd=="DescartesHuman"][["tests"]]` (`r res_summ$tmerged[ctd=="DescartesHuman"][["tests significant (%)"]]`%) tests were significant across `r res_summ$tmerged[ctd=="DescartesHuman"][["cell types significant"]]` cell types and `r res_summ$tmerged[ctd=="DescartesHuman"][["phenotypes significant"]]` phenotypes. In Human Cell Landscape, the corresponding values were `r res_summ$tmerged[ctd=="HumanCellLandscape"][["tests significant (%)"]]`% significant tests, `r res_summ$tmerged[ctd=="HumanCellLandscape"][["cell types significant"]]` cell types, and `r res_summ$tmerged[ctd=="HumanCellLandscape"][["phenotypes significant"]]` phenotypes, with more phenotypes linked to at least one cell type due to greater cell type diversity and life-stage coverage. Across both atlases, the median number of significant cell types per phenotype was `r res_summ_all[["cell types per phenotype (median)"]]`, indicating specificity of associations. Overall, `r res_summ_all[["diseases significant"]]`/`r res_summ_all[["diseases"]]` (>99.96\%) of diseases had significant cell type associations for at least one phenotype. Full stratified results are provided in @tbl-summary.

### Validation of expected phenotype-cell type relationships

```{r plot_bar_dendro, cache=FALSE}
plot_bar_dendro_out <- MSTExplorer::plot_bar_dendro(
  results = results, 
  hpo = hpo,
  cl = cl,
  show_plot = FALSE) 
overrep_dat <- plot_bar_dendro_out$ggbars_out$data_summary

# overrep_dat[,summary:=paste0(ancestor_name,": ", 
#                              n_celltypes_sig,"/",n_celltypes,
#                              " types of ",shQuote(target_celltypes),
#                              " were overrepresented",
#                              " ($N_{p}$=",phenotypes_per_ancestor,").")] 

overrep_df <- overrep_dat[,list("HPO branch"=ancestor_name,
                                "Phenotypes (total)"=phenotypes_per_ancestor,
                                "CL branch"=target_celltypes, 
                                "Cell types (overrepresented)"=n_celltypes_sig,
                                "Cell types (total)"=n_celltypes)]|>
  data.frame(check.names=FALSE)
```

```{r cor_pct_p}
cor.pct_p <- stats::cor.test(plot_bar_dendro_out$ggprop_out$data$pct, 
                             plot_bar_dendro_out$ggprop_out$data$minus_log_p)
nervous_pct <- plot_bar_dendro_out$ggprop_out$data[branch=="Abnormality of the\n nervous system" & minus_log_p==6]$pct
```

We intuit that organ system-specific abnormalities are often driven by cell types within that system.
The HPO’s high-level categories allow systematic testing; for example, heart phenotypes should typically involve cardiocytes, and nervous system abnormalities should involve neural cells.
All cell types in our single-cell atlases were mapped to the Cell Ontology (CL), a hierarchical vocabulary of cell types.

A cell type was considered *on-target* for an HPO branch if it belonged to a matching CL branch (@tbl-ontarget-celltypes).
For each HPO branch ([Fig. @fig-summary]b), we tested whether cell types were more often associated with phenotypes in that branch compared to all others, and identified those overrepresented at FDR\<0.05.
All `r length(unique(overrep_df[["HPO branch"]]))` HPO branches showed disproportionate associations with on-target cell types from their respective organ systems.

We hypothesised that more strongly significant phenotype–cell type associations are more likely to be on-target.
Grouping $-log_{10}(\text{p-values})$ into six bins, we calculated the proportion of on-target cell types per HPO–CL branch pairing.
Indeed, this proportion consistently increased with association significance ($r_{Pearson}=$`r cor.pct_p$estimate`, $p=$`r cor.pct_p$p.value`).
For example, in nervous system abnormalities neural cells constituted only `r round(plot_bar_dendro_out$ggprop_out$data[branch=="Abnormality of the\n nervous system" & minus_log_p==6]$pct_celltype)`% of all tested cell types, yet made up `r plot_bar_dendro_out$ggprop_out$data[branch=="Abnormality of the\n nervous system" & minus_log_p==6]$pct`% of associations within the $-log_{10}(\text{p-values}) \ge 6$ bin (which corresponds to $p\ge 10^{-7}$).
This confirms that stronger associations are more likely to involve on-target cell types, confirming our association strategy captures real relationships.

::: {#fig-summary .hide-subcaptions}
```{r fig-summary}
#| label: fig-summary
#| fig-cap: __High-throughput analysis reveals cell types underlying thousands of rare disease phenotypes.__ **a**, Some cell types are much more commonly associated with phenotypes than others. Bar height indicates the total number of significant phenotype enrichments per cell type (FDR<0.05) across all branches of the HPO. **b**, Analyses reveal expected and novel cell type associations within high-level HPO branches. Asterisks above each bar indicate whether that cell type was significantly more often enriched in that branch relative to all other HPO branches, including those not shown here, as a proxy for how specifically that cell type is associated with that branch; FDR<0.0001 (\*\*\*\*), FDR<0.001 (\*\*\*), FDR<0.01 (\*\*), FDR<0.05 (\*). **c**, Ontological relatedness of cell types in the Cell Ontology (CL) [@Diehl2016-gt]. **d**, The proportion of on-target associations (*y-axis*) increases with greater test significance (*x-axis*). Percentage of significant phenotype associations with on-target cell types (second row of facet labels), respective to the HPO branch. 
#| fig-height: 14
#| fig-width: 13

plot_bar_dendro_out$plot
```
:::

### Validation of inter- and intra-dataset consistency

```{r validate_associations_correlate_ctd}
library(data.table)

## Across CTD
validate_associations_correlate_ctd_out <- MSTExplorer::validate_associations_correlate_ctd(
  results=results, 
  point.args = list(alpha=.5),
  hpo_agg_lvl=NULL,
  marginal=TRUE,
  add_density = FALSE,
  group_var="ctd")

## Replace p-values of exactly 0 with smallest number R can compute
validate_associations_correlate_ctd_out$data_stats$p.all$summary_data$p.value <- max(validate_associations_correlate_ctd_out$data_stats$p.all$summary_data$p.value,
                  .Machine$double.xmin)

## Within CTD: across developmental stages
validate_associations_correlate_ctd_out_hcl <- MSTExplorer::validate_associations_correlate_ctd(
  results=results,
  hpo_agg_lvl=NULL,
  marginal=TRUE,
  add_density = FALSE,
  point.args = list(alpha=.5), 
  filters= list(ctd=c("HumanCellLandscape"), 
                stage=c("Fetus","Adult")),
  group_var="stage")


# dat <- data.table::copy(validate_associations_correlate_ctd_out$data$all)
# cor(dat$p_DescartesHuman, dat$p_HumanCellLandscape)
# cor(-log10(dat$p_DescartesHuman+1e-2), -log10(dat$p_HumanCellLandscape+1e-2))
# 
# dat <- data.table::copy(validate_associations_correlate_ctd_out_hcl$data$all)
# cor(dat$p_Fetus, dat$p_Adult)
# cor(-log10(dat$p_Fetus+1e-2), -log10(dat$p_Adult+1e-2))
# 
# 
# dat <- data.table::copy(validate_associations_correlate_ctd_out$data$all)
# # dat[,estimate_DescartesHuman_abs_log:=log10(abs(estimate_DescartesHuman))]
# # dat[,estimate_HumanCellLandscape_abs_log:=log10(abs(estimate_HumanCellLandscape))]
# # dat$estimate_DescartesHuman_log <- log2(dat[["estimate_DescartesHuman"]]+abs(min(dat[["estimate_DescartesHuman"]]))+1e-6)
# # dat$estimate_HumanCellLandscape_log <- log2(dat[["estimate_HumanCellLandscape"]]+abs(min(dat[["estimate_HumanCellLandscape"]]))+1e-6)
# # dat[,zscore_HumanCellLandscape:=log2(scales::rescale((estimate_HumanCellLandscape), to = c(1e-6,10)))]
# dat[,combined_DescartesHuman:=(1-pmax(p_DescartesHuman, .Machine$double.xmin)) * estimate_DescartesHuman]
# dat[,combined_HumanCellLandscape:=(1-pmax(p_HumanCellLandscape, .Machine$double.xmin)) * estimate_HumanCellLandscape]
# 
# p <- ggstatsplot::ggscatterstats(data = dat,#[q_DescartesHuman < 0.05 & q_HumanCellLandscape < 0.05],
#                             # x = estimate_DescartesHuman,
#                             # y = estimate_HumanCellLandscape,
#                            # x = estimate_DescartesHuman_abs_log,
#                            #  y = estimate_HumanCellLandscape_abs_log,
#                            x = combined_DescartesHuman,
#                             y = combined_HumanCellLandscape,
#                             cor.coef.name = "r",
#                             xlab = "Descartes estimate",
#                             ylab = "HumanCellLandscape estimate",
#                             title = "Correlation of effect sizes between Descartes and Human Cell Landscape",
#                             marginal.type = "histogram",
#                             xsidehistogram.args = list(fill="darkgrey", color="white"),
#                             ysidehistogram.args = list(fill="darkgrey", color="white"),
#                             point.args = list(alpha=.5),
#                             bf.message = FALSE,
#                             ggtheme = ggplot2::theme_minimal())
# p <- MSTExplorer:::filter_ggstatsplot_subtitle(p)
# p <- ggrastr::rasterize(p, layers = 1, dpi = 300)
# p
```

```{r replication-rates}
#### Compute replication rate ####

interCTD_rr <- MSTExplorer::get_replication_rate(validate_associations_correlate_ctd_out$data$all, 
                     x="q_DescartesHuman",
                     y="q_HumanCellLandscape")

intraCTD_rr <- MSTExplorer::get_replication_rate(validate_associations_correlate_ctd_out_hcl$data$all, 
                     x="q_Fetus",
                     y="q_Adult")

#### Permutation tests ####

#### Descartes vs HumanCellLandscape
interCTD_pval_permtest <- MSTExplorer::run_permutation_test_cor(df = validate_associations_correlate_ctd_out$data$all, 
                                      x_var="estimate_DescartesHuman",
                                      y_var="estimate_HumanCellLandscape",
                                      save_path = here::here("cache","interCTD_pval_permtest.rds")
                                      )
interCTD_pval_permtest$x_var <- "DescartesHuman: estimate (all)"
interCTD_pval_permtest$y_var <- "HumanCellLandscape: estimate (all)"


interCTD_effect_permtest <- MSTExplorer::run_permutation_test_cor(df = validate_associations_correlate_ctd_out$data$significant, 
                                      x_var="estimate_DescartesHuman",
                                      y_var="estimate_HumanCellLandscape",
                                      save_path = here::here("cache","interCTD_effect_permtest.rds")
                                      )
interCTD_effect_permtest$x_var <- "DescartesHuman: estimate (FDR<0.05)"
interCTD_effect_permtest$y_var <- "HumanCellLandscape: estimate (FDR<0.05)"

#### Fetus vs. Adult
intraCTD_pval_permtest <- MSTExplorer::run_permutation_test_cor(df = validate_associations_correlate_ctd_out_hcl$data$all, 
                                      x_var="estimate_Fetus",
                                      y_var="estimate_Adult",
                                      save_path = here::here("cache","intraCTD_pval_permtest.rds")
                                      )
intraCTD_pval_permtest$x_var <- "Fetus: estimate (all)"
intraCTD_pval_permtest$y_var <- "Adult: estimate (all)"

intraCTD_effect_permtest <- MSTExplorer::run_permutation_test_cor(df = validate_associations_correlate_ctd_out_hcl$data$significant, 
                                      x_var="estimate_Fetus",
                                      y_var="estimate_Adult",
                                      save_path = here::here("cache","intraCTD_effect_permtest.rds")
                                      )
intraCTD_effect_permtest$x_var <- "Fetus: estimate (FDR<0.05)"
intraCTD_effect_permtest$y_var <- "Adult: estimate (FDR<0.05)" 

ctd_validation_permtest <- rbind(
  interCTD_pval_permtest,
  interCTD_effect_permtest,
  intraCTD_pval_permtest,
  intraCTD_effect_permtest
)
ctd_validation_permtest[p_emp==0, p_emp:=.Machine$double.xmin]
```

If our methodology works, it should yield consistent phenotype-cell type associations across different datasets.
We therefore tested for the consistency of our results across the two single-cell reference datasets (Descartes Human vs. Human Cell Landscape) across the subset of overlapping cell types [Fig. @fig-ctd-correlation].
In total there were `r nrow(validate_associations_correlate_ctd_out$data$all)` phenotype-cell type associations to compare across the two datasets (across `r length(unique(validate_associations_correlate_ctd_out$data$all$hpo_id))` phenotypes and `r length(unique(validate_associations_correlate_ctd_out$data$all$cl_name))` cell types annotated to the exact same CL term.
We found that the correlation between p-values of the two datasets was high ($rho$=`r validate_associations_correlate_ctd_out$data_stats$p.all$summary_data$estimate`, $p$=`r validate_associations_correlate_ctd_out$data_stats$logFC.significant$summary_data$p.value`).
Within the subset of results that were significant in both single-cell datasets (FDR\<0.05), we found that degree of correlation between the association effect sizes across datasets was even stronger ($r_{Pearson}=$`r validate_associations_correlate_ctd_out$data_stats$logFC.significant$summary_data$estimate`, $p=$`r validate_associations_correlate_ctd_out$data_stats$logFC.significant$summary_data$p.value`).
We also checked for the intra-dataset consistency between the p-values of the foetal and adult samples in the Human Cell Landscape, showing a very similar degree of correlation as the inter-dataset comparison ($r_{Pearson}=$`r validate_associations_correlate_ctd_out_hcl$data_stats$p.all$summary_data$estimate`, $p=$`r validate_associations_correlate_ctd_out_hcl$data_stats$logFC.significant$summary_data$p.value`).
Together, these results suggest that our approach to identifying phenotype-cell type associations is highly replicable and generalisable to new datasets.

### More specific phenotypes are associated with fewer genes and cell types

```{r plot_ontology_levels, cache=FALSE}

plot_ontology_levels_out <- MSTExplorer::plot_ontology_levels(
  results = results, 
  ctd_list = ctd_list,
  
  x_vars = c("genes","cell types","estimate"), 
  stats_idx= c(1,3,4,7), # Filter stats in subtitle for better readability
  sig_vars= c(FALSE, TRUE, TRUE),
  log_vars = c(FALSE, FALSE, TRUE),
  # type="nonparametric", # Use parametric for IC, non-paramatric for levels
  nrow = 1,
  show_plot = FALSE)  

plot_ontology_levels_out_stats <- plot_ontology_levels_out$data_stats$summary_data|>
  data.table::setkeyv("parameter2")
## replace pvalues of exactly 0 with the minimum computable number in R
## This avoids creating -Inf when logging values.
plot_ontology_levels_out_stats[p.value==0, p.value:=.Machine$double.xmin]
plot_ontology_levels_out_stats[q.value==0, q.value:=.Machine$double.xmin]
```

```{r, plot_ontology_levels_perm_test}
pheno_df <- plot_ontology_levels_out$data
 
res_genes <- MSTExplorer::run_permutation_test_cor(
     pheno_df,
     x_var = "info_content",
     y_var = "genes",
     save_path = here::here("cache","run_permutation_test_cor.res_genes.rds"),
     )

res_cells <- MSTExplorer::run_permutation_test_cor( 
    # Remove duplicate rows (after aggregating by cell type count)
     pheno_df[q<=.05 & !duplicated(pheno_df, by = c("hpo_id","info_content", "cell types"))],
     x_var = "info_content",
     y_var = "cell types",
     save_path = here::here("cache","run_permutation_test_cor.res_cells.rds"),
     )

res_effect <- MSTExplorer::run_permutation_test_cor(
    pheno_df[q<=.05],
    x_var = "info_content",
    y_var = "estimate",
    save_path = here::here("cache","run_permutation_test_cor.res_effect.rds"),
      )

res_permtests <- rbind(res_genes, res_cells, res_effect)
# Replace p-values of exactly 0 with smallest number R can compute
res_permtests[p_emp==0, p_emp:=.Machine$double.xmin]
```

Higher levels of the ontology are broad classes of phenotype (e.g. 'Abnormality of the nervous system') while the lower levels can get very detailed (e.g. 'Spinocerebellar atrophy').
The higher level phenotypes inherit all genes associated with lower level phenotypes, so naturally they have more genes than the lower level phenotypes ([Fig. @fig-ontology-lvl]a; $r_{Pearson}=$`r plot_ontology_levels_out_stats["genes"]$estimate`, $p=$`r plot_ontology_levels_out_stats["genes"]$p.value`).

Next, we reasoned that the more detailed and specific a phenotype is, the more likely it is to be driven by one cell type.
For example, while 'Neurodevelopmental abnormality' could plausibly be driven by any/all cell types in the brain, it is more likely that 'Impaired visuospatial constructive cognition' is driven by fewer cell types.
This was indeed the case, as we observed a strongly significant negative correlation between the two variables ([Fig. @fig-ontology-lvl]b; $r_{Pearson}=$`r plot_ontology_levels_out_stats["cell types"]$estimate`, $p=$`r plot_ontology_levels_out_stats["cell types"]$p.value`).
We also found that the phenotype-cell type association effect size increased with greater phenotype specificity, reflecting the decreasing overall number of associated cell types at each ontological level ([Fig. @fig-ontology-lvl]c; $r_{Pearson}=$`r plot_ontology_levels_out_stats["estimate"]$estimate`, $p=$`r plot_ontology_levels_out_stats["estimate"]$p.value`).

::: {#fig-ontology-lvl .hide-subcaptions}
```{r fig-ontology-lvl}
#| label: fig-ontology-lvl
#| fig-cap: __More specific phenotypes are associated with fewer, more specific genes and cell types.__ Information content (IC), is a normalised measure of ontology term specificity. Terms with lower IC represent the broadest HPO terms (e.g. 'All'), while terms with higher IC indicate progressively more specific HPO terms (e.g. 'Contracture of proximal interphalangeal joints of 2nd-5th fingers'). Box plots show the relationship between HPO phenotype IC and **a**, the number of genes annotated to each phenotype, **b**, the number of significantly enriched cell types, **c**, the effect sizes (absolute model $R^2$ estimates after log-transformation) of significant phenotype-cell type association tests. Boxes are coloured by the mean value within each IC bin (after rounding continuous IC values to the nearest integer).
#| fig-height: 7
#| fig-width: 19

plot_ontology_levels_out$plot
```
:::

### Validation of phenotype-cell type associations using biomedical knowledge graphs

To validate phenotype–cell type associations without literature bias, we used the Monarch Knowledge Graph (MKG), a curated database of biomedical concepts and relationships containing `r nrow(kg_hp)` known associations [@Putman2024-et].\
The MKG served as a benchmark for the field’s current knowledge.
For each MKG association, we calculated the proportion of cell types recovered in our results at different ontological distances in the Cell Ontology.
Distance 0 indicates the closest possible match (e.g. “monocyte” vs. “monocyte”), with greater distances reflecting progressively broader matches (e.g. distance 1: “monocyte” vs. “classical monocyte”).
The theoretical maximum recall was capped by the percentage of MKG phenotypes for which we identified at least one significant association ($FDR_{pc}$).
In other words, since our results only contain significant cell type associations for 90% of the phenotypes in the MKG's phenotype-cell type associations, our maximum achievable performance was 90% recall.

Our results included at least one significant cell type for `r round(validate_associations_mkg_out$proportion_phenotypes_captured*100,1)`% of MKG phenotypes.
At distance `r validate_associations_mkg_out$cl_distance_results$data[1,]$dist`, we recalled `r round(validate_associations_mkg_out$cl_distance_results$data[1,]$pct,1)`% of associations; at distance `r validate_associations_mkg_out$cl_distance_results$data[2,]$dist`, recall rose to `r round(validate_associations_mkg_out$cl_distance_results$data[2,]$pct,1)`%, reaching a maximum of `r round(max(validate_associations_mkg_out$cl_distance_results$data$pct),1)`% at the largest allowed distance.
Precision could not be computed, as MKG lists only true positives.
Overall, these benchmarks show that our approach recovers most known phenotype–cell type associations while generating many novel ones.

### Phenome-wide analyses discover novel phenotype-cell type associations

```{r plot_bar_dendro_facets-rni}
target_branches <- list("Recurrent bacterial infections"="leukocyte")
lvl <- subset(hpo@elementMetadata,name==names(target_branches)[1])$ontLvl
results_tmp <- HPOExplorer::add_ancestor(data.table::copy(results),
                                         lvl = lvl,
                                         force_new = TRUE)
infections_out <- MSTExplorer::plot_bar_dendro_facets(
  results=results_tmp,
  target_branches=target_branches,
  facets = "hpo_name",
  lvl=lvl+1,
  ncol=2,
  vlines="hepatoblast",
  legend.position="top",
  facets_n=NULL,
  q_threshold=0.05,
  background_full=FALSE)
remove(results_tmp)

staph_res <- infections_out$data[hpo_name=="Recurrent staphylococcal infections"]
staph_res_top <- staph_res[,.SD[p %in% head(sort(p), 1)], by=c("hpo_id")]
   
recurrent_infections_ids <- KGExplorer::get_ontology_descendants(
  ont = hpo,
  terms = "Recurrent infections")[[1]]
hepatoblast_res <- results[q<0.05 & 
                           hpo_id %in% recurrent_infections_ids & 
                           cl_name=="hepatoblast"]
hepatocyte_res <- results[q<0.05 & 
                          ancestor_name=="Abnormality of the immune system" &
                          grepl("hepatocyte",CellType,ignore.case = TRUE)] 

rni_res <- infections_out$data[hpo_name=="Recurrent Neisserial infections" & cl_name!="hepatoblast"]
```

Having confirmed many phenotype-cell type associations match prior expectations, we explored novel links for undercharacterised phenotypes.
'Recurrent bacterial infections' (`r data.table::uniqueN(infections_out$data$hpo_id)-1` descendants, e.g. staphylococcal, streptococcal, Neisserial) mostly associated with immune cells (e.g. macrophages, dendritic cells, T cells, monocytes, neutrophils) ([Fig. @fig-rni]).
Known links include 'Recurrent staphylococcal infections' with myeloid cells [@Heim2014-du; @Pidwill2020-le; @Stoll2018-dc; @Tebartz2015-xs], where `r staph_res_top$cl_name`s were most strongly associated (FDR=`r staph_res_top$q`, $\beta$=`r staph_res_top$effect`). Notably, amongst the recurrent bacterial infection phenotypes, hepatoblasts were exclusively associated with 'Recurrent Neisserial infections', hereafter RNI (Descartes Human: FDR=`r hepatoblast_res[hpo_name=="Recurrent Neisserial infections" & CellType=="Hepatoblasts"]$q`, $\beta$=`r hepatoblast_res[hpo_name=="Recurrent Neisserial infections" & CellType=="Hepatoblasts"]$estimate`).

::: {#fig-rni .hide-subcaptions}
```{r fig-rni}
#| label: fig-rni
#| fig-cap: __Association tests reveal that hepatoblasts have a unique role in recurrent Neisserial infections.__ Significant phenotype-cell type tests for phenotypes within the branch 'Recurrent bacterial infections'. Amongst all different kinds of recurrent bacterial infections, hepatoblasts (highlighted by vertical dotted lines) are exclusively associated with 'Recurrent Neisserial infections'. Note that terms from multiple levels of the same ontology branch are shown as separate facets (e.g. 'Recurrent bacterial infections' and 'Recurrent gram−negative bacterial infections').
#| fig-height: 12
#| fig-width: 13

infections_out$plot + 
  ggplot2::guides(fill=ggplot2::guide_legend(ncol=2,
                                             title = NULL))
```
:::

```{r prioritise_targets_network-RNI, cache.lazy=FALSE}
## Annotate results with disease/symptom-level and gene-level information
## filtering q-values at this step yields the same results as filtering at the next step, 
## albeit with much fast computation.
results_annot <- HPOExplorer::add_disease(results[q<0.05],
                                          add_descriptions = TRUE,
                                          allow.cartesian = TRUE)
results_annot <- MSTExplorer::add_symptom_results(results = results_annot, 
                                                  ctd_list = ctd_list)
## Plot multi-scale mechanisms as an interactive network
phenotype <- "Recurrent Neisserial infections"
phenotype_id <- HPOExplorer::map_phenotypes(phenotype, to = "id")
results_annot[,estimate_mult:=estimate*25]
vn_rni <- MSTExplorer::prioritise_targets_network(
  top_targets = results_annot[hpo_name==phenotype], 
  edge_size_var = "estimate_mult",
  mediator_var = list(c(3,4),c(1,3),c(1,4)),
  main = NULL,  
  submain = NULL)

prioritise_targets_multiheatmap_out <- MSTExplorer::prioritise_targets_multiheatmap(
  top_targets=results_annot,
  prioritise_targets_network_out=vn_rni,
  ctd_list=ctd_list, 
  gene_order=NULL,
  hpo=hpo,
  gencc_extra=list("disease_id"="OMIM:614372",
                   "disease_name"="Mannose-binding lectin (MBL) deficiency",
                   "gene_symbol"="MBL2",
                   "evidence_score_sum"=5),
  show_plot=FALSE) 
```

To better understand the multi-scale mechanisms underlying RNI susceptibility, we visualised the putative causal relationships between genes, cell types and diseases associated with RNI as a network ([Fig. @fig-network-rni]).
The phenotype RNI was connected to cell types through the aforementioned association test results (FDR\<0.05).
Genes that were primarily driving these associations (i.e. genes that were both strongly linked with RNI and were highly specifically expressed in the given cell type) were designated as *driver genes"*and retained for plotting (see **Methods** for details).
While a gene being non-specifically expressed does not necessarily mean it is unimportant for the function of a cell (many ubiquitously expressed genes are essential for cell function), gene expression specificity is nevertheless a useful metric to unambiguously link cell types to other biological entities (e.g. phenotypes).
Diseases were then connected to phenotypes and gene nodes based on HPO annotations.
Finally, diseases were also connected to cell types as a transitive property of being connected to phenotypes, but only if the disease gene set overlapped with 25% or more of the driver genes for that particular phenotype-cell type relationship (see **Methods** subsection *Symptom-cell type associations* and [Fig. fig-diagram] for further explanation).
<!-- Across all phenotypes in the HPO, more specific phenotypes (terms in the HPO with greater IC) are not only more specific to certain cell types ([Fig. @fig-ontology-lvl]b), but are also associated with genes that have greater cell type-specific expression within those cell types. --> <!-- Even so, we should note that the choice of which specificity quantiles to include is arbitrary. --> 

Using this network-based approach, we found that RNI (a phenotype of `r length(sort(unique(results_annot[hpo_name=="Recurrent Neisserial infections"]$disease_name)))` diseases: `r paste(shQuote(sort(unique(results_annot[hpo_name=="Recurrent Neisserial infections"]$disease_name))), collapse=", ")`) was also linked to `r rni_res[cl_name==unique(rni_res$cl_name)[1]]$cl_name[1]`s (FDR=`r rni_res[cl_name==unique(rni_res$cl_name)[1]]$q[1]`, $\beta$=`r rni_res[cl_name==unique(rni_res$cl_name)[1]]$effect[1]`), `r rni_res[cl_name==unique(rni_res$cl_name)[2]]$cl_name[1]`s (FDR=`r rni_res[cl_name==unique(rni_res$cl_name)[2]]$q[1]`, $\beta$=`r rni_res[cl_name==unique(rni_res$cl_name)[2]]$effect[1]`), and `r rni_res[cl_name==unique(rni_res$cl_name)[3]]$cl_name[1]`s (FDR=`r rni_res[cl_name==unique(rni_res$cl_name)[3]]$q[1]`, $\beta$=`r round(rni_res[cl_name==unique(rni_res$cl_name)[3]]$effect[1],3)`).
All of the gene implicated in this causal network are part of the complement system (*C5*, *C6*, *C8B*, *CFB*, *CFI*, *MBL2*, and *C7*). Complement deficiences are known to cause a marked susceptibility to infection [@Ladhani2019-nf; @Rosain2017-ih]. However, the complement system comprises more than 56 genes and can be expressed in a wide variety of cell types [@Seal2023-pa]. 
Our analysis was able to decompose the multiple mechanisms underlying RNI into subsets of complement proteins and identify the specific cell types they each affect.
For example, disruptions in complement genes *C5*, *C8*, and *C7* cause RNI via hepatoblasts, stratified epithelial cells, and stromal cells, respectively ([Supp. Fig. @fig-network-rni]).
These four cell types may represent disease subtypes with distinct clinical courses or biomarkers, allowing us to begin resolving RNI-related disease mechanisms at cell-type resolution. 

In particular, we were intrigued by the finding that hepatoblasts were associated with RNI but their mature counterpart, hepatocytes, were not. 
Mature hepatocytes are well recognised as the principal source of complement proteins [@Zhou2016-kq]. However, emerging evidence indicates that foetal hepatoblasts also express detectable amounts of complement genes [@yu2001, @wesley2022], which we confirmed by querying the CellxGene browser [@CZI_Single-Cell_Biology_Program2023-fs] (see **Data Availability**). 
Upon further inspection, we found that these were in fact circulating hepatoblast-like cells found in liver, placenta and spleen, according to the authors of the Descartes scRNA-seq atlas [@cao2020].
These cells express high levels of alpha fetoprotein (AFB), serum albumin (ALB), and apolipoproteins (including APOE).
Our results support this hypothesis as these AFB+/ALB+ cells were significantly associated with `r length(unique(afb_cell_hepato$hpo_name))` liver-related phenotypes, as well as `r length(unique(afb_cells[ancestor_name=="Abnormality of blood and blood-forming tissues"]$hpo_name))` blood-related phenotypes. 
Together, these findings suggest that these hepatoblast-like cell subpopulations play a causal role in complement-related disorders during development. However, this relatively novel cell type must be better charactised before drawing any firm conclusions.

### Prioritising phenotypes based on severity

```{r gpt_annot_codify}
gpt_check <- HPOExplorer::gpt_annot_check(show_plot = FALSE)
gpt_annot <- HPOExplorer::gpt_annot_codify() 

gpt_annot$annot_weighted[,hpo_name:=gsub("^obsolete ","",hpo_name)]
least_severe_phenotype <- gpt_annot$annot_weighted[hpo_name=="Thin toenail" & severity_score_gpt==0,]
```

Some phenotypes are more severe than others and thus could be prioritised for treatment (e.g. 'Leukonychia' is far less severe than 'Leukodystrophy').
To systematically rank phenotypes, we used GPT-4 to annotate severity for `r length(unique(gpt_check$annot$hpo_id))`/`r hpo@n_terms` (`r length(unique(gpt_check$annot$hpo_id))/hpo@n_terms*100`%) HPO phenotypes [@murphyHarnessingGenerativeAI2024].
Benchmarking against ground-truth HPO branch annotations showed high accuracy (recall=`r mean(gpt_check$true_pos_rate)*100`%, min=`r min(gpt_check$true_pos_rate)*100`%, max=`r max(gpt_check$true_pos_rate)*100`%, SD=`r sd(gpt_check$true_pos_rate)*100`%) and strong consistency (`r mean(gpt_check$consistency_rate)*100`%).
From these, we computed weighted severity scores (0–100) for all phenotypes.
The most severe was `r shQuote(gpt_annot$annot_weighted[1,]$hpo_name)` (*`r gpt_annot$annot_weighted[1,]$hpo_id`*, score=`r gpt_annot$annot_weighted[1,]$severity_score_gpt`), followed by `r shQuote(gpt_annot$annot_weighted[2,]$hpo_name)` (*`r gpt_annot$annot_weighted[2,]$hpo_id`*, score=`r gpt_annot$annot_weighted[2,]$severity_score_gpt`).
There were `r data.table::uniqueN(gpt_annot$annot_weighted[severity_score_gpt==0]$hpo_name)` phenotypes with score 0 (e.g. `r paste(shQuote(least_severe_phenotype$hpo_name),collapse=", ")`), mean=`r mean(gpt_annot$annot_weighted$severity_score_gpt)` (median=`r median(gpt_annot$annot_weighted$severity_score_gpt)`).

```{r plot_celltype_severity}
## Identify cell types associated with the most severe phenotypes
plot_celltype_severity_out <- MSTExplorer::plot_celltype_severity(
  results = results,
  run_enrichment=TRUE,
  types=c("dot","bar"),
  run_prune_ancestors=TRUE,
  base_size = 6,
  nonsig_fill = ggplot2::alpha("transparent",0),
  save_path=here::here("cache","plot_celltype_severity.tests.rds"),
  force_new = FALSE
  )
```

```{r plot_severity_vs_nphenotypes}
plot_severity_vs_nphenotypes_out <- MSTExplorer::plot_severity_vs_nphenotypes(
  results = results,
  size=2,
  type="nonparametric",
  gpt_annot = gpt_annot)
plot_severity_vs_nphenotypes_out$plot <- MSTExplorer::filter_ggstatsplot_subtitle(plot_severity_vs_nphenotypes_out$plot)
```

Merging severity scores with significant (FDR\<0.05) phenotype–cell type associations revealed that `r plot_celltype_severity_out$bar$data$cl_name[1]`s had the highest average severity, followed by `r plot_celltype_severity_out$bar$data$cl_name[2]`s and `r plot_celltype_severity_out$bar$data$cl_name[3]`s, while `r plot_celltype_severity_out$bar$data$cl_name[length(plot_celltype_severity_out$bar$data$cl_name)]`s had the lowest.
Numerically encoding GPT annotations (0–3) and applying Wilcoxon tests confirmed expected links, e.g. `r (plot_celltype_severity_out$dot$data[variable=="blindness"]|>data.table::setorderv("p"))$cl_name[1]`s with blindness, `r (plot_celltype_severity_out$dot$data[variable=="death"]|>data.table::setorderv("p"))$cl_name[1]`s with death, and analogous patterns for reduced fertility, immunodeficiency, impaired mobility, and cancer.
Finally, we found that cell types associated with more phenotypes also tended to have higher mean composite severity ($p=$`r ggstatsplot::extract_stats(plot_severity_vs_nphenotypes_out$plot)$subtitle_data$p.value`, $\hat{\rho}_{Spearman}$=`r ggstatsplot::extract_stats(plot_severity_vs_nphenotypes_out$plot)$subtitle_data$estimate`), supporting the idea that broadly involved cell types perform critical physiological functions whose disruption causes more severe disease.

::: {#fig-celltype-severity-dot .hide-subcaptions}
```{r fig-celltype-severity-dot}
#| label: fig-celltype-severity-dot
#| fig-cap: __Genetic disruptions to some cell types cause more clinically severe phenotypes than others.__ **a**, Different cell types are associated with different aspects of phenotypic severity. The dot plot shows the mean encoded frequency value for a given severity annotation (0="never", 1="rarely", 2="often", 3="always"; shown as dot size), aggregated by the associated cell type. One-sided Wilcoxon rank-sum tests were performed for each cell type (within each GPT annotation) to determine which cell types more frequently caused severe phenotypes than all other cell types. Dots are colored by $-log_2(FDR)$ when Wilcoxon test FDR values were less than 0.05. All dots with non-significant Wilcoxon tests are not shown. Cell types (rows) are clustered according to the p-values of the Wilcoxon tests. **b**, Cell types that affect more phenotypes tend to have more clinically severe consequences. Specifically, the number of phenotypes each cell type is significantly associated with, and the mean composite severity score of each cell type. The cell types with the top/bottom three x/y axis values are labeled to illustrate the cell types that cause the most/least phenotypic disruption when dysfunctional. Side histograms show the density of data points along each axis. Summary statistics for the linear regression are shown in the title ($t_{Student}$ = Student t-test statistic, $p$ = p-value, $\hat{\rho}_{Spearman}$ = Spearman rank correlation coefficient, $CI_{95\%}$ = confidence intervals, $n_{pairs}$ = number of observed data pairs).
#| fig-width: 10
#| fig-height: 10

 (
   plot_celltype_severity_out$dot$plot | 
   (
     (patchwork::plot_spacer() / plot_severity_vs_nphenotypes_out$plot) +
      patchwork::plot_layout(heights = c(1,1.5))
    ) 
      
 )+
  patchwork::plot_layout(widths = c(.5,1)) +
   patchwork::plot_annotation(tag_levels = "a") #& 
   # ggplot2::theme(text=ggplot2::element_text(size=9))
 
```
:::

### Congenital phenotypes are associated with foetal cell types

```{r plot_congenital_annotations}
plot_congenital_annotations_out <- MSTExplorer::plot_congenital_annotations(
  results = results[ctd=="HumanCellLandscape"],
  x_var="fetal_celltype",
  hpo=hpo, 
  digits="signif1",
  sample.size.label.args = list(size=3)
  )
{
  plot_congenital_annotations_out$plot$labels$subtitle <- NULL
  plot_congenital_annotations_out$plot$labels$caption <- NULL
  plot_congenital_annotations_out$plot$labels$group <- NULL
  plot_congenital_annotations_out_data <- 
    data.table::data.table(plot_congenital_annotations_out$data, 
                           key="congenital_onset")
  plot_congenital_annotations_out_data[,summary:=paste0(
    shQuote(congenital_onset),"=",.label,
    " (n=",counts," associations)"
  )]
}
plot_congenital_annotations_out_stats <- data.table::data.table(
  plot_congenital_annotations_out$data_stats$summary_data
  )
# plot_congenital_annotations_out_fo <- MSTExplorer::plot_congenital_annotations(
#   results = results[ctd=="HumanCellLandscape"],
#   x_var="foetal_only",
#   hpo=hpo)
```

The life stage at which a phenotype manifests affects treatment options, as some interventions (e.g. gene therapies) may be ineffective once developmental defects occur.
In the DescartesHuman dataset all cells were foetal, while the Human Cell Landscape included both embryonic/foetal (`r data.table::data.table(plot_congenital_annotations_out$data)[,list(perc=mean(perc)),by="fetal_celltype"][fetal_celltype==TRUE]$perc`% of cell types), and adult tissues (`r data.table::data.table(plot_congenital_annotations_out$data)[,list(perc=mean(perc)),by="fetal_celltype"][fetal_celltype==FALSE]$perc`% of cell types).
Some cell types exist in both stages (e.g. chondrocytes), while others are foetal-specific (e.g. neural crest cells).
Congenital phenotypes (according to our severity annotations) were strongly associated with foetal cell types ($p=$`r plot_congenital_annotations_out_stats$p.value`, $\chi^2=$`r plot_congenital_annotations_out_stats$statistic`), consistent with their developmental origins.

```{r plot_congenital_annotation_branch}
#  Which branches of the HPO are most enriched for phenotypes with foetal cell type associations? 
plot_congenital_annotations_branch_out <- MSTExplorer::plot_congenital_annotations(
  results = results,
  by_branch=TRUE,
  keep_descendants="Phenotypic abnormality",
  hpo=hpo, 
  # sampling.plan="hypergeom",
   # type="nonparametric",
  digits="signif1",
  # label.args = list(alpha = 1, fill = "white", angle=90),
  sample.size.label.args = list(size=3)
  )

{
  plot_congenital_annotations_branch_out$plot$labels$subtitle <- NULL
  plot_congenital_annotations_branch_out$plot$labels$caption <- NULL
  plot_congenital_annotations_branch_out$plot$labels$group <- NULL
  if(plot_congenital_annotations_branch_out$data_stats$summary_data$p.value==0){
    plot_congenital_annotations_branch_out$data_stats$summary_data$p.value <- .Machine$double.xmin
  }
}

top_congenital <- unique(plot_congenital_annotations_branch_out$data[,c("ancestor_name","perc")])
## Create p-value heatmap
# one_sample_data <- data.table::data.table(
#   ggstatsplot::extract_stats(plot_congenital_annotations_branch_out$plot)$one_sample_data
# )[p.value==0,p.value:=.Machine$double.xmin]
# one_sample_heatmap <- one_sample_data|>
#   ggplot2::ggplot(ggplot2::aes(x=ancestor_name, y=1, fill=-log2(p.value))) +
#   ggplot2::geom_raster() +
#   ggplot2::labs(fill=expression(-log[2]~"(p-value)"), x=NULL, y=NULL) +
#    ggplot2::theme_minimal() +
#   ggplot2::theme(axis.text = ggplot2::element_blank()) 
```

::: {#fig-congenital .hide-subcaptions}
```{r fig-congenital}
#| label: fig-congenital
#| fig-cap: __Foetal vs. adult cell type references provide development context to phenotype etiology.__ **a**, Congenital phenotypes are more often associated with foetal cell types. As a phenotype is more often congenital in nature, the greater proportion of foetal cell types are significantly associated with it. **b**, The proportion of phenotype-cell type association tests that are enriched for foetal cell types within each HPO branch. The p-values to the right of each bar are the results of an additional series of $\chi^2$ tests to determine whether the proportion of foetal vs. non-foetal cell types significantly different differs from the proportions expected by chance (the dashed vertical line). The feotal silhouette was generated with DALL-E [@ramesh]. The adult silhouette is from phylopic.org and is freely available via CC0 1.0 Universal Public Domain Dedication.
#| fig-height: 8
#| fig-width: 11

legend_colors <- ggplot2::ggplot_build(
  plot_congenital_annotations_out$plot)$data[[1]][["fill"]]|>unique()

## Account for annoying path changes during rendering
img_fetus <- magick::image_read(here::here("img","fetus.png"))
img_adult <- magick::image_read(here::here("img","adult.png"))

fetus <- magick::image_ggplot(img_fetus, interpolate=TRUE) +
ggplot2::geom_line(data = data.frame(x = 1:2, y = 1:2),
                   linewidth=20,
                   ggplot2::aes(x, y, 
                                color = 'Foetal\ncell\ntypes')) +
ggplot2::scale_color_manual(NULL, values = legend_colors[1]) +
ggplot2::theme(legend.position = 'right',
               legend.key.width = ggplot2::unit(2.5, 'cm')) 
adult <- magick::image_ggplot(img_adult, interpolate=TRUE) +
ggplot2::geom_line(data = data.frame(x = 0:1, y = 1:2),
                   linewidth=20,
                   ggplot2::aes(x, y, 
                                color = 'Adult\ncell\ntypes')) +
ggplot2::scale_color_manual(NULL, values = legend_colors[2]) +
ggplot2::theme(legend.position = 'right',
               legend.key.width = ggplot2::unit(2.5, 'cm')) 
  
 

(
  (patchwork::plot_spacer()|adult|patchwork::plot_spacer()|fetus|patchwork::plot_spacer()) 
  ) /
(
  (
  (
  plot_congenital_annotations_out$plot +
    ggplot2::coord_flip() +
    
  ggplot2::labs(x="Congenital onset\nfrequency in phenotype",
                y=NULL,
                fill=NULL, group=NULL) +
  ggplot2::theme(axis.text.x = ggplot2::element_blank(), 
                 legend.position = "none") 
  ) /
  plot_congenital_annotations_branch_out$plot + 
    ggplot2::labs(x="HPO branch", y="Phenotype-cell type associations (%)") +
  ggplot2::theme(axis.text.x = ggplot2::element_text(size=12), 
                 legend.position = "none") +
    ggplot2::coord_flip()
)  + patchwork::plot_layout(heights  = c(data.table::uniqueN(plot_congenital_annotations_out$data$congenital_onset)/ 
data.table::uniqueN(plot_congenital_annotations_branch_out$data$ancestor_name),
1)
) 
  ) +
  patchwork::plot_layout(heights  = c(0.05, 1)) +
  patchwork::plot_annotation(tag_levels = list(c("", "", letters)))

```
:::

HPO branches varied significantly in the proportion of their significant assocations with foetal cell types ($\hat{V}_{Cramer}$=`r plot_congenital_annotations_branch_out$data_stats$summary_data$estimate`, $p<$`r plot_congenital_annotations_branch_out$data_stats$summary_data$p.value`).
Branches with the most disproportiate number of foetal cell type associations were `r shQuote(top_congenital$ancestor_name[1])` (`r top_congenital$perc[1]`%), `r shQuote(top_congenital$ancestor_name[2])` (`r top_congenital$perc[2]`%), and `r shQuote(top_congenital$ancestor_name[3])` (`r top_congenital$perc[3]`%).
The most adult-biased brances were `r shQuote(top_congenital$ancestor_name[nrow(top_congenital)])` (`r top_congenital$perc[nrow(top_congenital)]`%) and `r shQuote(top_congenital$ancestor_name[nrow(top_congenital)-1])` (`r top_congenital$perc[nrow(top_congenital)-1]`%).

```{r run_congenital_enrichment}
## Identify which phenotypes are more often associated with foetal version of 
## cell types than the adult versions.
run_congenital_enrichment_out <- MSTExplorer::run_congenital_enrichment(
  results = results[ctd=="HumanCellLandscape"]
  ) 
```

Some phenotypes involve only foetal or only adult versions of a cell type.
We quantified bias by comparing association p-values between foetal and adult versions of the same type (metric range: 1=foetal-only, -1=adult-only).
The top `r formals(MSTExplorer::run_congenital_enrichment)$top_n_hpo` foetal-biased phenotypes revealed were enriched for the HPO branches `r shQuote(run_congenital_enrichment_out$hpo_enrich_top$name[1])` ($p=$`r run_congenital_enrichment_out$hpo_enrich_top$p_value[1]`) and `r shQuote(run_congenital_enrichment_out$hpo_enrich_top$name[2])` ($p=$`r run_congenital_enrichment_out$hpo_enrich_top$p_value[2]`), which included specific phenotypes such as `r shQuote(unique(run_congenital_enrichment_out$foetal_pdiff_top$hpo_name)[1])`.
Adult-biased phenotypes were instead enriched for the branches `r shQuote(run_congenital_enrichment_out$hpo_enrich_bottom$name[1])` ($p=$`r run_congenital_enrichment_out$hpo_enrich_bottom$p_value[1]`) and `r shQuote(run_congenital_enrichment_out$hpo_enrich_bottom$name[2])` ($p=$`r run_congenital_enrichment_out$hpo_enrich_bottom$p_value[2]`), with examples like 'Excessive wrinkled skin' and 'Paroxysmal supraventricular tachycardia'.
These align with known developmental and age-related processes, supporting our approach for linking phenotypes to causal cell types.

### Therapeutic target identification

```{r prioritise_targets}
prioritise_targets_out <- MSTExplorer::prioritise_targets(
  results = results, 
  ctd_list = ctd_list,
  phenotype_to_genes = p2g,
  hpo = hpo,
  
  keep_deaths=NULL,
  keep_onsets=NULL,
  keep_specificity_quantiles = seq(30,40), ## NULL:70, 30-40:64 
  keep_mean_exp_quantiles = seq(30,40), ## NULL:65, 10:55
  info_content_threshold=8, ## 8:55, 5:64 
  effect_threshold=NULL, ## 1:39
  severity_score_gpt_threshold=NULL, ## 10:78, NULL:82
  symptom_intersection_threshold=.25, ## .25:57
  evidence_score_threshold=3,  ## 5:47, 4:47, 3:64
  top_n = 10, ## 5:38, 20:42, 30:45, 40:52, 50:55
  group_vars = "hpo_id", 
  save_path = here::here("cache","prioritise_targets.rds"),
  force_new = FALSE)
 
genes_per_pheno <- prioritise_targets_out$top_targets[,list(n=data.table::uniqueN(gene_symbol)),
                                                      by="hpo_id"]
p2g <- HPOExplorer::add_ont_lvl(p2g)
min_ont_lvl <- 3
genes_per_pheno_all <- p2g[ontLvl>min_ont_lvl,
                           list(n=data.table::uniqueN(gene_symbol)),
                           by="hpo_id"]


top_celltypes <- prioritise_targets_out$top_targets[,list(
  np=data.table::uniqueN(hpo_id)),
  by="cl_name"]|>data.table::setorderv("np",-1)
top_ancestors <- prioritise_targets_out$top_targets[,list(
  np=data.table::uniqueN(hpo_id),
  nc=data.table::uniqueN(CellType),
  ng=data.table::uniqueN(gene_symbol)
  ),
  by="ancestor_name"]|>
  data.table::setorderv("np",-1)
```

In the above sections, we demonstrated how gene association databases can be used to investigate the cell types underlying disease phenotypes at scale.
While these associations are informative on their own, we wished to take these results further in order to have a more translational impact.
Knowledge of the causal cell types underlying each phenotype can be highly informative for scientists and clinicians in their quest to study and treat them.
Therapeutic targets with supportive genetic evidence have 2.6x higher success rates in clinical trials [@nelsonSupportHumanGenetic2015; @ochoaHumanGeneticsEvidence2022; @minikelRefiningImpactGenetic2024].
Furthermore, knowing which cell types to target with gene therapy can maximise the efficacy of highly expensive payloads, and minimise side effects (e.g. immune reaction to viral vectors).
Recent biotechnological advances have greatly enhanced our ability to target specific cell types with gene therapy, making specific and accurate knowledge the correct underlying cell types more pertinent than ever [@kawabataImprovingCellspecificRecombination2024; @ocarrollAAVTargetingGlial2021].

Rather than consider phenotypes in isolation, or even phenotype associations with particular cell types, we sought to identify multi-scale therapeutic targets.
That is, specific genes to target in specific cell types in specific phenotypes as they present in particular diseases.
The confluence of these pieces of information is crucial for clinical utility, as it provides the much-needed context to develop effective therapeutics for real-world patient populations.
For example, the same phenotype may be caused by disruptions to one of several cell types, each of which are in turn caused by mutations to particular genes.
Networks are a natural way to visualize the complex relationships between these various biological entities, and using sensible filters (i.e. pruning) keeps the networks small enough to gain meaningful insights through visual exploration.

Towards this objective, we developed an automated pipeline to identify putative cell type–specific gene targets for each phenotype by integrating phenotype–cell type association results with primary resources such as GenCC gene–disease relationships and scRNA-seq atlas datasets, producing a table where each row represented a disease–phenotype–cell type–gene tetrad.
We applied sequential filters to retain only significant phenotype–cell type associations ($FDR<0.05$), phenotype–gene pairs with strong causal evidence (GenCC score \> 3), phenotypes with high specificity ($IC>8$), and gene–cell type links in the top 25% expression specificity quantile, and further required a symptom intersection \> 0.25 when linking cell types to diseases via phenotypes.
The filtered results were ranked by GPT-4 composite severity scores, with only the top 10 tetrads retained per phenotype, yielding compact, high-confidence networks suitable for manual inspection and visualization.

This yielded putative therapeutic targets for `r prioritise_targets_out$report[step=="end"][["Phenotypes"]]` phenotypes across `r prioritise_targets_out$report[step=="end"][["Diseases"]]` diseases in `r prioritise_targets_out$report[step=="end"][["Cell types"]]` cell types and `r prioritise_targets_out$report[step=="end"][["Genes"]]` genes ([Supp. Fig. @fig-therapy-filter]).
While this constitutes a large number of genes in total, each phenotype was assigned a median of `r round(median(genes_per_pheno$n))` gene targets (mean=`r mean(genes_per_pheno$n)`, min=`r min(genes_per_pheno$n)`, max=`r max(genes_per_pheno$n)`).
Relative to the number of genes annotations per phenotype in the HPO overall (median=`r median(genes_per_pheno_all$n)`, mean=`r mean(genes_per_pheno_all$n)`, min=`r min(genes_per_pheno_all$n)`, max=`r max(per_phenotype$ng)`) this represents a substantial decrease in the number of candidate target genes, even when excluding high-level phenotypes (HPO level\>`r min_ont_lvl`).
It is also important to note that the phenotypes in the prioritised targets list are ranked by their severity, allowing us to distinguish between phenotypes with a high medical urgency (e.g. `r shQuote(unique(prioritise_targets_out$top_targets[order(severity_score_gpt, decreasing = TRUE)]$hpo_name)[1])`) from those with lower medical urgency (e.g. `r shQuote(unique(prioritise_targets_out$top_targets[order(severity_score_gpt)]$hpo_name)[1])`).
This can be useful for clinicians, biomedical scientists, and pharmaceutical manufacturers who wish to focus their research efforts on phenotypes with the greatest need for intervention.

Across all phenotypes, `r top_celltypes[1,]$cl_name` were most commonly implicated (`r top_celltypes[1,]$np` phenotypes), followed by `r top_celltypes[2,]$cl_name` (`r top_celltypes[2,]$np` phenotypes), `r top_celltypes[2,]$cl_name` (`r top_celltypes[2,]$np` phenotypes), `r top_celltypes[3,]$cl_name` (`r top_celltypes[3,]$np` phenotypes), `r top_celltypes[4,]$cl_name` (`r top_celltypes[4,]$np` phenotypes), and `r top_celltypes[5,]$cl_name` (`r top_celltypes[5,]$np` phenotypes).
Grouped by higher-order ontology category, `r shQuote(top_ancestors[1,]$ancestor_name)` had the greatest number of enriched phenotypes (`r top_ancestors[1,]$np` phenotypes, `r top_ancestors[1,]$ng` genes), followed by `r shQuote(top_ancestors[2,]$ancestor_name)` (`r top_ancestors[2,]$np` phenotypes, `r top_ancestors[2,]$ng` genes), `r shQuote(top_ancestors[3,]$ancestor_name)` (`r top_ancestors[3,]$np` phenotypes, `r top_ancestors[3,]$ng` genes), `r shQuote(top_ancestors[4,]$ancestor_name)` (`r top_ancestors[4,]$np` phenotypes, `r top_ancestors[4,]$ng` genes), and `r shQuote(top_ancestors[5,]$ancestor_name)` (`r top_ancestors[5,]$np` phenotypes, `r top_ancestors[5,]$ng` genes).

```{r plot_report}
plot_report_out <- MSTExplorer::plot_report(
  results = results,
  rep_dt =prioritise_targets_out$report[(Rows_diff!=0) | step %in% c("start","end")],
  label.size = .1,
  add_tiers = FALSE,
  show_plot = FALSE, 
  save_path = NULL)
```

### Therapeutic target validation

```{r ttd_check}
# remove_status <- c("Application submitted",
#                    "Patented",
#                    "Preregistration","Registered",
#                    "Investigative","Preclinical",
#                    "Clinical trial","Phase 0",
#                    NA) 
run_map_genes = FALSE

## Gene therapy only
ttd_check_out <- MSTExplorer::ttd_check(
  top_targets=prioritise_targets_out$top_targets, 
  phenotype_to_genes=p2g,
  run_map_genes = run_map_genes,
  drug_types = "Gene therapy",
  as_percent = FALSE,
  # remove_status=remove_status,
  allow.cartesian = TRUE,
  show_plot = FALSE)

## All therapy types
ttd_check_all_out <- MSTExplorer::ttd_check(
  top_targets=prioritise_targets_out$top_targets,   
  phenotype_to_genes=p2g,
  run_map_genes = run_map_genes,
  as_percent = FALSE,
  # remove_status=remove_status,  
  allow.cartesian = TRUE,
  show_plot = FALSE)
```

To determine whether the genes prioritised by our therapeutic targets pipeline were plausible, we checked what percentage of gene therapy targets we recapitulated.
Data on therapeutic approval status was gathered from the Therapeutic Target Database (TTD; release `r Sys.Date()`) [@Liu2011-qd].
Overall, we prioritised `r sum(ttd_check_out$data[failed==FALSE]$prioritised)/nrow(ttd_check_out$data[failed==FALSE])*100`% (`r nrow(ttd_check_out$data[failed==FALSE])` total) of all non-failed existing gene therapy targets (ie. those which are currently approved, investigative, or undergoing clinical trials).
A hypergeometric test confirmed that our prioritised targets were significantly enriched for non-failed gene therapy targets ($p=$`r ttd_check_out$hypergeo[status=="nonfailed"]$p`, $\text{odds ratio=}$`r ttd_check_out$hypergeo[status=="nonfailed"]$OR`, $\text{sensitivity}$=`r ttd_check_out$hypergeo[status=="nonfailed"]$sensitivity`, $\text{specificity}$=`r ttd_check_out$hypergeo[status=="nonfailed"]$specificity`).
For these hypergeometric tests, the background gene set was composed of the union of all phenotype-associated genes in the HPO and all gene therapy targets listed in TTD.

Even when considering therapeutics of any kind ([Supp. Fig. @fig-therapy-validate-all]), not just gene therapies, we recapitulated `r sum(ttd_check_all_out$data[failed==FALSE]$prioritised)/nrow(ttd_check_all_out$data[failed==FALSE])*100`% of the non-failed therapeutic targets and 0% of the terminated/withdrawn therapeutic targets (n=`r nrow(ttd_check_all_out$data[failed==TRUE])`).
Here we found that our prioritised targets were highly significantly depleted for failed therapeutics (($p=$`r ttd_check_all_out$hypergeo[status=="failed"]$p`, $\text{odds ratio=}$`r ttd_check_all_out$hypergeo[status=="failed"]$OR`, $\text{sensitivity}$=`r ttd_check_all_out$hypergeo[status=="failed"]$sensitivity`, $\text{specificity}$=`r ttd_check_all_out$hypergeo[status=="failed"]$specificity`)).
This suggests that our multi-scale evidence-based prioritisation pipeline is capable of selectively identifying genes that are likely to be effective therapeutic targets.

```{r rpe65}
rpe65 <- prioritise_targets_out$top_targets[gene_symbol=="RPE65"]
```

In addition to aggregate enrichment results, we also provide specific examples of successful gene therapies whose cell type-specific mechanism were recapitulated by our phenotype-cell associations.
In particular, our pipeline nominated the gene *RPE65* within '`r names(sort(table(rpe65$cl_name), decreasing = TRUE))[1]`s' as the top target for '`r unique(rpe65$hpo_name)[1]`' vision-related phenotypes that are hallmarks of 'Leber congenital amaurosis, type II' and 'Severe early-childhood-onset retinal dystrophy'.
Indeed, gene therapies targeting *RPE65* within the retina of patients with these rare genetic conditions are some of the most successful clinical applications of this technology to date, able to restore vision in many cases [@chiu2021].
In other cases, a tissue (e.g. liver) may be known to be causally involved in disease genesis, but the precise causal cell types within that tissue remain unknown (e.g. heptocytes, Kupffer cells, Cholangiocytes, Hepatic stellate cells, Natural killer cells, etc.).
Tissue-level investigations (e.g. using bulk transcriptomics or epigenomics) would be dominated by hepatocytes, which comprise 75% of the liver.
Our prioritized gene therapy targets can aid in such scenarios by providing the cell type-resolution context most likely to be causal for a given phenotype or set of phenotypes.

::: {#fig-therapy-validate .hide-subcaptions}
```{r fig-therapy-validate}
#| label: fig-therapy-validate
#| fig-cap: __Prioritised targets recapitulate existing gene therapy targets.__ The proportion of existing gene therapy targets (documented in the Therapeutic Target Database) recapitulated by our prioritisation pipeline. Therapeutics are stratified by the stage of clinical development they were at during the time of writing. While our prioritized targets did not include any failed ('Terminated') therapies, the fact that only one such therapy exists in the dataset preclude us from making any conclusions about depletion of failed gene therapy targets in our prioritised targets list. 
#| fig-height: 7
#| fig-width: 6

ttd_check_out$plot 
```
:::

### Selected example targets

```{r prioritise_targets_grid}
res_class <- HPOExplorer::gpt_annot_class()
prioritise_targets_grid_out <- MSTExplorer::prioritise_targets_grid(
  top_targets = prioritise_targets_out$top_targets, 
  res_class = res_class,
  width = 65,
  keep_severity_class = c("profound","severe"),
  keep_physical_malformations = c(0,1), 
  show_plot=FALSE)
```

```{r top_targets}
severity_score_gpt_threshold <- 15
top_targets <- prioritise_targets_out$top_targets[severity_score_gpt>severity_score_gpt_threshold,]
top_phenotypes <- unique(top_targets$hpo_name)
```

```{r prioritise_targets_network-examples}
height <- "100vh"
width <- "100vw"
phenotypes_network <- c(
"GM2-ganglioside accumulation",
"Spinocerebellar atrophy",
"Neuronal loss in central nervous system")
vn_therapy <- lapply(stats::setNames(phenotypes_network, phenotypes_network), function(phenotype){
  MSTExplorer::prioritise_targets_network(
    top_targets = top_targets[hpo_name==phenotype], 
    main = NULL,
    height = height,
    width = width,
    submain = NULL)$plot
}) 
```

::: {#fig-therapy-examples .hide-subcaptions}
```{r fig-therapy-examples}
#| label: fig-therapy-examples
#| fig-cap: __Evidence-based pipeline nominates causal mechanisms to target for gene therapy.__ Shown here are the top 40 prioritised gene therapy targets at multiple biological scales, stratified by congenital (top row) vs. non-congential phenotypes (bottom row) as well as severity class ("profound" or "severe"). In this plot, only the top 10 most severe phenotypes within a given strata/substrata are shown **a,c**, Severity annotation generated by GPT-4. **b,d**, Composite severity scores computed across all severity metrics. **e,g**, Top mediator disease and cell type-specific target for each phenotype. **f,h** top target gene for each phenotype within humans (*Homo sapiens*). We also include the 1:1 ortholog of each human gene in several commonly used animal models, including monkey (*Macaca mulatta*), mouse (*Mus musculus*), zebrafish (*Danio rerio*), fly (*Drosophila melanogaster*) and nematode (*Caenorhabditis elegans*). Boxes are empty where no 1:1 ortholog is known. See supplement [Supp. Fig. @fig-therapy-examples2] for network plots of  cell type-specific gene therapy targets for several severe phenotypes and their associated diseases.
#| fig-height: 12
#| fig-width: 20

if(knitr::is_html_output()){
  knitr::opts_current$set(results="asis")
  prioritise_targets_grid_out$plot 
  vn_therapy
} else {
  # img_paths <- stats::setNames(
  #   here::here( 
  #     "img","fig-therapy-examples",
  #  c("tay-sachs.png",
  #    "spinocerebellar-atrophy.png",
  #    "neuronal-loss.png")),
  #   c("GM2-ganglioside accumulation",
  #     "Spinocerebellar atrophy",
  #     "Neuronal loss in central nervous system")
  # )
  # fig_networks <- lapply(img_paths, function(x){
  #   i = which(x==img_paths)
  #   magick::image_read(x, density=300)|>magick::image_ggplot(interpolate=TRUE) +
  #   ggplot2::theme_void() +
  #   ggplot2::labs(subtitle=names(img_paths)[i]) +
  #   ggplot2::theme(plot.margin = ggplot2::unit(c(0,0,0,0),"pt"), 
  #                  panel.background = ggplot2::element_rect(fill = "grey50"))
  # }) 
  
  
  
  # fig_therapy_examples <- ( (
  # (patchwork::plot_spacer()|prioritise_targets_grid_out$plot) +
  #   patchwork::plot_layout(widths = c(0,1)) &  
  #   ggplot2::theme(legend.direction = "vertical", 
  #                  legend.position = "bottom")
  # ) /
  # (
  #   (fig_networks$`GM2-ganglioside accumulation` | fig_networks$`Spinocerebellar atrophy`) / 
  #     fig_networks$`Neuronal loss in central nervous system`
  # ) )+ 
  #   patchwork::plot_layout(heights = c(1,3)) +
  #  patchwork::plot_annotation(tag_levels="a")
  # fig_therapy_examples 
  
  
  # ggplot2::ggsave(fig_therapy_examples, filename="~/Downloads/fig_therapy_examples.png", height=30, width=18, dpi=300)
  
  # ggplot2::ggsave("~/Downloads/prioritise_targets_grid.pdf",
  #                  (patchwork::plot_spacer()|prioritise_targets_grid_out$plot) +
  #   patchwork::plot_layout(widths = c(0,1)) + 
  #     patchwork::plot_annotation(tag_levels="a") &  
  #   ggplot2::theme(legend.direction = "vertical", 
  #                  legend.position = "bottom"), 
  #   height=10, width=19
  # ) 
  
  ## Combined plots in Illustrator for more flexibility
  # magick::image_read(here::here("img","fig-therapy-examples.png"), 
  #                    density=400) |>
  #   magick::image_scale(geometry = "2000x2000>")
    # magick::image_resize("1000x1000",)
  
  prioritise_targets_grid_out$plot +
   patchwork::plot_annotation(tag_levels="a") 
} 
```
:::

From our prioritised targets, we selected four phenotype or disease examples: `r paste(shQuote(phenotypes_network), collapse=", ")`.
To focus on clinically relevant phenotypes and reduce overplotting, we limited selection to those with GPT severity scores above `r severity_score_gpt_threshold` [Supp. Fig. @fig-therapy-examples2].
Selection was based on severity and network simplicity to allow compact visualisation.

Tay-Sachs disease (TSD) is a fatal neurodegenerative condition caused by *HEXA* deficiency and ganglioside buildup.
We identified alternatively activated macrophages as the cell type most associated with 'GM2-ganglioside accumulation' [Supp. Fig. @fig-therapy-examples2].
This aligns with prior findings of ganglioside accumulation in TSD macrophages [@fendersonChapterDevelopmentalGenetic2009; @vilcaesGangliosideSynthesisPlasma2020.; @yoheGangliosideAlterationsStimulated1985; @demirGM2GangliosideAccumulation2020].
Our results support macrophages as causal in TSD and the most promising therapeutic target.

```{r sca}
sca <- results[q<0.05 & hpo_name=='Spinocerebellar atrophy']
```

Spinocerebellar atrophy is a progressive neurodegenerative phenotype in disorders like Spinocerebellar ataxia.
Our pipeline implicates M2 macrophages ('Alternatively activated macrophages') as the only causal cell type [Supp. Fig. @fig-therapy-examples2].
This suggests Purkinje cell loss is downstream of macrophage dysfunction, consistent with microglial roles in neurodegeneration [@ferroRoleMicrogliaAtaxias2019; @holMicroglialTranscriptomicsMeets2022; @lopesAtlasGeneticEffects2020a].
Our findings provide the first statistically supported link between risk genes and this cell type, which is supported by relevant mouse models (e.g. *Atxn1*, *Pnpla6*) that replicate cellular and behavioural disease phenotypes.

```{r neuronal_loss}
neuronal_loss <- results[q<0.05 & hpo_name=='Neuronal loss in central nervous system']
```

'Neuronal loss in the central nervous system' is a phenotype by multiple serious diseases (e.g. Huntington disease, frontotemporal lobar degeneration, and certain mitochondrial disorders).
Across all of these diverse conditions with varying genetic causes (\>8 genes), these conditions converge on just 2 cell types: M2 macrophages and epithelial cells.

Additional examples of therapeutics targets include; cardiac muscle and endothelial cells in pheontypes associated with respiratory failure ([Supp. Fig. @fig-therapy-examples-supp-rf]), microglia in frontal lobe dementia ([Supp. Fig. @fig-therapy-examples-supp-dementia]), chondrocytes in lethal skeletal dyplasia ([Supp. Fig. @fig-therapy-examples-supp-lsd]), endothelial cells in small vessel disease ([Supp. Fig. @fig-therapy-examples-supp-svd]), oligodendrocytes and neurons in Parkinson's disease ([Supp. Fig. @fig-therapy-examples-supp-pd]).
and multiple gastrointestinal and immune cell types in Alzheimer's disease ([Supp. Fig. @fig-therapy-examples-supp-ad]).
For further details please refer to the Supplementary Results.

### Mappings

```{r maps_agg}
maps <- HPOExplorer::get_mappings(max_dist = NULL) 
maps_agg <- (
  data.table::rbindlist(maps, idcol = "source")
 )[,list("source terms"=data.table::uniqueN(curie_id), 
         "HPO terms"=data.table::uniqueN(mapped_curie),
         "mappings"=.N),
   keyby=c("source","distance")]
maps_agg_top <- maps_agg[distance==1][get("HPO terms")==max(get("HPO terms")),]

```

Mappings from HPO phenotypes and other commonly used medical ontologies (SNOMED, UMLS, ICD-9, and ICD-10) were gathered using the Ontology Xref Service (OxO; <https://www.ebi.ac.uk/spot/oxo/>) to facilitate others using our results in future work.
Direct mappings, with a cross-ontology distance of 1, are the most precise and reliable.
Counts of mappings at each distance are shown in @tbl-mappings.
In total, there were `r sum(maps_agg[distance==1][["HPO terms"]])` direct mappings between the HPO and other ontologies, with the largest number of mappings coming from the `r maps_agg_top$source[1]` ontology (`r maps_agg_top[["source terms"]]` UMLS terms).

## Discussion {#sec-discussion}

```{r}
notfail <- ttd_check_out$data[DRUGTYPE=="Gene therapy" & failed==FALSE, drop=FALSE]
ttd_nonfailed_val <- sum(notfail$prioritised)/nrow(notfail)*100 |> round(2)
```

Investigating rare diseases (RDs) at the phenotype level offers advantages in research and clinical medicine.
Most RDs have a single causal gene (`r nrow(per_disease[ng==1])`/`r nrow(per_disease)` = `r format(nrow(per_disease[ng==1])/nrow(per_disease)*100,digits=1)`%).
Therefore aggregating genes into phenotype-based sets enables well-powered analyses (mean \~`r mean(per_phenotype$ng)` genes/phenotype).
Phenotypes often converge on shared molecular pathways, and a phenotype-centric approach captures interindividual variation in disease presentations.
This requires mapping the molecular and cellular mechanisms behind each phenotype, which we achieve here at phenome scale.

Across `r res_summ_all[["cell types"]]` cell types and `r res_summ_all[["phenotypes"]]` phenotypes, we found \>`r res_summ_all[["tests significant"]]` significant phenotype–cell type relationships, enabling multi-scale mechanistic tracing.
Results replicate known links, add cellular context, and uncover novel associations.
Extensive benchmarking confirmed expected associations, aided by comprehensive phenotype and cell type ontologies.
Key findings include enrichment of anatomically matched associations, correlation of phenotype specificity with association strength, precise subtypes for recurrent infections, and links between congenital onset frequency and developmental cell types.

It is a matter of ongoing scientific debate as to whether the rare, more monogenic versions of diseases and phenotypes share the same genetic etiology as their common, polygenic counterparts [@katsanis2016, @freund2018].
Nevertheless, we identified several examples where these two disparate data sources converge on the same cell types.
These include the implication of smooth muscle cells and endothelial cells in small vessel disease [@ziegler2025, @bryois2020, @sargurupremraj2020], and the consistent association between macrophages and neurodegenerative diseases such as Alzheimer's and Parkinson’s [@nott2023, @novikova2019, @heritabi, @smajic2022, @ziegler2025].
Additionally, we recover genome-wide association study (GWAS) -derived associations between macular degeneration and specific cell types of the eye (photoreceptor cells, retinal pigment cells, Muller cells), immune system (dendritic cells, macrophages), and vascular system (endothelial cells) [@menon2019, @lyu2021, @lindström2023, @strunz2020].
A more systematic, phenome-wide comparison of cell type association in rare vs. common forms of diseases is warranted, though remains outside the scope of our present study.

Despite our growing knowledge of RD genetics, less than 5% of RDs have treatments [@Halley2022-pd].
However, advances in CRISPR, prime editing, antisense oligonucleotides, viral/lipid delivery [@Bueren2023-ma; @Bulaklak2020-ta; @Kohn2023-vh; @Zhao2023-qy] are accelerating.
The FDA’s new program [@Lu2024-kl] aims to expand gene/cell therapy approvals in years rather than decades [@Brown2022-ye], but success depends on understanding the causal mechanisms of each RD.
Here, we built a reproducible pipeline for nominating cell type–resolved therapeutic targets ([Fig. @fig-therapy-examples]), factoring in association strength, gene specificity, severity, therapy delivery suitability, and model translatability.
We recovered `r ttd_nonfailed_val`% of active gene therapies, confirming strong enrichment.
Highlighted cases include macrophage-driven phenotypes in Tay-Sachs, spinocerebellar ataxia, and Alzheimer’s disease, pinpointing specific phenotypes (e.g. neurofibrillary tangles) to causal cell types.

Current limitations of our study include missing certain rare cell subtypes and states (e.g. immune cell responses, diseased states, aging) and incomplete knowledge of gene–phenotype associations.
With the expectation that data will continue to improve over time, our pipeline is fully containerised and documented for end-to-end reproducibility.
Comprehensive, ontology-driven frameworks like ours enable discovery, diagnosis, and basket trial design for shared molecular etiologies across many diseases [@Zanello2023-zd].
Furthermore, we invite collaborations to validate and translate these predictions, and have publicly released all results via R packages and the Rare Disease Celltyping Portal (<https://neurogenomics-ukdri.dsi.ic.ac.uk/>) to support broad access for researchers, clinicians, and patients.

In summary, we present a scalable, cost-effective, and reproducible method for phenome-wide, cell type–specific mechanism prediction in RDs.
With advances in gene therapy and supportive regulatory changes, this approach can help realise the promise of genomic medicine for the global RD community.

## Methods {#sec-methods}

### Human Phenotype Ontology

```{r get_gencc}
gencc <- KGExplorer::get_gencc(agg_by = NULL)
gencc_version <- KGExplorer::get_version(gencc, return_version = TRUE)
```

The latest version of the HPO (release `r KGExplorer::get_version(hpo, return_version = TRUE)`) was downloaded from the EMBL-EBI Ontology Lookup Service [@Cote2010-gp] and imported into R using the `HPOExplorer` package.
This R object was used to extract ontological relationships between phenotypes as well as to assign absolute and relative ontological levels to each phenotype.
The latest version of the HPO phenotype-to-gene mappings and phenotype annotations were downloaded from the official HPO GitHub repository and imported into R using `HPOExplorer`.
This contains lists of genes associated with phenotypes via particular diseases, formatted as three columns in a table (gene, phenotype, disease).

However, not all genes have equally strong evidence of causality with a disease or phenotype, especially when considering that the variety of resources used to generate these annotations (OMIM, Orphanet, DECIPHER) use variable methodologies (e.g. expert-curated review of the medical literature vs. automated text mining of the literature).
Because of this, formalizing phenotypes (or any biological entity) as unweighted gene sets can lead to loss of information about the most relevant signals.
This is especially true when gene sets become large and poorly supported, essentially decreasing the signal-to-noise ratio [@mathur2018, @irizarry2009, @maleki2020].

Therefore we imported data from the Gene Curation Coalition (GenCC) [@DiStefano2022-ao; @DiStefano2023-np], which (as of `r gencc_version`) `r nrow(gencc)` evidence scores across `r data.table::uniqueN(gencc$disease_curie)` diseases and `r data.table::uniqueN(gencc$gene_curie)` genes.
Evidence scores are defined by GenCC using a standardised ordinal rubric which we then encoded as a semi-quantitative score ranging from 0 (no evidence of disease-gene relationship) to 6 (strongest evidence of disease-gene relationship) (see @tbl-gencc).
As each Disease-Gene pair can have multiple entries (from different studies) with different levels of evidence, we then summed evidence scores per Disease-Gene pair to generate aggregated Disease-by-Gene evidence scores.
This procedure can be described as follows.

Let us denote:

-   $D$ as diseases.

-   $P$ as phenotypes in the HPO.

-   $G$ as genes

-   $S$ as the evidence scores describing the strength of the relationship between each Disease-Gene pair.

-   $M_{ij}$ as the aggregated Disease-by-Gene evidence score matrix.

$$
M_{ij} = \sum_{k=1}^{\text{f}} D_i G_j S_k
$$

Next, we extracted Disease-Gene-Phenotype relationships from the annotations file distributed by the HPO (*phenotype_to_genes.txt*).
This provides a list of genes associated with phenotypes via particular diseases, but does not include any strength of evidence scores.

Here we define:  

-   $A_{ijk}$ as the Disease-Gene-Phenotype relationships.  

-   $D_i$ as the $i$th disease.  

-   $G_j$ as the $j$th gene.  

-   $P_k$ as the $k$th phenotype.

$$
A_{ijk} = D_i G_j P_k
$$

In order to assign evidence scores to each Phenotype-Gene relationship, we combined the aforementioned datasets from GenCC ($M_{ij}$) and HPO ($A_{ijk}$) by merging on the gene and disease ID columns.
For each phenotype, we then computed the mean of Disease-Gene scores across all diseases for which that phenotype is a symptom.
This resulted in a final 2D tensor of Phenotype-by-Gene evidence scores ($L_{ij}$):

::: {#eq-evidence-scores .content-hidden unless-format="html"}
![](equations/eq1.png){height="300px"}

\

**Construction of the tensor of Phenotype-by-Gene evidence scores.**
:::

\
\

::: {.content-visible unless-format="html"}
$$
 \eqnmarkbox[NavyBlue]{n1}{ L_{ij} } = 
 \begin{cases}
  \frac{\sum_{k=1}^{\text{f}} 
    \eqnmarkbox[Cerulean]{n3a}{D_i G_j} 
    \eqnmarkbox[BlueViolet]{n5}{P_k} 
    }{\text{f}}, 
  \text{if } \eqnmarkbox[Cerulean]{n3b}{D_i G_j} 
    \in \eqnmarkbox[blue]{n4a}{A},
  \\
  1, \hspace{2.2cm}
  \text{if } \eqnmarkbox[Cerulean]{n3c}{D_i G_j} 
    \notin\eqnmarkbox[blue]{n4b}{A}
 \end{cases}
$$ \annotate[yshift=1em]{left}{n1}{Tensor of Phenotype-by-Gene \\evidence scores} \annotate[yshift=2.5em]{right}{n3a,n3b,n3c}{Tensor of Disease-by-Gene \\evidence scores} \annotate[yshift=-2.5em]{below,left}{n4a,n4b}{Disease-by-Gene-by-Phenotype \\relationships} \annotate[yshift=1em, xshift=-1em]{right}{n5}{Phenotype}

\

**Construction of the tensor of Phenotype-by-Gene evidence scores.**
:::

\

Histograms of evidence score distributions at each step in processing can be found in [Fig. @fig-evidence-histograms].

### Single-cell transcriptomic atlases

In this study, the gene by cell type specificity matrix was constructed using the Descartes Human transcriptome atlas of foetal gene expression, which contains a mixture of single-nucleus and single-cell RNA-seq data (collected with sci-RNA-seq3) [@Cao2020-qz].
This dataset contains 377,456 cells representing 77 distinct cell types across 15 tissues.
All 121 human foetal samples ranged from 72 to 129 days in estimated postconceptual age.
To independently replicate our findings, we also used the Human Cell Landscape which contains single-cell transcriptomic data (collected with microwell-seq) from embryonic, foetal, and adult human samples across 49 tissues [@Han2020-iq].

Specificity matrices were generated separately for each transcriptomic atlas using the R package `EWCE` (v1.11.3) [@Skene2016-rb].
Within each atlas, cell types were defined using the authors' original freeform annotations in order to preserve the granularity of cell subtypes as well as incorporate expert-identified rare cell types.
Cell types were only aligned and aggregated to the level of corresponding Cell Ontology (CL) [@Diehl2016-gt] annotations afterwards when generating summary figures and performing cross-atlas analyses.
Using the original gene-by-cell count matrices from each single-cell atlas, we computed gene-by-cell type expression specificity matrices as follows.
Genes with very no expression across any cell types were considered to be uninformative and were therefore removed from the input gene-by-cell matrix $F(g,i,c)$.

Next, we calculated the mean expression per cell type and normalised the resulting matrix to transform it into a gene-by-cell type expression specificity matrix ($S_{g,c}$).
In other words, each gene in each cell type had a 0-1 score where 1 indicated the gene was mostly specifically expressed in that particular cell type relative to all other cell types.
This procedure was repeated separately for each of the single-cell atlases and can be summarised as:

::: {#eq-ctd-specificity .content-hidden unless-format="html"}
![](equations/eq3.png){height="300px"}

**Construction of the gene-by-cell type specificity matrix.**
:::

\

::: {.content-visible unless-format="html"}
\begin{equation*}
  \eqnmarkbox[orange]{s1}{S_{gc}}
  =
  \frac{
    \eqnmarkbox[purple]{s3a}{
      \frac{
        \sum_{i=1}^{|L|} F_{gic}
      }{
        N_c  
      }
    } 
  }{
   \eqnmarkbox[OrangeRed]{s6}{\sum_{r=1}^{k}}(
     \eqnmarkbox[purple]{s3b}{
      \frac{
        \sum_{i=1}^{|L|} F_{gic}
      }{
        N_c  
      }
    } 
   ) 
  }
\end{equation*} \annotate[yshift=1em]{left}{s1}{Gene-by-cell type specificity matrix} \annotate[yshift=2em]{left}{s3a,s3b}{Compute mean expression of each gene per cell type} \annotate{below,left}{s6}{Compute row sums of \\mean gene-by-cell type matrix}
:::

\

### Phenotype-cell type associations

To test for relationships between each pairwise combination of phenotype (n=`r res_summ_all[["phenotypes"]]`) and cell type (n=`r res_summ_all[["cell types"]]`) we ran a series of univariate generalised linear models implemented via the `MSTExplorer::run_phenomix` function in R (which uses `stats::glm` internally).
Framing association testing as a regression problem rather than a gene set enrichment problem has additional benefits, including speed.
This allowed us to complete all tests within \~30 min on a MacBook Pro with 8 cores, while early tests indicated that gene set enrichment-based methods like `EWCE` [@skene2016] would take days to weeks on a high-performance computing cluster to complete the same number of tests due to its computationally expensive bootstrapping procedure.
These tests also showed that `EWCE` was unable to recover nearly as many true positive results (as indicated by our MKG benchmark) with a comparable degree of precision (see related Issue for further details: <https://github.com/neurogenomics/rare_disease_celltyping/issues/51>).

First, we filtered the gene-by-phenotype evidence score matrix ($L_{ij}$) and the gene-by-cell type expression specificity matrix ($S_{gc}$) to only include genes present in both matrices (n=4,949 genes in the Descartes Human analyses; n=4,653 genes in the Human Cell Landscape analyses).
Then, within each matrix any rows or columns with a sum of 0 were removed as these were uninformative data points that did not vary.
To improve interpretability of the results $\beta$ coefficient estimates across models (i.e. effect size), we performed a scaling prestep on all dependent and independent variables.
Initial tests showed that this had virtually no impact on the total number of significant results or any of the benchmarking metrics based on p-value thresholds [Fig. @fig-summary].
This scaling prestep improved our ability to rank cell types by the strength of their association with a given phenotype as determined by separate linear models.

We repeated the aforementioned procedure separately for each of the single-cell references.
Once all results were generated using both cell type references (`r res_summ_all[["tests"]]` association tests total), we applied Benjamini-Hochberg false discovery rate [@Benjamini1995-vo] (denoted as $FDR_{pc}$) to account for multiple testing.
Of note, we applied this correction across all results at once (as opposed to each single-cell reference separately) to ensure the $FDR_{pc}$ was stringently controlled for across all tests performed in this study.

For the significant phenotype-cell type associations, we also wished to identify which genes were most strongly driving this association. We therefore designed a heuristic to consistently extract such *driver genes* . For a given phenotype-cell type pair, *driver genes* were defined as the intersect of genes that had a phenotype evidence score >0 and were within the top 75th expression specificity percentile (quantiles 30-40 out of 40) for the associated cell type. This be described as follows.

Let 

-  $G$ be the full set of genes,
-  $s_i$ be the phenotype evidence score for gene $i$,
-  $e_i$ be the expression specificity of gene $i$ in the associated cell type,
-  $q(e_i)$ be the empirical specificity percentile of $e_i$ (ranging from 1 to 40).


We define two subsets:
$$
A = \{\, i \in G : s_i > 0 \,\},
$$
$$
B = \{\, i \in G : q(e_i) \ge 30 \,\}.
$$

The set of driver genes for that phenotype--cell type pair is then
$$
D = A \cap B.
$$


A programmatic implementation of this heuristic can be found in the `MSTExplorer::add_driver_genes` R function.


### Symptom-cell type associations

```{r}
p2g_eg <- HPOExplorer::add_disease(p2g[hpo_name=="Headache"], add_descriptions = TRUE)

# dplyr::group_by(p2g_eg, hpo_id, hpo_name, disease_id, disease_name)|> 
#   dplyr::summarise(count=length(unique(gene_symbol)), 
#                    genes=paste(gene_symbol,collapse = ","))|>
#   dplyr::filter(count==3)|>
#   dplyr::arrange(disease_name) |>
#   data.frame()#|>head(30)
```


In the HPO, a phenotype can be associated with multiple diseases via different subsets of genes ([Supp. Fig. @fig-diagram]).
Here we define a symptom as a phenotype as it presents within the context of the specific disease.
A symptom’s gene set is therefore the subset of genes connecting a phenotype to a specific disease.
For example, the phenotype 'Headache' (*HP:0002315*) is associated with `r data.table::uniqueN(p2g_eg$gene_symbol)` genes across different `r data.table::uniqueN(p2g_eg$disease_id)` diseases. 
Our association tests described above use evidence scores from all `r data.table::uniqueN(p2g_eg$gene_symbol)` of these genes when testing for a relationship between 'Headache' and various cell types.
However, the disease 'Erythrocytosis, familial, 1' (*OMIM:133100*) is linked to the 'Headache' phenotype via a single gene; *EPOR*. Thus, the symptom 'Heachache due to Erythrocytosis, familial, 1' would have only *EPOR* as its gene list.
Another disease, 'Migraine with or without aura, susceptibility to, 1' (*OMIM:157300*) is linked with 'Headache' via 3 different genes (*ESR1*, *TNF*, *EDNRA*), so the symptom 'Heachache' in the presence of this disease would consist of these 3 genes.
 
More formally, the features of a given symptom can be described as the subset of genes annotated to phenotype $p$ via a particular disease $d$, denoted as $G_{dp}$ ([see Fig. @fig-diagram]).
We then computed the intersect between symptom genes ($G_{dp}$) and driver genes (described in the previous section, and denoted as $G_{pc}$), resulting in the gene subset $G_{d \cap p \cap c}$.
Only $G_{d \cap p \cap c}$ gene sets with 25% or greater overlap with the symptom gene subset ($G_{dp}$) were kept.
This procedure was repeated for all phenotype-cell type-disease triads, which can be summarised as follows:

::: {#eq-symptoms .content-hidden unless-format="html"}
![](equations/eq4.png){height="300px"}
:::

\

::: {.content-visible unless-format="html"}
\begin{equation*}
  \frac{
     \eqnmarkbox[Chartreuse3]{g1}{|G_{d \cap p \cap c} |}
    }{
       \eqnmarkbox[Emerald]{g2}{|G_{dp}|}} 
  \geq \eqnmarkbox[SeaGreen]{g3}{.25} 
\end{equation*} \annotate[yshift=1em]{left}{g1}{Intersect between \\symptom genes ($G_{dp}$) and driver genes ($G_{pc}$)} \annotate[yshift=-1em]{below,left}{g2}{Symptom genes \\(i.e. genes annotated to a phenotype\\ via a specific disease)} \annotate[yshift=-1em]{below,right}{g3}{Minimum proportion of overlap \\between $G_{dpc}$ and $G_{dp}$}
:::

\
\

### Validation of expected phenotype-cell type relationships

We first sought to confirm that our tests (across both single-cell references) were able to recover expected phenotype-cell type relationships across seven high-level branches within the HPO ([Fig. @fig-summary]), including abnormalities of the cardiovascular system, endocrine system, eye, immune system, musculoskeletal system, nervous system, and respiratory system.
Within each branch the number of significant tests in a given cell type were plotted ([Fig. @fig-summary]b).
Mappings between freeform annotations (the level at which we performed our phenotype- cell type association tests) provided by the original atlas authors and their closest CL term equivalents were provided by CellxGene [@CZI_Single-Cell_Biology_Program2023-fs].
CL terms along the *x-axis* of [Fig. @fig-summary]b were assigned colours corresponding to which HPO branch showed the greatest number of enrichments (after normalising within each branch to account for differences in scale).
The normalised colouring allows readers to quickly assess which HPO branch was most often associated with each cell type, while accounting for differences in the number of phenotypes across branches.
We then ran a series of Analysis of Variance (ANOVA) tests to determine whether (within a given branch) a given cell type was more often enriched (FDR\<0.05) within that branch relative to all of the other HPO branches of an equivalent level in the ontology (including all branches not shown in [Fig. @fig-summary]b).
After applying Benjamini-Hochberg multiple testing correction [@Benjamini1995-vo] (denoted as $FDR _{b,c}$), we annotated each respective branch-by-cell type bar according to the significance (\*\*\*\* : $FDR _{b,c}<1e-04$, \*\*\* : $FDR _{b,c}<0.001$, \*\* : $FDR _{b,c}<0.01$, \* : $FDR _{b,c}<0.05$).
Cell types in [Fig. @fig-summary]a-b were ordered along the *x-axis* according to a dendrogram derived from the CL ontology ([Fig. @fig-summary]c), which provides ground-truth semantic relationships between all cell types (e.g. different neuronal subtypes are grouped together).

As an additional measure of the accuracy of our phenotype-cell types test results we identified conceptually matched branches across the HPO and the CL ([Fig. @fig-summary]d and @tbl-celltypes).
For example, 'Abnormality of the cardiovascular system' in the HPO was matched with 'cardiocytes' in the CL which includes all cell types specific to the heart.
Analogously, 'Abnormality of the nervous system' in the HPO was matched with 'neural cell' in the CL which includes all descendant subtypes of neurons and glia.
This cross-ontology matching was repeated for each HPO branch and can be referred to as on-target cell types.
Within each branch, the $-log_{10}(FDR _{pc})$ values of on-target cell types were binned by rounding to the nearest integer (*x-axis*) and the percentage of tests for on-target cell types relative to all cell types were computed at each bin (*y-axis*) ([Fig. @fig-summary]d).
The baseline level (dotted horizontal line) illustrates the percentage of on-target cell types relative to the total number of observed cell types.
Any percentages above this baseline level represent greater than chance representation of the on-target cell types in the significant tests.

### Validation of inter- and intra-dataset consistency

We tested for inter-dataset consistency of our phenotype-cell type association results across different single-cell reference datasets (Descartes Human and Human Cell Landscape).
<!-- For all tests reported here, the relevant association metrics (p-values or effect size) were first averaged to the level of ancestral HPO terms (5 levels down the hierachy) to reduce figure size. --> For association tests with exactly matching Cell Ontology ID across the two references, we tested for a relationship between the effect sizes (each GLM model's $R^2$ estimates) generated with each of the references by fitting linear regression model (`stats::lm` via the R function `ggstatsplot::ggscatterstats`).
We repeated this regression analysis using only the phenotype-cell type associations that were significant (FDR \< 0.05) in both reference datasets.

We also tested for intra-dataset consistency within the Human Cell Landscape by running additional linear regressions between the phenotype-cell type association test statistics of the foetal and the adult samples (again using estimates from all results, and significant-only results).
While we would not expect the same exact cell type associations across different developmental stages, we would nevertheless expect there to be some degree of correlation between the developing and mature versions of the same cell types.

Finally, we compute the symmetric replication rate within each of the comparison described above (between scRNA-seq references, and between developmental stages within the Human Cell Landscape).
This was defined as the proportion of significant results (FDR \< 0.05) in dataset A that were also significant (FDR \< 0.05) in dataset B, and vice versa, averaged across both directions.
This can be described as follows.

**Variable descriptions**  

-   $S_{A}$: The set of phenotype–cell-type associations that pass the significance threshold (e.g., $FDR < 0.05$) in dataset $A$.

-   $S_{B}$: The set of phenotype–cell-type associations that pass the same significance threshold in dataset $B$.

-   $S_{A} \cap S_{B}$: The set of associations that are significant in *both* datasets.

-   $RR$: Replication rate.

**Directional replication rate (A → B)**

$$ 
\text{RR}_{A \rightarrow B}
=
\frac{
\left| 
S_{A} \cap S_{B}
\right|
}{
\left|
S_{A}
\right|
}
$$

**Directional replication rate (B → A)**

$$
\text{RR}_{B \rightarrow A}
=
\frac{
\left| 
S_{A} \cap S_{B}
\right|
}{
\left|
S_{B}
\right|
}
$$

**Symmetric replication rate**

$$
RR_{sym} 
=
\frac{
\left(
        RR_{A \rightarrow B}
        +
            RR_{B \rightarrow A}
        \right)
}{2}
$$

### More specific phenotypes are associated with fewer genes and cell types

To explore the relationship between HPO phenotype specificity and various metrics from our results, we computed the information content (IC) scores for each term in the HPO.
IC is a measure of how much specific information a term within an ontology contains.
In general, terms deeper in an ontology (closer to the leaves) are more specific, and thus informative, than terms at the very root of the ontology (e.g. 'Phenotypic abnormality').
Where $k$ denotes the number of offspring terms (including the term itself) and $N$ denotes the total number of terms in the ontology, IC can be calculated as:

$$
IC=-log(\frac{k}{N})
$$

Next, IC scores were quantised into 10 bins using the `ceiling` R function to improve visualisation.
We then performed a series of linear regressions between phenotype binned IC scores and: 1) number of genes annotated per HPO phenotype, 2) the number of significantly associated cell types per HPO phenotype, and 3) the model estimate of each significant phenotype-cell type associations (at FDR \< 0.05) after taking the log of the absolute value ($log_2(|estimate|)$).

### Monarch Knowledge Graph recall

Finally, we gathered known phenotype-cell type relationships from the Monarch Knowledge Graph (MKG), a comprehensive database of links between many aspects of disease biology [@Putman2024-et].
This currently includes `r nrow(kg_hp)` links between HPO phenotypes (n=`r data.table::uniqueN(kg_hp$from)`) and CL cell types (n=`r data.table::uniqueN(kg_hp$to)`).
Of these, we only considered the `r data.table::uniqueN(validate_associations_mkg_out$kg_filt)` phenotypes that we were able to test given that our ability to generate associations was dependent on the existence of gene annotations within the HPO.
We considered instances where we found a significant relationship between exactly matching pairs of HPO-CL terms as a hit.

However, as the cell types in MKG were not necessarily annotated at the same level as our single-cell references, we considered instances where the MKG cell type was an ancestor term of our cell type (e.g. 'myeloid cell' vs. 'monocyte'), or *vice versa*, as hits.
We also adjusted ontological distance by computing the ratio between the observed ontological distance and the smallest possible ontological distance for that cell type given the cell type that were available in our references ($dist_{adjusted}=(\frac{dist_{observed}+1}{dist_{minimum}+1})-1$).
This provides a way of accurately measuring how dissimilar our identified cell types were for each phenotype-cell type association ([Fig. @fig-monarch-recall]).

### Prioritising phenotypes based on severity

```{r code_dict}
gpt_codes <- formals(HPOExplorer::gpt_annot_codify)
code_dict <- paste0(shQuote(names(eval(gpt_codes$code_dict))),"=",
                     eval(gpt_codes$code_dict), collapse = ", ")
weights_dict <- paste0(shQuote(names(eval(gpt_codes$weights_dict))),"=",
                     eval(gpt_codes$weights_dict), collapse = ", ") 
```

Only a small fraction of the the phenotypes in HPO (\<1%) have metadata annotations containing information on their time course, consequences, and severity.
This is due to the time-consuming nature of manually annotating thousands of phenotypes.
To generate such annotations at scale, we previously used Generative Pre-trained Transformer 4 (GPT-4), a large language model (LLM) as implemented within OpenAI's Application Programming Interface (API) [@murphyHarnessingGenerativeAI2024].
After extensive prompt engineering and ground-truth benchmarking, we were able to acquire annotations on how often each phenotype directly causes intellectual disability, death, impaired mobility, physical malformations, blindness, sensory impairments, immunodeficiency, cancer, reduced fertility, or is associated with a congenital onset.
These criteria were previously defined in surveys of medical experts as a means of systematically assessing phenotype severity [@Lazarin2014-we].
Responses for each metric were provided in a consistent one-word format which could be one of: `r paste0(shQuote(names(eval(gpt_codes$code_dict))), collapse=", ")`.
This procedure was repeated in batches (to avoid exceeding token limits) until annotations were gathered for `r length(unique(gpt_annot$annot$hpo_id))`/`r hpo@n_terms` HPO phenotypes.

We then encoded these responses into a semi-quantitative scoring system (`r code_dict`), which were then weighted by multiplying a semi-subjective scoring of the relevance of each metric to the concept of severity on a scale from `r min(unlist(eval(gpt_codes$weights_dict)))`-`r max(unlist(eval(gpt_codes$weights_dict)))`, with `r max(unlist(eval(gpt_codes$weights_dict)))` being the most severe (`r weights_dict`).
Finally, the product of the score was normalised to a quantitative severity score ranging from 0-100, where 100 is the theoretical maximum severity score.
This phenotype severity scoring procedure can be expressed as follows.

Let us denote:

-   $p$ : a phenotype in the HPO.

-   $j$ : the identity of a given annotation metric (i.e. clinical characteristic, such as 'intellectual disability' or 'congenital onset').

-   $W_j$: the assigned weight of metric $j$.

-   $F_j$: the maximum possible value for metric $j$, equal to 3 ("always").
    This value is equivalent across all $j$ annotations.

-   $F_{pj}$ : the numerically encoded value of annotation metric $j$ for phenotype $p$.

-   $NSS_p$: the final composite severity score for phenotype $p$ after applying normalisation to align values to a 0-100 scale and ensure equivalent meaning regardless of which other phenotypes are being analysed in addition to $p$.
    This allows for direct comparability of severity scores across studies with different sets of phenotypes.

::: {#eq-gpt .content-hidden unless-format="html"}
![](equations/eq5.png){height="300px"}

**Computing normalised severity score from encoded GPT-4 annotations.**
:::

\
\

::: {.content-visible unless-format="html"}
\begin{equation*}
  \eqnmarkbox[Brown4]{nss}{NSS_p}
  =
  \frac{ 
    \eqnmarkbox[Goldenrod]{nss2}{\sum_{j=1}^{m}} 
    (
      \eqnmarkbox[Goldenrod4]{nss3}{F_{pj}}
      \times 
      \eqnmarkbox[IndianRed4]{nss4}{W_j}
    )
    }{
    \eqnmarkbox[Tan]{nss5}{\sum_{j=1}^{m}(\max\{F_j\} \times W_j)} 
  } \times 100
\end{equation*} \annotate[yshift=1em]{left}{nss}{Normalised Severity Score \\for each phenotype} \annotate[yshift=3em]{left}{nss2}{Sum of weighted annotation values \\across all metrics} \annotate[yshift=3em]{right}{nss3}{Numerically encoded annotation value \\of metric $j$ for phenotype $p$} \annotate[yshift=1em]{right}{nss4}{Weight for metric $j$} \annotate[yshift=-1em]{below,right}{nss5}{Theoretical maximum severity score}
:::

\

Using the numerically encoded GPT annotations (0="never", 1="rarely", 2="often", 3="always") we computed the mean encoded value per cell type within each annotation.
One-sided Wilcoxon rank-sum tests were run using the `rstatix::wilcox_test()` function to test whether each cell type was associated with more severe phenotypes relative to all other cell types.
This procedure was repeated for severity annotation independently (death, intellectual disability, impaired mobility, etc.) [Fig. @fig-celltype-severity-dot]a.
Next, we performed a Pearson correlation test between the number of phenotypes that a cell type is significantly associated with (at FDR\<0.05) has a relationship with the mean composite GPT severity score of those phenotypes ([Fig. @fig-celltype-severity-dot]b).
This was performed using the `ggstatsplot::ggscatterstats()` R function.

### Congenital phenotypes are associated with foetal cell types

```{r foetal_keywords}
foetal_keywords <- shQuote(eval(formals(MSTExplorer::plot_congenital_annotations)$foetal_keywords))
```

The GPT-4 annotations also enabled us to assess whether foetal cell types were more often significantly associated with congenital phenotypes in our Human Cell Landscape results as this single-cell reference contained both adult and foetal versions of cell types ([Fig. @fig-congenital]).
To do this, we performed a chi-squared ($\chi^2$) test on the proportion of significantly associated cell types containing any of the substrings `r paste(paste0(foetal_keywords[-length(foetal_keywords)], collapse = ", "),"or",tail(foetal_keywords,1))` (within cell types annotations from the original Human Cell Landscape authors [@Han2020-iq]) vs. those associated without, stratified by how often the corresponding phenotype had a congenital onset according to the GPT phenotype annotations (including `r paste0(shQuote(names(eval(gpt_codes$code_dict))), collapse=", ")`).
In addition, a series of $\chi^2$ tests were performed within each congenital onset frequency strata, to determine whether the observed proportion of foetal cell types vs. non-foetal cell types significantly deviated from the proportions expected by chance.

We next tested whether the proportion of tests with significant associations with foetal cell types varied across the major HPO branches using a $\chi^2$ test.
We also performed separate $\chi^2$ test within each branch to determine whether the proportion of significant associations with foetal cell types was significantly different from chance.

Next, we aimed to create a continuous metric from -1 to 1 that indicated how biased each phenotype is towards associations with the foetal or adult form of a cell type.
For each phenotype we calculated the foetal-adult bias score as the difference in the association p-values between the foetal and adult version of the equivalent cell type ($\text{foetal-adult bias}: p_{adult} - p_{foetal} = \Delta p \in [-1,1]$).
A score of 1 indicates the phenotype is only associated with the foetal version of the cell type and -1 indicates the phenotype is only associated with the adult version of the cell type.

In order to summarise higher-order HPO phenotype categories that were most biased towards foetal or adult cell types, ontological enrichment tests were run on the phenotypes with the top/bottom 50 greatest/smallest foetal-adult bias scores.
The enrichment tests were performed using the `simona::dag_enrich_on_offsprings` function, which uses a hypergeometric test to determine whether a list of terms in an ontology are enriched for offspring terms (descendants) of a given ancestor term within the ontology.
Phenotypes categories with an HPO ontological enrichment a p-value \< 0.05 were considered significant.

We were similarly interested in which higher-order cell type categories tended to be most commonly associated with these strongly foetal-/adult-biased phenotype s.
Another set of ontological enrichment tests were run on the cell types associated with the top/bottom 50 phenotypes from the previous analysis.
The CL ontology-aligned IDs for each group cell types were fed into the `simona::dag_enrich_on_offsprings` using the CL ontology.
Significiantly enriched cell type categories were defined as those with a CL ontological enrichment p-value \< 0.05.

### Therapeutic target identification

We developed a systematic and automated strategy for identifying putative cell type-specific gene targets for each phenotype based on a series of filters at phenotype, cell type, and gene levels.

First, we transformed our phenotype-cell type association results and merged them with primary data sources (e.g. GenCC gene-disease relationships, scRNA-seq atlas datasets) to create a large table of multi-scale relationships, where each row represented a tetrad of disease-phenotype-cell type-gene relationships.
We then filtered non-significant phenotype-cell type relationships (only associations with $FDR<0.05$) as well as phenotype-gene relationships with strong causal evidence ($\text{GenCC score}>3$).
We also removed any phenotypes that were too broad to be clinically useful, as quantified using the information content (IC) ($IC>8$), which measures the how specific each term is within an ontology (i.e. HPO).
Gene-cell type relationships were established by taking genes that had the top 25% expression specificity quantiles within each cell type.
When connecting cell types to diseases via phenotypes, we used a symptom intersection threshold of \>.25.
Next, we sorted the remaining results in descending order of phenotype severity using the GPT4 composite severity scores described earlier.
Finally, to limit the size of the resulting multi-scale networks we took only the top 10 rows, where each row represented a tetrad of disease-phenotype-cell type-gene relationships.
This resulted in number of relatively small, high-confidence disease-phenotype-cell type-gene networks that could be reasonably interrogated through manual inspection and network visualisation.
For example, if one was interested in the mechanisms causing 'Recurrent Neisserial infections', one would need only select all rows that include this phenotype to find all of its most relevant connection to diseases, cell types, and genes.

The entire target prioritisation procedure can be replicated with a single function: `MSTExplorer::prioritise_targets`.
This function automates all of the reference data gathering (e.g. phenotype metadata, cell type metadata, cell type signature reference, gene lengths, severity tiers) and takes a variety of arguments at each step for greater customisability.
Each step is described in detail in @tbl-filters.
Phenotypes that often or always caused physical malformations (according to the GPT-4 annotations) were also removed from the final prioritised targets list, as these were unlikely to be amenable to gene therapy interventions.
Finally, phenotypes were sorted by their composite severity scores such that the most severe phenotypes were ranked the highest.

### Therapeutic target validation

To assess whether our prioritised therapeutic targets were likely to be viable, we computed the overlap between our gene targets and those of existing gene therapies at various stages of clinical development ([Fig. @fig-therapy-validate]).
Gene targets were obtained for each therapy from the Therapeutic Target Database (TTD; release `r Sys.Date()`) and mapped onto standardised HUGO Gene Nomenclature Committee (HGNC) gene symbols using the `orthogene` R package.
We stratified our overlap metrics according to whether the therapies had failed (unsuccessful clinical trials or withdrawn), or were non-failed (successful or ongoing clinical trials).
We then conducted hypergeometric tests to determine whether the observed overlap between our prioritised targets and the non-failed therapy targets was significantly greater than expected by chance (i.e. enrichment).
We also conducted a second hypergeometric test to determine whether the observed overlap between our prioritised targets and the failed therapy targets was significantly less than expected by chance (i.e. depletion).
Finally, we repeated the analysis against all therapeutic targets, not just those of gene therapies, to determine whether our prioritised targets had relevance to other therapeutic modalities.

### Experimental model translatability

To improve the likelihood of successful translation between preclinical animal models and human patients, we created an interspecies translatability prediction tool for each phenotype nominated by our gene therapy prioritised pipeline ([Supp. Fig. @fig-animal-models]).
First, we extracted ontological similarity scores of homologous phenotypes across species from the MKG [@Putman2024-et].
Briefly, the ontological similarity scores ($SIM_o$) are computed for each homologous pair of phenotypes across two ontologies by calculating the overlap in homologous phenotypes that are ancestors or descendants of the target phenotype.
Next, we generated genotypic similarity scores ($SIM_g$) for each homologous phenotype pair by computing the proportion of 1:1 orthologous genes using gene annotation from their respective ontologies.
Interspecies orthologs were also obtained from the MKG.
Finally, both scores are multiplied together to yield a unified ontological-genotypic similarity score ($SIM_{og}$).

### Novel R packages

To facilitate all analyses described in this study and to make them more easily reproducible by others, we created several open-source R packages.
[`KGExplorer`](https://github.com/neurogenomics/KGExplorer) imports and analyses large-scale biomedical knowledge graphs and ontologies.
[`HPOExplorer`](https://github.com/neurogenomics/HPOExplorer) aids in managing and querying the directed acyclic ontology graph within the HPO.
[`MSTExplorer`](https://github.com/neurogenomics/MSTExplorer) facilitates the efficient analysis of many thousands of phenotype-cell type association tests, and provides a suite of multi-scale therapeutic target prioritisation and visualisation functions.
These R packages also include various functions for distributing the post-processed results from this study in an organised, tabular format.
Of note, `MSTExplorer::load_example_results` loads all summary statistics from our phenotype-cell type tests performed here.

### Rare Disease Celltyping Portal

To further increase the ease of access for stakeholders in the RD community without the need for programmatic experience, we developed a series of web apps to interactively explore, visualise, and download the results from our study.
Collectively, these web apps are called the Rare Disease Celltyping Portal.
The website can be accessed at <https://neurogenomics-ukdri.dsi.ic.ac.uk/>.

The Rare Disease Celltyping Portal integrates diverse datasets, including the HPO, cell types, genes, and phenotype severity, into a unified platform that allows users to perform flexible, bidirectional queries.
Users can start from any entry point: either phenotype, cell type, genes, or severity, and seamlessly trace relationships across these dimensions.

The portal provides a dynamic and intuitive exploration experience with its real-time interaction capabilities and responsive interface including network graphs, bar charts, and heat maps.
It has the ability to handle large datasets efficiently and offer fast query response by building with FARM stack (FastAPI, React, MongoDB).
The portal is designed for a broad audience, including researchers, clinicians, and biologists, by offering user-friendly navigation and interactive visual outputs.
By enabling users to intuitively explore complex biological relationships, the portal aims to accelerate rare disease research, enhance diagnostic accuracy, and drive therapeutic innovation.

All code used to generate the website can be found at <https://github.com/neurogenomics/Rare-Disease-Web-Portal>.

### Mappings

Mappings from the HPO to other medical ontologies were extracted from the EMBL-EBI Ontology Xref Service (OxO; <https://www.ebi.ac.uk/spot/oxo/>) by selecting the National Cancer Institute metathesaurus (NCIm) as the target ontology and either "SNOMED CT", "UMLS", "ICD-9" or "ICD-10CM" as the data source.
HPO terms were then selected as the ID framework with to mediate the cross-ontology mappings.
Mappings between each pair of ontologies were then downloaded, stored in a tabular format.
The mappings files can be accessed with the function `HPOExplorer::get_mappings` or directly via the `HPOExplorer` Releases page on GitHub (<https://github.com/neurogenomics/HPOExplorer/releases/tag/latest>).

## Data Availability

All data is publicly available through the following resources:

-   Human Phenotype Ontology (<https://hpo.jax.org>)
-   GenCC (<https://thegencc.org/>)
-   Descartes Human scRNA-seq atlas (<https://cellxgene.cziscience.com/collections/c114c20f-1ef4-49a5-9c2e-d965787fb90c>)
-   Human Cell Landscape scRNA-seq atlas (<https://cellxgene.cziscience.com/collections/38833785-fac5-48fd-944a-0f62a4c23ed1>)
-   Processed Cell Type Datasets (*ctd_DescartesHuman.rds* and *ctd_HumanCellLandscape.rds*; <https://github.com/neurogenomics/MSTExplorer/releases>)
-   Gene x Phenotype association matrix (*hpo_matrix.rds*; <https://github.com/neurogenomics/MSTExplorer/releases>)
-   GPT-4 phenotype severity annotations (<https://github.com/neurogenomics/rare_disease_celltyping/releases/download/latest/gpt_check_annot.csv.gz>)
-   Full phenotype-cell type association test results <https://github.com/neurogenomics/MSTExplorer/releases/download/v0.1.10/phenomix_results.tsv.gz>
-   Rare Disease Celltyping Portal (<https://neurogenomics-ukdri.dsi.ic.ac.uk/>)
-   Rare Disease Celltyping Portal data (<https://zenodo.org/records/15147825>)
-   Complement system gene list (<https://www.genenames.org/data/genegroup/#!/group/492>)
-   Therapeutic Target Database data (<http://db.idrblab.net/ttd/>)
-   CellxGene browser: hepatocytes & hepatoblast complement expression (<https://cellxgene.cziscience.com/gene-expression?genes=C5%2CCFI%2CC8B%2CCFB%2CMBL2%2CAPOE&cellTypes=hepatoblast%2Chepatocyte&ver=2>)

## Code Availability

All code is made freely available through the following GitHub repositories:

-   `KGExplorer` (<https://github.com/neurogenomics/KGExplorer>)
-   `HPOExplorer` (<https://github.com/neurogenomics/HPOExplorer>)
-   `MSTExplorer` (<https://github.com/neurogenomics/MSTExplorer>)
-   Code to replicate analyses (<https://github.com/neurogenomics/rare_disease_celltyping>)
-   Rare Disease Celltyping Portal code (<https://github.com/neurogenomics/Rare-Disease-Web-Portal>)

## Acknowledgements

We would like to thank the following individuals for their insightful feedback and assistance with data resources: Sarah J. Marzi, Gerton Lunter, Peter Robinson, Melissa Haendel, Ben Coleman, Nico Matentzoglu, Shawn T. O'Neil, Alan E. Murphy, Sarada Gurung.

### Funding

This work was supported by a UK Dementia Research Institute (UK DRI) Future Leaders Fellowship \[MR/T04327X/1\] and the UK DRI which receives its funding from UK DRI Ltd, funded by the UK Medical Research Council, Alzheimer's Society and Alzheimer's Research UK.

## References {.unnumbered}

::: {#refs}
:::

\

\newpage

## Supplementary Materials

### Supplementary Results

#### Congenital phenotypes are associated with foetal cell types

To test whether some cell types tend to show strong differences in their phenotype associations between their foetal and adult forms.
To test this, we performed an analogous enrichment procedure as with the phenotypes, except using Cell Ontology terms and the Cell Ontology graph.
This analysis identified the cell type category `r run_congenital_enrichment_out$cl_enrich_top$name[1]` ($p=$`r run_congenital_enrichment_out$cl_enrich_top$p_value[1]`, $log_2(\text{fold-change})$=`r run_congenital_enrichment_out$cl_enrich_top$log2_fold_enrichment[1]`) as the most foetal-biased cell type.
No cell type categories were significantly enriched for the most adult-biased cell types.
This is likely due to the fact that cell types can be disrupted at different stages of life, resulting in different phenotypes.
Thus there the same cell types may be involved in both the most foetal-biased and adult-biased phenotypes.

#### Selected example targets

From our prioritised targets, we selected the following four sets of phenotypes or diseases as examples: `r paste(shQuote(phenotypes_network), collapse=", ")`.
Only phenotypes with a GPT severity score greater than `r severity_score_gpt_threshold` were considered to avoid overplotting and to focus on the more clinically relevant phenotypes [Fig. @fig-therapy-examples]a-h.
These examples were then selected partly on the basis of severity rankings, and partly for their relatively smaller, simpler networks than lent themselves to compact visualisations.

Tay-Sachs disease (TSD) is a devastating hereditary condition in which children are born appearing healthy, which gradually degrades leading to death after 3-5 years.
The underlying cause is the toxic accumulation of gangliosides in the nervous system due to a loss of the enzyme produced by *HEXA*.
While this could in theory be corrected with gene editing technologies, there remain some outstanding challenges.
One of which is identifying which cell types should be targeted to ensure the most effective treatments.
Here we identified alternatively activated macrophages as the cell type most strongly associated with 'GM2-ganglioside accumulation' [Fig. @fig-therapy-examples]i.
The role of aberrant macrophage activity in the regulation of ganglioside levels is supported by observation that gangliosides accumulate within macrophages in TSD [@fendersonChapterDevelopmentalGenetic2009], as well as experimental evidence in rodent models [@vilcaesGangliosideSynthesisPlasma2020.; @yoheGangliosideAlterationsStimulated1985; @demirGM2GangliosideAccumulation2020].
Our results not only corroborate these findings, but propose macrophages as the primary causal cell type in TSD, making it the most promising cell type to target in therapies.

Another challenge in TSD is early detection and diagnosis, before irreversible damage has occurred.
Our pipeline implicated extravillous trophoblasts of the placenta in 'GM2-ganglioside accumulation'.
While not necessarily a target for gene therapy (as the child is detached from the placenta after birth), checking these cells *in utero* for an absence of *HEXA* may serve as a viable biomarker as these cells normally express the gene at high levels.
Early detection of TSD may lengthen the window of opportunity for therapeutic intervention [@solovyevaNewApproachesTaySachs2018], especially when genetic sequencing is not available or variants of unknown significance are found within *HEXA* [@hoffmanNextgenerationDNASequencing2013].

Spinocerebellar atrophy is a debilitating and lethal phenotype that occurs in diseases such as Spinocerebellar ataxia and Boucher-Nenhauser syndrome.
These diseases are characterised by progressive degeneration of the cerebellum and spinal cord, leading to severe motor and cognitive impairments.
Our pipeline identified M2 macrophages (labeled as the closest CL term 'Alternatively activated macrophages' in [Fig. @fig-therapy-examples]j) as the only causal cell type associated with 'Spinocerebellar atrophy'.
This strongly suggests that degeneration of cerebellar Purkinje cells are in fact downstream consequences of macrophage dysfunction, rather than being the primary cause themselves.
This is consistent with the known role of macrophages, especially microglia, in neuroinflammation and other neurodegenerative conditions such as Alzheimer's and Parkinsons' disease [@ferroRoleMicrogliaAtaxias2019; @holMicroglialTranscriptomicsMeets2022; @lopesAtlasGeneticEffects2020a].
While experimental and postmortem observational studies have implicated microglia in spinocerebellar atrophy previously [@ferroRoleMicrogliaAtaxias2019], our results provide a statistically-supported and unbiased genetic link between known risk genes and this cell type.
Therefore, targeting M2 microglia in the treatment of spinocerebellar atrophy may therefore represent a promising therapeutic strategy.
This is aided by the fact that there are mouse models that perturb the ortholog of human spinocerebellar atrophy risk genes (e.g. [*Atxn1*](https://www.informatics.jax.org/marker/MGI:104783), [*Pnpla6*](https://www.informatics.jax.org/marker/MGI:1354723)) and reliably recapitulate the effects of this diseases at the cellular (e.g. loss of Purkinje cells), morphological (e.g. atrophy of the cerebellum, spinal cord, and muscles), and functional (e.g. ataxia) levels.

Next, we investigated the phenotype 'Neuronal loss in the central nervous system'.
Despite the fact that this is a fairly broad phenotype, we found that it was only significantly associated with `r length(unique(neuronal_loss$cl_name))` cell types (`r paste(unique(neuronal_loss$cl_name), collapse=', ')`), specifically M2 macrophages and sinusoidal endothelial cells [Fig. @fig-therapy-examples]k.

Skeletal dysplasia is a heterogeneous group of over 450 disorders that affect the growth and development of bone and cartilage.
This phenotype can be lethal when deficient bone growth leads to the constriction of vital organs such as the lungs.
Even after surgical interventions, these complications continue to arise as the child develops.
Pharmacological interventions to treat this condition have largely been ineffective.
While there are various cell types involved in skeletal system development, our pipeline nominated chondrocytes as the causal cell type underlying the lethal form of this condition ([Fig. @fig-therapy-examples-supp]).
Assuringly, we found that the disease 'Achondrogenesis Type 1B' is caused by the genes *SLC26A2* and *COL2A1* via chondrocytes.
We also found that 'Platyspondylic lethal skeletal dysplasia, Torrance type'.
Thus, in cases where surgical intervention is insufficient, targeting these genes within chondrocytes may prove a viable long-term solution for children suffering from lethal skeletal dysplasia.

Alzheimer's disease (AD) is the most common neurodegenerative condition.
It is characterised by a set of variably penetrant phenotypes including memory loss, cognitive decline, and cerebral proteinopathy.
Interestingly, we found that different forms of early onset AD (which are defined by the presence of a specific disease gene) are each associated with different cell types via different phenotypes ([Fig. @fig-therapy-examples-supp]).
For example, AD 3 and AD 4 are primarily associated with cells of the digestive system (`r paste(shQuote(unique(top_targets[disease_name=="Alzheimer disease 4"]$cl_name)), collapse=", ")`) and are implied to be responsible for the phenotypes `r paste(shQuote(unique(top_targets[disease_name=="Alzheimer disease 4"]$hpo_name)), collapse=", ")`.
Meanwhile, AD 2 is primarily associated with immune cells (`r paste(shQuote(unique(top_targets[disease_name %in% c("Early-onset autosomal dominant Alzheimer disease","Alzheimer disease 2")]$cl_name)[1]), collapse=", ")`) and is implied to be responsible for the phenotypes `r paste(shQuote(unique(top_targets[disease_name %in% c("Early-onset autosomal dominant Alzheimer disease","Alzheimer disease 2")]$hpo_name)), collapse=", ")`.
This suggests that different forms of AD may be driven by different cell types and phenotypes, which may help to explain its variability in onset and clinical presentation.

Finally, Parkinson's disease (PD) is characterised by motor symptoms such as tremor, rigidity, and bradykinesia.
However there are a number of additional phenotypes associated with the disease that span multiple physiological systems.
PD 19a and PD 8 seemed to align most closely with the canonical understanding of PD as a disease of the central nervous system in that they implicated oligodendrocytes and neurons ([Fig. @fig-therapy-examples-supp]).
Though the reference datasets being used in this study were not annotated at sufficient resolution to distinguish between different subtypes of neurons, in particular dopaminergic neurons.
PD 19a/8 also suggested that risk variants in *LRRK2* mediate their effects on PD through both myeloid cells and oligodendrocytes by causing gliosis of the substantia nigra.
The remaining clusters of PD mechanisms revolved around chondrocytes (PD 20), amacrine cells of the eye (hereditary late-onset PD), and the respiratory/immune system (PD 14).
While the diversity in cell type-specific mechanisms is somewhat surprising, it may help to explain the wide variety of cross-system phenotypes frequently observed in PD.

It should be noted that the HPO only includes gene annotations for the monogenic forms of AD and PD.
However it has previously been shown that there is at least partial overlap in their phenotypic and genetic etiology with respect to their common forms.
Thus understanding the monogenic forms of these diseases may shed light onto their more common counterparts.

### Experimental model translatability

```{r map_upheno_data}
pheno_map_genes_match <- KGExplorer::map_upheno_data(pheno_map_method = "upheno", 
                                                     force_new = FALSE)
pheno_map_targets <- pheno_map_genes_match[
  id1 %in% unique(prioritise_targets_out$top_targets$hpo_id)
  ]|>
  data.table::setorderv("phenotype_genotype_score",-1)
taxa_count <- sort(table(pheno_map_targets$gene_taxon_label2), decreasing = TRUE)

pheno_map_targets_severe <- pheno_map_targets[
  id1 %in% unique(prioritise_targets_out$top_targets[severity_score_gpt>10,]$hpo_id)
  ]

pheno_map_targets_severe[,summary:=paste0(
  shQuote(object_label1),  
  " ($SIM_{og}=",round(phenotype_genotype_score,options()$digits),"$)"
  )] 
```

We computed interspecies translatability scores using a combination of both ontological ($SIM_{o}$) and genotypic ($SIM_{g}$) similarity relative to each homologous human phenotype and its associated genes [Supp. Fig. @fig-animal-models].
In total, we mapped `r length(unique(pheno_map_genes_match$id2))` non-human phenotypes (in `r paste0("*",sort(unique(pheno_map_genes_match$gene_taxon_label2)),"*",collapse=", ")`) to `r length(unique(pheno_map_genes_match$id1))` homologous human phenotypes.
Amongst the `r length(unique(prioritise_targets_out$top_targets$hpo_id))` phenotype within our prioritised therapy targets, `r length(intersect(prioritise_targets_out$top_targets$hpo_id, pheno_map_genes_match$id1))` had viable animal models in at least on non-human species.
Per species, the number of homologous phenotypes was: `r paste0("*",names(taxa_count),"*"," (n=",taxa_count,")", collapse = ", ")`.
Amongst our prioritised targets with a GPT-4 severity score of \>10, the phenotypes with the greatest animal model similarity were `r paste(pheno_map_targets_severe$summary[1:5],collapse = ", ")`.

### Supplementary Figures

::: {#fig-evidence-histograms .hide-subcaptions}
```{r fig-evidence-histograms, cache.lazy=FALSE}
#| label: fig-evidence-histograms
#| fig-cap: __Distribution of GenCC evidence scores at each processing step.__ GenCCC (<https://thegencc.org/>) is a database where semi-quantitative scores for the current strength of evidence attributing disruption of a gene as a causal factor in a given disease. "gcc_disease_raw" is the distribution of raw GenCC scores before any aggregation. "gcc_disease_agg" is the distribution of GenCC scores after aggregating by disease. "gcc_phenotype" is the distribution of scores after linking each phenotype to one or more disease. "gcc_phenotype_agg" is the distribution of scores after aggregating by phenotype, while "gcc_phenotype_agg_sd" is the standard deviation of those aggregated scores.
#| fig-height: 6
#| fig-weight: 8

evidence_plot <- HPOExplorer::plot_evidence(phenotype_to_genes = p2g,
                                            show_plot = FALSE)
evidence_plot$plot
```
:::

::: {#fig-diagram}
![](img/fig-diagram.png)

**Diagrammatic overview of multi-scale disease investigation strategy.** Here we provide an abstract example of differential disease etiology across multiple scales: diseases ($D$), phenotypes ($P$), cell types ($C$), genes ($G$), and clinical outcomes ($O$).
In the HPO, genes are assigned to phenotypes via particular diseases ($G_{dp}$).
Therefore, the final gene list for each phenotype is aggregated from across multiple diseases ($G_{p}$).
We performed association tests for all pairwise combinations of cell types and phenotypes and filtered results after multiple testing corrections (FDR\<0.05).
Each phenotype in the context of a given disease is referred to here as a symptom.
Links were established between symptoms and cell types through proportional gene set overlap at a minimum threshold of 25%.
:::

\newpage

::: {#fig-ctd-correlation .hide-subcaptions}
```{r fig-ctd-correlation, dev = "cairo_pdf"}
#| label: fig-ctd-correlation
#| fig-cap: __Inter- and intra-dataset validation across the different CellTypeDataset (CTD) and developmental stages.__ Correlations are computed using Pearson correlation coefficient. Point density is plotted using a 2D kernel density estimate. **a** Correlation between the uncorrected p-values from all phenotype-cell type association tests using the Descartes Human vs. Human Cell Landscape CTDs. **b** Correlation between the $log_{10}(fold-change)$ from significant phenotype-cell type association tests (FDR<0.05) using the Descartes Human vs. Human Cell Landscape CTDs. **c** Correlation between the uncorrected p-values from all phenotype-cell type association tests using the Human Cell Landscape foetal samples vs. Human Cell Landscape adult samples. **d** Correlation between the $log_{10}(fold-change)$ from significant phenotype-cell type association tests (FDR<0.05) using the Human Cell Landscape foetal samples vs. Human Cell Landscape adult samples.
#| fig-height: 12
#| fig-width: 12

library(ggplot2)
library(ggrastr)

fig_ctd_correlation <- (
  (
    validate_associations_correlate_ctd_out$plot$estimate.all + 
     ggplot2::labs(title = "Inter-dataset correlation: effect sizes (all)")|
    validate_associations_correlate_ctd_out$plot$estimate.significant + 
    ggplot2::labs(title = "Inter-dataset correlation: effect sizes (FDR<0.05)")
 
 )/
    
  (
    validate_associations_correlate_ctd_out_hcl$plot$estimate.all +
     ggplot2::labs(title = "Intra-dataset correlation: effect sizes (all)")|
    validate_associations_correlate_ctd_out_hcl$plot$estimate.significant +
    ggplot2::labs(title = "Intra-dataset correlation: effect sizes (FDR<0.05)")
  )
) +
  patchwork::plot_annotation(tag_levels = letters)

ggrastr::rasterize(fig_ctd_correlation, layers='Point', dpi=300)
```
:::

\newpage

::: {#fig-monarch-recall .hide-subcaptions}
```{r fig-monarch-recall}
#| label: fig-monarch-recall
#| fig-cap: __Recall of ground-truth Monarch Knowledge Graph phenotype-cell type relationships at each ontological distance between cell types according to the Cell Ontology.__

validate_associations_mkg_out$cl_distance_results$plot
```
:::

\newpage

::: {#fig-celltype-severity-bar .hide-subcaptions}
```{r fig-celltype-severity-bar}
#| label: fig-celltype-severity-bar
#| fig-cap: __Cell types ordered by the mean severity of the phenotypes they're associated with.__ **a**, The disribution of phenotype severity annotation frequencies aggregated by cell type. **b**, The composite severity score, averaged across all phenotypes associated with each cell type. 
#| fig-height: 12
#| fig-width: 16

plot_celltype_severity_out$bar$plot +
  patchwork::plot_annotation(tag_levels = "a") 
```
:::

\newpage

::: {#fig-network-rni .hide-subcaptions}
```{r fig-network-rni, cache.lazy=FALSE}
#| label: fig-network-rni
#| fig-cap: __Causal network of recurrent Neisserial infections (RNI) reveals multi-scale disease etiology.__ RNI is a phenotype in seven different monogenic diseases caused by disruptions to specific complement system genes. Four cell types were significantly associated with RNI. **a**, One can trace how genes causal for RNI (yellow boxes, bottom) mediate their effects through cell types (orange circles, middle) and diseases (blue cylinders, top). Cell types are connected to RNI via association testing (FDR\<0.05). Genes shown here have both strong evidence for a causal role in RNI and high expression specificity in the associated cell type. Cell types can be linked  to monogenic diseases via the genes specifically expressed in those cell types (i.e. are in the top 25\% of cell type specificity expression quantiles). Nodes are arranged using the Sugiyama algorithm [@Sugiyama1981-ev]. **b** Expression specificity quantiles (1-40 scale) of each driver gene in each cell type (darker = greater specificity). **c** GenCC-derived eevidence scores between the RNI phenotype and each gene. **d** Expression specificity (0 = least specific, 1 = most specific) of each gene in each cell type.
#| fig-height: 8
#| fig-width: 8

if(knitr::is_html_output()){
  knitr::opts_current$set(results="asis")
  vn_rni$plot
  prioritise_targets_multiheatmap_out$plot
} else { 
  # Read the image
  img <- magick::image_read(here::here("img","fig-rni.png"), 
                            density = 400)
  img_grob <- grid::rasterGrob(img, interpolate = TRUE)
  
  # Create a ggplot with the image as a background
  img_plot <- ggplot2::ggplot() +
    ggplot2::annotation_custom(img_grob, xmin = -Inf, xmax = Inf, 
                      ymin = -Inf, ymax = Inf) +
    ggplot2::theme_void()
  
  fig_network_rni <- patchwork::wrap_plots(
    img_plot,
    prioritise_targets_multiheatmap_out$plot,
    ncol = 1,
    heights = c(1, .6)
  ) + 
    patchwork::plot_annotation(tag_levels = "a") & 
    ggplot2::theme(plot.margin = ggplot2::margin(),
                   plot.tag = ggplot2::element_text(size = 12)) # works now
  fig_network_rni
  # ggplot2::ggsave("~/Downloads/fig_network_rni.png", fig_network_rni, dpi = 400, height = 8, width = 8)
} 
```
:::

\newpage

::: {#fig-therapy-filter .hide-subcaptions}
```{r fig-therapy-filter}
#| label: fig-therapy-filter
#| fig-cap: __Prioritised target filtering steps.__ This plot visualises the number of unique phenotype-cell type associations, cell types, genes, and phenotypes (*y-axis*) at each filtering step (*x-axis*) within the multi-scale therapeutic target prioritisation pipeline. Each step in the pipeline can be easily adjusted according to user preference and use case. See @tbl-filters for descriptions and criterion of each filtering step.
#| fig-width: 10
#| fig-height: 8

plot_report_out$plot
```
:::

\newpage

::: {#fig-therapy-validate-all .hide-subcaptions}
```{r fig-therapy-validate-all}
#| label: fig-therapy-validate-all
#| fig-cap: __Validation of prioritised therapeutic targets.__ Proportion of existing all therapy targets (documented in the Therapeutic Target Database) recapitulated by our prioritisation pipeline.
#| fig-height: 10
#| fig-width: 11

ttd_check_all_out$plot
```
:::

\newpage

::: {#fig-animal-models .hide-subcaptions}
```{r fig-animal-models}
#| label: fig-animal-models
#| fig-cap:  __Identification of translatable experimental models.__ Interspecies translatability of the top 200 human phenotypes nominated by the gene therapy prioritised pipeline. Above, the combined ontological-genotypic similarity score ($SIM_{og}$) is displayed as the heatmap fill colour stratified by the model organism (*x-axis*). An additional column ("n_genes_db1" on the far left) displays the total number of unique genes annotated to the phenotypic within the HPO. Phenotypes are clustered according to their ontological similarity in the HPO (*y-axis*).
#| fig-height: 12
#| fig-width: 10

library(ggplot2) # <-- Necessary due to bug in one of the plotting dependencies (tidyquant::scale_fill_tq) 
upheno <- KGExplorer::get_ontology(name="upheno")
top_ids <- unique(prioritise_targets_out$top_targets$hpo_id)[seq(200)]
plot_upheno_out <- KGExplorer::plot_upheno(
  pheno_map_genes_match = pheno_map_genes_match, 
  filters=list(id1=top_ids),
  types=c("heatmap"),
  name="Phenotype-\ngenotype\nsimilarity"
  )
plot_upheno_out$heatmap
```
:::

\newpage

::: {#fig-therapy-examples2 .hide-subcaptions}
```{r fig-therapy-examples2}
#| label: fig-therapy-examples2
#| fig-cap: __Causal multi-scale networks reveal cell type-specific therapeutic targets.__ Each disease (blue cylinders) is connected to its phenotype (purple cylinders) based on well-established clinical observations recorded within the HPO [@Gargano2024-fc]. Phenotypes are connected to cell types (orange circles) via association testing between weighted gene sets (FDR<0.05). Each cell type is connected to the prioritised gene targets (yellow boxes) based on the driver gene analysis.The thickness of the edges connecting the nodes represent the (mean) fold-change from the bootstrapped enrichment tests. Nodes were spatially arranged using the Sugiyama algorithm [@Sugiyama1981-ev].
#| fig-height: 8

img_paths <- stats::setNames(
   here::here( 
     "img","fig-therapy-examples",
  c("tay-sachs.png",
    "spinocerebellar-atrophy.png",
    "neuronal-loss.png")),
   c("GM2-ganglioside accumulation",
     "Spinocerebellar atrophy",
     "Neuronal loss in central nervous system")
 )
 fig_networks <- lapply(img_paths, function(x){
   i = which(x==img_paths)
   magick::image_read(x, density=300)|>magick::image_ggplot(interpolate=TRUE) +
   ggplot2::theme_void() +
   ggplot2::labs(subtitle=names(img_paths)[i]) +
   ggplot2::theme(plot.margin = ggplot2::unit(c(0,0,0,0),"pt"),
                  panel.background = ggplot2::element_rect(fill = "grey50"))
 })
 
{design <- "AB
           CC"
 patchwork::wrap_plots(fig_networks, ncol = 1, 
                       design=design, heights = c(.75, 1)) +
   patchwork::plot_annotation(tag_levels = "a")  
}
```
:::

\newpage

```{r prioritise_targets_network-supp-tmp}
height <- "100vh"
severity_score_gpt_threshold <- 15
top_targets <- prioritise_targets_out$top_targets[severity_score_gpt>severity_score_gpt_threshold]
phenotypes <- c("respiratory failure", 
                "amyotrophic lateral sclerosis", 
                "dementia",
                "lethal skeletal dysplasia",
                "small vessel disease",
                "Alzheimer disease",
                "Parkinson disease")
vn_therapy <- lapply(stats::setNames(phenotypes,phenotypes), function(phenotype){
  # cat("\nPreparing",phenotype,"\n")
  MSTExplorer::prioritise_targets_network(
    top_targets = top_targets[grepl(paste(phenotype,collapse = "|"), hpo_name,ignore.case = TRUE)|
                              grepl(paste(phenotype,collapse = "|"), disease_name,ignore.case = TRUE)],
    main = phenotype, 
    height = height,
    submain = NULL)$plot
}) 
```

::: {#fig-therapy-examples-supp .content-visible unless-format="html"}
![Respiratory failure](img/fig-therapy-examples-supp/respiratory_failure.png){#fig-therapy-examples-supp-rf}

**Example cell type-specific gene therapy targets for phenotypes associated with respiratory failure-related diseases.** Each disease (blue cylinders) is connected to its phenotype (purple cylinders) based on well-established clinical observations recorded within the HPO [@Gargano2024-fc].Phenotypes are connected to cell types (red circles) via association testing between weighted gene sets (FDR\<0.05).
Each cell type is connected to the prioritised gene targets (yellow boxes) based on the driver gene analysis.The thickness of the edges connecting the nodes represent the (mean) fold-change from the bootstrapped enrichment tests.
Nodes were spatially arranged using the Sugiyama algorithm [@Sugiyama1981-ev].
:::

\newpage

![**Causal multi-scale network for dementia phenotypes.**](img/fig-therapy-examples-supp/dementia.png){#fig-therapy-examples-supp-dementia}

\newpage

![**Causal multi-scale network for the phenotype lethal skeletal dysplasia.**](img/fig-therapy-examples-supp/lethal_skeletal_dysplasia.png){#fig-therapy-examples-supp-lsd}

\newpage

![**Causal multi-scale network for phenotypes associated with small vessel disease.**](img/fig-therapy-examples-supp/small_vessel_disease.png){#fig-therapy-examples-supp-svd}

\newpage

![**Causal multi-scale network for phenotypes associated with various subtypes of Parkinson's disease.**](img/fig-therapy-examples-supp/parkinson.png){#fig-therapy-examples-supp-pd}

\newpage

![**Causal multi-scale network for phenotypes associated with various subtypes of Alzheimer's disease.**](img/fig-therapy-examples-supp/alzheimer.png){#fig-therapy-examples-supp-ad}

\newpage

![**Causal multi-scale network for phenotypes associated with Amyotrophic Lateral Sclerosis (ALS).**](img/fig-therapy-examples-supp/als.png){#fig-therapy-examples-supp-als}

::: {#fig-therapy-examples-supp-html .content-hidden unless-format="html"}
```{r fig-therapy-examples-supp-html, results='asis'}
#| label: fig-therapy-examples-supp-html
#| fig-cap: __Example cell type-specific gene therapy targets for several severe phenotypes and their associated diseases.__ Each disease (blue cylinders) is connected to its phenotype (purple cylinders) based on well-established clinical observations recorded within the HPO [@Gargano2024-fc].Phenotypes are connected to cell types (red circles) via association testing between weighted gene sets (FDR<0.05). Each cell type is connected to the prioritised gene targets (yellow boxes) based on the driver gene analysis.The thickness of the edges connecting the nodes represent the (mean) fold-change from the bootstrapped enrichment tests. Nodes were spatially arranged using the Sugiyama algorithm [@Sugiyama1981-ev].
#| fig-height: 7
#| fig-width: 12
#| fig-subcap:
#| - "Respiratory failure"
#| - "Amyotrophic lateral sclerosis" 
#| - "Dementia"
#| - "Lethal skeletal dysplasia"
#| - "Small vessel disease"
#| - "Parkinson's disease"
#| - "Alzheimer's disease"

vn_therapy
```
:::

### Supplementary Tables

::: {#tbl-mappings}
```{r tbl-mappings}
#| label: tbl-mappings 

maps_agg|> 
  kableExtra::kbl() 
```

**Mappings between HPO phenotypes and other medical ontologies.** "source" indicates the medical ontology and "distance" indicates the cross-ontology distance.
"source terms" and "HPO terms" indicates the number of unique IDs mapped from the source ontology and HPO respectively.
"mappings" is the total number of cross-ontology mappings within a given distance.
Some IDs may have more than one mapping for a given source due to many-to-many relationships.
:::

\newpage

::: {#tbl-summary}
```{r tbl-summary}
#| label: tbl-summary  

data.frame(res_summ$tmerged[,-c("ctd")], 
             row.names = res_summ$tmerged$ctd,
             check.names = FALSE) |>
  t() |>
  kableExtra::kbl(label = NA)
```

**Summary statistics of enrichment results stratified by single-cell atlas.** Summary statistics at multiple levels (tests, cell types, phenotypes, diseases, cell types per phenotype, phenotypes per cell type) stratified by the single-cell atlas that was used as a cell type signature reference (Descartes Human or Human Cell Landscape).
:::

\newpage

::: {#tbl-filters tbl-colwidths="[20,20,60]"}
```{r tbl-filters}
#| label: tbl-filters

tbl_filters <- plot_report_out$data$filters[,.SD[1],by=c("step")][,c("level","step","description")]
tbl_filters[,step:=gsub("_"," ",step)]


tbl_filters |>
  as.data.frame() |>
  knitr::kable()
```

**Description of each filtering step performed in the multi-scale therapeutic target prioritisation pipleline.** 'level' indicates the biological scale at which the step is applied to.
:::

\newpage

::: {#tbl-ontarget-celltypes}
```{r tbl-ontarget-celltypes}
#| label: tbl-ontarget-celltypes

overrep_df |>
  knitr::kable()
#   # dplyr::select(-`HPO branch`) |>
  # kableExtra::kbl() 
  # kableExtra::kable_styling(full_width=FALSE)
  # kableExtra::group_rows(index=overrep_df$`HPO branch`)
```

**Cross-ontology mappings between HPO and CL branches.** The last two columns represent the number of cell types that were overrepresented in the on-target HPO branch and the total number of cell types in that branch.
A disaggregated version of this table with all descendant cell type names is available in @tbl-celltypes.
:::

::: {#tbl-gencc}
```{r tbl-gencc}
#| label: tbl-gencc 

gencc <- KGExplorer::get_gencc(agg_by=NULL)
dict <- eval(formals(KGExplorer::get_gencc)$dict)
gencc_tbl <- gencc[,list(classification_title=unique(classification_title)),
                   by="classification_curie"][,encoding:=dict[classification_title]]|>
  data.table::setorderv("encoding",-1)
knitr::kable(as.data.frame(gencc_tbl))
```

**Encodings for GenCC evidence scores.** Assigned numeric values for the GenCC evidence levels.
:::

\newpage

::: {#tbl-celltypes}
```{r tbl-celltypes}
#| label: tbl-celltypes

target_branches <- MSTExplorer:::get_target_branches()
target_celltypes <- MSTExplorer:::get_target_celltypes(target_branches=target_branches)
hpo <- HPOExplorer::get_hpo()
celltype_maps <- MSTExplorer::celltype_maps|> data.table::setkeyv("cl_id")

ct_dt <- lapply(names(target_celltypes), function(b){
  cl_ids <- intersect(target_celltypes[[b]],
                      celltype_maps$cl_id) 
  data.table::data.table(hpo_branch=b, 
                         # hpo_branch_id=HPOExplorer::map_phenotypes(b,
                         #                                           hpo = hpo,
                         #                                           to="id"),
                         cl_branch=paste0(target_branches[[b]],collapse=" / "),
                         celltype_maps[cl_ids,c("cl_name","cl_id")])
}) |> data.table::rbindlist(fill=TRUE) |> 
  unique() |>
  as.data.frame()

ct_dt |>
  dplyr::rename("CL branch"="cl_branch",
                "CL name"="cl_name",
                "CL ID"="cl_id" ) |>
  dplyr::select(-hpo_branch) |>
  kableExtra::kbl() |>
  kableExtra::kable_styling(font_size = 7, full_width = FALSE) |>
  kableExtra::pack_rows(index = table(ct_dt$hpo_branch))  

```

**On-target cell types for each Human Phenotype Ontology (HPO) ancestral branch.** Cell type-phenotype branch pairings were manually curated by comparing high-level HPO terms to terms within the Cell Ontology (CL).
Each HPO branch is shown as bolded row dividers.
Ancestral CL branch names are shown in the first column, along with the specific CL names and IDs.
:::

\newpage

::: {#tbl-ontology_permtests}
```{r tbl-ontology_permtests}
#| label: tbl-ontology_permtests

res_permtests[,CI_lower_upper:=paste0("[",paste(round(null_ci_lo,2), round(null_ci_hi,2), sep=", "),"]")]
res_permtests |>
  as.data.frame() |>
  dplyr::rename("X"="x_var",
                "y"="y_var",
                # "perm."="B", 
                "empirical P"="p_emp",
                "observed cor"="obs_cor",
                "null cor"="null_mean", 
                "null CI [lower, upper]"="CI_lower_upper", 
                ) |>
  dplyr::select(-B, -method, -null_ci_lo, -null_ci_hi) |>
  kableExtra::kbl() 
```

**Results of permutation tests evaluating relationships between phenotype information content and gene number, cell type number, and effect size.** Correlations between phenotypes and cell types may not be independent due to the hierarchical nature of the HPO.
To account for this, for each pair of variables we computed the observed correlation (*observed cor*), the mean null correlation (*nullc cor*) from 1000 permutations, and the 95% confidence interval of the null correlations (*null CI \[lower, upper\]*).
:::

\newpage

::: {#tbl-ctd_validation_permtests}
```{r}
#| label: tbl-ctd_validation_permtests

ctd_validation_permtest[,CI_lower_upper:=paste0("[",paste(round(null_ci_lo,2), round(null_ci_hi,2), sep=", "),"]")]
ctd_validation_permtest |>
    as.data.frame() |>
    dplyr::rename("X"="x_var",
                "y"="y_var",
                # "perm."="B", 
                "empirical P"="p_emp",
                "observed cor"="obs_cor",
                "null cor"="null_mean", 
                "null CI [lower, upper]"="CI_lower_upper", 
                ) |>
  dplyr::select(-B, -method, -null_ci_lo, -null_ci_hi) |>
  kableExtra::kbl() 
```

**Results of permutation tests evaluating phenotype-cell type association consistency across cell type references (Human Cell Landscape vs. Descartes Human) and across developmental stages (foetal vs. Adult).** To account for the non-independence between phenotypes, we computed empirical p-values for each Pearson correlation comparison over 1000 permutations.
Here, we report observed correlation (*observed cor*), the mean null correlation (*null cor*), 95% confidence interval of the null correlations (*null CI \[lower, upper\]*).
:::

\newpage

<!-- ::: {#tbl-death} -->

<!-- ```{r tbl-death} -->

<!-- #| label: tbl-death -->

<!-- hpo_deaths <- KGExplorer::get_ontology_descendants(ont = hpo, -->

<!--                                                    include_self = FALSE, -->

<!--                                                    terms = "Age of death")[[1]] -->

<!-- deaths_dt <- data.table::data.table(hpo_id=hpo_deaths) -->

<!-- deaths_dt <- HPOExplorer::add_hpo_name(deaths_dt) -->

<!-- dict <- HPOExplorer:::hpo_dict(type="AgeOfDeath") -->

<!-- deaths_dt[,encoding:=dict[hpo_name]] |> -->

<!--   data.table::setorderv("encoding") -->

<!-- knitr::kable(as.data.frame(deaths_dt)) -->

<!-- ``` -->

<!-- Encodings for Age of Death scores. -->

<!-- Assigned numeric values for the Age of Death scores within the HPO annotations. -->

<!-- ::: -->

<!-- \newpage -->

::: {#tbl-hpo_enrich}
```{r tbl-hpo_enrich}
#| label: tbl-hpo_enrich

foetal_biased_tbl <- (data.table::rbindlist(
  list(top=run_congenital_enrichment_out$hpo_enrich_top,
       bottom=run_congenital_enrichment_out$hpo_enrich_bottom), 
  idcol = "group")
 )[,c("group","term","name","p_adjust","log2_fold_enrichment","depth")]|>
  data.frame() |>
  dplyr::rename("FDR"="p_adjust", 
                "log2-fold enrichment"="log2_fold_enrichment") 
top_idx = which(foetal_biased_tbl$group=="top")
bottom_idx = which(foetal_biased_tbl$group=="bottom")

foetal_biased_tbl|>
  dplyr::select(-group) |>
  kableExtra::kbl() |> 
  kableExtra::group_rows("Foetal-biased", min(top_idx), max(top_idx)) |>
  kableExtra::group_rows("Adult-biased", min(bottom_idx), max(bottom_idx))
```

**Some HPO phenotype categories or more biased towards foetal- or adult- versions of the same cell type.** We took the top 50 phenotypes with the greatest bias towards foetal-cell type associations ("Foetal-biased") and the greatest bias towards adult-cell type associations ("Adult-biased") and fed each list of terms into ontological enrichment tests to get a summary of the representative HPO branches for each group.
The phenotypes most biased towards associations with only the foetal versions of cell type and those biased towards the adult versions of cell types.
"FDR" is the False Discovery Rate-adjusted p-value from the enrichment test, "log2-fold enrichment" is the log2 fold-change from the enrichment test, and "depth" is the depth of the enriched HPO term in the ontology.
:::

\newpage

::: {#tbl-fetal_examples}
```{r tbl-fetal_examples}
#| label: tbl-fetal_examples

fetal_examples_tbl <- data.table::rbindlist(list(
  top=run_congenital_enrichment_out$fetal_pdiff_top,
  bottom=run_congenital_enrichment_out$fetal_pdiff_bottom
), idcol = "group")[,c("group","hpo_name","hpo_id","cl_id","cl_name","fetal_nonfetal_pdiff")]|>unique()|>
  data.frame()|>
  dplyr::rename("p-value difference"="fetal_nonfetal_pdiff") 

top_idx = which(fetal_examples_tbl$group=="top")
bottom_idx = which(fetal_examples_tbl$group=="bottom")

fetal_examples_tbl|>
  dplyr::select(-group) |>
  dplyr::rename("HPO name"="hpo_name",
                "HPO ID"="hpo_id",
                "CL ID"="cl_id",
                "CL name"="cl_name") |>
  kableExtra::kbl() |> 
  kableExtra::group_rows("Foetal-biased", min(top_idx), max(top_idx)) |>
  kableExtra::group_rows("Adult-biased", min(bottom_idx), max(bottom_idx))
```

**Examples of specific phenotypes that are most biased towards associations with only the foetal versions of cell types ("Foetal-biased") and those biased towards the adult versions of cell types ("Adult-biased").** "p-value difference" is the difference in the association p-values between the foetal and adult version of the equivalent cell type ($\text{foetal-adult bias}: p_{adult} - p_{foetal} = \Delta p \in [-1,1]$).
:::

\newpage

::: {#tbl-ttd_hypergeo}
```{r tbl-ttd_hypergeo}
#| label: tbl-ttd_hypergeo

rbind(
  ttd_check_out$hypergeo,
  ttd_check_all_out$hypergeo
) |>
  data.frame() |> 
  dplyr::select(-targets_prioritised, -targets_in_status) |>
  kableExtra::kbl() |>
   kableExtra::group_rows("Gene therapies", 1, 2) |>
   kableExtra::group_rows("All therapies", 3, 4) 
```

**Hypergeometric test results evaluating the enrichment of existing therapy targets documented in the Therapeutic Target Database (TTD) amongst our prioritised targets.** Results are shown separately for gene therapies and all therapies.
Column keys: universe=gene universe size, p=hypergeometric test p-value, OR=odds ratio, NPV=negative predictive value, PPV=positive predictive value, FDR=false discovery rate, TP=true positives, FP=false positives, FN=false negatives, TN=true negatives.
:::

<!-- \newpage -->

<!-- ::: {#tbl-cl_enrich} -->

<!-- ```{r tbl-cl_enrich} -->

<!-- #| label: tbl-cl_enrich -->

<!-- cbind( -->

<!--   group="top", -->

<!--   run_congenital_enrichment_out$cl_enrich_top -->

<!--   )[,c("group","term","name","p_adjust","log2_fold_enrichment","depth")] -->

<!--   |>data.frame()|> -->

<!--   dplyr::mutate(group = ifelse(group=="top","foetal-biased","adult-biased"))|> -->

<!--   knitr::kable() -->

<!-- ``` -->

<!-- Cell type enrichment results. -->

<!-- The cell types that most strongly show a difference between their foetal and adult versions in their phenotype associations. -->

<!-- "p_adjust" is the adjusted p-value from the enrichment test, "log2_fold_enrichment" is the $log_2$ fold-change from the enrichment test, and "depth" is the depth of the CL term in the ontology. -->

<!-- ::: -->
